---
title: "Differential Expression Analysis - Symbiont Genes"
author: "Kelsey Beavers"
date: "`r Sys.Date()`"
output: html_document
editor_options:
  chunk_output_type: console
---

# Required Packages
```{r}
library(tximport)
library(tibble)
library(DESeq2)
library(dplyr)
library(PCAtools)
library(ggalt)
library(ggplot2)
library(lifecycle)
library(ggvenn)
library(stringr)
library(ggpubr)
library(pheatmap)
library(utils)
library(corrplot)
```

# Symbiont Composition Bar Charts

## C. natans
```{r}
metaData <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/metaData.csv")
cnatData <- metaData[metaData$species == 'cnat',]
names(cnatData)
colnames(cnatData)[colnames(cnatData)=="X.unambiguousReads_host"] <- "Host"
colnames(cnatData)[colnames(cnatData)=="X.unambiguousReads_cladeA"] <- "A"
colnames(cnatData)[colnames(cnatData)=="X.unambiguousReads_cladeB"] <- "B"
colnames(cnatData)[colnames(cnatData)=="X.unambiguousReads_cladeC"] <- "C"
colnames(cnatData)[colnames(cnatData)=="X.unambiguousReads_cladeD"] <- "D"
colnames(cnatData)[colnames(cnatData)=="X.unambiguousReads_unmatched"] <- "Unmatched"
names(cnatData)
cnatData <- cnatData[,c(3,4,10:13)] #name,treatment,A:D
names(cnatData)

cnat_reshape <- melt(cnatData,id=c("name","treatment"))
names(cnat_reshape)
colnames(cnat_reshape)[colnames(cnat_reshape)=="variable"] <- "Transcriptome"
colnames(cnat_reshape)[colnames(cnat_reshape)=="value"] <- "Reads"
colnames(cnat_reshape)[colnames(cnat_reshape)=="name"] <- "Sample"
colnames(cnat_reshape)[colnames(cnat_reshape)=="treatment"] <- "Treatment"
cnat_reshape

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/BBSplit_stats/")
pdf(file="Cnat_indiv_SCTLD.pdf",height=6,width=7)
cnat <- 
ggplot(cnat_reshape,aes(fill=Transcriptome, y = Reads, x =Sample))+
  geom_bar(position="fill",stat="identity")+
  labs(x = "Sample",y="Proportion of Symbiont Reads")+
  scale_fill_manual(values = c("#fff7bc","#a1dab4","#41b6c4","#225ea8"))+
  facet_wrap(~Treatment,scales = 'free_x')+
  theme_light()+
  theme(axis.text.x = element_text(angle = 90), plot.title = element_text(size=10))
cnat
dev.off()
```

## M. cavernosa
```{r}
metaData <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/metaData.csv")
mcavData <- metaData[metaData$species == 'mcav',]
names(mcavData)
colnames(mcavData)[colnames(mcavData)=="X.unambiguousReads_host"] <- "Host"
colnames(mcavData)[colnames(mcavData)=="X.unambiguousReads_cladeA"] <- "A"
colnames(mcavData)[colnames(mcavData)=="X.unambiguousReads_cladeB"] <- "B"
colnames(mcavData)[colnames(mcavData)=="X.unambiguousReads_cladeC"] <- "C"
colnames(mcavData)[colnames(mcavData)=="X.unambiguousReads_cladeD"] <- "D"
colnames(mcavData)[colnames(mcavData)=="X.unambiguousReads_unmatched"] <- "Unmatched"
names(mcavData)
mcavData <- mcavData[,c(3,4,10:13)] #name,treatment,A:D
names(mcavData)

mcav_reshape <- melt(mcavData,id=c("name","treatment"))
names(mcav_reshape)
colnames(mcav_reshape)[colnames(mcav_reshape)=="variable"] <- "Transcriptome"
colnames(mcav_reshape)[colnames(mcav_reshape)=="value"] <- "Reads"
colnames(mcav_reshape)[colnames(mcav_reshape)=="name"] <- "Sample"
colnames(mcav_reshape)[colnames(mcav_reshape)=="treatment"] <- "Treatment"

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/BBSplit_stats/")
pdf(file="mcav_indiv_SCTLD.pdf",height=6,width=7)
mcav <- 
ggplot(mcav_reshape,aes(fill=Transcriptome, y = Reads, x =Sample))+
  geom_bar(position="fill",stat="identity")+
  labs(x = "Sample",y="Proportion of Symbiont Reads")+
  scale_fill_manual(values = c("#fff7bc","#a1dab4","#41b6c4","#225ea8"))+
  facet_wrap(~Treatment,scales = 'free_x')+
  theme_light()+
  theme(axis.text.x = element_text(angle = 90), plot.title = element_text(size=10))
mcav
dev.off()
```

## O. annularis
```{r}
metaData <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/metaData.csv")
oannData <- metaData[metaData$species == 'oann',]
names(oannData)
colnames(oannData)[colnames(oannData)=="X.unambiguousReads_host"] <- "Host"
colnames(oannData)[colnames(oannData)=="X.unambiguousReads_cladeA"] <- "A"
colnames(oannData)[colnames(oannData)=="X.unambiguousReads_cladeB"] <- "B"
colnames(oannData)[colnames(oannData)=="X.unambiguousReads_cladeC"] <- "C"
colnames(oannData)[colnames(oannData)=="X.unambiguousReads_cladeD"] <- "D"
colnames(oannData)[colnames(oannData)=="X.unambiguousReads_unmatched"] <- "Unmatched"
names(oannData)
oannData <- oannData[,c(3,4,10:13)] #name,treatment,A:D
names(oannData)

oann_reshape <- melt(oannData,id=c("name","treatment"))
names(oann_reshape)
colnames(oann_reshape)[colnames(oann_reshape)=="variable"] <- "Transcriptome"
colnames(oann_reshape)[colnames(oann_reshape)=="value"] <- "Reads"
colnames(oann_reshape)[colnames(oann_reshape)=="name"] <- "Sample"
colnames(oann_reshape)[colnames(oann_reshape)=="treatment"] <- "Treatment"

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/BBSplit_stats/")
pdf(file="oann_indiv_SCTLD.pdf",height=6,width=7)
oann <- 
ggplot(oann_reshape,aes(fill=Transcriptome, y = Reads, x =Sample))+
  geom_bar(position="fill",stat="identity")+
  labs(x = "Sample",y="Proportion of Symbiont Reads")+
  scale_fill_manual(values = c("#fff7bc","#a1dab4","#41b6c4","#225ea8"))+
  facet_wrap(~Treatment,scales = 'free_x')+
  theme_light()+
  theme(axis.text.x = element_text(angle = 90), plot.title = element_text(size=10))
oann
dev.off()
```

## P. astreoides
```{r}
metaData <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/metaData.csv")
pastData <- metaData[metaData$species == 'past',]
names(pastData)
colnames(pastData)[colnames(pastData)=="X.unambiguousReads_host"] <- "Host"
colnames(pastData)[colnames(pastData)=="X.unambiguousReads_cladeA"] <- "A"
colnames(pastData)[colnames(pastData)=="X.unambiguousReads_cladeB"] <- "B"
colnames(pastData)[colnames(pastData)=="X.unambiguousReads_cladeC"] <- "C"
colnames(pastData)[colnames(pastData)=="X.unambiguousReads_cladeD"] <- "D"
colnames(pastData)[colnames(pastData)=="X.unambiguousReads_unmatched"] <- "Unmatched"
names(pastData)
pastData <- pastData[,c(3,4,10:13)] #name,treatment,A:D
names(pastData)

past_reshape <- melt(pastData,id=c("name","treatment"))
names(past_reshape)
colnames(past_reshape)[colnames(past_reshape)=="variable"] <- "Transcriptome"
colnames(past_reshape)[colnames(past_reshape)=="value"] <- "Reads"
colnames(past_reshape)[colnames(past_reshape)=="name"] <- "Sample"
colnames(past_reshape)[colnames(past_reshape)=="treatment"] <- "Treatment"

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/BBSplit_stats/")
pdf(file="past_indiv_SCTLD.pdf",height=6,width=7)
past <- 
ggplot(past_reshape,aes(fill=Transcriptome, y = Reads, x =Sample))+
  geom_bar(position="fill",stat="identity")+
  labs(x = "Sample",y="Proportion of Symbiont Reads")+
  scale_fill_manual(values = c("#fff7bc","#a1dab4","#41b6c4","#225ea8"))+
  facet_wrap(~Treatment,scales = 'free_x')+
  theme_light()+
  theme(axis.text.x = element_text(angle = 90), plot.title = element_text(size=10))
past
dev.off()
```

## P. strigosa
```{r}
metaData <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/metaData.csv")
pstrData <- metaData[metaData$species == 'pstr',]
names(pstrData)
colnames(pstrData)[colnames(pstrData)=="X.unambiguousReads_host"] <- "Host"
colnames(pstrData)[colnames(pstrData)=="X.unambiguousReads_cladeA"] <- "A"
colnames(pstrData)[colnames(pstrData)=="X.unambiguousReads_cladeB"] <- "B"
colnames(pstrData)[colnames(pstrData)=="X.unambiguousReads_cladeC"] <- "C"
colnames(pstrData)[colnames(pstrData)=="X.unambiguousReads_cladeD"] <- "D"
colnames(pstrData)[colnames(pstrData)=="X.unambiguousReads_unmatched"] <- "Unmatched"
names(pstrData)
pstrData <- pstrData[,c(3,4,10:13)] #name,treatment,A:D
names(pstrData)

pstr_reshape <- melt(pstrData,id=c("name","treatment"))
names(pstr_reshape)
colnames(pstr_reshape)[colnames(pstr_reshape)=="variable"] <- "Transcriptome"
colnames(pstr_reshape)[colnames(pstr_reshape)=="value"] <- "Reads"
colnames(pstr_reshape)[colnames(pstr_reshape)=="name"] <- "Sample"
colnames(pstr_reshape)[colnames(pstr_reshape)=="treatment"] <- "Treatment"

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/BBSplit_stats/")
pdf(file="pstr_indiv_SCTLD.pdf",height=6,width=7)
pstr <- 
  ggplot(pstr_reshape,aes(fill=Transcriptome, y = Reads, x =Sample))+
  geom_bar(position="fill",stat="identity")+
  labs(x = "Sample",y="Proportion of Symbiont Reads")+
  scale_fill_manual(values = c("#fff7bc","#a1dab4","#41b6c4","#225ea8"))+
  facet_wrap(~Treatment,scales = 'free_x')+
  theme_light()+
  theme(axis.text.x = element_text(angle = 90), plot.title = element_text(size=10))
pstr
dev.off()
```

## Combined
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/BBSplit_stats/")
pdf(file="symbiont_composition_all.pdf",height=6,width=10)
ggarrange(past,
          pstr,
          mcav,
          oann,
          cnat,
          ncol=3, nrow = 2, common.legend = TRUE, legend = "right",align = "hv")
dev.off()
```

# Clade A

## Tximport

Read in the raw counts from Salmon for all samples dominant in Clade A symbionts

Past: c1,c5,c6,c7,d1,d2,d3,d4,d5,d6,d7

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeA/salmon/")
cladeA_samples <- read.table("cladeA_samples.csv", header = TRUE)
cladeA_samples
cladeA_files <- file.path(cladeA_samples$sample, "quant.sf")
names(cladeA_files) <- paste0("sample", 1:11)
all(file.exists(cladeA_files))
CladeA_tx2gene <- read.csv("cladeA_tx2gene.csv")
head(CladeA_tx2gene)
cladeA_txi <- tximport(cladeA_files, type = 'salmon',tx2gene=CladeA_tx2gene)
names(cladeA_txi)
tail(cladeA_txi$counts)
dim(cladeA_txi$counts)
# 10,675 contigs with evalue < 0.000001
write.csv(cladeA_txi, file = "CladeA_txi.csv")
write.csv(cladeA_txi$counts, file="CladeA_counts.csv",quote = FALSE)
cladeA_counts <- cladeA_txi$counts
colnames(cladeA_counts)
cladeA_counts <- as.data.frame(cladeA_counts)
colnames(cladeA_counts) <- c("past_c1","past_c5","past_c6","past_c7","past_d1","past_d2","past_d3","past_d4","past_d5","past_d6","past_d7")
cladeA_counts <- tibble::rownames_to_column(cladeA_counts,"Entry")
cladeA_counts[2:12] <-  round(cladeA_counts[2:12], digits=0)
cladeA_counts <- as.data.frame(cladeA_counts)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeA/DESeq2/")
write.csv(cladeA_counts, file = "cladeA_counts.csv",quote = FALSE)
```

## DESeq2 

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeA/DESeq2/")

countData <- read.csv("CladeA_counts.csv", row.names="Entry")
colnames(countData)
countData <- countData[,-c(1)]
colData <- read.csv("./../../../metaData.csv", row.names = "name")
cladeAColData <- colData[colData$dominant_clade == 'A',]
rownames(cladeAColData)
colnames(countData)

# Set the design. This design will test for the differences in expression caused by treatment 
# Filter out columns with an average number of reads less than 10. 
dds_treatment <- DESeqDataSetFromMatrix(countData = countData, 
                                        colData = cladeAColData, 
                                        design = ~treatment )
dds_treatment <- dds_treatment[ rowMeans(counts(dds_treatment)) > 10, ] 
test <- DESeq(dds_treatment)
res <- results(test)
res
resorderd <-res[order(res$padj),] 
write.csv(resorderd, file = "DESeq_treatment.csv")
rld <- rlog(dds_treatment, blind=FALSE) 
rrld <- assay(rld)
write.csv(rrld, file = "treatment_rlog.csv")  

# Keep significant DEGs (padj < 0.05)
DESeq_treatment_master <- read.csv("DESeq_treatment.csv")
DEGs <- DESeq_treatment_master %>% filter(padj < 0.05)
# 0 DEGs
write.csv(DEGs, file = "treatment_sig_DEGs.csv")
```

## PCA

```{r}
# Remove all columns except for "Entry ID" and the counts
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeA/DESeq2/")
countData <- read.csv("treatment_rlog.csv", row.names="X")
colnames(countData)
Gene_expression <- as.matrix(countData)
Metadata <- as.data.frame(read.csv("./../../../metaData.csv", header = TRUE, row.names = "name"))
cladeAMetadata <- Metadata[Metadata$dominant_clade == 'A',]
rownames(cladeAMetadata)
colnames(Gene_expression)
p <- pca(Gene_expression, metadata = cladeAMetadata,center = TRUE,removeVar = 0.1) #Remove the lower 10% of variables based on variance. 
screeplot(p)

cladeAMetadata
pdf(file="rlog_PCA.pdf",height=6,width=6)
biplot(p,
       lab = cladeAMetadata$sample_number,
       colby = "treatment", colkey = c('control'='blue','disease' = 'red'), colLegendTitle = 'Treatment',
       ellipse = TRUE,
       xlim=c(-60,50), ylim = c(-40,50),
       title = 'Rlog Gene Expression PCA',
       legendPosition = 'right') 
  dev.off()
```

# Clade B

## Tximport

Read in the raw counts from Salmon for all samples dominant in Clade B symbionts

Pstr: c3,c5,c7,c8,d3,d5,d7

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeB/salmon/")
cladeB_samples <- read.table("cladeB_samples.csv", header = TRUE)
cladeB_samples
cladeB_files <- file.path(cladeB_samples$sample, "quant.sf")
names(cladeB_files) <- paste0("sample", 1:7)
all(file.exists(cladeB_files))
CladeB_tx2gene <- read.csv("CladeB_tx2gene.csv")
head(CladeB_tx2gene)
cladeB_txi <- tximport(cladeB_files, type = 'salmon',tx2gene=CladeB_tx2gene)
names(cladeB_txi)
tail(cladeB_txi$counts)
dim(cladeB_txi$counts)
# 10,072 contigs 
write.csv(cladeB_txi, file = "CladeB_txi.csv")
write.csv(cladeB_txi$counts, file="CladeB_counts.csv",quote = FALSE)
cladeB_counts <- cladeB_txi$counts
colnames(cladeB_counts)
cladeB_counts <- as.data.frame(cladeB_counts)
colnames(cladeB_counts) <- c("pstr_c3","pstr_c5","pstr_c7","pstr_c8","pstr_d3","pstr_d5","pstr_d7")
cladeB_counts <- tibble::rownames_to_column(cladeB_counts,"Entry")
cladeB_counts[2:8] <-  round(cladeB_counts[2:8], digits=0)
cladeB_counts <- as.data.frame(cladeB_counts)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeB/DESeq2/")
write.csv(cladeB_counts, file = "cladeB_counts.csv",quote = FALSE)
```

## DESeq2 

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeB/DESeq2/")

countData <- read.csv("cladeB_counts.csv", row.names="Entry")
colnames(countData)
countData <- countData[,-c(1)]
colData <- read.csv("./../../../metaData.csv", row.names = "name")
cladeBColData <- colData[colData$dominant_clade == 'B',]
rownames(cladeBColData)
colnames(countData)

# Set the design. This design will test for the differences in expression caused by treatment 
# Filter out columns with an average number of reads less than 10. 
dds_treatment <- DESeqDataSetFromMatrix(countData = countData, 
                                        colData = cladeBColData, 
                                        design = ~treatment )
dds_treatment <- dds_treatment[ rowMeans(counts(dds_treatment)) > 10, ] 
test <- DESeq(dds_treatment)
res <- results(test)
res
resorderd <-res[order(res$padj),] 
write.csv(resorderd, file = "DESeq_treatment.csv")
rld <- rlog(dds_treatment, blind=FALSE) 
rrld <- assay(rld)
write.csv(rrld, file = "treatment_rlog.csv")  

# Keep significant DEGs (padj < 0.05)
DESeq_treatment_master <- read.csv("DESeq_treatment.csv")
DEGs <- DESeq_treatment_master %>% filter(padj < 0.05)
# 18 DEGs
write.csv(DEGs, file = "treatment_sig_DEGs.csv")
```

## PCA

```{r}
# Remove all columns except for "Entry ID" and the counts
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeB/DESeq2/")
countData <- read.csv("treatment_rlog.csv", row.names="X")
colnames(countData)
Gene_expression <- as.matrix(countData)
Metadata <- as.data.frame(read.csv("./../../../metaData.csv", header = TRUE, row.names = "name"))
cladeBMetadata <- Metadata[Metadata$dominant_clade == 'B',]
rownames(cladeBMetadata)
colnames(Gene_expression)
p <- pca(Gene_expression, metadata = cladeBMetadata,center = TRUE,removeVar = 0.1) #Remove the lower 10% of variables based on variance. 
screeplot(p)

cladeBMetadata
pdf(file="rlog_PCA.pdf",height=6,width=6)
biplot(p,
       lab = cladeBMetadata$sample_number,
       colby = "treatment", colkey = c('control'='blue','disease' = 'red'), colLegendTitle = 'Treatment',
       ellipse = TRUE,
       xlim=c(-60,50), ylim = c(-40,50),
       title = 'Rlog Gene Expression PCA',
       legendPosition = 'right') 
  dev.off()
```

# Clade C

## Tximport

Read in the raw counts from Salmon for all samples dominant in Clade C symbionts

Cnat: c3,c4,d3,d4
Mcav: c1,c3,c5,c6,c7,c8,d1,d2,d3,d4,d5,d6,d8
Oann: c1,c3,c6,c7,c8,d1,d3,d6,d7,d8

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeC/salmon/")
cladeC_samples <- read.table("cladeC_samples.csv", header = TRUE)
cladeC_samples
cladeC_files <- file.path(cladeC_samples$sample, "quant.sf")
names(cladeC_files) <- paste0("sample", 1:27)
all(file.exists(cladeC_files))
CladeC_tx2gene <- read.csv("cladeC_tx2gene.csv")
head(CladeC_tx2gene)
cladeC_txi <- tximport(cladeC_files, type = 'salmon',tx2gene=CladeC_tx2gene)
names(cladeC_txi)
tail(cladeC_txi$counts)
dim(cladeC_txi$counts)
# 11,939 contigs with evalue < 0.000001
write.csv(cladeC_txi, file = "CladeC_txi.csv")
write.csv(cladeC_txi$counts, file="CladeC_counts.csv",quote = FALSE)
cladeC_counts <- cladeC_txi$counts
colnames(cladeC_counts)
cladeC_counts <- as.data.frame(cladeC_counts)
colnames(cladeC_counts) <- c("cnat_c3","cnat_c4","cnat_d3","cnat_d4","mcav_c1","mcav_c3","mcav_c5","mcav_c6","mcav_c7","mcav_c8","mcav_d1","mcav_d2","mcav_d3","mcav_d4","mcav_d5","mcav_d6","mcav_d8","oann_c1","oann_c3","oann_c6","oann_c7","oann_c8","oann_d1","oann_d3","oann_d6","oann_d7","oann_d8")
cladeC_counts <- tibble::rownames_to_column(cladeC_counts,"Entry")
cladeC_counts[2:28] <-  round(cladeC_counts[2:28], digits=0)
cladeC_counts <- as.data.frame(cladeC_counts)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeC/DESeq2/")
write.csv(cladeC_counts, file = "cladeC_counts.csv",quote = FALSE)
```

## DESeq2 

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeC/DESeq2/")

countData <- read.csv("cladeC_counts.csv", row.names="Entry")
colnames(countData)
countData <- countData[,-c(1)]
colData <- read.csv("./../../../metaData.csv", row.names = "name")
cladeCColData <- colData[colData$dominant_clade == 'C',]
# Remove oann_c2, outlier
cladeCColData <- cladeCColData[-c(19),]
rownames(cladeCColData)
colnames(countData)

# Set the design. This design will test for the differences in expression caused by treatment 
# Filter out columns with an average number of reads less than 10. 
dds_treatment <- DESeqDataSetFromMatrix(countData = countData, 
                                        colData = cladeCColData, 
                                        design = ~treatment )
dds_treatment <- dds_treatment[ rowMeans(counts(dds_treatment)) > 10, ] 
test <- DESeq(dds_treatment)
res <- results(test)
res
resorderd <-res[order(res$padj),] 
write.csv(resorderd, file = "DESeq_treatment.csv")
rld <- rlog(dds_treatment, blind=FALSE) 
rrld <- assay(rld)
write.csv(rrld, file = "treatment_rlog.csv")  

# Keep significant DEGs (padj < 0.05)
DESeq_treatment_master <- read.csv("DESeq_treatment.csv")
DEGs <- DESeq_treatment_master %>% filter(padj < 0.05)
# 3 DEGs
write.csv(DEGs, file = "treatment_sig_DEGs.csv")
```

## PCA

```{r}
# Remove all columns except for "Entry ID" and the counts
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeC/DESeq2/")
countData <- read.csv("treatment_rlog.csv", row.names="X")
colnames(countData)
Gene_expression <- as.matrix(countData)
Metadata <- as.data.frame(read.csv("./../../../metaData.csv", header = TRUE, row.names = "name"))
cladeCMetadata <- Metadata[Metadata$dominant_clade == 'C',]
cladeCMetadata <- cladeCMetadata[-c(19),]
rownames(cladeCMetadata)
colnames(Gene_expression)
p <- pca(Gene_expression, metadata = cladeCMetadata,center = TRUE,removeVar = 0.1) #Remove the lower 10% of variables based on variance. 
screeplot(p)

cladeCMetadata
pdf(file="rlog_PCA.pdf",height=6,width=6)
biplot(p,
       lab = cladeCMetadata$sample_number,
       colby = "treatment", colkey = c('control'='blue','disease' = 'red'), colLegendTitle = 'Treatment',
       ellipse = TRUE,
       xlim=c(-60,60), ylim = c(-40,50),
       title = 'Rlog Gene Expression PCA',
       legendPosition = 'right') 
  dev.off()
```

# Clade D

## Tximport

Read in the raw counts from Salmon for all samples dominant in Clade D symbionts

Cnat: c2,c6,c8,d2,d6,d8

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeD/salmon/")
cladeD_samples <- read.table("cladeD_samples.csv", header = TRUE)
cladeD_samples
cladeD_files <- file.path(cladeD_samples$sample, "quant.sf")
names(cladeD_files) <- paste0("sample", 1:6)
all(file.exists(cladeD_files))
CladeD_tx2gene <- read.csv("cladeD_tx2gene.csv")
head(CladeD_tx2gene)
cladeD_txi <- tximport(cladeD_files, type = 'salmon',tx2gene=CladeD_tx2gene)
names(cladeD_txi)
tail(cladeD_txi$counts)
dim(cladeD_txi$counts)
# 10,328 contigs 
write.csv(cladeD_txi, file = "CladeD_txi.csv")
write.csv(cladeD_txi$counts, file="CladeD_counts.csv",quote = FALSE)
cladeD_counts <- cladeD_txi$counts
colnames(cladeD_counts)
cladeD_counts <- as.data.frame(cladeD_counts)
colnames(cladeD_counts) <- c("cnat_c2","cnat_c6","cnat_c8","cnat_d2","cnat_d6","cnat_d8")
cladeD_counts <- tibble::rownames_to_column(cladeD_counts,"Entry")
cladeD_counts[2:7] <-  round(cladeD_counts[2:7], digits=0)
cladeD_counts <- as.data.frame(cladeD_counts)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeD/DESeq2/")
write.csv(cladeD_counts, file = "cladeD_counts.csv",quote = FALSE)
```

## DESeq2 

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeD/DESeq2/")

countData <- read.csv("cladeD_counts.csv", row.names="Entry")
colnames(countData)
countData <- countData[,-c(1)]
colData <- read.csv("./../../../metaData.csv", row.names = "name")
cladeDColData <- colData[colData$dominant_clade == 'D',]
rownames(cladeDColData)
colnames(countData)

# Set the design. This design will test for the differences in expression caused by treatment 
# Filter out columns with an average number of reads less than 10. 
dds_treatment <- DESeqDataSetFromMatrix(countData = countData, 
                                        colData = cladeDColData, 
                                        design = ~treatment )
dds_treatment <- dds_treatment[ rowMeans(counts(dds_treatment)) > 10, ] 
test <- DESeq(dds_treatment)
res <- results(test)
res
resorderd <-res[order(res$padj),] 
write.csv(resorderd, file = "DESeq_treatment.csv")
rld <- rlog(dds_treatment, blind=FALSE) 
rrld <- assay(rld)
write.csv(rrld, file = "treatment_rlog.csv")  

# Keep significant DEGs (padj < 0.05)
DESeq_treatment_master <- read.csv("DESeq_treatment.csv")
DEGs <- DESeq_treatment_master %>% filter(padj < 0.05)
# 13 DEGs
write.csv(DEGs, file = "treatment_sig_DEGs.csv")
```

## PCA

```{r}
# Remove all columns except for "Entry ID" and the counts
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeD/DESeq2/")
countData <- read.csv("treatment_rlog.csv", row.names="X")
colnames(countData)
Gene_expression <- as.matrix(countData)
Metadata <- as.data.frame(read.csv("./../../../metaData.csv", header = TRUE, row.names = "name"))
cladeDMetadata <- Metadata[Metadata$dominant_clade == 'D',]
rownames(cladeDMetadata)
colnames(Gene_expression)
p <- pca(Gene_expression, metadata = cladeDMetadata,center = TRUE,removeVar = 0.1) #Remove the lower 10% of variables based on variance. 
screeplot(p)

cladeDMetadata
pdf(file="rlog_PCA.pdf",height=6,width=6)
biplot(p,
       lab = cladeDMetadata$sample_number,
       colby = "treatment", colkey = c('control'='blue','disease' = 'red'), colLegendTitle = 'Treatment',
       ellipse = TRUE,
       xlim=c(-60,60), ylim = c(-40,50),
       title = 'Rlog Gene Expression PCA',
       legendPosition = 'right') 
  dev.off()
```

# Analysis of Shared Genes (inferred homologs)

## Venn Diagram - Shared Genes
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/")
cladeA <- read.csv("./cladeA/DESeq2/cladeA_counts.csv")
cladeA <- cladeA[,-c(1)]

cladeB <- read.csv("./cladeB/DESeq2/cladeB_counts.csv")
cladeB <- cladeB[,-c(1)]
names(cladeB)

cladeC <- read.csv("./cladeC/DESeq2/cladeC_counts.csv")
cladeC <- cladeC[,-c(1)]
names(cladeC)

cladeD <- read.csv("./cladeD/DESeq2/cladeD_counts.csv")
cladeD <- cladeD[,-c(1)]
names(cladeD)


dim(cladeA) #10,675
dim(cladeB) #10,072
dim(cladeC) #11,939
dim(cladeD) #10,328

cladeA_cladeB <- merge(cladeA,cladeB,by="Entry") 
names(cladeA_cladeB)
dim(cladeA_cladeB) #4,965

cladeA_cladeB_cladeC <- merge(cladeA_cladeB,cladeC,by="Entry")
names(cladeA_cladeB_cladeC)
dim(cladeA_cladeB_cladeC) #4,037

cladeA_cladeB_cladeC_cladeD <- merge(cladeA_cladeB_cladeC, cladeD, by="Entry")
names(cladeA_cladeB_cladeC_cladeD)
dim(cladeA_cladeB_cladeC_cladeD) #3,379

master <- cladeA_cladeB_cladeC_cladeD[ !duplicated(cladeA_cladeB_cladeC_cladeD$Entry), ]  
dim(master) #3,379

getwd()
setwd("./all/DESeq2/")
write.csv(master,file = "sharedGenes_counts.csv")

x <- list(
  cladeA = cladeA$Entry,
  cladeB = cladeB$Entry,
  cladeC = cladeC$Entry,
  cladeD = cladeD$Entry
)

ggvenn(
  x,
  fill_color = c("#FFF7BC", "#A1DAB4", "#41B6C4", "#225EA8"),
  stroke_size = 0.5, set_name_size = 5
)
dev.off()
```

## Venn Diagram - DEGs

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/")
cladeA <- read.csv("./cladeA/DESeq2/treatment_sig_DEGs.csv")
cladeB <- read.csv("./cladeB/DESeq2/treatment_sig_DEGs.csv")
cladeC <- read.csv("./cladeC/DESeq2/treatment_sig_DEGs.csv")
cladeD <- read.csv("./cladeD/DESeq2/treatment_sig_DEGs.csv")
colnames(cladeA)[colnames(cladeA)=="X"] <- "Entry"
colnames(cladeB)[colnames(cladeB)=="X"] <- "Entry"
colnames(cladeC)[colnames(cladeC)=="X"] <- "Entry"
colnames(cladeD)[colnames(cladeD)=="X"] <- "Entry"

x <- list(
  cladeA = cladeA$Entry,
  cladeB = cladeB$Entry,
  cladeC = cladeC$Entry,
  cladeD = cladeD$Entry
)

ggvenn(
  x,
  fill_color = c("#FFF7BC", "#A1DAB4", "#41B6C4", "#225EA8"),
  stroke_size = 0.5, set_name_size = 5
)
dev.off()
```

# Rab7 Correlation Analysis
Correlate Coral host Rab7 expression to histology, the three shared symbiont photosystem genes (psaA, psbB and psbC) and four representative metabolism genes
psaA = Q9XQV3
psbB = P49471
psbC = P49472
High affinity nitrate transporter 2.5 (NRT2.5) = Q9LPV5
Superoxide dismutase = Q59094
Carbonic anhydrase = Q5BCC5
Apoptosis-inducing factor 3 = Q96NN9
Heat shock 70 kDa protein = P11144
Two pore calcium channel protein 1 = Q5QM84

## Clade A
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/DESeq2")
coralShared_rlogs <- read.csv("treatment_rlog.csv",row.names = "X")
colnames(coralShared_rlogs)[colnames(coralShared_rlogs)=="X"] <- "Entry"
rab7 <- coralShared_rlogs[c("H9BW96"),]
names(rab7)
t_rab7 <- as.data.frame(t(rab7))
t_rab7 <- tibble::rownames_to_column(t_rab7, "name")

histology <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/histologyData.csv")
names(histology)
cladeA_histology <- histology[histology$dominant_genus == 'Symbiodinium',]
rownames(cladeA_histology)
cladeA_histology <- cladeA_histology[,c(3,7:9)]
cladeA_histology

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeA/DESeq2/")
cladeA_rlog <- read.csv("treatment_rlog.csv",row.names = "X")
names(cladeA_rlog)
curatedGenes <- cladeA_rlog[c("Q9XQV3","P49471","P49472","Q9LPV5","Q59094","Q5BCC5","Q96NN9","P11144","Q5QM84"),]
t_rlogs <- as.data.frame(t(curatedGenes))
t_rlogs <- tibble::rownames_to_column(t_rlogs, "name")
cladeA_correlations <- merge(t_rab7,cladeA_histology,by="name")
cladeA_correlations <- merge(cladeA_correlations,t_rlogs,by="name")

names(cladeA_correlations)
cladeA_correlations <- cladeA_correlations[,-c(1)]
colnames(cladeA_correlations)
colnames(cladeA_correlations)[1] <- "Host Rab7"
colnames(cladeA_correlations)[5] <- "psaA"
colnames(cladeA_correlations)[6] <- "psbB"
colnames(cladeA_correlations)[7] <- "psbC"
colnames(cladeA_correlations)[8] <- "High affinity nitrate transporter 2.5"
colnames(cladeA_correlations)[9] <- "Superoxide dismutase"
colnames(cladeA_correlations)[10] <- "Carbonic anhydrase"
colnames(cladeA_correlations)[11] <- "Apoptosis-inducing factor 3"
colnames(cladeA_correlations)[12] <- "Heat shock protein 70 "
colnames(cladeA_correlations)[13] <- "Two pore calcium channel protein 1"
colnames(cladeA_correlations)

A = cor(cladeA_correlations)
A.testRes = cor.mtest(cladeA_correlations, conf.level = 0.95)


  corrplot(A,p.mat = A.testRes$p,
         method = 'circle',
         diag = FALSE,
         type = 'upper',
         sig.level = c(0.001,0.01,0.05),
         pch.cex = 1,
         insig = 'label_sig',
         pch.col = 'white',
         title = "Symbiodinium Correlations",
         tl.col = "black",
         tl.cex = 0.5,
        # cl.ratio = 0.15,
         tl.srt = 45,
         colorRampPalette(c("#0571b0","#92c5de","white","#f4a582","#ca0020"))(200))
)
```

## Clade B
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/DESeq2")
coralShared_rlogs <- read.csv("treatment_rlog.csv",row.names = "X")
colnames(coralShared_rlogs)[colnames(coralShared_rlogs)=="X"] <- "Entry"
rab7 <- coralShared_rlogs[c("H9BW96"),]
names(rab7)
t_rab7 <- as.data.frame(t(rab7))
t_rab7 <- tibble::rownames_to_column(t_rab7, "name")

histology <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/histologyData.csv")
names(histology)
cladeB_histology <- histology[histology$dominant_genus == 'Breviolum',]
cladeB_histology <- cladeB_histology[,c(3,7:9)]
cladeB_histology

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeB/DESeq2/")
cladeB_rlog <- read.csv("treatment_rlog.csv",row.names = "X")
names(cladeB_rlog)
curatedGenes <- cladeB_rlog[c("Q9XQV3","P49471","P49472","Q9LPV5","Q59094","Q5BCC5","Q96NN9","P11144","Q5QM84"),]
t_rlogs <- as.data.frame(t(curatedGenes))
t_rlogs <- tibble::rownames_to_column(t_rlogs, "name")
cladeB_correlations <- merge(t_rab7,cladeB_histology,by="name")
cladeB_correlations <- merge(cladeB_correlations,t_rlogs,by="name")

names(cladeB_correlations)
cladeB_correlations <- cladeB_correlations[,-c(1)]
colnames(cladeB_correlations)
colnames(cladeB_correlations)[1] <- "Host Rab7"
colnames(cladeB_correlations)[5] <- "psaA"
colnames(cladeB_correlations)[6] <- "psbB"
colnames(cladeB_correlations)[7] <- "psbC"
colnames(cladeB_correlations)[8] <- "High affinity nitrate transporter 2.5"
colnames(cladeB_correlations)[9] <- "Superoxide dismutase"
colnames(cladeB_correlations)[10] <- "Carbonic anhydrase"
colnames(cladeB_correlations)[11] <- "Apoptosis-inducing factor 3"
colnames(cladeB_correlations)[12] <- "Heat shock protein 70 "
colnames(cladeB_correlations)[13] <- "Two pore calcium channel protein 1"
colnames(cladeB_correlations)

B = cor(cladeB_correlations)
B.testRes = cor.mtest(cladeB_correlations, conf.level = 0.95)


  corrplot(B,p.mat = B.testRes$p,
         method = 'circle',
         diag = FALSE,
         type = 'upper',
         sig.level = c(0.001,0.01,0.05),
         pch.cex = 1,
         insig = 'label_sig',
         pch.col = 'white',
         title = "Breviolum Correlations",
         tl.col = "black",
         tl.cex = 0.5,
        # cl.ratio = 0.15,
         tl.srt = 45,
         colorRampPalette(c("#0571b0","#92c5de","white","#f4a582","#ca0020"))(200))
)
```


## Clade C
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/DESeq2")
coralShared_rlogs <- read.csv("treatment_rlog.csv",row.names = "X")
colnames(coralShared_rlogs)[colnames(coralShared_rlogs)=="X"] <- "Entry"
rab7 <- coralShared_rlogs[c("H9BW96"),]
names(rab7)
t_rab7 <- as.data.frame(t(rab7))
t_rab7 <- tibble::rownames_to_column(t_rab7, "name")

histology <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/histologyData.csv")
names(histology)
cladeC_histology <- histology[histology$dominant_genus == 'Cladocopium',]
cladeC_histology <- cladeC_histology[,c(3,7:9)]
cladeC_histology

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeC/DESeq2/")
cladeC_rlog <- read.csv("treatment_rlog.csv",row.names = "X")
names(cladeC_rlog)
curatedGenes <- cladeC_rlog[c("Q9XQV3","P49471","P49472","Q9LPV5","Q59094","Q5BCC5","Q96NN9","P11144","Q5QM84"),]
t_rlogs <- as.data.frame(t(curatedGenes))
t_rlogs <- tibble::rownames_to_column(t_rlogs, "name")
cladeC_correlations <- merge(t_rab7,cladeC_histology,by="name")
cladeC_correlations <- merge(cladeC_correlations,t_rlogs,by="name")

names(cladeC_correlations)
cladeC_correlations <- cladeC_correlations[,-c(1)]
colnames(cladeC_correlations)
colnames(cladeC_correlations)[1] <- "Host Rab7"
colnames(cladeC_correlations)[5] <- "psaA"
colnames(cladeC_correlations)[6] <- "psbB"
colnames(cladeC_correlations)[7] <- "psbC"
colnames(cladeC_correlations)[8] <- "High affinity nitrate transporter 2.5"
colnames(cladeC_correlations)[9] <- "Superoxide dismutase"
colnames(cladeC_correlations)[10] <- "Carbonic anhydrase"
colnames(cladeC_correlations)[11] <- "Apoptosis-inducing factor 3"
colnames(cladeC_correlations)[12] <- "Heat shock protein 70 "
colnames(cladeC_correlations)[13] <- "Two pore calcium channel protein 1"
colnames(cladeC_correlations)

C = cor(cladeC_correlations)
C.testRes = cor.mtest(cladeC_correlations, conf.level = 0.95)


  corrplot(C,#p.mat = C.testRes$p,
         method = 'circle',
         diag = FALSE,
         type = 'upper',
         sig.level = c(0.001,0.01,0.05),
         pch.cex = 1,
         insig = 'label_sig',
         pch.col = 'white',
         title = "Breviolum Correlations",
         tl.col = "black",
         tl.cex = 0.5,
        # cl.ratio = 0.15,
         tl.srt = 45,
         colorRampPalette(c("#0571b0","#92c5de","white","#f4a582","#ca0020"))(200))
)
```


## Clade D
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/DESeq2")
coralShared_rlogs <- read.csv("treatment_rlog.csv",row.names = "X")
colnames(coralShared_rlogs)[colnames(coralShared_rlogs)=="X"] <- "Entry"
rab7 <- coralShared_rlogs[c("H9BW96"),]
names(rab7)
t_rab7 <- as.data.frame(t(rab7))
t_rab7 <- tibble::rownames_to_column(t_rab7, "name")

histology <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/histologyData.csv")
names(histology)
cladeD_histology <- histology[histology$dominant_genus == 'Durusdinium',]
cladeD_histology <- cladeD_histology[,c(3,7:9)]
cladeD_histology

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/symbiont/cladeD/DESeq2/")
cladeD_rlog <- read.csv("treatment_rlog.csv",row.names = "X")
names(cladeD_rlog)
curatedGenes <- cladeD_rlog[c("Q9XQV3","P49471","P49472","Q9LPV5","Q59094","Q5BCC5","Q96NN9","P11144","Q5QM84"),]
t_rlogs <- as.data.frame(t(curatedGenes))
t_rlogs <- tibble::rownames_to_column(t_rlogs, "name")
cladeD_correlations <- merge(t_rab7,cladeD_histology,by="name")
cladeD_correlations <- merge(cladeD_correlations,t_rlogs,by="name")

names(cladeD_correlations)
cladeD_correlations <- cladeD_correlations[,-c(1)]
colnames(cladeD_correlations)
colnames(cladeD_correlations)[1] <- "Host Rab7"
colnames(cladeD_correlations)[5] <- "psaA"
colnames(cladeD_correlations)[6] <- "psbB"
colnames(cladeD_correlations)[7] <- "psbC"
colnames(cladeD_correlations)[8] <- "High affinity nitrate transporter 2.5"
colnames(cladeD_correlations)[9] <- "Superoxide dismutase"
colnames(cladeD_correlations)[10] <- "Carbonic anhydrase"
colnames(cladeD_correlations)[11] <- "Apoptosis-inducing factor 3"
colnames(cladeD_correlations)[12] <- "Heat shock protein 70 "
colnames(cladeD_correlations)[13] <- "Two pore calcium channel protein 1"
colnames(cladeD_correlations)

D = cor(cladeD_correlations)
D.testRes = cor.mtest(cladeD_correlations, conf.level = 0.95)


  corrplot(D,#p.mat = D.testRes$p,
         method = 'circle',
         diag = FALSE,
         type = 'upper',
         sig.level = c(0.001,0.01,0.05),
         pch.cex = 1,
         insig = 'label_sig',
         pch.col = 'white',
         title = "Breviolum Correlations",
         tl.col = "black",
         tl.cex = 0.5,
        # cl.ratio = 0.15,
         tl.srt = 45,
         colorRampPalette(c("#0571b0","#92c5de","white","#f4a582","#ca0020"))(200))
)
```


