---
title: "Coral Orthogroup Analysis"
author: "Kelsey Beavers"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/")
knitr::opts_chunk$set(dev='ragg_png')
library(ggpubr)
library(DESeq2)
library(tibble)
library(tidyverse)
library(tximport)
library(GOplot)
library(dplyr)
library(tidyr)
library(reshape2)
library(ggalt)
library(ggplot2)
library(lifecycle)
library(stringr)
library(utils)
library(corrplot)
library(repr) 
library(data.table)
```

## Step 1: Obtain Orthologs from Orthofinder

To obtain orthologs, we filter our Orthofinder results to only contain Orthogroups with at least one sequence per species. 

This can be done in excel like so:
1. Open Orthogroups.tsv in Excel
2. Click *Find & Select*
3. Click *Go to Special*
4. Choose *Blanks*
5. Click OK and then all the *blank rows/cells will be highlighted*
6. Choose the *Delete under Cells* section on the Home Tab
7. Click *Delete Sheet Rows*
8. Save as "shared_orthogroups.csv"

### Make tx2gene files using shared orthogroups

For this analysis we will use "Orthogroup_Sequences.csv" from the "Command_Line_Code.rmd" file to create new tx2gene files for each species

```{r}
Orthogroups <- read.csv("./all/Orthofinder/Orthogroups.csv")
cnat_orthogroups <- Orthogroups[,c(1,2)]
cnat_orthogroups <- cnat_orthogroups %>% separate_rows(cnat_ref_proteome_cdhit, sep = ", ")
cnat_orthogroups <- cnat_orthogroups[,c(2,1)]
colnames(cnat_orthogroups)[1] <- "Transcript"
cnat_orthogroups$Transcript <- gsub("\\..*","",cnat_orthogroups$Transcript)
write.csv(cnat_orthogroups,file="./cnat/salmon/cnat_orthogroup_tx2gene.csv",row.names = FALSE)

mcav_orthogroups <- Orthogroups[,c(1,3)]
mcav_orthogroups <- mcav_orthogroups %>% separate_rows(mcav_ref_proteome_cdhit, sep = ", ")
mcav_orthogroups <- mcav_orthogroups[,c(2,1)]
colnames(mcav_orthogroups)[1] <- "Transcript"
mcav_orthogroups$Transcript <- gsub("\\..*","",mcav_orthogroups$Transcript)
write.csv(mcav_orthogroups,file="./mcav/salmon/mcav_orthogroup_tx2gene.csv",row.names = FALSE)

oann_orthogroups <- Orthogroups[,c(1,4)]
oann_orthogroups <- oann_orthogroups %>% separate_rows(oann_ref_proteome_cdhit, sep = ", ")
oann_orthogroups <- oann_orthogroups[,c(2,1)]
colnames(oann_orthogroups)[1] <- "Transcript"
oann_orthogroups$Transcript <- gsub("\\..*","",oann_orthogroups$Transcript)
write.csv(oann_orthogroups,file="./oann/salmon/oann_orthogroup_tx2gene.csv",row.names = FALSE)

past_orthogroups <- Orthogroups[,c(1,5)]
past_orthogroups <- past_orthogroups %>% separate_rows(past_ref_proteome_cdhit, sep = ", ")
past_orthogroups <- past_orthogroups[,c(2,1)]
colnames(past_orthogroups)[1] <- "Transcript"
past_orthogroups$Transcript <- gsub("\\..*","",past_orthogroups$Transcript)
write.csv(past_orthogroups,file="./past/salmon/past_orthogroup_tx2gene.csv",row.names = FALSE)

pstr_orthogroups <- Orthogroups[,c(1,6)]
pstr_orthogroups <- pstr_orthogroups %>% separate_rows(pstr_ref_proteome_cdhit, sep = ", ")
pstr_orthogroups <- pstr_orthogroups[,c(2,1)]
colnames(pstr_orthogroups)[1] <- "Transcript"
pstr_orthogroups$Transcript <- gsub("\\..*","",pstr_orthogroups$Transcript)
write.csv(pstr_orthogroups,file="./pstr/salmon/pstr_orthogroup_tx2gene.csv",row.names = FALSE)
```

## Step 2: Count Matrix Generation

Read in the raw counts from Salmon for all coral samples and assign Orthogroups

### C. natans
```{r,results='hide',tidy=TRUE}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/cnat/salmon")
cnat_samples <- read.table("cnat_samples.csv", header = TRUE)
cnat_files <- file.path(cnat_samples$sample, "quant.sf")
names(cnat_files) <- paste0("sample", 1:10)
all(file.exists(cnat_files))
# Do not continue if output shows [FALSE]
cnat_tx2gene <- read.csv("cnat_orthogroup_tx2gene.csv")
cnat_txi <- tximport(cnat_files, type = 'salmon',tx2gene=cnat_tx2gene)
dim(cnat_txi$counts)
# 20,966 Orthogroups

cnat_counts <- cnat_txi$counts
cnat_counts <- as.data.frame(cnat_counts)
colnames(cnat_counts) <- c("cnat_c2","cnat_c3","cnat_c4","cnat_c6","cnat_c8","cnat_d2","cnat_d3","cnat_d4","cnat_d6","cnat_d8")
cnat_counts <- tibble::rownames_to_column(cnat_counts,"Orthogroup")
cnat_counts[2:11] <-  round(cnat_counts[2:11], digits=0)
cnat_counts <- as.data.frame(cnat_counts)
```

### M. cavernosa
```{r,results='hide',tidy=TRUE}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/mcav/salmon")
mcav_samples <- read.table("mcav_samples.csv", header = TRUE)
mcav_files <- file.path(mcav_samples$sample, "quant.sf")
names(mcav_files) <- paste0("sample", 1:13)
all(file.exists(mcav_files))
# Do not continue if output shows [FALSE]
mcav_tx2gene <- read.csv("mcav_orthogroup_tx2gene.csv")
mcav_txi <- tximport(mcav_files, type = 'salmon',tx2gene=mcav_tx2gene)
dim(mcav_txi$counts)
# 18,397 Orthogroups

mcav_counts <- mcav_txi$counts
mcav_counts <- as.data.frame(mcav_counts)
colnames(mcav_counts) <- c("mcav_c1","mcav_c3","mcav_c5","mcav_c6","mcav_c7","mcav_c8","mcav_d1","mcav_d2","mcav_d3","mcav_d4","mcav_d5","mcav_d6","mcav_d8")
mcav_counts <- tibble::rownames_to_column(mcav_counts,"Orthogroup")
mcav_counts[2:14] <-  round(mcav_counts[2:14], digits=0)
mcav_counts <- as.data.frame(mcav_counts)
```

### O. annularis
```{r,results='hide',tidy=TRUE}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/oann/salmon")
oann_samples <- read.table("oann_samples.csv", header = TRUE)
oann_files <- file.path(oann_samples$sample, "quant.sf")
names(oann_files) <- paste0("sample", 1:11)
all(file.exists(oann_files))
# Do not continue if output shows [FALSE]
oann_tx2gene <- read.csv("oann_orthogroup_tx2gene.csv")
oann_txi <- tximport(oann_files, type = 'salmon',tx2gene=oann_tx2gene)
dim(oann_txi$counts)
# 18,459 Orthogroups

oann_counts <- oann_txi$counts
oann_counts <- as.data.frame(oann_counts)
colnames(oann_counts) <- c("oann_c1","oann_c2","oann_c3","oann_c6","oann_c7","oann_c8","oann_d1","oann_d3","oann_d6","oann_d7","oann_d8")
oann_counts <- tibble::rownames_to_column(oann_counts,"Orthogroup")
oann_counts[2:12] <-  round(oann_counts[2:12], digits=0)
oann_counts <- as.data.frame(oann_counts)
```

### P. astreoides
```{r,results='hide',tidy=TRUE}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/past/salmon")
past_samples <- read.table("past_samples.csv", header = TRUE)
past_files <- file.path(past_samples$sample, "quant.sf")
names(past_files) <- paste0("sample", 1:11)
all(file.exists(past_files))
# Do not continue if output shows [FALSE]
past_tx2gene <- read.csv("past_orthogroup_tx2gene.csv")
past_txi <- tximport(past_files, type = 'salmon',tx2gene=past_tx2gene)
dim(past_txi$counts)
# 18,255 Orthogroups

past_counts <- past_txi$counts
past_counts <- as.data.frame(past_counts)
colnames(past_counts) <- c("past_c1","past_c5","past_c6","past_c7","past_d1","past_d2","past_d3","past_d4","past_d5","past_d6","past_d7")
past_counts <- tibble::rownames_to_column(past_counts,"Orthogroup")
past_counts[2:12] <-  round(past_counts[2:12], digits=0)
past_counts <- as.data.frame(past_counts)
```
### P. strigosa
```{r,results='hide',tidy=TRUE}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/pstr/salmon")
pstr_samples <- read.table("pstr_samples_outlier_removed.csv", header = TRUE)
pstr_files <- file.path(pstr_samples$sample, "quant.sf")
names(pstr_files) <- paste0("sample", 1:7)
all(file.exists(pstr_files))
# Do not continue if output shows [FALSE]
pstr_tx2gene <- read.csv("pstr_orthogroup_tx2gene.csv")
pstr_txi <- tximport(pstr_files, type = 'salmon',tx2gene=pstr_tx2gene)
dim(pstr_txi$counts)
# 14,621 Orthogroups

pstr_counts <- pstr_txi$counts
pstr_counts <- as.data.frame(pstr_counts)
colnames(pstr_counts) <- c("pstr_c3","pstr_c5","pstr_c7","pstr_c8","pstr_d3","pstr_d5","pstr_d7")
pstr_counts <- tibble::rownames_to_column(pstr_counts,"Orthogroup")
pstr_counts[2:8] <-  round(pstr_counts[2:8], digits=0)
pstr_counts <- as.data.frame(pstr_counts)
```

## Step 3: Analysis of Shared Orthogroups
### Merge the count data
```{r,results='hide',tidy=TRUE}
Cnat_Mcav <- merge(cnat_counts,mcav_counts, by ="Orthogroup")
Cnat_Mcav_Oann <- merge(Cnat_Mcav,oann_counts,by="Orthogroup")
Cnat_Mcav_Oann_Past <- merge(Cnat_Mcav_Oann,past_counts,by="Orthogroup")
all_species <- merge(Cnat_Mcav_Oann_Past,pstr_counts,by="Orthogroup")
# 6,242 Orthogroups expressed and shared between all 5 species
```

### DESeq2

Convert counts to integers.
Run DESeq2 on Orthogroups to get Rlog normalized expression.

```{r,results='hide',tidy=TRUE}
countData <- all_species
countData <- tibble::column_to_rownames(countData,"Orthogroup")
colData <- read.csv("./all/metaData.csv",row.names = "name")
# Remove pstr_d8
colData <- colData[-c(53),]

# Make counts in countData integers:
countData[1:52] <-  round(countData[1:52], digits=0)
countData <- as.data.frame(countData)

# Set the design
dds_treatment <- DESeqDataSetFromMatrix(countData = countData, 
                                        colData = colData, 
                                        design = ~species + treatment )

# Filter out columns with an average number of reads less than 10
dds_treatment <- dds_treatment[ rowMeans(counts(dds_treatment)) > 10, ] 

# Run DESeq2
test <- DESeq(dds_treatment)
rld <- rlog(dds_treatment, blind=FALSE) 
rrld <- assay(rld)
# 6170 Orthogroups after filtering out low counts
write.csv(rrld, file = "./all/DESeq2/treatment_rlog_orthogroups.csv")  
```

### Annotate the Orthogroups

To annotate the orthogroups, first we need to pull out the first sequence per Orthogroup from one species (M.cavernosa). Then we will assign a Unipot ID using the M.cavernosa transcriptome annotation. Finally we will merge Entry ID with Uniprot info to obtain full annotation for each orthogroup.

```{r}
# Use one representative seqeuence from Mcav to assign Entry ID to each orthogroup
mcav_orthogroups <- Orthogroups[,c(1,3)]
colnames(mcav_orthogroups)[2] <- "Transcript"
mcav_orthogroups$Transcript <- gsub("\\..*","",mcav_orthogroups$Transcript)
mcav_annot <- read.csv("./annotations/mcav_annotation.csv")
mcav_annot <- merge(mcav_orthogroups,mcav_annot,by="Transcript") #12,548
uniprot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/uniprot.csv")
mcav_annot <- left_join(mcav_annot,uniprot,by="Entry")
mcav_annot <- mcav_annot %>% filter(mcav_annot$Evalue < 0.000001)
# 12,546 Orthogroups with good annotation

# Merge with expression matrix
expr_mat <- as.data.frame(rrld)
expr_mat <- tibble::rownames_to_column(expr_mat,"Orthogroup")
expr_mat_annot <- merge(expr_mat,mcav_annot,by="Orthogroup")
expr_mat_annot <- expr_mat_annot %>% relocate(Transcript, .after = Orthogroup)
expr_mat_annot <- expr_mat_annot %>% relocate(Entry, .after = Transcript)
expr_mat_annot <- expr_mat_annot %>% relocate(Evalue, .after = Entry)
expr_mat_annot <- expr_mat_annot %>% relocate(Entry.name, .after = Evalue)
expr_mat_annot <- expr_mat_annot %>% relocate(Gene.names, .after = Entry.name)
expr_mat_annot <- expr_mat_annot %>% relocate(Protein.names, .after = Gene.names)
expr_mat_annot <- expr_mat_annot %>% relocate(Organism, .after = Protein.names)
expr_mat_annot <- expr_mat_annot %>% relocate(Gene.ontology..GO., .after = Organism)
expr_mat_annot <- expr_mat_annot[,-c(4,8)]
colnames(expr_mat_annot)[7] <- "GO"
```

### Pull out Orthogroups with immune function

```{r}
immunity <- expr_mat_annot[expr_mat_annot$GO %like% "immunity",]
dim(immunity) #2
immune <- expr_mat_annot[expr_mat_annot$GO %like% "immune",]
dim(immune) #132
all_immunity <- rbind(dplyr::filter(immunity, !Orthogroup %in% intersect(immunity$Orthogroup, immune$Orthogroup)), immune)
dim(all_immunity) #133 immune-related Orthogroups

virus <- expr_mat_annot[expr_mat_annot$GO %like% "virus",]
dim(virus) #57
viral <- expr_mat_annot[expr_mat_annot$GO %like% "viral",]
dim(viral) #54
all_viral <- rbind(dplyr::filter(virus, !Orthogroup %in% intersect(virus$Orthogroup, viral$Orthogroup)), viral)
dim(all_viral) #103

all <- rbind(dplyr::filter(all_immunity, !Orthogroup %in% intersect(all_immunity$Orthogroup, all_viral$Orthogroup)), all_viral)
dim(all) # 212

write.csv(all,file = "./all/immune_expression.csv",row.names = FALSE)
```

## Step 4: ANOVA on Immune Orthogroups

We run a one-way ANOVA on the Orthogroups with immune function to test for differences in expression between disease state (control, exposed, infected)
```{r}
t_all_immunity <- setNames(data.frame(t(all[,-1])), all[,1])
row.names(t_all_immunity)
t_all_immunity <- tibble::rownames_to_column(t_all_immunity, "name")
t_all_immunity <- t_all_immunity[-c(1:6),]

label <- read.csv("./all/metaData.csv")
colnames(label)
label <- label[-c(53),c(3,5)] # remove pstr_d8; keep name and status
master <- merge(label,t_all_immunity,by="name")

# Perform ANOVA on the 213 immune genes
names(master)
# ANOVA loop significant results
for(i in 3:ncol(master))
{
  column <- names(master[i])
  #tidy will summarize and return neat format
  avz <- broom::tidy(aov(master[,i] ~ status, data = master))
  
  # Add this condition if you only want aov with P < 0.05
  if(avz$p.value[1] < 0.05) {
    
    print(column)
  }
}
```

[1] "OG0002316"
```{r}
OG0002316 <- aov(master$OG0002316 ~ master$status, data = master)
summary(OG0002316) 
TukeyHSD(OG0002316) 
```
[1] "OG0002812"
```{r}
OG0002812 <- aov(master$OG0002812 ~ master$status, data = master)
summary(OG0002812) 
TukeyHSD(OG0002812)
```
[1] "OG0002951"
```{r}
OG0002951 <- aov(master$OG0002951 ~ master$status, data = master)
summary(OG0002951) 
TukeyHSD(OG0002951) 
```
[1] "OG0003092"
```{r}
OG0003092 <- aov(master$OG0003092 ~ master$status, data = master)
summary(OG0003092) 
TukeyHSD(OG0003092) 
```
[1] "OG0003270"
```{r}
OG0003270 <- aov(master$OG0003270 ~ master$status, data = master)
summary(OG0003270) 
TukeyHSD(OG0003270) 
```
[1] "OG0003771"
```{r}
OG0003771 <- aov(master$OG0003771 ~ master$status, data = master)
summary(OG0003771) 
TukeyHSD(OG0003771) 
```
[1] "OG0003940"
```{r}
OG0003940 <- aov(master$OG0003940 ~ master$status, data = master)
summary(OG0003940) 
TukeyHSD(OG0003940) 
```
[1] "OG0003997"
```{r}
OG0003997 <- aov(master$OG0003997 ~ master$status, data = master)
summary(OG0003997) 
TukeyHSD(OG0003997) 
```
[1] "OG0004300"
```{r}
OG0004300 <- aov(master$OG0004300 ~ master$status, data = master)
summary(OG0004300) 
TukeyHSD(OG0004300) 
```
[1] "OG0004350"
```{r}
OG0004350 <- aov(master$OG0004350 ~ master$status, data = master)
summary(OG0004350) 
TukeyHSD(OG0004350) 
```
[1] "OG0005460"
```{r}
OG0005460 <- aov(master$OG0005460 ~ master$status, data = master)
summary(OG0005460) 
TukeyHSD(OG0005460) 
```
[1] "OG0006200"
```{r}
OG0006200 <- aov(master$OG0006200 ~ master$status, data = master)
summary(OG0006200) 
TukeyHSD(OG0006200) 
```
[1] "OG0007817"
```{r}
OG0007817 <- aov(master$OG0007817 ~ master$status, data = master)
summary(OG0007817) 
TukeyHSD(OG0007817) 
```
[1] "OG0007911"
```{r}
OG0007911 <- aov(master$OG0007911 ~ master$status, data = master)
summary(OG0007911) 
TukeyHSD(OG0007911) 
```
[1] "OG0011317"
```{r}
OG0011317 <- aov(master$OG0011317 ~ master$status, data = master)
summary(OG0011317) 
TukeyHSD(OG0011317) 
```
[1] "OG0012254"
```{r}
OG0012254 <- aov(master$OG0012254 ~ master$status, data = master)
summary(OG0012254) 
TukeyHSD(OG0012254) 
```
[1] "OG0000274"
```{r}
OG0000274 <- aov(master$OG0000274 ~ master$status, data = master)
summary(OG0000274) 
TukeyHSD(OG0000274) 
```
[1] "OG0002491"
```{r}
OG0002491 <- aov(master$OG0002491 ~ master$status, data = master)
summary(OG0002491) 
TukeyHSD(OG0002491) 
```
[1] "OG0005125"
```{r}
OG0005125 <- aov(master$OG0005125 ~ master$status, data = master)
summary(OG0005125) 
TukeyHSD(OG0005125) 
```
[1] "OG0005462"
```{r}
OG0005462 <- aov(master$OG0005462 ~ master$status, data = master)
summary(OG0005462) 
TukeyHSD(OG0005462) 
```
[1] "OG0007162"
```{r}
OG0007162 <- aov(master$OG0007162 ~ master$status, data = master)
summary(OG0007162) 
TukeyHSD(OG0007162) 
```
[1] "OG0012356"
```{r}
OG0012356 <- aov(master$OG0012356 ~ master$status, data = master)
summary(OG0012356) 
TukeyHSD(OG0012356) 
```
[1] "OG0003182"
```{r}
OG0003182 <- aov(master$OG0003182 ~ master$status, data = master)
summary(OG0003182) 
TukeyHSD(OG0003182) 
```
[1] "OG0003597"
```{r}
OG0003597 <- aov(master$OG0003597 ~ master$status, data = master)
summary(OG0003597) 
TukeyHSD(OG0003597) 
```
[1] "OG0004181"
```{r}
OG0004181 <- aov(master$OG0004181 ~ master$status, data = master)
summary(OG0004181) 
TukeyHSD(OG0004181) 
```
[1] "OG0005274"
```{r}
OG0005274 <- aov(master$OG0005274 ~ master$status, data = master)
summary(OG0005274) 
TukeyHSD(OG0005274) 
```
[1] "OG0005285"
```{r}
OG0005285 <- aov(master$OG0005285 ~ master$status, data = master)
summary(OG0005285) 
TukeyHSD(OG0005285) 
```
[1] "OG0005977"
```{r}
OG0005977 <- aov(master$OG0005977 ~ master$status, data = master)
summary(OG0005977) 
TukeyHSD(OG0005977) 
```
[1] "OG0007458"
```{r}
OG0007458 <- aov(master$OG0007458 ~ master$status, data = master)
summary(OG0007458) 
TukeyHSD(OG0007458) 
```
[1] "OG0011228"
```{r}
OG0011228 <- aov(master$OG0011228 ~ master$status, data = master)
summary(OG0011228) 
TukeyHSD(OG0011228) 
```

### Significant 

Make a dataframe including only the 30 significant highly variable orthologs

```{r,results='hide',tidy=TRUE}
all
sig_orthogroups <- c("OG0002951","OG0003092","OG0003270","OG0003771","OG0003940","OG0003997","OG0004300","OG0004350","OG0005460","OG0006200","OG0007817","OG0011317","OG0000274","OG0002491","OG0005125","OG0005462","OG0007162","OG0012356","OG0003182","OG0003597","OG0004181","OG0005274","OG0005285","OG0005977","OG0007458","OG0011228","OG0002316","OG0002812","OG0007911","OG0012254")
sig_orthogroups <- all[all$Orthogroup %in% sig_orthogroups,]
write.csv(sig_orthogroups,file="./all/sig_immune_orthogroups.csv",row.names = FALSE)
```
Add the Tukey p value results to this file in 3 columns: EvI,CvI and CvE.
Rename this file "sig_immune_orthogroups_ANOVA.csv"

### Boxplots
```{r}
immune_expression <- read.csv("./all/sig_immune_orthogroups_ANOVA.csv")
names(immune_expression)
immune_expression <- immune_expression[,-c(2:10)]
t_immune_expression <- setNames(as.data.frame(t(immune_expression[,-1])), immune_expression[,1])
row.names(t_immune_expression)
view(t_immune_expression)
t_immune_expression <- tibble::rownames_to_column(t_immune_expression, "name")
view(t_immune_expression)

label <- read.csv("./all/metaData.csv")
colnames(label)
label <- label[-c(53),c(1,5)] # remove pstr_d8; keep species, status
t_immune_expression$species <- label$species
t_immune_expression <- t_immune_expression %>% relocate(species, .before = everything())
t_immune_expression$status <- label$status
t_immune_expression <- t_immune_expression %>% relocate(status, .after = name)


t_immune_expression$status <- gsub("healthy","control", t_immune_expression$status)
t_immune_expression$status <- gsub("disease","infected", t_immune_expression$status)
t_immune_expression$status <- factor(t_immune_expression$status, levels = c("control","exposed","infected"))
t_immune_expression$species <- factor(t_immune_expression$species, levels = c("mcav","past","pstr","oann","cnat"))
master <- t_immune_expression
```

#### Pancreatic secretory granule membrane major glycoprotein GP2
Functions as an intestinal M-cell transcytotic receptor specific of type-I-piliated bacteria that participates in the mucosal immune response toward these bacteria. At the apical membrane of M-cells it binds fimH, a protein of the bacteria type I pilus tip. Internalizes bound bacteria, like E.coli and S.typhimurium, from the lumen of the intestine and delivers them, through M-cells, to the underlying organized lymphoid follicles where they are captured by antigen-presenting dendritic cells to elicit a mucosal immune response.
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0002951 <- compare_means(OG0002951~status,data=master,method = "t.test")
OG0002951

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0002951_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0002951, fill = status)) +
  geom_point(aes(y = OG0002951, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Pancreatic secretory granule membrane major glycoprotein GP2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0002951, y.position = 11, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0002951_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0002951, fill = status)) +
  geom_point(aes(y = OG0002951, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Pancreatic secretory granule membrane major glycoprotein GP2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0002951, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0002951_species,OG0002951_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```
#### Polyglutamylase complex subunit TTLL1
Catalytic subunit of a polyglutamylase complex which modifies tubulin, generating side chains of glutamate on the gamma-carboxyl group of specific glutamate residues within the C-terminal tail of tubulin. Probably involved in the side-chain elongation step of the polyglutamylation reaction rather than the initiation step. Modifies both alpha- and beta-tubulins with a preference for the alpha-tail. Unlike most polyglutamylases of the tubulin--tyrosine ligase family, only displays a catalytic activity when in complex with other proteins as it is most likely lacking domains important for autonomous activity. Part of the neuronal tubulin polyglutamylase complex. Mediates cilia and flagella polyglutamylation which is essential for their biogenesis and motility. Involved in respiratory motile cilia function through the regulation of beating asymmetry. Essential for sperm flagella biogenesis, motility and male fertility. Involved in KLF4 glutamylation which impedes its ubiquitination, thereby leading to somatic cell reprogramming, pluripotency maintenance and embryogenesis.
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0003092 <- compare_means(OG0003092~status,data=master,method = "t.test")
OG0003092

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0003092_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0003092, fill = status)) +
  geom_point(aes(y = OG0003092, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Ubiquitin carboxyl-terminal hydrolase 14") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003092, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0003092_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0003092, fill = status)) +
  geom_point(aes(y = OG0003092, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Ubiquitin carboxyl-terminal hydrolase 14") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003092, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0003092_species,OG0003092_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```

#### Ubiquitin carboxyl-terminal hydrolase 14
Proteasome-associated deubiquitinase which releases ubiquitin from the proteasome targeted ubiquitinated proteins. Ensures the regeneration of ubiquitin at the proteasome. Is a reversibly associated subunit of the proteasome and a large fraction of proteasome-free protein exists within the cell. Required for the degradation of the chemokine receptor CXCR4 which is critical for CXCL12-induced cell chemotaxis. Serves also as a physiological inhibitor of endoplasmic reticulum-associated degradation (ERAD) under the non-stressed condition by inhibiting the degradation of unfolded endoplasmic reticulum proteins via interaction with ERN1 (By similarity).
Indispensable for synaptic development and function at neuromuscular junctions (NMJs) (By similarity).
Plays a role in the innate immune defense against viruses by stabilizing the viral DNA sensor CGAS and thus inhibiting its autophagic degradation (By similarity).
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0003270 <- compare_means(OG0003270~status,data=master,method = "t.test")
OG0003270

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0003270_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0003270, fill = status)) +
  geom_point(aes(y = OG0003270, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Ubiquitin carboxyl-terminal hydrolase 14") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0003270_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0003270, fill = status)) +
  geom_point(aes(y = OG0003270, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Ubiquitin carboxyl-terminal hydrolase 14") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0003270_species,OG0003270_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### RNA-binding protein 1/2/3
Removed - chloroplasic gene

#### Aryl hydrocarbon receptor
Ligand-activated transcription factor that enables cells to adapt to changing conditions by sensing compounds from the environment, diet, microbiome and cellular metabolism, and which plays important roles in development, immunity and cancer. Upon ligand binding, translocates into the nucleus, where it heterodimerizes with ARNT and induces transcription by binding to xenobiotic response elements (XRE). Regulates a variety of biological processes, including angiogenesis, hematopoiesis, drug and lipid metabolism, cell motility and immune modulation. Xenobiotics can act as ligands: upon xenobiotic-binding, activates the expression of multiple phase I and II xenobiotic chemical metabolizing enzyme genes (such as the CYP1A1 gene). Mediates biochemical and toxic effects of halogenated aromatic hydrocarbons. Next to xenobiotics, natural ligands derived from plants, microbiota, and endogenous metabolism are potent AHR agonists. Tryptophan (Trp) derivatives constitute an important class of endogenous AHR ligands. Acts as a negative regulator of anti-tumor immunity: indoles and kynurenic acid generated by Trp catabolism act as ligand and activate AHR, thereby promoting AHR-driven cancer cell motility and suppressing adaptive immunity. Regulates the circadian clock by inhibiting the basal and circadian expression of the core circadian component PER1. Inhibits PER1 by repressing the CLOCK-BMAL1 heterodimer mediated transcriptional activation of PER1. The heterodimer ARNT:AHR binds to core DNA sequence 5'-TGCGTG-3' within the dioxin response element (DRE) of target gene promoters and activates their transcription.
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0003771 <- compare_means(OG0003771~status,data=master,method = "t.test")
OG0003771

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0003771_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0003771, fill = status)) +
  geom_point(aes(y = OG0003771, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Aryl hydrocarbon receptor") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0003771_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0003771, fill = status)) +
  geom_point(aes(y = OG0003771, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Aryl hydrocarbon receptor") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0003771_species,OG0003771_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### ATPase WRNIP1
Functions as a modulator of initiation or reinitiation events during DNA polymerase delta-mediated DNA synthesis. In the presence of ATP, stimulation of DNA polymerase delta-mediated DNA synthesis is decreased. Also plays a role in the innate immune defense against viruses. Stabilizes the RIGI dsRNA interaction and promotes RIGI 'Lys-63'-linked polyubiquitination. In turn, RIGI transmits the signal through mitochondrial MAVS.
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0003997 <- compare_means(OG0003997~status,data=master,method = "t.test")
OG0003997

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0003997_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0003997, fill = status)) +
  geom_point(aes(y = OG0003997, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "ATPase WRNIP1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0003997_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0003997, fill = status)) +
  geom_point(aes(y = OG0003997, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "ATPase WRNIP1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0003997_species,OG0003997_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### Hamartin
In complex with TSC2, inhibits the nutrient-mediated or growth factor-stimulated phosphorylation of S6K1 and EIF4EBP1 by negatively regulating mTORC1 signaling (PubMed:12271141, PubMed:28215400).
Seems not to be required for TSC2 GAP activity towards RHEB (PubMed:15340059).
Implicated as a tumor suppressor. Involved in microtubule-mediated protein transport, but this seems to be due to unregulated mTOR signaling (By similarity).
Acts as a co-chaperone for HSP90AA1 facilitating HSP90AA1 chaperoning of protein clients such as kinases, TSC2 and glucocorticoid receptor NR3C1 (PubMed:29127155).
Increases ATP binding to HSP90AA1 and inhibits HSP90AA1 ATPase activity (PubMed:29127155).
Competes with the activating co-chaperone AHSA1 for binding to HSP90AA1, thereby providing a reciprocal regulatory mechanism for chaperoning of client proteins (PubMed:29127155).
Recruits TSC2 to HSP90AA1 and stabilizes TSC2 by preventing the interaction between TSC2 and ubiquitin ligase HERC1 (PubMed:16464865, PubMed:29127155).
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0004300 <- compare_means(OG0004300~status,data=master,method = "t.test")
OG0004300

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0004300_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0004300, fill = status)) +
  geom_point(aes(y = OG0004300, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Hamartin") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 11, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0004300_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0004300, fill = status)) +
  geom_point(aes(y = OG0004300, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Hamartin") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0004300_species,OG0004300_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### Phosphoinositide phospholipase C-gamma-2
The production of the second messenger molecules diacylglycerol (DAG) and inositol 1,4,5-trisphosphate (IP3) is mediated by activated phosphatidylinositol-specific phospholipase C enzymes. It is a crucial enzyme in transmembrane signaling.
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0004350 <- compare_means(OG0004350~status,data=master,method = "t.test")
OG0004350

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0004350_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0004350, fill = status)) +
  geom_point(aes(y = OG0004350, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Phosphoinositide phospholipase C-gamma-2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 11, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0004350_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0004350, fill = status)) +
  geom_point(aes(y = OG0004350, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Phosphoinositide phospholipase C-gamma-2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0004350_species,OG0004350_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### ** SMAD family member 6
Transforming growth factor-beta superfamily receptors signaling occurs through the Smad family of intracellular mediators. SMAD6 is an inhibitory Smad (i-Smad) that negatively regulates signaling downstream of type I transforming growth factor-beta (PubMed:9436979, PubMed:16951688, PubMed:22275001, PubMed:9759503, PubMed:10647776, PubMed:10708948, PubMed:10708949, PubMed:30848080).
Acts as a mediator of TGF-beta and BMP anti-inflammatory activities. Suppresses IL1R-TLR signaling through its direct interaction with PEL1, preventing NF-kappa-B activation, nuclear transport and NF-kappa-B-mediated expression of pro-inflammatory genes (PubMed:16951688).
Blocks the BMP-SMAD1 signaling pathway by competing with SMAD4 for receptor-activated SMAD1-binding (PubMed:9436979, PubMed:30848080).
Binds to regulatory elements in target promoter regions (PubMed:16491121).
```{r, fig.width=4,fig.height=2,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0005460 <- compare_means(OG0005460~status,data=master,method = "t.test")
OG0005460

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0005460_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0005460, fill = status)) +
  geom_point(aes(y = OG0005460, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "SMAD family member 6") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0005460, y.position = 11, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0005460_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0005460, fill = status)) +
  geom_point(aes(y = OG0005460, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "SMAD family member 6") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0005460, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0005460_species,OG0005460_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### Adhesion G protein-coupled receptor E1
Orphan receptor involved in cell adhesion and probably in cell-cell interactions specifically involving cells of the immune system. May play a role in regulatory T-cells (Treg) development.
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0006200 <- compare_means(OG0006200~status,data=master,method = "t.test")
OG0006200

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0006200_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0006200, fill = status)) +
  geom_point(aes(y = OG0006200, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Adhesion G protein-coupled receptor E1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 12, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0006200_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0006200, fill = status)) +
  geom_point(aes(y = OG0006200, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Adhesion G protein-coupled receptor E1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0006200_species,OG0006200_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### E3 ubiquitin-protein ligase MARCHF8
E3 ubiquitin-protein ligase that mediates ubiquitination of CD86 and MHC class II proteins, such as HLA-DR alpha and beta, and promotes their subsequent endocytosis and sorting to lysosomes via multivesicular bodies. May also promote ubiquitination and endocytosis of TFRC and FAS (By similarity).
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0007817 <- compare_means(OG0007817~status,data=master,method = "t.test")
OG0007817

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0007817_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0007817, fill = status)) +
  geom_point(aes(y = OG0007817, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "E3 ubiquitin-protein ligase MARCHF8") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 9.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0007817_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0007817, fill = status)) +
  geom_point(aes(y = OG0007817, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "E3 ubiquitin-protein ligase MARCHF8") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0007817_species,OG0007817_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### Deformed epidermal autoregulatory factor 1
Transcription factor that binds the homeotic Deformed (Dfd) response element. High affinity binding sites contain at least 1 TTCG motif surrounded by additional TCG sequences. May be involved in the selective action of Dfd on these sites without binding directly to the Dfd protein. Requirement of DEAF1 activity may be a common feature of enhancers targeted by Dfd.
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0011317 <- compare_means(OG0011317~status,data=master,method = "t.test")
OG0011317

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0011317_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0011317, fill = status)) +
  geom_point(aes(y = OG0011317, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Deformed epidermal autoregulatory factor 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 10, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0011317_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0011317, fill = status)) +
  geom_point(aes(y = OG0011317, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Deformed epidermal autoregulatory factor 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0011317_species,OG0011317_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### ** TNF receptor-associated factor 3
Cytoplasmic E3 ubiquitin ligase that regulates various signaling pathways, such as the NF-kappa-B, mitogen-activated protein kinase (MAPK) and interferon regulatory factor (IRF) pathways, and thus controls a lot of biological processes in both immune and non-immune cell types (PubMed:17015635).
In TLR and RLR signaling pathways, acts as an E3 ubiquitin ligase promoting the synthesis of 'Lys-63'-linked polyubiquitin chains on several substrates such as ASC that lead to the activation of the type I interferon response or the inflammasome (PubMed:19898473, PubMed:23871208, PubMed:26305951, PubMed:23150880).
Following the activation of certain TLRs such as TLR4, acts as a negative NF-kappa-B regulator, possibly to avoid unregulated inflammatory response, and its degradation via 'Lys-48'-linked polyubiquitination is required for MAPK activation and production of inflammatory cytokines (PubMed:16306937).
Alternatively, when TLR4 orchestrates bacterial expulsion, TRAF3 undergoes 'Lys-33'-linked polyubiquitination and subsequently binds to RALGDS, mobilizing the exocyst complex to rapidly expel intracellular bacteria back for clearance. Acts also as a constitutive negative regulator of the alternative NF-kappa-B pathway, which controls B-cell survival and lymphoid organ development (PubMed:17723217).
Required for normal antibody isotype switching from IgM to IgG (PubMed:19228877).
Plays a role T-cell dependent immune responses (PubMed:8934568).
Down-regulates proteolytic processing of NFKB2, and thereby inhibits non-canonical activation of NF-kappa-B. Promotes ubiquitination and proteasomal degradation of MAP3K14
```{r, fig.width=4,fig.height=2,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0000274 <- compare_means(OG0000274~status,data=master,method = "t.test")
OG0000274

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0000274_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0000274, fill = status)) +
  geom_point(aes(y = OG0000274, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "TNF receptor-associated factor 3") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0000274, y.position = 12, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0000274_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0000274, fill = status)) +
  geom_point(aes(y = OG0000274, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "TNF receptor-associated factor 3") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0000274_species,OG0000274_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### E3 ubiquitin-protein ligase Itchy 
Acts as an E3 ubiquitin-protein ligase which accepts ubiquitin from an E2 ubiquitin-conjugating enzyme in the form of a thioester and then directly transfers the ubiquitin to targeted substrates (PubMed:15358865, PubMed:16446428, PubMed:17592138, PubMed:18628966, PubMed:20392206, PubMed:25632008).
It catalyzes 'Lys-29'-, 'Lys-48'- and 'Lys-63'-linked ubiquitin conjugation (By similarity).
Involved in the control of inflammatory signaling pathways (By similarity).
Is an essential component of a ubiquitin-editing protein complex, comprising also TNFAIP3, TAX1BP1 and RNF11, that ensures the transient nature of inflammatory signaling pathways (By similarity).
Promotes the association of the complex after TNF stimulation (By similarity).
Once the complex is formed, TNFAIP3 deubiquitinates 'Lys-63' polyubiquitin chains on RIPK1 and catalyzes the formation of 'Lys-48'-polyubiquitin chains (By similarity).
This leads to RIPK1 proteasomal degradation and consequently termination of the TNF- or LPS-mediated activation of NFKB1 (By similarity).
Ubiquitinates RIPK2 by 'Lys-63'-linked conjugation and influences NOD2-dependent signal transduction pathways (By similarity).
Regulates the transcriptional activity of several transcription factors involved in immune response (PubMed:15358865, PubMed:11828324).
Ubiquitinates NFE2 by 'Lys-63' linkages and is implicated in the control of the development of hematopoietic lineages (By similarity).
Mediates JUN ubiquitination and degradation (PubMed:15358865).
Mediates JUNB ubiquitination and degradation (PubMed:11828324, PubMed:15358865).
Critical regulator of type 2 helper T (Th2) cell cytokine production by inducing JUNB ubiquitination and degradation (PubMed:11828324).
Involved in the negative regulation of MAVS-dependent cellular antiviral responses (By similarity).
Ubiquitinates MAVS through 'Lys-48'-linked conjugation resulting in MAVS proteasomal degradation (By similarity).
Following ligand stimulation, regulates sorting of Wnt receptor FZD4 to the degradative endocytic pathway probably by modulating PI42KA activity (By similarity).
Ubiquitinates PI4K2A and negatively regulates its catalytic activity (By similarity).
Ubiquitinates chemokine receptor CXCR4 and regulates sorting of CXCR4 to the degradative endocytic pathway following ligand stimulation by ubiquitinating endosomal sorting complex required for transport ESCRT-0 components HGS and STAM (By similarity).
Targets DTX1 for lysosomal degradation and controls NOTCH1 degradation, in the absence of ligand, through 'Lys-29'-linked polyubiquitination (PubMed:18628966).
Ubiquitinates SNX9 (By similarity).
Ubiquitinates MAP3K7 through 'Lys-48'-linked conjugation (PubMed:25632008).
Involved in the regulation of apoptosis and reactive oxygen species levels through the ubiquitination and proteasomal degradation of TXNIP (By similarity).
Mediates the antiapoptotic activity of epidermal growth factor through the ubiquitination and proteasomal degradation of p15 BID (PubMed:20392206).
Ubiquitinates BRAT1 and this ubiquitination is enhanced in the presence of NDFIP1 (By similarity).
Ubiquitinates NEDD9/HEF1, resulting in proteosomal degradation of NEDD9/HEF1 (By similarity).
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0002491 <- compare_means(OG0002491~status,data=master,method = "t.test")
OG0002491

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0002491_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0002491, fill = status)) +
  geom_point(aes(y = OG0002491, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "E3 ubiquitin-protein ligase Itchy") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 12, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0002491_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0002491, fill = status)) +
  geom_point(aes(y = OG0002491, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "E3 ubiquitin-protein ligase Itchy") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0002491_species,OG0002491_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### Paired amphipathic helix protein Sin3a
Acts as a transcriptional repressor. Corepressor for REST. Interacts with MXI1 to repress MYC responsive genes and antagonize MYC oncogenic activities. Also interacts with MXD1-MAX heterodimers to repress transcription by tethering SIN3A to DNA. Acts cooperatively with OGT to repress transcription in parallel with histone deacetylation. Involved in the control of the circadian rhythms. Required for the transcriptional repression of circadian target genes, such as PER1, mediated by the large PER complex through histone deacetylation. Cooperates with FOXK1 to regulate cell cycle progression probably by repressing cell cycle inhibitor genes expression (By similarity).
Required for cortical neuron differentiation and callosal axon elongation (By similarity).
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0005125 <- compare_means(OG0005125~status,data=master,method = "t.test")
OG0005125

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0005125_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0005125, fill = status)) +
  geom_point(aes(y = OG0005125, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Paired amphipathic helix protein Sin3a") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 12, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0005125_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0005125, fill = status)) +
  geom_point(aes(y = OG0005125, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Paired amphipathic helix protein Sin3a") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0005125_species,OG0005125_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### Mitochondrial import receptor subunit TOM70
Acts as receptor of the preprotein translocase complex of the outer mitochondrial membrane (TOM complex) (PubMed:12526792).
Recognizes and mediates the translocation of mitochondrial preproteins from the cytosol into the mitochondria in a chaperone dependent manner (PubMed:12526792, PubMed:35025629).
Mediates TBK1 and IRF3 activation induced by MAVS in response to Sendai virus infection and promotes host antiviral responses during virus infection (PubMed:20628368, PubMed:25609812, PubMed:32728199).
Upon Sendai virus infection, recruits HSP90AA1:IRF3:BAX in mitochondrion and the complex induces apoptosis (PubMed:25609812).
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0005462 <- compare_means(OG0005462~status,data=master,method = "t.test")
OG0005462

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0005462_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0005462, fill = status)) +
  geom_point(aes(y = OG0005462, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Mitochondrial import receptor subunit TOM70") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 12, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0005462_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0005462, fill = status)) +
  geom_point(aes(y = OG0005462, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Mitochondrial import receptor subunit TOM70") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0005462_species,OG0005462_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### (maybe) E3 ubiquitin-protein ligase RNF26
E3 ubiquitin-protein ligase that plays a key role in endosome organization by retaining vesicles in the perinuclear cloud. Acts as a platform for perinuclear positioning of the endosomal system by mediating ubiquitination of SQSTM1 through interaction with the ubiquitin conjugating enzyme UBE2J1. Ubiquitinated SQSTM1 attracts specific vesicle-associated adapters, forming a molecular bridge that restrains cognate vesicles in the perinuclear region and organizes the endosomal pathway for efficient cargo transport. Also acts as a regulator of type I interferon production in response to viral infection by mediating the formation of 'Lys-11'-linked polyubiquitin chains on TMEM173/STING, leading to stabilize TMEM173/STING. Also required to limit type I interferon response by promoting autophagic degradation of IRF3.
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0007162 <- compare_means(OG0007162~status,data=master,method = "t.test")
OG0007162

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0007162_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0007162, fill = status)) +
  geom_point(aes(y = OG0007162, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "E3 ubiquitin-protein ligase RNF26") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 8.2, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0007162_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0007162, fill = status)) +
  geom_point(aes(y = OG0007162, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "E3 ubiquitin-protein ligase RNF26") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0007162_species,OG0007162_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### Macrophage mannose receptor 1
Mediates the endocytosis of glycoproteins by macrophages. Binds both sulfated and non-sulfated polysaccharide chains.
(Microbial infection) Acts as phagocytic receptor for bacteria, fungi and other pathogens.
(Microbial infection) Acts as a receptor for Dengue virus envelope protein E.
(Microbial infection) Interacts with Hepatitis B virus envelope protein.
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0012356 <- compare_means(OG0012356~status,data=master,method = "t.test")
OG0012356

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0012356_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0012356, fill = status)) +
  geom_point(aes(y = OG0012356, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Macrophage mannose receptor 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 12, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0012356_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0012356, fill = status)) +
  geom_point(aes(y = OG0012356, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Macrophage mannose receptor 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0012356_species,OG0012356_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### UV radiation resistance-associated gene protein (p63)
Versatile protein that is involved in regulation of different cellular pathways implicated in membrane trafficking. Involved in regulation of the COPI-dependent retrograde transport from Golgi and the endoplasmic reticulum by associating with the NRZ complex; the function is dependent on its binding to phosphatidylinositol 3-phosphate (PtdIns3P) (PubMed:16799551, PubMed:18552835, PubMed:20643123, PubMed:24056303, PubMed:28306502).
During autophagy acts as regulatory subunit of the alternative PI3K complex II (PI3KC3-C2) that mediates formation of phosphatidylinositol 3-phosphate and is believed to be involved in maturation of autophagosomes and endocytosis. Activates lipid kinase activity of PIK3C3 (PubMed:16799551, PubMed:20643123, PubMed:24056303, PubMed:28306502).
Involved in the regulation of degradative endocytic trafficking and cytokinesis, and in regulation of ATG9A transport from the Golgi to the autophagosome; the functions seems to implicate its association with PI3KC3-C2 (PubMed:16799551, PubMed:20643123, PubMed:24056303).
*Involved in maturation of autophagosomes and degradative endocytic trafficking independently of BECN1 but depending on its association with a class C Vps complex (possibly the HOPS complex); the association is also proposed to promote autophagosome recruitment and activation of Rab7 and endosome-endosome fusion events (PubMed:18552835, PubMed:28306502).*
Enhances class C Vps complex (possibly HOPS complex) association with a SNARE complex and promotes fusogenic SNARE complex formation during late endocytic membrane fusion (PubMed:24550300).
In case of negative-strand RNA virus infection is required for efficient virus entry, promotes endocytic transport of virions and is implicated in a VAMP8-specific fusogenic SNARE complex assembly (PubMed:24550300).
Involved in maintaining chromosomal stability. Promotes DNA double-strand break (DSB) repair by association with DNA-dependent protein kinase complex DNA-PK and activating it in non-homologous end joining (NHEJ) (PubMed:22542840).
Required for centrosome stability and proper chromosome segregation (PubMed:22542840).
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0003182 <- compare_means(OG0003182~status,data=master,method = "t.test")
OG0003182

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0003182_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0003182, fill = status)) +
  geom_point(aes(y = OG0003182, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "UV radiation resistance-associated gene protein (p63)") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0003182_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0003182, fill = status)) +
  geom_point(aes(y = OG0003182, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "UV radiation resistance-associated gene protein (p63)") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0003182_species,OG0003182_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### Integrin alpha-V
The alpha-V (ITGAV) integrins are receptors for vitronectin, cytotactin, fibronectin, fibrinogen, laminin, matrix metalloproteinase-2, osteopontin, osteomodulin, prothrombin, thrombospondin and vWF. They recognize the sequence R-G-D in a wide array of ligands. ITGAV:ITGB3 binds to fractalkine (CX3CL1) and may act as its coreceptor in CX3CR1-dependent fractalkine signaling (PubMed:23125415).
ITGAV:ITGB3 binds to NRG1 (via EGF domain) and this binding is essential for NRG1-ERBB signaling (PubMed:20682778).
ITGAV:ITGB3 binds to FGF1 and this binding is essential for FGF1 signaling (PubMed:18441324).
ITGAV:ITGB3 binds to FGF2 and this binding is essential for FGF2 signaling (PubMed:28302677).
ITGAV:ITGB3 binds to IGF1 and this binding is essential for IGF1 signaling (PubMed:19578119).
ITGAV:ITGB3 binds to IGF2 and this binding is essential for IGF2 signaling (PubMed:28873464).
ITGAV:ITGB3 binds to IL1B and this binding is essential for IL1B signaling (PubMed:29030430).
ITGAV:ITGB3 binds to PLA2G2A via a site (site 2) which is distinct from the classical ligand-binding site (site 1) and this induces integrin conformational changes and enhanced ligand binding to site 1 (PubMed:18635536, PubMed:25398877).
ITGAV:ITGB3 and ITGAV:ITGB6 act as receptors for fibrillin-1 (FBN1) and mediate R-G-D-dependent cell adhesion to FBN1 (PubMed:12807887, PubMed:17158881).
Integrin alpha-V/beta-6 or alpha-V/beta-8 (ITGAV:ITGB6 or ITGAV:ITGB8) mediates R-G-D-dependent release of transforming growth factor beta-1 (TGF-beta-1) from regulatory Latency-associated peptide (LAP), thereby playing a key role in TGF-beta-1 activation (PubMed:15184403, PubMed:22278742, PubMed:28117447).
ITGAV:ITGB3 acts as a receptor for CD40LG (PubMed:31331973).
(Microbial infection) Integrin ITGAV:ITGB5 acts as a receptor for Adenovirus type C.
(Microbial infection) Integrin ITGAV:ITGB5 and ITGAV:ITGB3 act as receptors for Coxsackievirus A9 and B1.
(Microbial infection) Integrin ITGAV:ITGB3 acts as a receptor for Herpes virus 8/HHV-8.
(Microbial infection) Integrin ITGAV:ITGB6 acts as a receptor for herpes simplex 1/HHV-1.
(Microbial infection) Integrin ITGAV:ITGB3 acts as a receptor for Human parechovirus 1.
(Microbial infection) Integrin ITGAV:ITGB3 acts as a receptor for West nile virus.
(Microbial infection) In case of HIV-1 infection, the interaction with extracellular viral Tat protein seems to enhance angiogenesis in Kaposi's sarcoma lesions.
```{r, fig.width=4,fig.height=2,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0003597 <- compare_means(OG0003597~status,data=master,method = "t.test")
OG0003597

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0003597_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0003597, fill = status)) +
  geom_point(aes(y = OG0003597, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Integrin alpha-V") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003597, y.position = 10, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0003597_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0003597, fill = status)) +
  geom_point(aes(y = OG0003597, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Integrin alpha-V") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003597, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0003597_species,OG0003597_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### ** Charged multivesicular body protein 4b
Probable core component of the endosomal sorting required for transport complex III (ESCRT-III) which is involved in multivesicular bodies (MVBs) formation and sorting of endosomal cargo proteins into MVBs. MVBs contain intraluminal vesicles (ILVs) that are generated by invagination and scission from the limiting membrane of the endosome and mostly are delivered to lysosomes enabling degradation of membrane proteins, such as stimulated growth factor receptors, lysosomal enzymes and lipids (By similarity).
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0004181 <- compare_means(OG0004181~status,data=master,method = "t.test")
OG0004181

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0004181_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0004181, fill = status)) +
  geom_point(aes(y = OG0004181, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Charged multivesicular body protein 4b") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0004181, y.position = 10.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0004181_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0004181, fill = status)) +
  geom_point(aes(y = OG0004181, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Charged multivesicular body protein 4b") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0004181, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0004181_species,OG0004181_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### IST1 homolog
ESCRT-III-like protein involved in cytokinesis, nuclear envelope reassembly and endosomal tubulation (By similarity).
Is required for efficient abscission during cytokinesis (By similarity).
Involved in recruiting VPS4A and/or VPS4B to the midbody of dividing cells (By similarity).
During late anaphase, involved in nuclear envelope reassembly and mitotic spindle disassembly together with the ESCRT-III complex: IST1 acts by mediating the recruitment of SPAST to the nuclear membrane, leading to microtubule severing (By similarity).
Recruited to the reforming nuclear envelope (NE) during anaphase by LEMD2 (By similarity).
Regulates early endosomal tubulation together with the ESCRT-III complex by mediating the recruitment of SPAST (By similarity).
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0005274 <- compare_means(OG0005274~status,data=master,method = "t.test")
OG0005274

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0005274_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0005274, fill = status)) +
  geom_point(aes(y = OG0005274, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "IST1 homolog") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 11.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0005274_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0005274, fill = status)) +
  geom_point(aes(y = OG0005274, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "IST1 homolog") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0005274_species,OG0005274_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### Proprotein convertase subtilisin/kexin type 5
Serine endoprotease that processes various proproteins by cleavage at paired basic amino acids, recognizing the RXXX[KR]R consensus motif. Likely functions in the constitutive and regulated secretory pathways. Plays an essential role in pregnancy establishment by proteolytic activation of a number of important factors such as BMP2, CALD1 and alpha-integrins. May be responsible for the maturation of gastrointestinal peptides. May be involved in the cellular proliferation of adrenal cortex via the activation of growth factors.
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0005285 <- compare_means(OG0005285~status,data=master,method = "t.test")
OG0005285

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0005285_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0005285, fill = status)) +
  geom_point(aes(y = OG0005285, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Proprotein convertase subtilisin/kexin type 5") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 11.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0005285_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0005285, fill = status)) +
  geom_point(aes(y = OG0005285, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Proprotein convertase subtilisin/kexin type 5") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0005285_species,OG0005285_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### Protein TASOR
Component of the HUSH complex, a multiprotein complex that mediates epigenetic repression. The HUSH complex is recruited to genomic loci rich in H3K9me3 and is required to maintain transcriptional silencing by promoting recruitment of SETDB1, a histone methyltransferase that mediates further deposition of H3K9me3, as well as MORC2. Also represses L1 retrotransposons in collaboration with MORC2 and, probably, SETDB1, the silencing is dependent of repressive epigenetic modifications, such as H3K9me3 mark. Silencing events often occur within introns of transcriptionally active genes, and lead to the down-regulation of host gene expression. The HUSH complex is also involved in the silencing of unintegrated retroviral DNA by being recruited by ZNF638: some part of the retroviral DNA formed immediately after infection remains unintegrated in the host genome and is transcriptionally repressed (By similarity).
Plays a crucial role in early embryonic development (PubMed:31112734, PubMed:24781204, PubMed:28839193).
Involved in the organization of spindle poles and spindle apparatus assembly during zygotic division (PubMed:31112734).
Plays an important role in maintaining epiblast fitness or potency (PubMed:28839193).
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0005977 <- compare_means(OG0005977~status,data=master,method = "t.test")
OG0005977

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0005977_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0005977, fill = status)) +
  geom_point(aes(y = OG0005977, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Protein TASOR") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0005977_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0005977, fill = status)) +
  geom_point(aes(y = OG0005977, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Protein TASOR") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0005977_species,OG0005977_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### ** Ras-related protein rab7
Key regulator in endo-lysosomal trafficking. Governs early-to-late endosomal maturation, microtubule minus-end as well as plus-end directed endosomal migration and positioning, and endosome-lysosome transport through different protein-protein interaction cascades. Plays important roles in microbial pathogen infection and survival, as well as in participating in the life cycle of viruses (By similarity).
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0007458 <- compare_means(OG0007458~status,data=master,method = "t.test")
OG0007458

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0007458_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0007458, fill = status)) +
  geom_point(aes(y = OG0007458, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Ras-related protein rab7") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0007458, y.position = 11.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0007458_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0007458, fill = status)) +
  geom_point(aes(y = OG0007458, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Ras-related protein rab7") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0007458_species,OG0007458_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### F-box/LRR-repeat protein 2
Calcium-activated substrate recognition component of the SCF (SKP1-cullin-F-box protein) E3 ubiquitin-protein ligase complex, SCF(FBXL2), which mediates the ubiquitination and subsequent proteasomal degradation of target proteins. Unlike many F-box proteins, FBXL2 does not seem to target phosphodegron within its substrates but rather calmodulin-binding motifs and is thereby antagonized by calmodulin. This is the case for the cyclins CCND2 and CCND3 which polyubiquitination and subsequent degradation are inhibited by calmodulin. Through CCND2 and CCND3 degradation induces cell-cycle arrest in G0. SCF(FBXL2) also mediates PIK3R2 ubiquitination and proteasomal degradation thereby regulating phosphatidylinositol 3-kinase signaling and autophagy (By similarity).
PCYT1A monoubiquitination by SCF(FBXL2) and subsequent degradation regulates synthesis of phosphatidylcholine, which is utilized for formation of membranes and of pulmonary surfactant (PubMed:21343341).
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0011228 <- compare_means(OG0011228~status,data=master,method = "t.test")
OG0011228

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0011228_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0011228, fill = status)) +
  geom_point(aes(y = OG0011228, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "F-box/LRR-repeat protein 2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 8, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0011228_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0011228, fill = status)) +
  geom_point(aes(y = OG0011228, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "F-box/LRR-repeat protein 2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0011228_species,OG0011228_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### ** Toll-like receptor 6 
Participates in the innate immune response to Gram-positive bacteria and fungi. Specifically recognizes diacylated and, to a lesser extent, triacylated lipopeptides (PubMed:20037584).
In response to diacylated lipopeptides, forms the activation cluster TLR2:TLR6:CD14:CD36, this cluster triggers signaling from the cell surface and subsequently is targeted to the Golgi in a lipid-raft dependent pathway (PubMed:16880211).
Acts via MYD88 and TRAF6, leading to NF-kappa-B activation, cytokine secretion and the inflammatory response. Recognizes mycoplasmal macrophage-activating lipopeptide-2kD (MALP-2), soluble tuberculosis factor (STF), phenol-soluble modulin (PSM) and B.burgdorferi outer surface protein A lipoprotein (OspA-L) cooperatively with TLR2 (PubMed:11441107).
In complex with TLR4, promotes sterile inflammation in monocytes/macrophages in response to oxidized low-density lipoprotein (oxLDL) or amyloid-beta 42. In this context, the initial signal is provided by oxLDL- or amyloid-beta 42-binding to CD36. This event induces the formation of a heterodimer of TLR4 and TLR6, which is rapidly internalized and triggers inflammatory response, leading to the NF-kappa-B-dependent production of CXCL1, CXCL2 and CCL9 cytokines, via MYD88 signaling pathway, and CCL5 cytokine, via TICAM1 signaling pathway, as well as IL1B secretion (PubMed:11441107, PubMed:20037584).
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0002316 <- compare_means(OG0002316~status,data=master,method = "t.test")
OG0002316

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0002316_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0002316, fill = status)) +
  geom_point(aes(y = OG0002316, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Toll-like receptor 6 ") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0002316, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0002316_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0002316, fill = status)) +
  geom_point(aes(y = OG0002316, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Toll-like receptor 6 ") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0002316_species,OG0002316_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### ** Deleted in malignant brain tumors 1 protein
May play roles in mucosal defense system and cellular immune defense. May play a role in liver regeneration. May be an important factor in fate decision and differentiation of transit-amplifying ductular (oval) cells within the hepatic lineage. May function as a binding protein in saliva for the regulation of taste sensation. May play a role as an opsonin receptor for SFTPD and SPAR in macrophage tissues throughout the body, including epithelial cells lining the gastrointestinal tract (By similarity).
Required for terminal differentiation of columnar epithelial cells during early embryogenesis. Displays a broad calcium-dependent binding spectrum against both Gram-positive and Gram-negative bacteria, suggesting a role in defense against bacterial pathogens. Binds to a range of poly-sulfated and poly-phosphorylated ligands which may explain its broad bacterial-binding specificity. Inhibits cytoinvasion of S.enterica. Associates with the actin cytoskeleton and is involved in its remodeling during regulated exocytosis. Interacts with pancreatic zymogens in a pH-dependent manner and may act as a Golgi cargo receptor in the regulated secretory pathway of the pancreatic acinar cell
```{r, fig.width=4,fig.height=2,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0002812 <- compare_means(OG0002812~status,data=master,method = "t.test")
OG0002812

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0002812_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0002812, fill = status)) +
  geom_point(aes(y = OG0002812, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Deleted in malignant brain tumors 1 protein") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0002812, y.position = 10.2, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0002812_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0002812, fill = status)) +
  geom_point(aes(y = OG0002812, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Deleted in malignant brain tumors 1 protein") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0002812_species,OG0002812_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### Protein Wnt-5a
Ligand for members of the frizzled family of seven transmembrane receptors (PubMed:17117926).
Can activate or inhibit canonical Wnt signaling, depending on receptor context (PubMed:16602827).
In the presence of FZD4, activates beta-catenin signaling. In the presence of ROR2, inhibits the canonical Wnt pathway by promoting beta-catenin degradation through a GSK3-independent pathway which involves down-regulation of beta-catenin-induced reporter gene expression (PubMed:16602827).
Suppression of the canonical pathway allows chondrogenesis to occur and inhibits tumor formation. Stimulates cell migration (PubMed:17117926).
Decreases proliferation, migration, invasiveness and clonogenicity of carcinoma cells and may act as a tumor suppressor. Mediates motility of melanoma cells (By similarity).
Required during embryogenesis for extension of the primary anterior-posterior axis and for outgrowth of limbs and the genital tubercle (PubMed:10021340).
Inhibits type II collagen expression in chondrocytes (By similarity).
```{r, fig.width=4,fig.height=2,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0007911 <- compare_means(OG0007911~status,data=master,method = "t.test")
OG0007911

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0007911_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0007911, fill = status)) +
  geom_point(aes(y = OG0007911, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Protein Wnt-5a") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0007911, y.position = 11.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0007911_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0007911, fill = status)) +
  geom_point(aes(y = OG0007911, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Protein Wnt-5a") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0007911_species,OG0007911_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### Platelet glycoprotein 4
Multifunctional glycoprotein that acts as receptor for a broad range of ligands. Ligands can be of proteinaceous nature like thrombospondin, fibronectin, collagen or amyloid-beta as well as of lipidic nature such as oxidized low-density lipoprotein (oxLDL), anionic phospholipids, long-chain fatty acids and bacterial diacylated lipopeptides. They are generally multivalent and can therefore engage multiple receptors simultaneously, the resulting formation of CD36 clusters initiates signal transduction and internalization of receptor-ligand complexes. The dependency on coreceptor signaling is strongly ligand specific. Cellular responses to these ligands are involved in angiogenesis, inflammatory response, fatty acid metabolism, taste and dietary fat processing in the intestine (Probable). Binds long-chain fatty acids and facilitates their transport into cells, thus participating in muscle lipid utilization, adipose energy storage, and gut fat absorption (By similarity) (PubMed:18353783, PubMed:21610069).
Mechanistically, binding of fatty acids activates downstream kinase LYN, which phosphorylates the palmitoyltransferase ZDHHC5 and inactivates it, resulting in the subsequent depalmitoylation of CD36 and caveolar endocytosis (PubMed:32958780).
In the small intestine, plays a role in proximal absorption of dietary fatty acid and cholesterol for optimal chylomicron formation, possibly through the activation of MAPK1/3 (ERK1/2) signaling pathway (By similarity) (PubMed:18753675).
Involved in oral fat perception and preferences (PubMed:22240721, PubMed:25822988).
Detection into the tongue of long-chain fatty acids leads to a rapid and sustained rise in flux and protein content of pancreatobiliary secretions (By similarity).
In taste receptor cells, mediates the induction of an increase in intracellular calcium levels by long-chain fatty acids, leading to the activation of the gustatory neurons in the nucleus of the solitary tract (By similarity).
Important factor in both ventromedial hypothalamus neuronal sensing of long-chain fatty acid and the regulation of energy and glucose homeostasis (By similarity).
Receptor for thrombospondins, THBS1 and THBS2, mediating their antiangiogenic effects (By similarity).
As a coreceptor for TLR4:TLR6 heterodimer, promotes inflammation in monocytes/macrophages. Upon ligand binding, such as oxLDL or amyloid-beta 42, interacts with the heterodimer TLR4:TLR6, the complex is internalized and triggers inflammatory response, leading to NF-kappa-B-dependent production of CXCL1, CXCL2 and CCL9 cytokines, via MYD88 signaling pathway, and CCL5 cytokine, via TICAM1 signaling pathway, as well as IL1B secretion, through the priming and activation of the NLRP3 inflammasome (By similarity) (PubMed:20037584).
Selective and nonredundant sensor of microbial diacylated lipopeptide that signal via TLR2:TLR6 heterodimer, this cluster triggers signaling from the cell surface, leading to the NF-kappa-B-dependent production of TNF, via MYD88 signaling pathway and subsequently is targeted to the Golgi in a lipid-raft dependent pathway (By similarity) (PubMed:16880211).
(Microbial infection) Directly mediates cytoadherence of Plasmodium falciparum parasitized erythrocytes and the internalization of particles independently of TLR signaling.
```{r, fig.width=4,fig.height=2,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0012254 <- compare_means(OG0012254~status,data=master,method = "t.test")
OG0012254

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0012254_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0012254, fill = status)) +
  geom_point(aes(y = OG0012254, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Platelet glycoprotein 4") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0012254, y.position = 9.2, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0012254_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0012254, fill = status)) +
  geom_point(aes(y = OG0012254, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Platelet glycoprotein 4") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0012254, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0012254_species,OG0012254_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```
### Pull out Orthogroups with ECM function

```{r}
ECM <- expr_mat_annot[expr_mat_annot$GO %like% "extracellular matrix",]
dim(ECM) #107

write.csv(ECM,file = "./all/ECM_expression.csv",row.names = FALSE)
```

## Step 7: ANOVA on ECM Orthogroups

We run a one-way ANOVA on the Orthogroups with immune function to test for differences in expression between disease state (control, exposed, infected)
```{r}
t_all_ECM <- setNames(data.frame(t(ECM[,-1])), ECM[,1])
row.names(t_all_ECM)
t_all_ECM <- tibble::rownames_to_column(t_all_ECM, "name")
t_all_ECM <- t_all_ECM[-c(1:6),]

label <- read.csv("./all/metaData.csv")
colnames(label)
label <- label[-c(53),c(3,5)] # remove pstr_d8; keep name and status
master <- merge(label,t_all_ECM,by="name")

# Perform ANOVA on the 213 immune genes
names(master)
# ANOVA loop significant results
for(i in 3:ncol(master))
{
  column <- names(master[i])
  #tidy will summarize and return neat format
  avz <- broom::tidy(aov(master[,i] ~ status, data = master))
  
  # Add this condition if you only want aov with P < 0.05
  if(avz$p.value[1] < 0.05) {
    
    print(column)
  }
}
```
[1] "OG0000071"
```{r}
OG0000071 <- aov(master$OG0000071 ~ master$status, data = master)
summary(OG0000071) 
TukeyHSD(OG0000071) # Significant between exposed and disease
```
[2] "OG0000128"
```{r}
OG0000128 <- aov(master$OG0000128 ~ master$status, data = master)
summary(OG0000128) 
TukeyHSD(OG0000128) # Significant between exposed and disease; healthy and disease
```
[3] "OG0001191"
```{r}
OG0001191 <- aov(master$OG0001191 ~ master$status, data = master)
summary(OG0001191) 
TukeyHSD(OG0001191) # Significant between exposed and disease; healthy and disease
```
[4] "OG0001630"
```{r}
OG0001630 <- aov(master$OG0001630 ~ master$status, data = master)
summary(OG0001630) 
TukeyHSD(OG0001630) # Significant between healthy and disease
```
[5] "OG0001920"
```{r}
OG0001920 <- aov(master$OG0001920 ~ master$status, data = master)
summary(OG0001920) 
TukeyHSD(OG0001920) # Significant between exposed and disease; healthy and disease
```
[6] "OG0002093"
```{r}
OG0002093 <- aov(master$OG0002093 ~ master$status, data = master)
summary(OG0002093) 
TukeyHSD(OG0002093)
Tukey <- TukeyHSD(OG0002093) # Significant between exposed and disease; healthy and disease
print(Tukey,digits = 15)
```
[7] "OG0002812"
```{r}
OG0002812 <- aov(master$OG0002812 ~ master$status, data = master)
summary(OG0002812) 
TukeyHSD(OG0002812) # Significant between healthy and disease; healthy and exposed
```
[8] "OG0002951"
```{r}
OG0002951 <- aov(master$OG0002951 ~ master$status, data = master)
summary(OG0002951) 
TukeyHSD(OG0002951) # Significant between healthy and exposed
```
[9] "OG0003597"
```{r}
OG0003597 <- aov(master$OG0003597 ~ master$status, data = master)
summary(OG0003597) 
TukeyHSD(OG0003597) # Significant between exposed and disesase
```
[10] "OG0004233"
```{r}
OG0004233 <- aov(master$OG0004233 ~ master$status, data = master)
summary(OG0004233) 
TukeyHSD(OG0004233) # Significant between healthy and disesase
```
[-] "OG0005350"
```{r}
OG0005350 <- aov(master$OG0005350 ~ master$status, data = master)
summary(OG0005350) 
TukeyHSD(OG0005350) # NOT Significant!
```
[11] "OG0007699"
```{r}
OG0007699 <- aov(master$OG0007699 ~ master$status, data = master)
summary(OG0007699) 
TukeyHSD(OG0007699) # significant between healthy and disease
```
[12] "OG0007911"
```{r}
OG0007911 <- aov(master$OG0007911 ~ master$status, data = master)
summary(OG0007911) 
TukeyHSD(OG0007911) # significant between exposed and disease; healthy and disease
```
[13] "OG0008154"
```{r}
OG0008154 <- aov(master$OG0008154 ~ master$status, data = master)
summary(OG0008154) 
TukeyHSD(OG0008154) # significant between exposed and disease
```
[14] "OG0011025"
```{r}
OG0011025 <- aov(master$OG0011025 ~ master$status, data = master)
summary(OG0011025) 
TukeyHSD(OG0011025) # significant between healthy and disease
```
[15] "OG0011028"
```{r}
OG0011028 <- aov(master$OG0011028 ~ master$status, data = master)
summary(OG0011028) 
TukeyHSD(OG0011028) # significant between healthy and disease; healthy and exposed
```
[16] "OG0011034"
```{r}
OG0011034 <- aov(master$OG0011034 ~ master$status, data = master)
summary(OG0011034) 
TukeyHSD(OG0011034) # significant between healthy and disease
```
[17] "OG0011714"
```{r}
OG0011714 <- aov(master$OG0011714 ~ master$status, data = master)
summary(OG0011714) 
TukeyHSD(OG0011714) # significant between healthy and disease
```
[18] "OG0011721"
```{r}
OG0011721 <- aov(master$OG0011721 ~ master$status, data = master)
summary(OG0011721) 
TukeyHSD(OG0011721) # significant between healthy and disease
```

### Significant 

Make a dataframe including only the 18 significant highly variable orthologs

```{r,results='hide',tidy=TRUE}
ECM
sig_orthogroups <- c("OG0000071","OG0000128","OG0001191","OG0001630","OG0001920","OG0002093","OG0002812","OG0002951","OG0003597","OG0004233","OG0007699","OG0007911","OG0008154","OG0011025","OG0011028","OG0011034","OG0011714","OG0011721")
sig_orthogroups <- ECM[ECM$Orthogroup %in% sig_orthogroups,]
write.csv(sig_orthogroups,file="./all/sig_ECM_orthogroups.csv",row.names = FALSE)
```
Add the Tukey p value results to this file in 3 columns: EvI,CvI and CvE.
Rename this file "sig_immune_orthogroups_ANOVA.csv"

### Boxplots
```{r}
ECM_expression <- read.csv("./all/sig_ECM_orthogroups_ANOVA.csv")
names(ECM_expression)
ECM_expression <- ECM_expression[,-c(2:12)]
t_ECM_expression <- setNames(as.data.frame(t(ECM_expression[,-1])), ECM_expression[,1])
row.names(t_ECM_expression)
view(t_ECM_expression)
t_ECM_expression <- tibble::rownames_to_column(t_ECM_expression, "name")
view(t_ECM_expression)

label <- read.csv("./all/metaData.csv")
colnames(label)
label <- label[-c(53),c(1,5)] # remove pstr_d8; keep species, status
t_ECM_expression$species <- label$species
t_ECM_expression <- t_ECM_expression %>% relocate(species, .before = everything())
t_ECM_expression$status <- label$status
t_ECM_expression <- t_ECM_expression %>% relocate(status, .after = name)


t_ECM_expression$status <- gsub("healthy","control", t_ECM_expression$status)
t_ECM_expression$status <- gsub("disease","infected", t_ECM_expression$status)
t_ECM_expression$status <- factor(t_ECM_expression$status, levels = c("control","exposed","infected"))
t_ECM_expression$species <- factor(t_ECM_expression$species, levels = c("mcav","past","pstr","oann","cnat"))
master <- t_ECM_expression
```

#### Hemicentin-1
Involved in transforming growth factor beta-mediated rearrangement of the podocyte cytoskeleton which includes reduction of F-actin fibers and broadening, flattening and elongation of podocytes (By similarity).
Plays a role in basement membrane organization (PubMed:34504132).
May promote cleavage furrow maturation during cytokinesis in preimplantation embryos (PubMed:21215633).
May play a role in the architecture of adhesive and flexible epithelial cell junctions (PubMed:17015624).
May play a role during myocardial remodeling by imparting an effect on cardiac fibroblast migration (PubMed:24951538).
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0000071 <- compare_means(OG0000071~status,data=master,method = "t.test")
OG0000071

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0000071_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0000071, fill = status)) +
  geom_point(aes(y = OG0000071, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Hemicentin-1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0000071, y.position = 11, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0000071_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0000071, fill = status)) +
  geom_point(aes(y = OG0000071, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Hemicentin-1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0000071, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0000071_species,OG0000071_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### ** Alpha-2 type I collagen
Type I collagen is a member of group I collagen (fibrillar forming collagen).
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0000128 <- compare_means(OG0000128~status,data=master,method = "t.test")
OG0000128

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0000128_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0000128, fill = status)) +
  geom_point(aes(y = OG0000128, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Alpha-2 type I collagen") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0000128, y.position = 16, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0000128_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0000128, fill = status)) +
  geom_point(aes(y = OG0000128, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Alpha-2 type I collagen") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0000128, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0000128_species,OG0000128_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```



#### Matrix metalloproteinase-24
Metalloprotease that mediates cleavage of N-cadherin (CDH2) and acts as a regulator of neuro-immune interactions and neural stem cell quiescence (PubMed:19805319, PubMed:24952463).
Involved in cell-cell interactions between nociceptive neurites and mast cells, possibly by mediating cleavage of CDH2, thereby acting as a mediator of peripheral thermal nociception and inflammatory hyperalgesia (PubMed:19805319).
Key regulator of neural stem cells quiescence by mediating cleavage of CDH2, affecting CDH2-mediated anchorage of neural stem cells to ependymocytes in the adult subependymal zone, leading to modulate their quiescence (PubMed:24952463).
May play a role in axonal growth (PubMed:11714638).
Able to activate progelatinase A. May also be a proteoglycanase involved in degradation of proteoglycans, such as dermatan sulfate and chondroitin sulfate proteoglycans. Cleaves partially fibronectin, but not collagen type I, nor laminin (PubMed:10622708).
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0001191 <- compare_means(OG0001191~status,data=master,method = "t.test")
OG0001191

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0001191_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0001191, fill = status)) +
  geom_point(aes(y = OG0001191, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Matrix metalloproteinase-24") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0001191, y.position = 10, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0001191_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0001191, fill = status)) +
  geom_point(aes(y = OG0001191, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Matrix metalloproteinase-24") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0001191, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0001191_species,OG0001191_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```




#### Basement membrane-specific heparan sulfate proteoglycan core protein
Integral component of basement membranes. Component of the glomerular basement membrane (GBM), responsible for the fixed negative electrostatic membrane charge, and which provides a barrier which is both size- and charge-selective. It serves as an attachment substrate for cells. Plays essential roles in vascularization. Critical for normal heart development and for regulating the vascular response to injury. Also required for avascular cartilage development.
Endorepellin in an anti-angiogenic and anti-tumor peptide that inhibits endothelial cell migration, collagen-induced endothelial tube morphogenesis and blood vessel growth in the chorioallantoic membrane. Blocks endothelial cell adhesion to fibronectin and type I collagen. Anti-tumor agent in neovascularization. Interaction with its ligand, integrin alpha2/beta1, is required for the anti-angiogenic properties. Evokes a reduction in phosphorylation of receptor tyrosine kinases via alpha2/beta1 integrin-mediated activation of the tyrosine phosphatase, PTPN6.
The LG3 peptide has anti-angiogenic properties that require binding of calcium ions for full activity.
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0001630 <- compare_means(OG0001630~status,data=master,method = "t.test")
OG0001630

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0001630_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0001630, fill = status)) +
  geom_point(aes(y = OG0001630, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "HSPG2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0001630, y.position = 10, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0001630_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0001630, fill = status)) +
  geom_point(aes(y = OG0001630, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "HSPG2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0001630, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0001630_species,OG0001630_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```

#### Hemicentin-1
Involved in transforming growth factor beta-mediated rearrangement of the podocyte cytoskeleton which includes reduction of F-actin fibers and broadening, flattening and elongation of podocytes (PubMed:29488390).
Plays a role in basement membrane organization (By similarity).
May promote cleavage furrow maturation during cytokinesis in preimplantation embryos (By similarity).
May play a role in the architecture of adhesive and flexible epithelial cell junctions (By similarity).
May play a role during myocardial remodeling by imparting an effect on cardiac fibroblast migration (By similarity).
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0001920 <- compare_means(OG0001920~status,data=master,method = "t.test")
OG0001920

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0001920_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0001920, fill = status)) +
  geom_point(aes(y = OG0001920, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Hemicentin-1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0001920, y.position = 13, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0001920_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0001920, fill = status)) +
  geom_point(aes(y = OG0001920, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Hemicentin-1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0001920, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0001920_species,OG0001920_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```

#### Alpha-1 type II collagen

```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0002093 <- compare_means(OG0002093~status,data=master,method = "t.test")
OG0002093

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0002093_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0002093, fill = status)) +
  geom_point(aes(y = OG0002093, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Alpha-1 type II collagen") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0002093, y.position = 15, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0002093_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0002093, fill = status)) +
  geom_point(aes(y = OG0002093, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Alpha-1 type II collagen") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0002093, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0002093_species,OG0002093_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### Deleted in malignant brain tumors 1 protein
May play roles in mucosal defense system and cellular immune defense. May play a role in liver regeneration. May be an important factor in fate decision and differentiation of transit-amplifying ductular (oval) cells within the hepatic lineage. May function as a binding protein in saliva for the regulation of taste sensation. May play a role as an opsonin receptor for SFTPD and SPAR in macrophage tissues throughout the body, including epithelial cells lining the gastrointestinal tract (By similarity).
Required for terminal differentiation of columnar epithelial cells during early embryogenesis. Displays a broad calcium-dependent binding spectrum against both Gram-positive and Gram-negative bacteria, suggesting a role in defense against bacterial pathogens. Binds to a range of poly-sulfated and poly-phosphorylated ligands which may explain its broad bacterial-binding specificity. Inhibits cytoinvasion of S.enterica. Associates with the actin cytoskeleton and is involved in its remodeling during regulated exocytosis. Interacts with pancreatic zymogens in a pH-dependent manner and may act as a Golgi cargo receptor in the regulated secretory pathway of the pancreatic acinar cell
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0002812 <- compare_means(OG0002812~status,data=master,method = "t.test")
OG0002812

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0002812_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0002812, fill = status)) +
  geom_point(aes(y = OG0002812, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Deleted in malignant brain tumors 1 protein") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 10.2, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0002812_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0002812, fill = status)) +
  geom_point(aes(y = OG0002812, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Deleted in malignant brain tumors 1 protein") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0002812_species,OG0002812_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```


#### Pancreatic secretory granule membrane major glycoprotein GP2
Functions as an intestinal M-cell transcytotic receptor specific of type-I-piliated bacteria that participates in the mucosal immune response toward these bacteria. At the apical membrane of M-cells it binds fimH, a protein of the bacteria type I pilus tip. Internalizes bound bacteria, like E.coli and S.typhimurium, from the lumen of the intestine and delivers them, through M-cells, to the underlying organized lymphoid follicles where they are captured by antigen-presenting dendritic cells to elicit a mucosal immune response.
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0002951 <- compare_means(OG0002951~status,data=master,method = "t.test")
OG0002951

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0002951_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0002951, fill = status)) +
  geom_point(aes(y = OG0002951, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Pancreatic secretory granule membrane major glycoprotein GP2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0002951, y.position = 11, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0002951_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0002951, fill = status)) +
  geom_point(aes(y = OG0002951, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Pancreatic secretory granule membrane major glycoprotein GP2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0002951, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0002951_species,OG0002951_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```
#### Integrin alpha-V
The alpha-V (ITGAV) integrins are receptors for vitronectin, cytotactin, fibronectin, fibrinogen, laminin, matrix metalloproteinase-2, osteopontin, osteomodulin, prothrombin, thrombospondin and vWF. They recognize the sequence R-G-D in a wide array of ligands. ITGAV:ITGB3 binds to fractalkine (CX3CL1) and may act as its coreceptor in CX3CR1-dependent fractalkine signaling (PubMed:23125415).
ITGAV:ITGB3 binds to NRG1 (via EGF domain) and this binding is essential for NRG1-ERBB signaling (PubMed:20682778).
ITGAV:ITGB3 binds to FGF1 and this binding is essential for FGF1 signaling (PubMed:18441324).
ITGAV:ITGB3 binds to FGF2 and this binding is essential for FGF2 signaling (PubMed:28302677).
ITGAV:ITGB3 binds to IGF1 and this binding is essential for IGF1 signaling (PubMed:19578119).
ITGAV:ITGB3 binds to IGF2 and this binding is essential for IGF2 signaling (PubMed:28873464).
ITGAV:ITGB3 binds to IL1B and this binding is essential for IL1B signaling (PubMed:29030430).
ITGAV:ITGB3 binds to PLA2G2A via a site (site 2) which is distinct from the classical ligand-binding site (site 1) and this induces integrin conformational changes and enhanced ligand binding to site 1 (PubMed:18635536, PubMed:25398877).
ITGAV:ITGB3 and ITGAV:ITGB6 act as receptors for fibrillin-1 (FBN1) and mediate R-G-D-dependent cell adhesion to FBN1 (PubMed:12807887, PubMed:17158881).
Integrin alpha-V/beta-6 or alpha-V/beta-8 (ITGAV:ITGB6 or ITGAV:ITGB8) mediates R-G-D-dependent release of transforming growth factor beta-1 (TGF-beta-1) from regulatory Latency-associated peptide (LAP), thereby playing a key role in TGF-beta-1 activation (PubMed:15184403, PubMed:22278742, PubMed:28117447).
ITGAV:ITGB3 acts as a receptor for CD40LG (PubMed:31331973).
(Microbial infection) Integrin ITGAV:ITGB5 acts as a receptor for Adenovirus type C.
(Microbial infection) Integrin ITGAV:ITGB5 and ITGAV:ITGB3 act as receptors for Coxsackievirus A9 and B1.
(Microbial infection) Integrin ITGAV:ITGB3 acts as a receptor for Herpes virus 8/HHV-8.
(Microbial infection) Integrin ITGAV:ITGB6 acts as a receptor for herpes simplex 1/HHV-1.
(Microbial infection) Integrin ITGAV:ITGB3 acts as a receptor for Human parechovirus 1.
(Microbial infection) Integrin ITGAV:ITGB3 acts as a receptor for West nile virus.
(Microbial infection) In case of HIV-1 infection, the interaction with extracellular viral Tat protein seems to enhance angiogenesis in Kaposi's sarcoma lesions.
```{r, fig.width=4,fig.height=2,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0003597 <- compare_means(OG0003597~status,data=master,method = "t.test")
OG0003597

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0003597_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0003597, fill = status)) +
  geom_point(aes(y = OG0003597, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Integrin alpha-V") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003597, y.position = 10, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0003597_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0003597, fill = status)) +
  geom_point(aes(y = OG0003597, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Integrin alpha-V") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003597, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0003597_species,OG0003597_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```



#### ** Superoxide dismutase [Cu-Zn] 1
```{r, fig.width=6,fig.height=3,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0004233 <- compare_means(OG0004233~status,data=master,method = "t.test")
OG0004233

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0004233_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0004233, fill = status)) +
  geom_point(aes(y = OG0004233, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Superoxide dismutase [Cu-Zn] 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0004233, y.position = 11.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0004233_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0004233, fill = status)) +
  geom_point(aes(y = OG0004233, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Superoxide dismutase [Cu-Zn] 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0004233, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0004233_species,OG0004233_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```




#### Matrix metalloproteinase-25
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0007699 <- compare_means(OG0007699~status,data=master,method = "t.test")
OG0007699

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0007699_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0007699, fill = status)) +
  geom_point(aes(y = OG0007699, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Matrix metalloproteinase-25") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0007699, y.position = 10, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0007699_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0007699, fill = status)) +
  geom_point(aes(y = OG0007699, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Matrix metalloproteinase-25") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0007699, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0007699_species,OG0007699_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```





#### Protein Wnt-5a
Ligand for members of the frizzled family of seven transmembrane receptors (PubMed:17117926).
Can activate or inhibit canonical Wnt signaling, depending on receptor context (PubMed:16602827).
In the presence of FZD4, activates beta-catenin signaling. In the presence of ROR2, inhibits the canonical Wnt pathway by promoting beta-catenin degradation through a GSK3-independent pathway which involves down-regulation of beta-catenin-induced reporter gene expression (PubMed:16602827).
Suppression of the canonical pathway allows chondrogenesis to occur and inhibits tumor formation. Stimulates cell migration (PubMed:17117926).
Decreases proliferation, migration, invasiveness and clonogenicity of carcinoma cells and may act as a tumor suppressor. Mediates motility of melanoma cells (By similarity).
Required during embryogenesis for extension of the primary anterior-posterior axis and for outgrowth of limbs and the genital tubercle (PubMed:10021340).
Inhibits type II collagen expression in chondrocytes (By similarity).
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0007911 <- compare_means(OG0007911~status,data=master,method = "t.test")
OG0007911

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0007911_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0007911, fill = status)) +
  geom_point(aes(y = OG0007911, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Protein Wnt-5a") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 11.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0007911_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0007911, fill = status)) +
  geom_point(aes(y = OG0007911, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Protein Wnt-5a") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0007911_species,OG0007911_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```



#### Metalloproteinase inhibitor 3
Ligand for members of the frizzled family of seven transmembrane receptors (PubMed:17117926).
Can activate or inhibit canonical Wnt signaling, depending on receptor context (PubMed:16602827).
In the presence of FZD4, activates beta-catenin signaling. In the presence of ROR2, inhibits the canonical Wnt pathway by promoting beta-catenin degradation through a GSK3-independent pathway which involves down-regulation of beta-catenin-induced reporter gene expression (PubMed:16602827).
Suppression of the canonical pathway allows chondrogenesis to occur and inhibits tumor formation. Stimulates cell migration (PubMed:17117926).
Decreases proliferation, migration, invasiveness and clonogenicity of carcinoma cells and may act as a tumor suppressor. Mediates motility of melanoma cells (By similarity).
Required during embryogenesis for extension of the primary anterior-posterior axis and for outgrowth of limbs and the genital tubercle (PubMed:10021340).
Inhibits type II collagen expression in chondrocytes (By similarity).
```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0008154 <- compare_means(OG0008154~status,data=master,method = "t.test")
OG0008154

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0008154_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0008154, fill = status)) +
  geom_point(aes(y = OG0008154, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Metalloproteinase inhibitor 3") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 11.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0008154_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0008154, fill = status)) +
  geom_point(aes(y = OG0008154, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Metalloproteinase inhibitor 3") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0008154_species,OG0008154_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```




#### Collagen triple helix repeat-containing protein 1

```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0011025 <- compare_means(OG0011025~status,data=master,method = "t.test")
OG0011025

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0011025_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0011025, fill = status)) +
  geom_point(aes(y = OG0011025, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Collagen triple helix repeat-containing protein 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 11.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0011025_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0011025, fill = status)) +
  geom_point(aes(y = OG0011025, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Collagen triple helix repeat-containing protein 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0011025_species,OG0011025_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```





#### Chymotrypsin-like elastase family member 3B

```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0011028 <- compare_means(OG0011028~status,data=master,method = "t.test")
OG0011028

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0011028_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0011028, fill = status)) +
  geom_point(aes(y = OG0011028, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Chymotrypsin-like elastase family member 3B") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 11.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0011028_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0011028, fill = status)) +
  geom_point(aes(y = OG0011028, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Chymotrypsin-like elastase family member 3B") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0011028_species,OG0011028_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```






#### Collagen triple helix repeat-containing protein 1

```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0011034 <- compare_means(OG0011034~status,data=master,method = "t.test")
OG0011034

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0011034_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0011034, fill = status)) +
  geom_point(aes(y = OG0011034, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Collagen triple helix repeat-containing protein 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 11.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0011034_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0011034, fill = status)) +
  geom_point(aes(y = OG0011034, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Collagen triple helix repeat-containing protein 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0011034_species,OG0011034_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```

#### Tenascin-R

```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0011714 <- compare_means(OG0011714~status,data=master,method = "t.test")
OG0011714

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0011714_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0011714, fill = status)) +
  geom_point(aes(y = OG0011714, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Tenascin-R") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 11.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0011714_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0011714, fill = status)) +
  geom_point(aes(y = OG0011714, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Tenascin-R") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0011714_species,OG0011714_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```








#### Bone morphogenetic protein 1

```{r, fig.width=8,fig.height=4,dpi=200}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0011721 <- compare_means(OG0011721~status,data=master,method = "t.test")
OG0011721

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0011721_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0011721, fill = status)) +
  geom_point(aes(y = OG0011721, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Bone morphogenetic protein 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0003270, y.position = 11.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0011721_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0011721, fill = status)) +
  geom_point(aes(y = OG0011721, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Bone morphogenetic protein 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0011721_species,OG0011721_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```









### Combined Figure
```{r immune orthogroups boxplots, fig.width=6, fig.height=20, dpi=200}
pdf("immunePlots_2.pdf",width = 6,height = 20)
ggarrange(OG0005460_species,OG0005460_status,
          OG0004181_species,OG0004181_status,
          OG0002812_species,OG0002812_status,
          OG0007458_species,OG0007458_status,
          OG0002316_species,OG0002316_status,
          OG0000274_species,OG0000274_status,
          OG0002093_species,OG0002093_status,
          OG0004233_species,OG0004233_status,
          ncol = 2,nrow = 8,
          widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",
          align = "hv"
          )
dev.off()
# Remember to change the pvalues and add the ones from the TukeyHSD post hoc test
```

## Step 8: Rab7 Correlations
Correlate the Rab7 Orthogroup to symbiont genes

OG0007458 = Rab7
psaA = Q9XQV3 (DEG in Breviolum and Durusdinium)
psbB = P49471 (DEG in Breviolum and Durusdinium)
psbC = P49472 (DEG in Breviolum and Durusdinium)

High affinity nitrate transporter 2.5 (NRT2.5) = Q9LPV5
Superoxide dismutase = Q59094
Carbonic anhydrase = Q5BCC5
Apoptosis-inducing factor 3 = Q96NN9
Heat shock 70 kDa protein = P11144
Two pore calcium channel protein 1 = Q5QM84

```{r}

```

