---
title: "Coral EVE Analysis"
author: "Kelsey Beavers"
date: "`r Sys.Date()`"
output:
  html_document:
    keep_md: true
editor_options: 
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = "~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/")
knitr::opts_chunk$set(dev='ragg_png')
library(tximport)
library(ape)
library(tibble)
library(reshape2)
library(DESeq2)
library(dplyr)
library(PCAtools)
library(ggalt)
library(ggplot2)
library(lifecycle)
library(evemodel)
library(ggvenn)
library(stringr)
library(ggpubr)
library(pheatmap)
library(utils)
library(corrplot)
library(repr) 
library(data.table)
library(dplyr)
library(GOplot)
```

# TXimport
## Step 1: Set up 

This time, instead of using Uniprot IDs as our gene ID in our tx2gene files, we will use the single-copy ortholog ID from OrthoFinder. We will use "Orthogroup_Sequences.csv" from the "Command_Line_Code.Rmd" file to create these new tx2 genes files for each species. 

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/all/Orthofinder/")

orthogroupSequenceNames <- read.csv("Orthogroup_Sequences.csv",header = TRUE)

#1817 single copy orthologs

cnat <- orthogroupSequenceNames[,c(1,2)]
write.csv(cnat,file="cnat_contig_to_orthogroup.csv")

mcav <- orthogroupSequenceNames[,c(1,3)]
write.csv(mcav,file="mcav_contig_to_orthogroup.csv")

oann <- orthogroupSequenceNames[,c(1,4)]
write.csv(oann,file="oann_contig_to_orthogroup.csv")

past <- orthogroupSequenceNames[,c(1,5)]
write.csv(past,file="past_contig_to_orthogroup.csv")

pstr <- orthogroupSequenceNames[,c(1,6)]
write.csv(pstr,file="pstr_contig_to_orthogroup.csv")
```

## Step 2: Import read quants from Salmon

Read in the raw counts from Salmon for ALL coral samples

### C. natans
```{r,results='hide',tidy=TRUE}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/cnat/salmon")
cnat_samples <- read.table("cnat_samples.csv", header = TRUE)
cnat_files <- file.path(cnat_samples$sample, "quant.sf")
names(cnat_files) <- paste0("sample", 1:10)
all(file.exists(cnat_files))
# Do not continue if output shows [FALSE]
cnat_tx2gene <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/all/Orthofinder/cnat_contig_to_orthogroup.csv")
cnat_tx2gene <- cnat_tx2gene[,c(3,2)]
cnat_tx2gene$cnat_ref_proteome_cdhit <- gsub("\\..*","",cnat_tx2gene$cnat_ref_proteome_cdhit)
cnat_txi <- tximport(cnat_files, type = 'salmon',tx2gene=cnat_tx2gene)
dim(cnat_txi$counts)
# 13,692 transcripts with evalue ≤ 1.00E-06

cnat_counts <- cnat_txi$counts
cnat_counts <- as.data.frame(cnat_counts)
colnames(cnat_counts) <- c("cnat_c2","cnat_c3","cnat_c4","cnat_c6","cnat_c8","cnat_d2","cnat_d3","cnat_d4","cnat_d6","cnat_d8")
cnat_counts <- tibble::rownames_to_column(cnat_counts,"Orthogroup")
cnat_counts[2:11] <-  round(cnat_counts[2:11], digits=0)
cnat_counts <- as.data.frame(cnat_counts)
```

### M. cavernosa
```{r,results='hide',tidy=TRUE}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/mcav/salmon")
mcav_samples <- read.table("mcav_samples.csv", header = TRUE)
mcav_files <- file.path(mcav_samples$sample, "quant.sf")
names(mcav_files) <- paste0("sample", 1:13)
all(file.exists(mcav_files))
# Do not continue if output shows [FALSE]
mcav_tx2gene <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/all/Orthofinder/mcav_contig_to_orthogroup.csv")
mcav_tx2gene <- mcav_tx2gene[,c(3,2)]
mcav_tx2gene$mcav_ref_proteome_cdhit <- gsub("\\..*","",mcav_tx2gene$mcav_ref_proteome_cdhit)
mcav_txi <- tximport(mcav_files, type = 'salmon',tx2gene=mcav_tx2gene)
dim(mcav_txi$counts)
# 13,692 transcripts with evalue ≤ 1.00E-06

mcav_counts <- mcav_txi$counts
mcav_counts <- as.data.frame(mcav_counts)
colnames(mcav_counts) <- c("mcav_c1","mcav_c3","mcav_c5","mcav_c6","mcav_c7","mcav_c8","mcav_d1","mcav_d2","mcav_d3","mcav_d4","mcav_d5","mcav_d6","mcav_d8")
mcav_counts <- tibble::rownames_to_column(mcav_counts,"Orthogroup")
mcav_counts[2:14] <-  round(mcav_counts[2:14], digits=0)
mcav_counts <- as.data.frame(mcav_counts)
```

### O. annularis
```{r,results='hide',tidy=TRUE}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/oann/salmon")
oann_samples <- read.table("oann_samples.csv", header = TRUE)
oann_files <- file.path(oann_samples$sample, "quant.sf")
names(oann_files) <- paste0("sample", 1:11)
all(file.exists(oann_files))
# Do not continue if output shows [FALSE]
oann_tx2gene <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/all/Orthofinder/oann_contig_to_orthogroup.csv")
oann_tx2gene <- oann_tx2gene[,c(3,2)]
oann_tx2gene$oann_ref_proteome_cdhit <- gsub("\\..*","",oann_tx2gene$oann_ref_proteome_cdhit)
oann_txi <- tximport(oann_files, type = 'salmon',tx2gene=oann_tx2gene)
dim(oann_txi$counts)
# 13,692 transcripts with evalue ≤ 1.00E-06

oann_counts <- oann_txi$counts
oann_counts <- as.data.frame(oann_counts)
colnames(oann_counts) <- c("oann_c1","oann_c2","oann_c3","oann_c6","oann_c7","oann_c8","oann_d1","oann_d3","oann_d6","oann_d7","oann_d8")
oann_counts <- tibble::rownames_to_column(oann_counts,"Orthogroup")
oann_counts[2:12] <-  round(oann_counts[2:12], digits=0)
oann_counts <- as.data.frame(oann_counts)
```

### P. astreoides
```{r,results='hide',tidy=TRUE}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/past/salmon")
past_samples <- read.table("past_samples.csv", header = TRUE)
past_files <- file.path(past_samples$sample, "quant.sf")
names(past_files) <- paste0("sample", 1:11)
all(file.exists(past_files))
# Do not continue if output shows [FALSE]
past_tx2gene <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/all/Orthofinder/past_contig_to_orthogroup.csv")
past_tx2gene <- past_tx2gene[,c(3,2)]
past_tx2gene$past_ref_proteome_cdhit <- gsub("\\..*","",past_tx2gene$past_ref_proteome_cdhit)
past_txi <- tximport(past_files, type = 'salmon',tx2gene=past_tx2gene)
dim(past_txi$counts)
# 13,692 transcripts with evalue ≤ 1.00E-06

past_counts <- past_txi$counts
past_counts <- as.data.frame(past_counts)
colnames(past_counts) <- c("past_c1","past_c5","past_c6","past_c7","past_d1","past_d2","past_d3","past_d4","past_d5","past_d6","past_d7")
past_counts <- tibble::rownames_to_column(past_counts,"Orthogroup")
past_counts[2:12] <-  round(past_counts[2:12], digits=0)
past_counts <- as.data.frame(past_counts)
```

### P. strigosa
```{r,results='hide',tidy=TRUE}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/pstr/salmon")
pstr_samples <- read.table("pstr_samples_outlier_removed.csv", header = TRUE)
pstr_files <- file.path(pstr_samples$sample, "quant.sf")
names(pstr_files) <- paste0("sample", 1:7)
all(file.exists(pstr_files))
# Do not continue if output shows [FALSE]
pstr_tx2gene <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/all/Orthofinder/pstr_contig_to_orthogroup.csv")
pstr_tx2gene <- pstr_tx2gene[,c(3,2)]
pstr_tx2gene$pstr_ref_proteome_cdhit <- gsub("\\..*","",pstr_tx2gene$pstr_ref_proteome_cdhit)
pstr_txi <- tximport(pstr_files, type = 'salmon',tx2gene=pstr_tx2gene)
dim(pstr_txi$counts)
# 13,692 transcripts with evalue ≤ 1.00E-06

pstr_counts <- pstr_txi$counts
pstr_counts <- as.data.frame(pstr_counts)
colnames(pstr_counts) <- c("pstr_c3","pstr_c5","pstr_c7","pstr_c8","pstr_d3","pstr_d5","pstr_d7")
pstr_counts <- tibble::rownames_to_column(pstr_counts,"Orthogroup")
pstr_counts[2:8] <-  round(pstr_counts[2:8], digits=0)
pstr_counts <- as.data.frame(pstr_counts)
```

## Step 3: Merge the data

Now merge the count data
```{r,results='hide',tidy=TRUE}
Cnat_Mcav <- merge(cnat_counts,mcav_counts, by ="Orthogroup")
Cnat_Mcav_Oann <- merge(Cnat_Mcav,oann_counts,by="Orthogroup")
Cnat_Mcav_Oann_Past <- merge(Cnat_Mcav_Oann,past_counts,by="Orthogroup")
all_species <- merge(Cnat_Mcav_Oann_Past,pstr_counts,by="Orthogroup")
# 1781 single-copy orthologs expressed and shared between all 5 species
```

## Step 4: Annotate the single-copy orthologs

We annotate using the names of the mcav proteome sequences

```{r,results='hide',tidy=TRUE}
# Merge in the names of the mcav proteome sequences
mcav_ortho_to_contig <- read.csv("./all/Orthofinder/mcav_SC_ortholog_sequences.csv")
names(mcav_ortho_to_contig)[2]<-paste("mcav_contig")

all_species_contig <- left_join(all_species,mcav_ortho_to_contig,by="Orthogroup")
all_species_contig <- all_species_contig %>% relocate(mcav_contig, .after = Orthogroup)

# Annotate with Uniprot Entry ID's with "mcav_sc_orthologs_annotated.csv"
annot <- read.csv("./all/Orthofinder/mcav_sc_orthologs_annotated.csv")
master <- left_join(all_species_contig,annot,by="mcav_contig")
master <- master %>% relocate(Entry, .after = mcav_contig)
master <- master %>% relocate(Entry.name, .after = Entry)
master <- master %>% relocate(Evalue, .after = Entry.name)
names(master)[1]<-paste("Orthogroup")
write.csv(master,file="./all/masterCounts.csv",row.names = FALSE)
```

# DESeq2

Make counts integers.
Run DESeq2 on Orthogroups to get Rlog normalized counts.

```{r,results='hide',tidy=TRUE}
countData <- read.csv("./all/masterCounts.csv", row.names="Orthogroup")
countData <- countData[,-c(1:4,57)]
colData <- read.csv("./all/metaData.csv",row.names = "name")
# Remove pstr_d8
colData <- colData[-c(53),]

# Make counts in countData integers:
countData[1:52] <-  round(countData[1:52], digits=0)
countData <- as.data.frame(countData)

# Set the design
dds_treatment <- DESeqDataSetFromMatrix(countData = countData, 
                                        colData = colData, 
                                        design = ~species + treatment )

# Filter out columns with an average number of reads less than 10
dds_treatment <- dds_treatment[ rowMeans(counts(dds_treatment)) > 10, ] 

# Run DESeq2
test <- DESeq(dds_treatment)
rld <- rlog(dds_treatment, blind=FALSE) 
rrld <- assay(rld)
# 1766 after filtering out low counts
write.csv(rrld, file = "./all/DESeq2/treatment_rlog.csv") 
getwd()
```

## EVE

```{r EVE_phyloTree,fig.align='center',fig.height=3,fig.width=3,dpi=100}
exprTbl <- read.csv("./all/DESeq2/treatment_rlog.csv")
colnames(exprTbl)[colnames(exprTbl)=="X"] <- "Ortholog"

# The table needs to be converted to a matrix:
exprMat <- as.matrix(exprTbl[,-1])
rownames(exprMat) <- exprTbl$Ortholog
exprMat
dim(exprMat)  
#Should return 1766 x 52

# Species phylogeny can be obtained from OrthoFinder output in DIR=/OrthoFinder/Results/Species_Tree/SpeciesTree_rooted.txt

# I find it is best to run OrthoFinder with an outgroup species to get a better species tree. I included a Cassiopea proteome (generated the same way as the other proteomes) as my outgroup to obtain the following rooted species tree:
# (Cassiopea_proteome:0.405291,(past_reference_proteome:0.207024,((mcav_reference_proteome:0.0938366,oann_reference_proteome:0.0723831)0.262775:0.0844971,(pstr_reference_proteome:0.0954775,cnat_reference_proteome:0.106944)0.514097:0.105377)0.45022:0.114876)1:0.405291); 
# I then remove the out group from the rooted tree and abbreviate the species names
speciesTree <-read.tree(text="((past:0.207024,((mcav:0.0938366,oann:0.0723831)0.262775:0.0844971,(pstr:0.0954775,cnat:0.106944)0.514097:0.105377)0.45022:0.114876)1:0.405291);")

# Check validity of species tree
plot(speciesTree,
     type = "phylogram",
     use.edge.length = TRUE,
     node.pos = NULL,
     show.tip.label = TRUE,
     root.edge = TRUE)
```

EVE needs to know which species each column in the expression matrix belongs to. This is done by creating a vector of species names corresponding to the tip labels in the species tree

```{r,results='hide',tidy=TRUE}
speciesTree$tip.label
colnames(exprMat)
colSpecies <- sub("_.*$","",colnames(exprMat))
colSpecies

# Run EVE
res <- betaSharedTest(tree = speciesTree, gene.data = exprMat, colSpecies = colSpecies)
```


```{r,results='hide',tidy=TRUE}
# P-value can then be calculated using:
pval = pchisq(res$LRT,df = 1,lower.tail = F)

# The shared beta:
res$sharedBeta
# [1] 0.3348927
log(0.3348927)
# [1] -1.093945

#Combine LRT, beta, theta, sigma2, alpha:
head(cbind(res$LRT,res$indivBetaRes$par,pval))
colnames(exprMat)
dim(exprMat)
# [1]  1766   52
a <- cbind(res$LRT,res$indivBetaRes$par,pval,exprMat[,1:52])
rownames(a) <- exprTbl$Ortholog

# Give Column Name to LRT column
colnames(a)
colnames(a)[colnames(a)==""] <- "LRT"
head(a)
EVE_results <- as.data.frame(a)
EVE_results <- tibble::rownames_to_column(EVE_results, "Orthogroup")

EVE_results$type <- ifelse(EVE_results$beta<0.3348927,"Lineage Specific","Highly Variable")
EVE_results$significant <- ifelse(EVE_results$pval<=0.05,"Significant","Not Significant")
EVE_results$category <- ifelse(EVE_results$significant == "Significant",EVE_results$type,"NS")
colnames(EVE_results)
EVE_results <- EVE_results %>% relocate(type, .after = pval)
EVE_results <- EVE_results %>% relocate(significant, .after = type)
EVE_results <- EVE_results %>% relocate(category, .after = significant)

write.csv(EVE_results, file="./all/EVE/R_EVE_results_SCTLD_5species.csv",row.names = FALSE)

#Visualize LRT v Beta by volcano plot for gene data
Shared_EVE <- read.csv("./all/EVE/R_EVE_results_SCTLD_5species.csv")
Shared_EVE_sig <- Shared_EVE %>% filter(pval <= 0.05)
write.csv(Shared_EVE_sig, file="./all/EVE/Signficant_SCTLD_eve.csv")
# Make a basic volcano plot
Shared_EVE_LS <- Shared_EVE_sig %>% filter(log(beta)< (-1.093945))
dim(Shared_EVE_LS)  #84
Shared_EVE_HV <- Shared_EVE_sig %>% filter(log(beta)> (-1.093945))
dim(Shared_EVE_HV)  #217
216+85  # 301
```

```{r EVE_volcano,fig.height=6,fig.width=6,dpi=200}
p <- ggplot(data = Shared_EVE,
       aes(x=log(beta),y=-log(pval),color=category))+
  geom_point()+
  scale_color_manual(values=c("#f37225","#00b9e3","black"))+
  geom_vline(xintercept = -1.093945, col="black", linetype="dashed")+
  annotate("text", x = 0.9, y = 9.5, label = "Shared beta", size = 3)+
  geom_hline(yintercept = -log(0.05), col="black",linetype = "dashed")+
  annotate("text", x = -4, y = 2.5, label = "p = 0.05", size = 3)+
  xlim(-5,5)+
  theme_light()
p + labs(color = "EVE Gene Category",title = "EVE Genes Volcano Plot",subtitle = "84 Lineage-Specific and 217 Highly Variable Genes") +
  theme(plot.title = element_text(face = "bold"), plot.subtitle = element_text(size = 9))
```

## Highly Variable Genes

```{r,results='hide',tidy=TRUE}
sig_EVE <- read.csv("./all/EVE/Signficant_SCTLD_eve.csv")
colnames(sig_EVE)
sig_EVE <- sig_EVE[,c(2,7,8,11)]   #Orthogroup,beta, pval, category
dim(sig_EVE)   #301 significant EVE genes
names(sig_EVE)
rlogs <- read.csv("./all/DESeq2/treatment_rlog.csv")
names(rlogs)
colnames(rlogs)[1] <- "Orthogroup"
sig_EVE_rlogs <- left_join(sig_EVE,rlogs,by="Orthogroup")    #301
names(sig_EVE_rlogs)
write.csv(sig_EVE_rlogs,file="./all/EVE/sig_EVE_rlogs.csv")

plasticGenes <- subset(sig_EVE_rlogs, sig_EVE$category == "Highly Variable")
names(plasticGenes)
plasticGenes <- plasticGenes[,-c(2:4)]
dim(plasticGenes)
# [1] 217  53
names(plasticGenes)
t_plasticGenes <- setNames(data.frame(t(plasticGenes[,-1])), plasticGenes[,1])
row.names(t_plasticGenes)
t_plasticGenes <- tibble::rownames_to_column(t_plasticGenes, "name")

label <- read.csv("./all/metaData.csv")
colnames(label)
label <- label[-c(53),c(3,4,5,7,8)] # remove pstr_d8; keep name,treatment,status,LGR,dominant_clade
master <- merge(label,t_plasticGenes,by="name")
master
dim(master)
# [1]  52 222
write.csv(master,file="./all/EVE/highlyVariable_rlogs_ortholog.csv")
```

### ANOVA

```{r,results='hide',tidy=TRUE}
results <- read.csv("./all/EVE/highlyVariable_rlogs_ortholog.csv")
results <- results[,c(4,7:ncol(results))]
dim(results)
# 52 218

# ANOVA loop significant results
for(i in 2:ncol(results))
{
  column <- names(results[i])
  #tidy will summarize and return neat format
  avz <- broom::tidy(aov(results[,i] ~ status, data = results))
  
  # Add this condition if you only want aov with P < 0.05
  if(avz$p.value[1] < 0.05) {
    
    print(column)
  }
}
```
[1] "OG0009610"
```{r}
OG0009610 <- aov(results$OG0009610 ~ results$status, data = results)
summary(OG0009610) 
TukeyHSD(OG0009610) # Significant between healthy and exposed
```
[2] "OG0009628"
```{r}
OG0009628 <- aov(results$OG0009628 ~ results$status, data = results)
summary(OG0009628) 
TukeyHSD(OG0009628) # Significant between exposed and disease
```
[3] "OG0009900"
```{r}
OG0009900 <- aov(results$OG0009900 ~ results$status, data = results)
summary(OG0009900) 
TukeyHSD(OG0009900) # Significant between healthy and disease
```
[4] "OG0009952"
```{r}
OG0009952 <- aov(results$OG0009952 ~ results$status, data = results)
summary(OG0009952) 
TukeyHSD(OG0009952) # Significant between healthy and disease
```
[5] "OG0009999"
```{r}
OG0009999 <- aov(results$OG0009999 ~ results$status, data = results)
summary(OG0009999) 
TukeyHSD(OG0009999) # Significant between healthy and disease
```
[6] "OG0010018"
```{r}
OG0010018 <- aov(results$OG0010018 ~ results$status, data = results)
summary(OG0010018) 
TukeyHSD(OG0010018) # Significant between healthy and disease
```
[7] "OG0010040"
```{r}
OG0010040 <- aov(results$OG0010040 ~ results$status, data = results)
summary(OG0010040) 
TukeyHSD(OG0010040) # Significant between healthy and disease
```
[-] "OG0010080"
```{r}
OG0010080 <- aov(results$OG0010080 ~ results$status, data = results)
summary(OG0010080) 
TukeyHSD(OG0010080) # NOT significant
```
[8] "OG0010117"
```{r}
OG0010117 <- aov(results$OG0010117 ~ results$status, data = results)
summary(OG0010117) 
TukeyHSD(OG0010117) # Significant between healthy and disease
```
[9] "OG0010145"
```{r}
OG0010145 <- aov(results$OG0010145 ~ results$status, data = results)
summary(OG0010145) 
TukeyHSD(OG0010145) # Significant between healthy and disease;healthy and exposed
```
[20] "OG0010176"
```{r}
OG0010176 <- aov(results$OG0010176 ~ results$status, data = results)
summary(OG0010176) 
TukeyHSD(OG0010176) # Significant between healthy and disease
```
[11] "OG0010177"
```{r}
OG0010177 <- aov(results$OG0010177 ~ results$status, data = results)
summary(OG0010177) 
TukeyHSD(OG0010177) # Significant between healthy and disease; exposed and disease
```
[12] "OG0010196"
```{r}
OG0010196 <- aov(results$OG0010196 ~ results$status, data = results)
summary(OG0010196) 
TukeyHSD(OG0010196) # Significant between healthy and disease
```
[13] "OG0010223"
```{r}
OG0010223 <- aov(results$OG0010223 ~ results$status, data = results)
summary(OG0010223) 
TukeyHSD(OG0010223) # Significant between healthy and disease
```
[-] "OG0010246"
```{r}
OG0010246 <- aov(results$OG0010246 ~ results$status, data = results)
summary(OG0010246) 
TukeyHSD(OG0010246) # NOT significant
```
[-] "OG0010267"
```{r}
OG0010267 <- aov(results$OG0010267 ~ results$status, data = results)
summary(OG0010267) 
TukeyHSD(OG0010267) # NOT significant
```
[14] "OG0010270"
```{r}
OG0010270 <- aov(results$OG0010270 ~ results$status, data = results)
summary(OG0010270) 
TukeyHSD(OG0010270) # Significant between healthy and disease
```
[15] "OG0010317"
```{r}
OG0010317 <- aov(results$OG0010317 ~ results$status, data = results)
summary(OG0010317) 
TukeyHSD(OG0010317) # Significant between healthy and disease
```
[16] "OG0010335"
```{r}
OG0010335 <- aov(results$OG0010335 ~ results$status, data = results)
summary(OG0010335) 
TukeyHSD(OG0010335) # Significant between healthy and disease
```
[17] "OG0010347"
```{r}
OG0010347 <- aov(results$OG0010347 ~ results$status, data = results)
summary(OG0010347) 
TukeyHSD(OG0010347) # Significant between healthy and disease
```
[18] "OG0010377"
```{r}
OG0010377 <- aov(results$OG0010377 ~ results$status, data = results)
summary(OG0010377) 
TukeyHSD(OG0010377) # Significant between healthy and disease
```
[19] "OG0010385"
```{r}
OG0010385 <- aov(results$OG0010385 ~ results$status, data = results)
summary(OG0010385) 
TukeyHSD(OG0010385) # Significant between healthy and disease
```
[20] "OG0010386"
```{r}
OG0010386 <- aov(results$OG0010386 ~ results$status, data = results)
summary(OG0010386) 
TukeyHSD(OG0010386) # Significant between healthy and disease
```
[21] "OG0010389"
```{r}
OG0010389 <- aov(results$OG0010389 ~ results$status, data = results)
summary(OG0010389) 
TukeyHSD(OG0010389) # Significant between healthy and disease
```
[22] "OG0010405"
```{r}
OG0010405 <- aov(results$OG0010405 ~ results$status, data = results)
summary(OG0010405) 
TukeyHSD(OG0010405) # Significant between healthy and disease
```
[23] "OG0010416"
```{r}
OG0010416 <- aov(results$OG0010416 ~ results$status, data = results)
summary(OG0010416) 
TukeyHSD(OG0010416) # Significant between healthy and disease
```
[24] "OG0010441"
```{r}
OG0010441 <- aov(results$OG0010441 ~ results$status, data = results)
summary(OG0010441) 
TukeyHSD(OG0010441) # Significant between healthy and disease
```
[25] "OG0010453"
```{r}
OG0010453 <- aov(results$OG0010453 ~ results$status, data = results)
summary(OG0010453) 
TukeyHSD(OG0010453) # Significant between healthy and disease
```
[26] "OG0010458"
```{r}
OG0010458 <- aov(results$OG0010458 ~ results$status, data = results)
summary(OG0010458) 
TukeyHSD(OG0010458) # Significant between healthy and disease
```
[27] "OG0010466"
```{r}
OG0010466 <- aov(results$OG0010466 ~ results$status, data = results)
summary(OG0010466) 
TukeyHSD(OG0010466) # Significant between healthy and disease; exposed and disease
```
[28] "OG0010477"
```{r}
OG0010477 <- aov(results$OG0010477 ~ results$status, data = results)
summary(OG0010477) 
TukeyHSD(OG0010477) # Significant between healthy and disease
```
[29] "OG0010479"
```{r}
OG0010479 <- aov(results$OG0010479 ~ results$status, data = results)
summary(OG0010479) 
TukeyHSD(OG0010479) # Significant between healthy and exposed
```
[30] "OG0010517"
```{r}
OG0010517 <- aov(results$OG0010517 ~ results$status, data = results)
summary(OG0010517) 
TukeyHSD(OG0010517) # Significant between healthy and disease
```
[31] "OG0010529"
```{r}
OG0010529 <- aov(results$OG0010529 ~ results$status, data = results)
summary(OG0010529) 
TukeyHSD(OG0010529) # Significant between healthy and disease
```
[32] "OG0010687"
```{r}
OG0010687 <- aov(results$OG0010687 ~ results$status, data = results)
summary(OG0010687) 
TukeyHSD(OG0010687) # Significant between healthy and disease; exposed and disease
```
[33] "OG0010700"
```{r}
OG0010700 <- aov(results$OG0010700 ~ results$status, data = results)
summary(OG0010700) 
TukeyHSD(OG0010700) # Significant between healthy and exposed
```
[34] "OG0010712"
```{r}
OG0010712 <- aov(results$OG0010712 ~ results$status, data = results)
summary(OG0010712) 
TukeyHSD(OG0010712) # Significant between healthy and disease
```
[-] "OG0010714"
```{r}
OG0010714 <- aov(results$OG0010714 ~ results$status, data = results)
summary(OG0010714) 
TukeyHSD(OG0010714) # NOT significant
```
[35] "OG0010721"
```{r}
OG0010721 <- aov(results$OG0010721 ~ results$status, data = results)
summary(OG0010721) 
TukeyHSD(OG0010721) # Significant between healthy and exposed
```
[36] "OG0010742"
```{r}
OG0010742 <- aov(results$OG0010742 ~ results$status, data = results)
summary(OG0010742) 
TukeyHSD(OG0010742) # Significant between healthy and disease
```
[37] "OG0010781"
```{r}
OG0010781 <- aov(results$OG0010781 ~ results$status, data = results)
summary(OG0010781) 
TukeyHSD(OG0010781) # Significant between healthy and disease
```
[38] "OG0010806"
```{r}
OG0010806 <- aov(results$OG0010806 ~ results$status, data = results)
summary(OG0010806) 
TukeyHSD(OG0010806) # Significant between healthy and disease; exposed and disease
```
[39] "OG0010856"
```{r}
OG0010856 <- aov(results$OG0010856 ~ results$status, data = results)
summary(OG0010856) 
TukeyHSD(OG0010856) # Significant between healthy and disease
```
[40] "OG0010888"
```{r}
OG0010888 <- aov(results$OG0010888 ~ results$status, data = results)
summary(OG0010888) 
TukeyHSD(OG0010888) # Significant between exposed and disease
```
[41] "OG0010904"
```{r}
OG0010904 <- aov(results$OG0010904 ~ results$status, data = results)
summary(OG0010904) 
TukeyHSD(OG0010904) # Significant between healthy and disease; exposed and disease
```
[-] "OG0010954"
```{r}
OG0010954 <- aov(results$OG0010954 ~ results$status, data = results)
summary(OG0010954) 
TukeyHSD(OG0010954) # NOT significant
```
[42] "OG0010962"
```{r}
OG0010962 <- aov(results$OG0010962 ~ results$status, data = results)
summary(OG0010962) 
TukeyHSD(OG0010962) # Significant between healthy and disease
```
[43] "OG0011004"
```{r}
OG0011004 <- aov(results$OG0011004 ~ results$status, data = results)
summary(OG0011004) 
TukeyHSD(OG0011004) # Significant between healthy and disease
```
[44] "OG0011009"
```{r}
OG0011009 <- aov(results$OG0011009 ~ results$status, data = results)
summary(OG0011009) 
TukeyHSD(OG0011009) # Significant between healthy and disease
```
[45] "OG0011043"
```{r}
OG0011043 <- aov(results$OG0011043 ~ results$status, data = results)
summary(OG0011043) 
TukeyHSD(OG0011043) # Significant between healthy and disease
```
[46] "OG0011062"
```{r}
OG0011062 <- aov(results$OG0011062 ~ results$status, data = results)
summary(OG0011062) 
TukeyHSD(OG0011062) # Significant between healthy and disease; exposed and disease
```
[47] "OG0011066"
```{r}
OG0011066 <- aov(results$OG0011066 ~ results$status, data = results)
summary(OG0011066) 
TukeyHSD(OG0011066) # Significant between healthy and disease; exposed and disease
```
[48] "OG0011070"
```{r}
OG0011070 <- aov(results$OG0011070 ~ results$status, data = results)
summary(OG0011070) 
TukeyHSD(OG0011070) # Significant between healthy and exposed
```
[49] "OG0011132"
```{r}
OG0011132 <- aov(results$OG0011132 ~ results$status, data = results)
summary(OG0011132) 
TukeyHSD(OG0011132) # Significant between healthy and disease
```
[50] "OG0011136"
```{r}
OG0011136 <- aov(results$OG0011136 ~ results$status, data = results)
summary(OG0011136) 
TukeyHSD(OG0011136) # Significant between healthy and disease
```
[51] "OG0011140"
```{r}
OG0011140 <- aov(results$OG0011140 ~ results$status, data = results)
summary(OG0011140) 
TukeyHSD(OG0011140) # Significant between healthy and disease
```
[52] "OG0011223"
```{r}
OG0011223 <- aov(results$OG0011223 ~ results$status, data = results)
summary(OG0011223) 
TukeyHSD(OG0011223) # Significant between healthy and disease
```
[53] "OG0011246"
```{r}
OG0011246 <- aov(results$OG0011246 ~ results$status, data = results)
summary(OG0011246) 
TukeyHSD(OG0011246) # Significant between exposed and disease
```
[54] "OG0011249"
```{r}
OG0011249 <- aov(results$OG0011249 ~ results$status, data = results)
summary(OG0011249) 
TukeyHSD(OG0011249) # Significant between healthy and disease
```
[55] "OG0011283"
```{r}
OG0011283 <- aov(results$OG0011283 ~ results$status, data = results)
summary(OG0011283) 
TukeyHSD(OG0011283) # Significant between healthy and disease; exposed and disease
```
[56] "OG0011345"
```{r}
OG0011345 <- aov(results$OG0011345 ~ results$status, data = results)
summary(OG0011345) 
TukeyHSD(OG0011345) # Significant between healthy and disease; exposed and disease
```
[57] "OG0011346"
```{r}
OG0011346 <- aov(results$OG0011346 ~ results$status, data = results)
summary(OG0011346) 
TukeyHSD(OG0011346) # Significant between healthy and disease
```
[58] "OG0011380"
```{r}
OG0011380 <- aov(results$OG0011380 ~ results$status, data = results)
summary(OG0011380) 
TukeyHSD(OG0011380) # Significant between exposed and disease
```
[59] "OG0011444"
```{r}
OG0011444 <- aov(results$OG0011444 ~ results$status, data = results)
summary(OG0011444) 
TukeyHSD(OG0011444) # Significant between healthy and disease
```
[60] "OG0011563"
```{r}
OG0011563 <- aov(results$OG0011563 ~ results$status, data = results)
summary(OG0011563) 
TukeyHSD(OG0011563) # Significant between healthy and disease
```
[61] "OG0011572"
```{r}
OG0011572 <- aov(results$OG0011572 ~ results$status, data = results)
summary(OG0011572) 
TukeyHSD(OG0011572) # Significant between healthy and disease; exposed and disease
```
[62] "OG0011610"
```{r}
OG0011610 <- aov(results$OG0011610 ~ results$status, data = results)
summary(OG0011610) 
TukeyHSD(OG0011610) # Significant between healthy and disease; healthy and exposed
```
[63] "OG0011693"
```{r}
OG0011693 <- aov(results$OG0011693 ~ results$status, data = results)
summary(OG0011693) 
TukeyHSD(OG0011693) # Significant between healthy and disease
```
[64] "OG0011783"
```{r}
OG0011783 <- aov(results$OG0011783 ~ results$status, data = results)
summary(OG0011783) 
TukeyHSD(OG0011783) # Significant between healthy and disease; exposed and disease
```
[65] "OG0011934"
```{r}
OG0011934 <- aov(results$OG0011934 ~ results$status, data = results)
summary(OG0011934) 
TukeyHSD(OG0011934) # Significant between healthy and disease; exposed and disease
```
[66] "OG0011988"
```{r}
OG0011988 <- aov(results$OG0011988 ~ results$status, data = results)
summary(OG0011988) 
TukeyHSD(OG0011988) # Significant between healthy and exposed
```
[67] "OG0012035"
```{r}
OG0012035 <- aov(results$OG0012035 ~ results$status, data = results)
summary(OG0012035) 
TukeyHSD(OG0012035) # Significant between healthy and disease; healthy and exposed
```
[68] "OG0012049"
```{r}
OG0012049 <- aov(results$OG0012049 ~ results$status, data = results)
summary(OG0012049) 
TukeyHSD(OG0012049) # Significant between healthy and exposed
```
[69] "OG0012059"
```{r}
OG0012059 <- aov(results$OG0012059 ~ results$status, data = results)
summary(OG0012059) 
TukeyHSD(OG0012059) # Significant between healthy and disease
```
[70] "OG0012131"
```{r}
OG0012131 <- aov(results$OG0012131 ~ results$status, data = results)
summary(OG0012131) 
TukeyHSD(OG0012131) # Significant between healthy and disease
```
[71] "OG0012135"
```{r}
OG0012135 <- aov(results$OG0012135 ~ results$status, data = results)
summary(OG0012135) 
TukeyHSD(OG0012135) # Significant between healthy and disease
```
[72] "OG0012234"
```{r}
OG0012234 <- aov(results$OG0012234 ~ results$status, data = results)
summary(OG0012234) 
TukeyHSD(OG0012234) # Significant between healthy and disease
```
[73] "OG0012259"
```{r}
OG0012259 <- aov(results$OG0012259 ~ results$status, data = results)
summary(OG0012259) 
TukeyHSD(OG0012259) # Significant between healthy and disease
```
[74] "OG0012265"
```{r}
OG0012265 <- aov(results$OG0012265 ~ results$status, data = results)
summary(OG0012265) 
TukeyHSD(OG0012265) # Significant between healthy and disease; exposed and disease
```
[75] "OG0012277"
```{r}
OG0012277 <- aov(results$OG0012277 ~ results$status, data = results)
summary(OG0012277) 
TukeyHSD(OG0012277) # Significant between healthy and exposed
```
[76] "OG0012396"
```{r}
OG0012396 <- aov(results$OG0012396 ~ results$status, data = results)
summary(OG0012396) 
TukeyHSD(OG0012396) # Significant between healthy and disease; exposed and disease
```
[77] "OG0012472"
```{r}
OG0012472 <- aov(results$OG0012472 ~ results$status, data = results)
summary(OG0012472) 
TukeyHSD(OG0012472) # Significant between healthy and disease
```
[78] "OG0012839"
```{r}
OG0012839 <- aov(results$OG0012839 ~ results$status, data = results)
summary(OG0012839) 
TukeyHSD(OG0012839) # Significant between healthy and disease
```

#### Significant 

Make a dataframe including only the 78 significant highly variable orthologs

```{r,results='hide',tidy=TRUE}
names(results)
sig_orthologs <- select(results,c(OG0009610,OG0009628,OG0009900,OG0009952,OG0009999,OG0010018,OG0010040,OG0010117,OG0010145,OG0010176,OG0010177,OG0010196,OG0010223,OG0010270,OG0010317,OG0010335,OG0010347,OG0010377,OG0010385,OG0010386,OG0010389,OG0010405,OG0010416,OG0010441,OG0010453,OG0010458,OG0010466,OG0010477,OG0010479,OG0010517,OG0010529,OG0010687,OG0010700,OG0010712,OG0010721,OG0010742,OG0010781,OG0010806,OG0010856,OG0010888,OG0010904,OG0010962,OG0011004,OG0011009,OG0011043,OG0011062,OG0011066,OG0011070,OG0011132,OG0011136,OG0011140,OG0011223,OG0011246,OG0011249,OG0011283,OG0011345,OG0011346,OG0011380,OG0011444,OG0011563,OG0011572,OG0011610,OG0011693,OG0011783,OG0011934,OG0011988,OG0012035,OG0012049,OG0012059,OG0012131,OG0012135,OG0012234,OG0012259,OG0012265,OG0012277,OG0012396,OG0012472,OG0012839))

write.csv(sig_orthologs,file="./all/EVE/sig_HV_orthologs.csv")

dim(sig_orthologs)
t_sig_orthologs <- as.data.frame(t(sig_orthologs))
label <- read.csv("./all/metaData.csv")
names(label)
label <- label[-c(53),] # remove pstr_d8
names(t_sig_orthologs) <- label$name
dim(t_sig_orthologs)

mcav_annot <- read.csv("./all/Orthofinder/mcav_sc_orthologs_annotated.csv")
mcav_contig <- read.csv("./all/Orthofinder/mcav_contig_to_orthogroup.csv")
mcav_contig <- mcav_contig[,-c(1)]
colnames(mcav_contig)[colnames(mcav_contig)=="mcav_ref_proteome_cdhit"] <- "mcav_contig"
mcav_info <- merge(mcav_annot,mcav_contig,by="mcav_contig")
t_sig_orthologs <- tibble::rownames_to_column(t_sig_orthologs, "Orthogroup")

results <- left_join(t_sig_orthologs,mcav_info,by="Orthogroup")
dim(results)
# [1] 78 57
# 78 HV genes significantly differentially expressed between disease states
uniprot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/uniprot.csv")
annot <- left_join(results,uniprot,by="Entry")
colnames(annot)
annot <- annot[,-c(58)]
colnames(annot)[56] <- "Entry.name"
annot <- annot %>% relocate(mcav_contig, .after = Orthogroup)
annot <- annot %>% relocate(Entry, .after = mcav_contig)
annot <- annot %>% relocate(Entry.name, .after = Entry)
annot <- annot %>% relocate(Evalue, .after = Entry.name)
annot <- annot %>% relocate(Gene.names, .after = Evalue)
annot <- annot %>% relocate(Protein.names, .after = Gene.names)
annot <- annot %>% relocate(Organism, .after = Protein.names)
annot <- annot %>% relocate(Gene.ontology..GO., .after = Organism)

write.csv(annot,file="./all/EVE/sig_HV_annot.csv",row.names = FALSE)
```
Add the Tukey p value results to this file in 3 columns: EvI,CvI and CvE.
Rename this file "sig_HV_annot_ANOVA.csv"
```{r,results='hide',tidy=TRUE}
sig_HV_annot_ANOVA.csv <- read.csv("./all/EVE/sig_HV_annot_ANOVA.csv")
goodEval <- sig_HV_annot_ANOVA.csv %>% filter(sig_HV_annot_ANOVA.csv$Evalue < 0.000001)
dim(goodEval)
# 53 genes significantly differentially expressed between disease states with good evalue
write.csv(goodEval,file="./all/EVE/sig_HV_annot_ANOVA_goodEval.csv",row.names = FALSE)
```

##### Heatmap

###### P ≤ 0.05

Plot the average expression of all highly variable orthologs with significant differential expression between disease states (p ≤ 0.05)

```{r,results='hide',tidy=TRUE}
HV_genes <- read.csv("./all/EVE/highlyVariable_rlogs_ortholog.csv")
sig_orthologs <- select(HV_genes,c(name,status,OG0012259,OG0011004,OG0011070,OG0011066,OG0010040,OG0010712,OG0009952,OG0012131,OG0012472,OG0011444,OG0011693,OG0012839,OG0011988,OG0010529,OG0012234,OG0012135,OG0010888,OG0010177,OG0010517,OG0011246,OG0012396,OG0012059,OG0010721,OG0010904,OG0010386,OG0009628,OG0011572,OG0011380,OG0011345,OG0012035,OG0012049,OG0010453,OG0010176,OG0010962,OG0010458,OG0010806,OG0010117,OG0010385,OG0009999,OG0010405,OG0010317,OG0010196,OG0012265,OG0011563,OG0011934,OG0010687,OG0011062,OG0010466,OG0011283,OG0012277,OG0010477,OG0010223,OG0010335,OG0010479,OG0010856))

dim(sig_orthologs)
sig_orthologs_ordered <- sig_orthologs[order(sig_orthologs$status),]
sig_orthologs_ordered$status
HV_sig_infected <- sig_orthologs_ordered[c(1:19),-c(1:2)]
HV_sig_exposed <- sig_orthologs_ordered[c(20:27),-c(1:2)]
HV_sig_control <- sig_orthologs_ordered[c(28:52),-c(1:2)]

HV_sig_control_avg <- colMeans(HV_sig_control)
HV_sig_exposed_avg <- colMeans(HV_sig_exposed)
HV_sig_infected_avg <- colMeans(HV_sig_infected)
HV_sig_genes_avg <- rbind(HV_sig_control_avg,HV_sig_exposed_avg,HV_sig_infected_avg)
dim(HV_sig_genes_avg)
row.names(HV_sig_genes_avg) <- c("control","exposed","infected")
t_HV_sig_genes_avg <- data.frame(t(HV_sig_genes_avg))
```

```{r heatmap_1,fig.height=6,fig.width=3,dpi=200}
pheatmap(t_HV_sig_genes_avg,
         scale="row",
         cluster_cols = FALSE,
         cluster_rows = TRUE,
         cellwidth = 10,
         cellheight = 8,
         color = colorRampPalette(c("#0571b0","#92c5de","white","#f4a582","#ca0020"))(200),
         treeheight_row = 20,
         treeheight_col = NULL,
         fontsize_row = 6,
         fontsize_col = 6
)
```
###### P ≤ 0.01
Plot the average expression of all highly variable orthologs with significant differential expression between disease states (p ≤ 0.01)

```{r,results='hide',tidy=TRUE}
HV_genes <- read.csv("./all/EVE/highlyVariable_rlogs_ortholog.csv")
sig_orthologs <- select(HV_genes,c(name,status,OG0011062,OG0010466,OG0011246,OG0012396,OG0011283,OG0010806,OG0010177,OG0011345,OG0010904,OG0012265,OG0011934,OG0011066,OG0012839,OG0010117,OG0010405,OG0010040,OG0010517,OG0010176,OG0012059,OG0010477,OG0010453,OG0012131,OG0010458,OG0011693,OG0010962,OG0012035,OG0012234,OG0011444,OG0009999,OG0010712,OG0012259))

dim(sig_orthologs)
sig_orthologs_ordered <- sig_orthologs[order(sig_orthologs$status),]
sig_orthologs_ordered$status
HV_sig_infected <- sig_orthologs_ordered[c(1:19),-c(1:2)]
HV_sig_exposed <- sig_orthologs_ordered[c(20:27),-c(1:2)]
HV_sig_control <- sig_orthologs_ordered[c(28:52),-c(1:2)]

HV_sig_control_avg <- colMeans(HV_sig_control)
HV_sig_exposed_avg <- colMeans(HV_sig_exposed)
HV_sig_infected_avg <- colMeans(HV_sig_infected)
HV_sig_genes_avg <- rbind(HV_sig_control_avg,HV_sig_exposed_avg,HV_sig_infected_avg)
dim(HV_sig_genes_avg)
row.names(HV_sig_genes_avg) <- c("control","exposed","infected")
t_HV_sig_genes_avg <- data.frame(t(HV_sig_genes_avg))
```

```{r heatmap_sig,fig.height=4,fig.width=3,dpi=200}
pheatmap(t_HV_sig_genes_avg,
         scale = "row",
         cluster_cols = FALSE,
         cluster_rows = TRUE,
         color = colorRampPalette(c("#0571b0","#92c5de","white","#f4a582","#ca0020"))(200),
         cellwidth = 10,
         cellheight = 8,
         treeheight_row = 20,
         treeheight_col = NULL,
         fontsize_row = 6
         )
```

##### Boxplots

```{r,results='hide',tidy=TRUE}
results <- read.csv("./all/EVE/sig_HV_annot_ANOVA.csv")
names(results)
dim(results)    # 78 significant Highly Variable EVE genes

resorderd <-results[order(results$CvI),] 
resorderd <- resorderd %>% filter(CvI < (0.05)) 
dim(resorderd)  #66 significantly different between control and infected

resorderd <-results[order(results$CvE),] 
resorderd <- resorderd %>% filter(CvE < (0.05)) 
dim(resorderd)  #11 significantly different between control and exposed

resorderd <-results[order(results$EvI),] 
resorderd <- resorderd %>% filter(EvI < (0.05)) 
dim(resorderd)  #19 significantly different between exposed and infected

names(results)
results <- results[,-c(2:12)]
t_resordered_rlogs <- setNames(as.data.frame(t(results[,-1])), results[,1])
row.names(t_resordered_rlogs)
t_resordered_rlogs <- tibble::rownames_to_column(t_resordered_rlogs, "name")
view(t_resordered_rlogs)


label <- read.csv("./all/metadata.csv")
label <- label[-c(53),] # remove pstr_d8
colnames(label)
label <- label[,c(1,3,5)] # keep species,name, and status
results <- merge(label,t_resordered_rlogs,by="name")
view(results)


results$status <- gsub("healthy","control", results$status)
results$status <- gsub("disease","infected", results$status)
results$status <- factor(results$status, levels = c("control","exposed","infected"))
results$species <- factor(results$species, levels = c("mcav","past","pstr","oann","cnat"))
```

###### Tpm4
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0010466 <- compare_means(OG0010466~status,data=results,method = "t.test")

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0010466_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0010466, fill = status)) +
  geom_point(aes(y = OG0010466, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Tropomyosin alpha-4 chain") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0010466, y.position = 12, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0010466_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0010466, fill = status)) +
  geom_point(aes(y = OG0010466, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Tropomyosin alpha-4 chain") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0010466, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

ggarrange(OG0010466_species,OG0010466_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```

###### FOXL2

```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0010177 <- compare_means(OG0010177~status,data=results,method = "t.test")

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0010177_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0010177, fill = status)) +
  geom_point(aes(y = OG0010177, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Forkhead box protein L2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0010177, y.position = 10, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0010177_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0010177, fill = status)) +
  geom_point(aes(y = OG0010177, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Forkhead box protein L2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0010177, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

ggarrange(OG0010177_species,OG0010177_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```

###### Klhl18 
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0011934 <- compare_means(OG0011934~status,data=results,method = "t.test")

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0011934_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0011934, fill = status)) +
  geom_point(aes(y = OG0011934, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Kelch-like protein 18") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0011934, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0011934_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0011934, fill = status)) +
  geom_point(aes(y = OG0011934, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Kelch-like protein 18") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0011934, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

ggarrange(OG0011934_species,OG0011934_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```

###### ADAMTS7 
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0012265 <- compare_means(OG0012265~status,data=results,method = "t.test")

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0012265_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0012265, fill = status)) +
  geom_point(aes(y = OG0012265, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "ADAM-TS 7") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0012265, y.position = 12.7, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0012265_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0012265, fill = status)) +
  geom_point(aes(y = OG0012265, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "ADAM-TS 7") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0012265, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

ggarrange(OG0012265_species,OG0012265_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```

###### Pacrg
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0011066 <- compare_means(OG0011066~status,data=results,method = "t.test")

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0011066_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0011066, fill = status)) +
  geom_point(aes(y = OG0011066, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Parkin coregulated gene protein homolog") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0011066, y.position = 11, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0011066_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0011066, fill = status)) +
  geom_point(aes(y = OG0011066, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Parkin coregulated gene protein homolog") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0011066, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

ggarrange(OG0011066_species,OG0011066_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```

###### P4HTM
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0010904 <- compare_means(OG0010904~status,data=results,method = "t.test")

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0010904_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0010904, fill = status)) +
  geom_point(aes(y = OG0010904, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Transmembrane prolyl 4-hydroxylase, P4H-TM") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0010904, y.position = 10, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0010904_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0010904, fill = status)) +
  geom_point(aes(y = OG0010904, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Transmembrane prolyl 4-hydroxylase, P4H-TM") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0010904, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

ggarrange(OG0010904_species,OG0010904_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```

###### NSMCE4A
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0012396 <- compare_means(OG0012396~status,data=results,method = "t.test")

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0012396_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0012396, fill = status)) +
  geom_point(aes(y = OG0012396, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Non-SMC element 4 homolog A") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0012396, y.position = 8.3, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0012396_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0012396, fill = status)) +
  geom_point(aes(y = OG0012396, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Non-SMC element 4 homolog A") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0012396, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

ggarrange(OG0012396_species,OG0012396_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```

###### Pdcl3
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0012839 <- compare_means(OG0012839~status,data=results,method = "t.test")

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0012839_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0012839, fill = status)) +
  geom_point(aes(y = OG0012839, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Phosducin-like protein 3") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0012839, y.position = 8.9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0012839_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0012839, fill = status)) +
  geom_point(aes(y = OG0012839, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Phosducin-like protein 3") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0012839, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

ggarrange(OG0012839_species,OG0012839_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```

###### ELK1
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0010405 <- compare_means(OG0010405~status,data=results,method = "t.test")

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0010405_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0010405, fill = status)) +
  geom_point(aes(y = OG0010405, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "ETS domain-containing protein Elk-1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0010405, y.position = 11, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0010405_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0010405, fill = status)) +
  geom_point(aes(y = OG0010405, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "ETS domain-containing protein Elk-1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0010405, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

ggarrange(OG0010405_species,OG0010405_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```

###### ADORA2A
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0011062 <- compare_means(OG0011062~status,data=results,method = "t.test")

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0011062_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0011062, fill = status)) +
  geom_point(aes(y = OG0011062, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Adenosine receptor A2a") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0011062, y.position = 9.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0011062_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0011062, fill = status)) +
  geom_point(aes(y = OG0011062, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Adenosine receptor A2a") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0011062, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

ggarrange(OG0011062_species,OG0011062_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```

###### Folh1
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0012035 <- compare_means(OG0012035~status,data=results,method = "t.test")

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0012035_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0012035, fill = status)) +
  geom_point(aes(y = OG0012035, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Glutamate carboxypeptidase 2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0012035, y.position = 11.3, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0012035_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0012035, fill = status)) +
  geom_point(aes(y = OG0012035, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Glutamate carboxypeptidase 2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0012035, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

ggarrange(OG0012035_species,OG0012035_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```

###### Irf2bp1

```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0012131 <- compare_means(OG0012131~status,data=results,method = "t.test")

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0012131_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0012131, fill = status)) +
  geom_point(aes(y = OG0012131, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Interferon regulatory factor 2-binding protein 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0012131, y.position = 11.3, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0012131_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0012131, fill = status)) +
  geom_point(aes(y = OG0012131, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Interferon regulatory factor 2-binding protein 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0012131, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

ggarrange(OG0012131_species,OG0012131_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```

###### BTG1
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0010176 <- compare_means(OG0010176~status,data=results,method = "t.test")

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0010176_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0010176, fill = status)) +
  geom_point(aes(y = OG0010176, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Protein BTG1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0010176, y.position = 12, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0010176_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0010176, fill = status)) +
  geom_point(aes(y = OG0010176, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Protein BTG1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0010176, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

ggarrange(OG0010176_species,OG0010176_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")
```

###### Together
```{r allBoxplots, fig.height=14,fig.width=6,dpi=200}
ggarrange(OG0010466_species,OG0010466_status,
          OG0011066_species,OG0011066_status,
          OG0010904_species,OG0010904_status,
          OG0012265_species,OG0012265_status,
          OG0011062_species,OG0011062_status,
          OG0010176_species,OG0010176_status,
  ncol=2, nrow=6,widths = c(1.75,1.25),
  common.legend = TRUE, legend = "right",align = "hv")
```

#### Venn Diagram

```{r,results='hide',tidy=TRUE}
sharedHV <- read.csv("./all/EVE/sig_HV_annot_ANOVA.csv")

EvI <-sharedHV[order(sharedHV$EvI),] 
EvI <- EvI %>% filter(EvI <= 0.05)
dim(EvI)  #19

CvI <-sharedHV[order(sharedHV$CvI),] 
CvI <- CvI %>% filter(CvI <= 0.05)
dim(CvI)  #66

CvE <-sharedHV[order(sharedHV$CvE),] 
CvE <- CvE %>% filter(CvE <= 0.05)
dim(CvE)  #11

x <- list(
  EvI = EvI$Orthogroup,
  CvI = CvI$Orthogroup,
  CvE = CvE$Orthogroup
  )
```

```{r HV_vennDiagram, fig.height=3,fig.width=3,dpi=200}
ggvenn(
  x,
  fill_color = c("#2ECFCA", "#F8DF77", "#FF4E62"),
  show_percentage = FALSE,
  stroke_size = 0.3, set_name_size = 3
)
```

## Lineage Specific Genes

```{r,results='hide',tidy=TRUE}
sig_EVE <- read.csv("./all/EVE/sig_EVE_rlogs.csv")
LineageGenes <- subset(sig_EVE, sig_EVE$category == "Lineage Specific")
names(LineageGenes)
dim(LineageGenes)  # 84 57
LineageGenes <- LineageGenes[,-c(1,3:5)]

#Average the rlog counts across species
colnames(LineageGenes)
LineageGenes$cnat <- rowMeans(LineageGenes[,c(2:11)])
LineageGenes$mcav <- rowMeans(LineageGenes[,c(12:24)])
LineageGenes$oann <- rowMeans(LineageGenes[,c(25:35)])
LineageGenes$past <- rowMeans(LineageGenes[,c(36:46)])
LineageGenes$pstr <- rowMeans(LineageGenes[,c(47:53)])
names(LineageGenes)
LSavg <- LineageGenes[,c(1,54:58)]
names(LSavg)
write.csv(LSavg,file="./all/EVE/lineage_species_avg.csv")


mcav_annot <- read.csv("./all/Orthofinder/mcav_sc_orthologs_annotated.csv")
mcav_contig <- read.csv("./all/Orthofinder/mcav_contig_to_orthogroup.csv")
mcav_contig <- mcav_contig[,-c(1)]
colnames(mcav_contig)[colnames(mcav_contig)=="mcav_ref_proteome_cdhit"] <- "mcav_contig"
mcav_info <- merge(mcav_annot,mcav_contig,by="mcav_contig")
names(mcav_info)
names(LSavg)
LSavg <- left_join(LSavg,mcav_info,by="Orthogroup")

LSavg <- LSavg %>% relocate(mcav_contig, .after = Orthogroup)
LSavg <- LSavg %>% relocate(Entry, .after = mcav_contig)
LSavg <- LSavg %>% relocate(Entry.name, .after = Entry)
LSavg <- LSavg %>% relocate(Evalue, .after = Entry.name)

write.csv(LSavg,file="./all/EVE/sig_LS_avg_annot.csv")

goodEval <- LSavg %>% filter(LSavg$Evalue < 0.000001)
write.csv(goodEval,file="./all/EVE/sig_LS_avg_annot_goodEval.csv")
```

### Correlate to Relative Risk
```{r,tidy=TRUE}
LSavg <- read.csv("./all/EVE/sig_LS_avg_annot_goodEval.csv",row.names = "Entry")
names(LSavg)
LSavg <- LSavg[,-c(1:5)]
LSavg <- LSavg %>% relocate(oann, .after = cnat)
LSavg <- LSavg %>% relocate(pstr, .after = oann)
LSavg <- LSavg %>% relocate(past, .after = pstr)

LS_rlogs <- as.data.frame(t(LSavg))
rownames(LS_rlogs)

Species <- c("cnat","oann","pstr","past","mcav")
Median_RR <- c(12.19,12.09,9.3,7.91,5.1)
Avg_LGR <- c(0.42,0.197214741,0.136086957,0.027850797,0.087824786)
label <- data.frame(Species,Median_RR,Avg_LGR)
label <- tibble::column_to_rownames(label,"Species")

rownames(label)
rownames(LS_rlogs)

LS_rlogs$Species <- rownames(LS_rlogs)
LS_rlogs$RelativeRisk <- label$Median_RR
names(LS_rlogs)
LS_rlogs <- LS_rlogs %>% relocate(Species, .before = everything())
LS_rlogs <- LS_rlogs %>% relocate(RelativeRisk, .after = Species)

##### Correlate to Relative Risk

# Loop through all tests, print estimates and p-values if p <=0.05
for (i in 3:length(LS_rlogs)){
   a <- cor.test(LS_rlogs$RelativeRisk, LS_rlogs[,i],method = c("pearson"))
   if (a$p.value <= 0.05){
      print(paste(colnames(LS_rlogs)[i], " corr:", a$estimate, " p-value:", a$p.value))
   }
}
```

##### Scatterplots

```{r,tidy=TRUE}
# Plot the genes
LS_rlogs$Species <- factor(LS_rlogs$Species, levels = c("cnat","oann","pstr","past","mcav"))

Q96LD1 <- 
   ggplot(LS_rlogs,aes(x=RelativeRisk, y=Q96LD1))+
   geom_point(aes(x=RelativeRisk, y=Q96LD1, color=Species), position = "identity",size=3.5)+
   geom_smooth(method=glm, color="black")+
   #geom_text(label=LS_rlogs$species, nudge_y = -0.5)+
   labs(y="Mean Expression", x = "Relative Risk", title = "Zeta-sarcoglycan")+
   stat_cor(label.x.npc = "left",label.y = 10.1)

Q4R5P6 <- 
   ggplot(LS_rlogs,aes(x=RelativeRisk, y=Q4R5P6))+
   geom_point(aes(x=RelativeRisk, y=Q4R5P6, color=Species), position = "identity",size=3)+
   geom_smooth(method=glm, color="black")+
   #geom_text(label=row.names(LS_rlogs), nudge_y = -0.5, size =3 )+
   labs(y="Mean Expression", x = "Relative Risk", title = "Cytoplasmic dynein 1 light intermediate chain 2")+
   stat_cor(label.x.npc = "left",label.y = 10.5)
   
Q6NWF4 <- 
   ggplot(LS_rlogs,aes(x=RelativeRisk, y=Q6NWF4))+
   geom_point(aes(x=RelativeRisk, y=Q6NWF4, color=Species), position = "identity",size=3)+
   geom_smooth(method=glm, color="black")+
   #geom_text(label=row.names(LS_rlogs), nudge_y = -0.5,size=3)+
   labs(y="Mean Expression", x = "Relative Risk", title = "Vacuolar protein-sorting-associated protein 25")+
   stat_cor(label.x.npc = "left",label.y = 8.3)
```

### Correlate to LGR
```{r,tidy=TRUE}
LSavg <- read.csv("./all/EVE/sig_LS_avg_annot_goodEval.csv",row.names = "Entry")
names(LSavg)
LSavg <- LSavg[,-c(1:5)]
LSavg <- LSavg %>% relocate(oann, .after = cnat)
LSavg <- LSavg %>% relocate(pstr, .after = oann)
LSavg <- LSavg %>% relocate(past, .after = pstr)

LS_rlogs <- as.data.frame(t(LSavg))
rownames(LS_rlogs)

Species <- c("cnat","oann","pstr","past","mcav")
Median_RR <- c(12.19,12.09,9.3,7.91,5.1)
Avg_LGR <- c(0.42,0.197214741,0.136086957,0.027850797,0.087824786)
label <- data.frame(Species,Median_RR,Avg_LGR)
label <- tibble::column_to_rownames(label,"Species")

rownames(label)
rownames(LS_rlogs)

LS_rlogs$Species <- rownames(LS_rlogs)
LS_rlogs$LGR <- label$Avg_LGR
LS_rlogs <- LS_rlogs %>% relocate(Species, .before = everything())
LS_rlogs <- LS_rlogs %>% relocate(LGR, .after = Species)

##### Correlate to Lesion Growth Rate

# Loop through all tests, print estimates and p-values if p <=0.05
for (i in 3:length(LS_rlogs)){
   a <- cor.test(LS_rlogs$LGR, LS_rlogs[,i],method = c("pearson"))
   if (a$p.value <= 0.05){
      print(paste(colnames(LS_rlogs)[i], " corr:", a$estimate, " p-value:", a$p.value))
   }
}
```

##### Scatterplots
```{r,tidy=TRUE}
LS_rlogs$Species <- factor(LS_rlogs$Species, levels = c("cnat","oann","pstr","past","mcav"))

Q63ZP1 <- 
   ggplot(LS_rlogs,aes(x=LGR, y=Q63ZP1))+
   geom_point(aes(x=LGR, y=Q63ZP1, color=Species), position = "identity",size=3)+
   geom_smooth(method=glm, color="black")+
   #geom_text(label=row.names(LS_rlogs), nudge_y = -0.5)+
   labs(y="Mean Expression", x = "Avg. Lesion Growth Rate", title = "PHD finger protein 10")+
   stat_cor(label.x.npc = "left",label.y = 3.7)

Q8VEE0 <- 
   ggplot(LS_rlogs,aes(x=LGR, y=Q8VEE0))+
   geom_point(aes(x=LGR, y=Q8VEE0, color=Species), position = "identity",size=3)+
   geom_smooth(method=glm, color="black")+
   #geom_text(label=row.names(LS_rlogs), nudge_y = -0.5)+
   labs(y="Mean Expression", x = "Avg. Lesion Growth Rate", title = "Ribulose-phosphate 3-epimerase")+
   stat_cor(label.x.npc = "left",label.y = .8)

Q8VBV7 <- 
   ggplot(LS_rlogs,aes(x=LGR, y=Q8VBV7))+
   geom_point(aes(x=LGR, y=Q8VBV7, color=Species), position = "identity",size=3)+
   geom_smooth(method=glm, color="black")+
   #geom_text(label=row.names(LS_rlogs), nudge_y = -0.5)+
   labs(y="Mean Expression", x = "Avg. Lesion Growth Rate", title = "COP9 signalosome complex subunit 8")+
   stat_cor(label.x.npc = "left",label.y = 6.5)

Q9D428 <- 
   ggplot(LS_rlogs,aes(x=LGR, y=Q9D428))+
   geom_point(aes(x=LGR, y=Q9D428, color=Species), position = "identity",size=3)+
   geom_smooth(method=glm, color="black")+
#   geom_text(label=row.names(LS_rlogs), nudge_y = -0.5)+
   labs(y="Mean Expression", x = "Avg. Lesion Growth Rate", title = "Golgin subfamily A member 7B")+
   stat_cor(label.x.npc = "left",label.y = .2)

Q8TBH0 <- 
   ggplot(LS_rlogs,aes(x=LGR, y=Q8TBH0))+
   geom_point(aes(x=LGR, y=Q8TBH0, color=Species), position = "identity",size=4)+
   geom_smooth(method=glm, color="black")+
#   geom_text(label=row.names(LS_rlogs), nudge_y = -0.5)+
   labs(y="Mean Expression", x = "Avg. Lesion Growth Rate", title = "Arrestin domain-containing protein 2")+
   stat_cor(label.x.npc = "left",label.y = 9.3)
```

```{r LGR,fig.height=8,fig.width=7,dpi=200}  
ggarrange(Q96LD1,Q4R5P6,Q6NWF4,
          Q63ZP1,Q8VEE0,Q8VBV7,Q9D428,Q8TBH0,
          ncol = 3, nrow = 3, legend = "none",
          common.legend = FALSE,
          align = "hv"
          )
```

#### Heatmaps

```{r,tidy=TRUE}
LSavg <- read.csv("./all/EVE/sig_LS_avg_annot_goodEval.csv",row.names = "Entry")
names(LSavg)
LSavg <- LSavg[,-c(1:5)]
LSavg <- LSavg %>% relocate(oann, .after = cnat)
LSavg <- LSavg %>% relocate(pstr, .after = oann)
LSavg <- LSavg %>% relocate(past, .after = pstr)
write.csv(LSavg,file="./all/EVE/LS_species_avg_entry.csv",row.names = FALSE)

breaksList=seq(0,12,by=1)
```
Plot Relative Risk Heatmap
```{r RR_Heatmap,fig.height=6,fig.width=6,dpi=200}
LSavg <- LSavg[,c("mcav","past","pstr","oann","cnat")]
RRGenes <- LSavg[c("Q96LD1","Q4R5P6","Q6NWF4"),]
pheatmap(RRGenes,
         cluster_cols = FALSE,
         cluster_rows = TRUE,
         scale = "none",
         color = colorRampPalette(c("white","#fed976","#fc4e2a","#67000d"))(12),
         border_color = "grey50",
         annotation_col = label,
         legend = TRUE,
         breaks = breaksList,
         legend_labels = label,
         fontsize_row = 7,
         fontsize_col = 10,
         cellwidth = 20,
         cellheight = 20
         )
```

Plot Lesion Growth Rate Heatmap
```{r LGR_Heatmap,fig.height=6,fig.width=6,dpi=200}
LSavg <- LSavg[,c("past","mcav","pstr","oann","cnat")]
LGRgenes <- LSavg[c("Q63ZP1","Q8VEE0","Q8VBV7","Q9D428","Q8TBH0"),]
pheatmap(LGRgenes,
         cluster_cols = FALSE,
         cluster_rows = TRUE,
         scale = "none",
         color = colorRampPalette(c("white","#fed976","#fc4e2a","#67000d"))(12),
         border_color = "grey50",
         annotation_col = label,
         legend = TRUE,
         breaks = breaksList,
         legend_labels = label,
         fontsize_row = 7,
         fontsize_col = 10,
         cellwidth = 20,
         cellheight = 20
         )
```
