---
title: "Coral Orthogroup Analysis"
author: "Kelsey Beavers"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/")
knitr::opts_chunk$set(dev='ragg_png')
library(ggpubr)
library(DESeq2)
library(tibble)
library(tidyverse)
library(tximport)
library(GOplot)
library(dplyr)
library(tidyr)
library(reshape2)
library(ggalt)
library(ggplot2)
library(lifecycle)
library(stringr)
library(utils)
library(corrplot)
library(rstatix)
library(repr) 
library(data.table)
```

## Step 1: Obtain Orthologs from Orthofinder

To obtain orthologs, we filter our Orthofinder results to only contain Orthogroups with at least one sequence per species. 

This can be done in excel like so:
1. Open Orthogroups.tsv in Excel
2. Click *Find & Select*
3. Click *Go to Special*
4. Choose *Blanks*
5. Click OK and then all the *blank rows/cells will be highlighted*
6. Choose the *Delete under Cells* section on the Home Tab
7. Click *Delete Sheet Rows*
8. Save as "shared_orthogroups.csv"

### Make tx2gene files using shared orthogroups

For this analysis we will use "Orthogroup_Sequences.csv" from the "Command_Line_Code.rmd" file to create new tx2gene files for each species

```{r}
Orthogroups <- read.csv("./all/Orthofinder/Orthogroups.csv")
cnat_orthogroups <- Orthogroups[,c(1,2)]
cnat_orthogroups <- cnat_orthogroups %>% separate_rows(cnat_ref_proteome_cdhit, sep = ", ")
cnat_orthogroups <- cnat_orthogroups[,c(2,1)]
colnames(cnat_orthogroups)[1] <- "Transcript"
cnat_orthogroups$Transcript <- gsub("\\..*","",cnat_orthogroups$Transcript)
write.csv(cnat_orthogroups,file="./cnat/salmon/cnat_orthogroup_tx2gene.csv",row.names = FALSE)

mcav_orthogroups <- Orthogroups[,c(1,3)]
mcav_orthogroups <- mcav_orthogroups %>% separate_rows(mcav_ref_proteome_cdhit, sep = ", ")
mcav_orthogroups <- mcav_orthogroups[,c(2,1)]
colnames(mcav_orthogroups)[1] <- "Transcript"
mcav_orthogroups$Transcript <- gsub("\\..*","",mcav_orthogroups$Transcript)
write.csv(mcav_orthogroups,file="./mcav/salmon/mcav_orthogroup_tx2gene.csv",row.names = FALSE)

oann_orthogroups <- Orthogroups[,c(1,4)]
oann_orthogroups <- oann_orthogroups %>% separate_rows(oann_ref_proteome_cdhit, sep = ", ")
oann_orthogroups <- oann_orthogroups[,c(2,1)]
colnames(oann_orthogroups)[1] <- "Transcript"
oann_orthogroups$Transcript <- gsub("\\..*","",oann_orthogroups$Transcript)
write.csv(oann_orthogroups,file="./oann/salmon/oann_orthogroup_tx2gene.csv",row.names = FALSE)

past_orthogroups <- Orthogroups[,c(1,5)]
past_orthogroups <- past_orthogroups %>% separate_rows(past_ref_proteome_cdhit, sep = ", ")
past_orthogroups <- past_orthogroups[,c(2,1)]
colnames(past_orthogroups)[1] <- "Transcript"
past_orthogroups$Transcript <- gsub("\\..*","",past_orthogroups$Transcript)
write.csv(past_orthogroups,file="./past/salmon/past_orthogroup_tx2gene.csv",row.names = FALSE)

pstr_orthogroups <- Orthogroups[,c(1,6)]
pstr_orthogroups <- pstr_orthogroups %>% separate_rows(pstr_ref_proteome_cdhit, sep = ", ")
pstr_orthogroups <- pstr_orthogroups[,c(2,1)]
colnames(pstr_orthogroups)[1] <- "Transcript"
pstr_orthogroups$Transcript <- gsub("\\..*","",pstr_orthogroups$Transcript)
write.csv(pstr_orthogroups,file="./pstr/salmon/pstr_orthogroup_tx2gene.csv",row.names = FALSE)
```

## Step 2: Count Matrix Generation

Read in the raw counts from Salmon for all coral samples and assign Orthogroups

### C. natans
```{r,results='hide',tidy=TRUE}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/cnat/salmon")
cnat_samples <- read.table("cnat_samples.csv", header = TRUE)
cnat_files <- file.path(cnat_samples$sample, "quant.sf")
names(cnat_files) <- paste0("sample", 1:10)
all(file.exists(cnat_files))
# Do not continue if output shows [FALSE]
cnat_tx2gene <- read.csv("cnat_orthogroup_tx2gene.csv")
cnat_txi <- tximport(cnat_files, type = 'salmon',tx2gene=cnat_tx2gene)
dim(cnat_txi$counts)
# 20,966 Orthogroups

cnat_counts <- cnat_txi$counts
cnat_counts <- as.data.frame(cnat_counts)
colnames(cnat_counts) <- c("cnat_c2","cnat_c3","cnat_c4","cnat_c6","cnat_c8","cnat_d2","cnat_d3","cnat_d4","cnat_d6","cnat_d8")
cnat_counts <- tibble::rownames_to_column(cnat_counts,"Orthogroup")
cnat_counts[2:11] <-  round(cnat_counts[2:11], digits=0)
cnat_counts <- as.data.frame(cnat_counts)
```

### M. cavernosa
```{r,results='hide',tidy=TRUE}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/mcav/salmon")
mcav_samples <- read.table("mcav_samples.csv", header = TRUE)
mcav_files <- file.path(mcav_samples$sample, "quant.sf")
names(mcav_files) <- paste0("sample", 1:13)
all(file.exists(mcav_files))
# Do not continue if output shows [FALSE]
mcav_tx2gene <- read.csv("mcav_orthogroup_tx2gene.csv")
mcav_txi <- tximport(mcav_files, type = 'salmon',tx2gene=mcav_tx2gene)
dim(mcav_txi$counts)
# 18,397 Orthogroups

mcav_counts <- mcav_txi$counts
mcav_counts <- as.data.frame(mcav_counts)
colnames(mcav_counts) <- c("mcav_c1","mcav_c3","mcav_c5","mcav_c6","mcav_c7","mcav_c8","mcav_d1","mcav_d2","mcav_d3","mcav_d4","mcav_d5","mcav_d6","mcav_d8")
mcav_counts <- tibble::rownames_to_column(mcav_counts,"Orthogroup")
mcav_counts[2:14] <-  round(mcav_counts[2:14], digits=0)
mcav_counts <- as.data.frame(mcav_counts)
```

### O. annularis
```{r,results='hide',tidy=TRUE}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/oann/salmon")
oann_samples <- read.table("oann_samples.csv", header = TRUE)
oann_files <- file.path(oann_samples$sample, "quant.sf")
names(oann_files) <- paste0("sample", 1:11)
all(file.exists(oann_files))
# Do not continue if output shows [FALSE]
oann_tx2gene <- read.csv("oann_orthogroup_tx2gene.csv")
oann_txi <- tximport(oann_files, type = 'salmon',tx2gene=oann_tx2gene)
dim(oann_txi$counts)
# 18,459 Orthogroups

oann_counts <- oann_txi$counts
oann_counts <- as.data.frame(oann_counts)
colnames(oann_counts) <- c("oann_c1","oann_c2","oann_c3","oann_c6","oann_c7","oann_c8","oann_d1","oann_d3","oann_d6","oann_d7","oann_d8")
oann_counts <- tibble::rownames_to_column(oann_counts,"Orthogroup")
oann_counts[2:12] <-  round(oann_counts[2:12], digits=0)
oann_counts <- as.data.frame(oann_counts)
```

### P. astreoides
```{r,results='hide',tidy=TRUE}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/past/salmon")
past_samples <- read.table("past_samples.csv", header = TRUE)
past_files <- file.path(past_samples$sample, "quant.sf")
names(past_files) <- paste0("sample", 1:11)
all(file.exists(past_files))
# Do not continue if output shows [FALSE]
past_tx2gene <- read.csv("past_orthogroup_tx2gene.csv")
past_txi <- tximport(past_files, type = 'salmon',tx2gene=past_tx2gene)
dim(past_txi$counts)
# 18,255 Orthogroups

past_counts <- past_txi$counts
past_counts <- as.data.frame(past_counts)
colnames(past_counts) <- c("past_c1","past_c5","past_c6","past_c7","past_d1","past_d2","past_d3","past_d4","past_d5","past_d6","past_d7")
past_counts <- tibble::rownames_to_column(past_counts,"Orthogroup")
past_counts[2:12] <-  round(past_counts[2:12], digits=0)
past_counts <- as.data.frame(past_counts)
```
### P. strigosa
```{r,results='hide',tidy=TRUE}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Experiment/host/pstr/salmon")
pstr_samples <- read.table("pstr_samples_outlier_removed.csv", header = TRUE)
pstr_files <- file.path(pstr_samples$sample, "quant.sf")
names(pstr_files) <- paste0("sample", 1:7)
all(file.exists(pstr_files))
# Do not continue if output shows [FALSE]
pstr_tx2gene <- read.csv("pstr_orthogroup_tx2gene.csv")
pstr_txi <- tximport(pstr_files, type = 'salmon',tx2gene=pstr_tx2gene)
dim(pstr_txi$counts)
# 14,621 Orthogroups

pstr_counts <- pstr_txi$counts
pstr_counts <- as.data.frame(pstr_counts)
colnames(pstr_counts) <- c("pstr_c3","pstr_c5","pstr_c7","pstr_c8","pstr_d3","pstr_d5","pstr_d7")
pstr_counts <- tibble::rownames_to_column(pstr_counts,"Orthogroup")
pstr_counts[2:8] <-  round(pstr_counts[2:8], digits=0)
pstr_counts <- as.data.frame(pstr_counts)
```

## Step 3: Analysis of Shared Orthogroups
### Merge the count data
```{r,results='hide',tidy=TRUE}
Cnat_Mcav <- merge(cnat_counts,mcav_counts, by ="Orthogroup")
Cnat_Mcav_Oann <- merge(Cnat_Mcav,oann_counts,by="Orthogroup")
Cnat_Mcav_Oann_Past <- merge(Cnat_Mcav_Oann,past_counts,by="Orthogroup")
all_species <- merge(Cnat_Mcav_Oann_Past,pstr_counts,by="Orthogroup")
# 6,242 Orthogroups expressed and shared between all 5 species
```

### DESeq2

Convert counts to integers.
Run DESeq2 on Orthogroups to get Rlog normalized expression.

```{r,results='hide',tidy=TRUE}
countData <- all_species
countData <- tibble::column_to_rownames(countData,"Orthogroup")
colData <- read.csv("./all/metaData.csv",row.names = "name")
# Remove pstr_d8
colData <- colData[-c(53),]

# Make counts in countData integers:
countData[1:52] <-  round(countData[1:52], digits=0)
countData <- as.data.frame(countData)

# Set the design
dds_treatment <- DESeqDataSetFromMatrix(countData = countData, 
                                        colData = colData, 
                                        design = ~species + treatment )

# Filter out columns with an average number of reads less than 10
dds_treatment <- dds_treatment[ rowMeans(counts(dds_treatment)) > 10, ] 

# Run DESeq2
test <- DESeq(dds_treatment)
rld <- rlog(dds_treatment, blind=FALSE) 
rrld <- assay(rld)
# 6170 Orthogroups after filtering out low counts
write.csv(rrld, file = "./all/DESeq2/treatment_rlog_orthogroups.csv")  
```

### Annotate the Orthogroups

To annotate the orthogroups, first we need to pull out the first sequence per Orthogroup from one species (M.cavernosa). Then we will assign a Unipot ID using the M.cavernosa transcriptome annotation. Finally we will merge Entry ID with Uniprot info to obtain full annotation for each orthogroup.

```{r}
# Use one representative seqeuence from Mcav to assign Entry ID to each orthogroup
mcav_orthogroups <- Orthogroups[,c(1,3)]
colnames(mcav_orthogroups)[2] <- "Transcript"
mcav_orthogroups$Transcript <- gsub("\\..*","",mcav_orthogroups$Transcript)
mcav_annot <- read.csv("./annotations/mcav_annotation.csv")
mcav_annot <- merge(mcav_orthogroups,mcav_annot,by="Transcript") #12,548
uniprot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/uniprot.csv")
mcav_annot <- left_join(mcav_annot,uniprot,by="Entry")
mcav_annot <- mcav_annot %>% filter(mcav_annot$Evalue < 0.000001)
# 12,546 Orthogroups with good annotation

# Merge with expression matrix
expr_mat <- as.data.frame(rrld)
expr_mat <- tibble::rownames_to_column(expr_mat,"Orthogroup")
expr_mat_annot <- merge(expr_mat,mcav_annot,by="Orthogroup")
expr_mat_annot <- expr_mat_annot %>% relocate(Transcript, .after = Orthogroup)
expr_mat_annot <- expr_mat_annot %>% relocate(Entry, .after = Transcript)
expr_mat_annot <- expr_mat_annot %>% relocate(Evalue, .after = Entry)
expr_mat_annot <- expr_mat_annot %>% relocate(Entry.name, .after = Evalue)
expr_mat_annot <- expr_mat_annot %>% relocate(Gene.names, .after = Entry.name)
expr_mat_annot <- expr_mat_annot %>% relocate(Protein.names, .after = Gene.names)
expr_mat_annot <- expr_mat_annot %>% relocate(Organism, .after = Protein.names)
expr_mat_annot <- expr_mat_annot %>% relocate(Gene.ontology..GO., .after = Organism)
expr_mat_annot <- expr_mat_annot[,-c(4,8)]
colnames(expr_mat_annot)[7] <- "GO"
```

### Pull out Orthogroups with immune function

```{r}
immunity <- expr_mat_annot[expr_mat_annot$GO %like% "immunity",]
dim(immunity) #2
immune <- expr_mat_annot[expr_mat_annot$GO %like% "immune",]
dim(immune) #132
all_immunity <- rbind(dplyr::filter(immunity, !Orthogroup %in% intersect(immunity$Orthogroup, immune$Orthogroup)), immune)
dim(all_immunity) #133 immune-related Orthogroups

virus <- expr_mat_annot[expr_mat_annot$GO %like% "virus",]
dim(virus) #57
viral <- expr_mat_annot[expr_mat_annot$GO %like% "viral",]
dim(viral) #54
all_viral <- rbind(dplyr::filter(virus, !Orthogroup %in% intersect(virus$Orthogroup, viral$Orthogroup)), viral)
dim(all_viral) #103

all <- rbind(dplyr::filter(all_immunity, !Orthogroup %in% intersect(all_immunity$Orthogroup, all_viral$Orthogroup)), all_viral)
dim(all) # 212

write.csv(all,file = "./all/immune_expression.csv",row.names = FALSE)
```

## Step 4: ANOVA on Immune Orthogroups

We run a one-way ANOVA on the Orthogroups with immune function to test for differences in expression between disease state (control, exposed, infected)
```{r}
t_all_immunity <- setNames(data.frame(t(all[,-1])), all[,1])
row.names(t_all_immunity)
t_all_immunity <- tibble::rownames_to_column(t_all_immunity, "name")
t_all_immunity <- t_all_immunity[-c(1:6),]

label <- read.csv("./all/metaData.csv")
colnames(label)
label <- label[-c(53),c(3,5)] # remove pstr_d8; keep name and status
master <- merge(label,t_all_immunity,by="name")

# Perform ANOVA on the 213 immune genes
names(master)
# ANOVA loop significant results
for(i in 3:ncol(master))
{
  column <- names(master[i])
  #tidy will summarize and return neat format
  avz <- broom::tidy(aov(master[,i] ~ status, data = master))
  
  # Add this condition if you only want aov with P < 0.05
  if(avz$p.value[1] < 0.05) {
    
    print(column)
  }
}
```

[1] "OG0002316"
```{r}
OG0002316 <- aov(master$OG0002316 ~ master$status, data = master)
summary(OG0002316) 
TukeyHSD(OG0002316) 
```
[1] "OG0002812"
```{r}
OG0002812 <- aov(master$OG0002812 ~ master$status, data = master)
summary(OG0002812) 
TukeyHSD(OG0002812)
```
[1] "OG0002951"
```{r}
OG0002951 <- aov(master$OG0002951 ~ master$status, data = master)
summary(OG0002951) 
TukeyHSD(OG0002951) 
```
[1] "OG0003092"
```{r}
OG0003092 <- aov(master$OG0003092 ~ master$status, data = master)
summary(OG0003092) 
TukeyHSD(OG0003092) 
```
[1] "OG0003270"
```{r}
OG0003270 <- aov(master$OG0003270 ~ master$status, data = master)
summary(OG0003270) 
TukeyHSD(OG0003270) 
```
[1] "OG0003771"
```{r}
OG0003771 <- aov(master$OG0003771 ~ master$status, data = master)
summary(OG0003771) 
TukeyHSD(OG0003771) 
```
[1] "OG0003940"
```{r}
OG0003940 <- aov(master$OG0003940 ~ master$status, data = master)
summary(OG0003940) 
TukeyHSD(OG0003940) 
```
[1] "OG0003997"
```{r}
OG0003997 <- aov(master$OG0003997 ~ master$status, data = master)
summary(OG0003997) 
TukeyHSD(OG0003997) 
```
[1] "OG0004300"
```{r}
OG0004300 <- aov(master$OG0004300 ~ master$status, data = master)
summary(OG0004300) 
TukeyHSD(OG0004300) 
```
[1] "OG0004350"
```{r}
OG0004350 <- aov(master$OG0004350 ~ master$status, data = master)
summary(OG0004350) 
TukeyHSD(OG0004350) 
```
[1] "OG0005460"
```{r}
OG0005460 <- aov(master$OG0005460 ~ master$status, data = master)
summary(OG0005460) 
TukeyHSD(OG0005460) 
```
[1] "OG0006200"
```{r}
OG0006200 <- aov(master$OG0006200 ~ master$status, data = master)
summary(OG0006200) 
TukeyHSD(OG0006200) 
```
[1] "OG0007817"
```{r}
OG0007817 <- aov(master$OG0007817 ~ master$status, data = master)
summary(OG0007817) 
TukeyHSD(OG0007817) 
```
[1] "OG0007911"
```{r}
OG0007911 <- aov(master$OG0007911 ~ master$status, data = master)
summary(OG0007911) 
TukeyHSD(OG0007911) 
```
[1] "OG0011317"
```{r}
OG0011317 <- aov(master$OG0011317 ~ master$status, data = master)
summary(OG0011317) 
TukeyHSD(OG0011317) 
```
[1] "OG0012254"
```{r}
OG0012254 <- aov(master$OG0012254 ~ master$status, data = master)
summary(OG0012254) 
TukeyHSD(OG0012254) 
```
[1] "OG0000274"
```{r}
OG0000274 <- aov(master$OG0000274 ~ master$status, data = master)
summary(OG0000274) 
TukeyHSD(OG0000274) 
```
[1] "OG0002491"
```{r}
OG0002491 <- aov(master$OG0002491 ~ master$status, data = master)
summary(OG0002491) 
TukeyHSD(OG0002491) 
```
[1] "OG0005125"
```{r}
OG0005125 <- aov(master$OG0005125 ~ master$status, data = master)
summary(OG0005125) 
TukeyHSD(OG0005125) 
```
[1] "OG0005462"
```{r}
OG0005462 <- aov(master$OG0005462 ~ master$status, data = master)
summary(OG0005462) 
TukeyHSD(OG0005462) 
```
[1] "OG0007162"
```{r}
OG0007162 <- aov(master$OG0007162 ~ master$status, data = master)
summary(OG0007162) 
TukeyHSD(OG0007162) 
```
[1] "OG0012356"
```{r}
OG0012356 <- aov(master$OG0012356 ~ master$status, data = master)
summary(OG0012356) 
TukeyHSD(OG0012356) 
```
[1] "OG0003182"
```{r}
OG0003182 <- aov(master$OG0003182 ~ master$status, data = master)
summary(OG0003182) 
TukeyHSD(OG0003182) 
```
[1] "OG0003597"
```{r}
OG0003597 <- aov(master$OG0003597 ~ master$status, data = master)
summary(OG0003597) 
TukeyHSD(OG0003597) 
```
[1] "OG0004181"
```{r}
OG0004181 <- aov(master$OG0004181 ~ master$status, data = master)
summary(OG0004181) 
TukeyHSD(OG0004181) 
```
[1] "OG0005274"
```{r}
OG0005274 <- aov(master$OG0005274 ~ master$status, data = master)
summary(OG0005274) 
TukeyHSD(OG0005274) 
```
[1] "OG0005285"
```{r}
OG0005285 <- aov(master$OG0005285 ~ master$status, data = master)
summary(OG0005285) 
TukeyHSD(OG0005285) 
```
[1] "OG0005977"
```{r}
OG0005977 <- aov(master$OG0005977 ~ master$status, data = master)
summary(OG0005977) 
TukeyHSD(OG0005977) 
```
[1] "OG0007458"
```{r}
OG0007458 <- aov(master$OG0007458 ~ master$status, data = master)
summary(OG0007458) 
TukeyHSD(OG0007458) 
```
[1] "OG0011228"
```{r}
OG0011228 <- aov(master$OG0011228 ~ master$status, data = master)
summary(OG0011228) 
TukeyHSD(OG0011228) 
```

### Significant 

Make a dataframe including only the 30 significant highly variable orthologs

```{r,results='hide',tidy=TRUE}
all
sig_orthogroups <- c("OG0002951","OG0003092","OG0003270","OG0003771","OG0003940","OG0003997","OG0004300","OG0004350","OG0005460","OG0006200","OG0007817","OG0011317","OG0000274","OG0002491","OG0005125","OG0005462","OG0007162","OG0012356","OG0003182","OG0003597","OG0004181","OG0005274","OG0005285","OG0005977","OG0007458","OG0011228","OG0002316","OG0002812","OG0007911","OG0012254")
sig_orthogroups <- all[all$Orthogroup %in% sig_orthogroups,]
write.csv(sig_orthogroups,file="./all/sig_immune_orthogroups.csv",row.names = FALSE)
```
Add the Tukey p value results to this file in 3 columns: EvI,CvI and CvE.
Rename this file "sig_immune_orthogroups_ANOVA.csv"

### Boxplots
```{r}
immune_expression <- read.csv("./all/sig_immune_orthogroups_ANOVA.csv")
names(immune_expression)
immune_expression <- immune_expression[,-c(2:10)]
t_immune_expression <- setNames(as.data.frame(t(immune_expression[,-1])), immune_expression[,1])
row.names(t_immune_expression)
view(t_immune_expression)
t_immune_expression <- tibble::rownames_to_column(t_immune_expression, "name")
view(t_immune_expression)

label <- read.csv("./all/metaData.csv")
colnames(label)
label <- label[-c(53),c(1,5)] # remove pstr_d8; keep species, status
t_immune_expression$species <- label$species
t_immune_expression <- t_immune_expression %>% relocate(species, .before = everything())
t_immune_expression$status <- label$status
t_immune_expression <- t_immune_expression %>% relocate(status, .after = name)


t_immune_expression$status <- gsub("healthy","control", t_immune_expression$status)
t_immune_expression$status <- gsub("disease","infected", t_immune_expression$status)
t_immune_expression$status <- factor(t_immune_expression$status, levels = c("control","exposed","infected"))
t_immune_expression$species <- factor(t_immune_expression$species, levels = c("mcav","past","pstr","oann","cnat"))
master <- t_immune_expression

# Create dataframe for Source Data file
## OG0005460 - SMAD family member 6
## OG0002316 - Toll-like receptor 6
## OG0000274 - TNF receptor-associated factor 3
## OG0002812 - Deleted in malignant brain tumors 1 protein
## OG0007458 - Ras-related protein rab 7
## OG0004181 - Charged multivesicular body protein 4b

source_data <- master[,c("species","name","status","OG0005460","OG0002316","OG0000274","OG0002812","OG0007458","OG0004181")]
```

#### ** SMAD family member 6
Transforming growth factor-beta superfamily receptors signaling occurs through the Smad family of intracellular mediators. SMAD6 is an inhibitory Smad (i-Smad) that negatively regulates signaling downstream of type I transforming growth factor-beta (PubMed:9436979, PubMed:16951688, PubMed:22275001, PubMed:9759503, PubMed:10647776, PubMed:10708948, PubMed:10708949, PubMed:30848080).
Acts as a mediator of TGF-beta and BMP anti-inflammatory activities. Suppresses IL1R-TLR signaling through its direct interaction with PEL1, preventing NF-kappa-B activation, nuclear transport and NF-kappa-B-mediated expression of pro-inflammatory genes (PubMed:16951688).
Blocks the BMP-SMAD1 signaling pathway by competing with SMAD4 for receptor-activated SMAD1-binding (PubMed:9436979, PubMed:30848080).
Binds to regulatory elements in target promoter regions (PubMed:16491121).
```{r, fig.width=4,fig.height=2,dpi=200}
OG0005460 <- aov(OG0005460~status,data=master) %>%
  tukey_hsd()
OG0005460

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

#OG0005460_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0005460, fill = status)) +
  geom_point(aes(y = OG0005460, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "SMAD family member 6") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0005460,label="p.adj",y.position=10.5,step.increase=0.15,inherit.aes=FALSE,size=3)

OG0005460_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0005460, fill = status)) +
  geom_point(aes(y = OG0005460, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "SMAD family member 6") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))


ggarrange(OG0005460_species,OG0005460_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")


OG0005460_species_stats <- ggplot_build(OG0005460_species)
OG0005460_species
OG0005460_species_stats$data

OG0005460_status_stats <- ggplot_build(OG0005460_status)
OG0005460_status
OG0005460_status_stats$data
```

#### ** TNF receptor-associated factor 3
Cytoplasmic E3 ubiquitin ligase that regulates various signaling pathways, such as the NF-kappa-B, mitogen-activated protein kinase (MAPK) and interferon regulatory factor (IRF) pathways, and thus controls a lot of biological processes in both immune and non-immune cell types (PubMed:17015635).
In TLR and RLR signaling pathways, acts as an E3 ubiquitin ligase promoting the synthesis of 'Lys-63'-linked polyubiquitin chains on several substrates such as ASC that lead to the activation of the type I interferon response or the inflammasome (PubMed:19898473, PubMed:23871208, PubMed:26305951, PubMed:23150880).
Following the activation of certain TLRs such as TLR4, acts as a negative NF-kappa-B regulator, possibly to avoid unregulated inflammatory response, and its degradation via 'Lys-48'-linked polyubiquitination is required for MAPK activation and production of inflammatory cytokines (PubMed:16306937).
Alternatively, when TLR4 orchestrates bacterial expulsion, TRAF3 undergoes 'Lys-33'-linked polyubiquitination and subsequently binds to RALGDS, mobilizing the exocyst complex to rapidly expel intracellular bacteria back for clearance. Acts also as a constitutive negative regulator of the alternative NF-kappa-B pathway, which controls B-cell survival and lymphoid organ development (PubMed:17723217).
Required for normal antibody isotype switching from IgM to IgG (PubMed:19228877).
Plays a role T-cell dependent immune responses (PubMed:8934568).
Down-regulates proteolytic processing of NFKB2, and thereby inhibits non-canonical activation of NF-kappa-B. Promotes ubiquitination and proteasomal degradation of MAP3K14
```{r, fig.width=4,fig.height=2,dpi=200}
OG0000274 <- aov(OG0000274~status,data=master) %>%
  tukey_hsd()
OG0000274

#OG0000274_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0000274, fill = status)) +
  geom_point(aes(y = OG0000274, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "TNF receptor-associated factor 3") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0005460,label="p.adj",y.position=12,step.increase=0.15,inherit.aes=FALSE,size=3)

OG0000274_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0000274, fill = status)) +
  geom_point(aes(y = OG0000274, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "TNF receptor-associated factor 3") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0000274_species,OG0000274_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")


OG0000274_species_stats <- ggplot_build(OG0000274_species)
OG0000274_species
OG0000274_species_stats$data

OG0000274_status_stats <- ggplot_build(OG0000274_status)
OG0000274_status
OG0000274_status_stats$data
```


#### ** Charged multivesicular body protein 4b
Probable core component of the endosomal sorting required for transport complex III (ESCRT-III) which is involved in multivesicular bodies (MVBs) formation and sorting of endosomal cargo proteins into MVBs. MVBs contain intraluminal vesicles (ILVs) that are generated by invagination and scission from the limiting membrane of the endosome and mostly are delivered to lysosomes enabling degradation of membrane proteins, such as stimulated growth factor receptors, lysosomal enzymes and lipids (By similarity).
```{r, fig.width=8,fig.height=4,dpi=200}
OG0004181 <- aov(OG0004181~status,data=master) %>%
  tukey_hsd()
OG0004181

#OG0004181_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0004181, fill = status)) +
  geom_point(aes(y = OG0004181, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Charged multivesicular body protein 4b") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0005460,label="p.adj",y.position=10.5,step.increase=0.15,inherit.aes=FALSE,size=3)

OG0004181_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0004181, fill = status)) +
  geom_point(aes(y = OG0004181, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Charged multivesicular body protein 4b") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0004181, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0004181_species,OG0004181_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")


OG0004181_species_stats <- ggplot_build(OG0004181_species)
OG0004181_species
OG0004181_species_stats$data

OG0004181_status_stats <- ggplot_build(OG0004181_status)
OG0004181_status
OG0004181_status_stats$data
```


#### ** Ras-related protein rab7
Key regulator in endo-lysosomal trafficking. Governs early-to-late endosomal maturation, microtubule minus-end as well as plus-end directed endosomal migration and positioning, and endosome-lysosome transport through different protein-protein interaction cascades. Plays important roles in microbial pathogen infection and survival, as well as in participating in the life cycle of viruses (By similarity).
```{r, fig.width=8,fig.height=4,dpi=200}
OG0007458 <- aov(OG0007458~status,data=master) %>%
  tukey_hsd()
OG0007458

#OG0007458_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0007458, fill = status)) +
  geom_point(aes(y = OG0007458, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Ras-related protein rab7") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0005460,label="p.adj",y.position=11.5,step.increase=0.15,inherit.aes=FALSE,size=3)

OG0007458_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0007458, fill = status)) +
  geom_point(aes(y = OG0007458, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Ras-related protein rab7") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0007458_species,OG0007458_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")


OG0007458_species_stats <- ggplot_build(OG0007458_species)
OG0007458_species
OG0007458_species_stats$data

OG0007458_status_stats <- ggplot_build(OG0007458_status)
OG0007458_status
OG0007458_status_stats$data
```


#### ** Toll-like receptor 6 
Participates in the innate immune response to Gram-positive bacteria and fungi. Specifically recognizes diacylated and, to a lesser extent, triacylated lipopeptides (PubMed:20037584).
In response to diacylated lipopeptides, forms the activation cluster TLR2:TLR6:CD14:CD36, this cluster triggers signaling from the cell surface and subsequently is targeted to the Golgi in a lipid-raft dependent pathway (PubMed:16880211).
Acts via MYD88 and TRAF6, leading to NF-kappa-B activation, cytokine secretion and the inflammatory response. Recognizes mycoplasmal macrophage-activating lipopeptide-2kD (MALP-2), soluble tuberculosis factor (STF), phenol-soluble modulin (PSM) and B.burgdorferi outer surface protein A lipoprotein (OspA-L) cooperatively with TLR2 (PubMed:11441107).
In complex with TLR4, promotes sterile inflammation in monocytes/macrophages in response to oxidized low-density lipoprotein (oxLDL) or amyloid-beta 42. In this context, the initial signal is provided by oxLDL- or amyloid-beta 42-binding to CD36. This event induces the formation of a heterodimer of TLR4 and TLR6, which is rapidly internalized and triggers inflammatory response, leading to the NF-kappa-B-dependent production of CXCL1, CXCL2 and CCL9 cytokines, via MYD88 signaling pathway, and CCL5 cytokine, via TICAM1 signaling pathway, as well as IL1B secretion (PubMed:11441107, PubMed:20037584).
```{r, fig.width=8,fig.height=4,dpi=200}
OG0002316 <- aov(OG0002316~status,data=master) %>%
  tukey_hsd()
OG0002316


#OG0002316_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0002316, fill = status)) +
  geom_point(aes(y = OG0002316, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Toll-like receptor 6 ") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0005460,label="p.adj",y.position=9,step.increase=0.15,inherit.aes=FALSE,size=3)

OG0002316_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0002316, fill = status)) +
  geom_point(aes(y = OG0002316, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Toll-like receptor 6 ") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))

ggarrange(OG0002316_species,OG0002316_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")


OG0002316_species_stats <- ggplot_build(OG0002316_species)
OG0002316_species
OG0002316_species_stats$data

OG0002316_status_stats <- ggplot_build(OG0002316_status)
OG0002316_status
OG0002316_status_stats$data
```


#### ** Deleted in malignant brain tumors 1 protein
May play roles in mucosal defense system and cellular immune defense. May play a role in liver regeneration. May be an important factor in fate decision and differentiation of transit-amplifying ductular (oval) cells within the hepatic lineage. May function as a binding protein in saliva for the regulation of taste sensation. May play a role as an opsonin receptor for SFTPD and SPAR in macrophage tissues throughout the body, including epithelial cells lining the gastrointestinal tract (By similarity).
Required for terminal differentiation of columnar epithelial cells during early embryogenesis. Displays a broad calcium-dependent binding spectrum against both Gram-positive and Gram-negative bacteria, suggesting a role in defense against bacterial pathogens. Binds to a range of poly-sulfated and poly-phosphorylated ligands which may explain its broad bacterial-binding specificity. Inhibits cytoinvasion of S.enterica. Associates with the actin cytoskeleton and is involved in its remodeling during regulated exocytosis. Interacts with pancreatic zymogens in a pH-dependent manner and may act as a Golgi cargo receptor in the regulated secretory pathway of the pancreatic acinar cell
```{r, fig.width=4,fig.height=2,dpi=200}
OG0002316 <- aov(OG0002316~status,data=master) %>%
  tukey_hsd()
OG0002316

OG0002812_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0002812, fill = status)) +
  geom_point(aes(y = OG0002812, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Deleted in malignant brain tumors 1 protein") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0002316,label="p.adj",y.position=10.2,step.increase=0.15,inherit.aes=FALSE,size=3)

OG0002812_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0002812, fill = status)) +
  geom_point(aes(y = OG0002812, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Deleted in malignant brain tumors 1 protein") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0003270, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0002812_species,OG0002812_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")


OG0002812_species_stats <- ggplot_build(OG0002812_species)
OG0002812_species
OG0002812_species_stats$data

OG0002812_status_stats <- ggplot_build(OG0002812_status)
OG0002812_status
OG0002812_status_stats$data
```

### Pull out Orthogroups with ECM function

```{r}
ECM <- expr_mat_annot[expr_mat_annot$GO %like% "extracellular matrix",]
dim(ECM) #107

write.csv(ECM,file = "./all/ECM_expression.csv",row.names = FALSE)
```

## Step 7: ANOVA on ECM Orthogroups

We run a one-way ANOVA on the Orthogroups with immune function to test for differences in expression between disease state (control, exposed, infected)
```{r}
t_all_ECM <- setNames(data.frame(t(ECM[,-1])), ECM[,1])
row.names(t_all_ECM)
t_all_ECM <- tibble::rownames_to_column(t_all_ECM, "name")
t_all_ECM <- t_all_ECM[-c(1:6),]

label <- read.csv("./all/metaData.csv")
colnames(label)
label <- label[-c(53),c(3,5)] # remove pstr_d8; keep name and status
master <- merge(label,t_all_ECM,by="name")

# Perform ANOVA on the 213 immune genes
names(master)
# ANOVA loop significant results
for(i in 3:ncol(master))
{
  column <- names(master[i])
  #tidy will summarize and return neat format
  avz <- broom::tidy(aov(master[,i] ~ status, data = master))
  
  # Add this condition if you only want aov with P < 0.05
  if(avz$p.value[1] < 0.05) {
    
    print(column)
  }
}
```
[1] "OG0000071"
```{r}
OG0000071 <- aov(master$OG0000071 ~ master$status, data = master)
summary(OG0000071) 
TukeyHSD(OG0000071) # Significant between exposed and disease
```
[2] "OG0000128"
```{r}
OG0000128 <- aov(master$OG0000128 ~ master$status, data = master)
summary(OG0000128) 
TukeyHSD(OG0000128) # Significant between exposed and disease; healthy and disease
```
[3] "OG0001191"
```{r}
OG0001191 <- aov(master$OG0001191 ~ master$status, data = master)
summary(OG0001191) 
TukeyHSD(OG0001191) # Significant between exposed and disease; healthy and disease
```
[4] "OG0001630"
```{r}
OG0001630 <- aov(master$OG0001630 ~ master$status, data = master)
summary(OG0001630) 
TukeyHSD(OG0001630) # Significant between healthy and disease
```
[5] "OG0001920"
```{r}
OG0001920 <- aov(master$OG0001920 ~ master$status, data = master)
summary(OG0001920) 
TukeyHSD(OG0001920) # Significant between exposed and disease; healthy and disease
```
[6] "OG0002093"
```{r}
OG0002093 <- aov(master$OG0002093 ~ master$status, data = master)
summary(OG0002093) 
TukeyHSD(OG0002093)
Tukey <- TukeyHSD(OG0002093) # Significant between exposed and disease; healthy and disease
print(Tukey,digits = 15)
```
[7] "OG0002812"
```{r}
OG0002812 <- aov(master$OG0002812 ~ master$status, data = master)
summary(OG0002812) 
TukeyHSD(OG0002812) # Significant between healthy and disease; healthy and exposed
```
[8] "OG0002951"
```{r}
OG0002951 <- aov(master$OG0002951 ~ master$status, data = master)
summary(OG0002951) 
TukeyHSD(OG0002951) # Significant between healthy and exposed
```
[9] "OG0003597"
```{r}
OG0003597 <- aov(master$OG0003597 ~ master$status, data = master)
summary(OG0003597) 
TukeyHSD(OG0003597) # Significant between exposed and disesase
```
[10] "OG0004233"
```{r}
OG0004233 <- aov(master$OG0004233 ~ master$status, data = master)
summary(OG0004233) 
TukeyHSD(OG0004233) # Significant between healthy and disesase
```
[-] "OG0005350"
```{r}
OG0005350 <- aov(master$OG0005350 ~ master$status, data = master)
summary(OG0005350) 
TukeyHSD(OG0005350) # NOT Significant!
```
[11] "OG0007699"
```{r}
OG0007699 <- aov(master$OG0007699 ~ master$status, data = master)
summary(OG0007699) 
TukeyHSD(OG0007699) # significant between healthy and disease
```
[12] "OG0007911"
```{r}
OG0007911 <- aov(master$OG0007911 ~ master$status, data = master)
summary(OG0007911) 
TukeyHSD(OG0007911) # significant between exposed and disease; healthy and disease
```
[13] "OG0008154"
```{r}
OG0008154 <- aov(master$OG0008154 ~ master$status, data = master)
summary(OG0008154) 
TukeyHSD(OG0008154) # significant between exposed and disease
```
[14] "OG0011025"
```{r}
OG0011025 <- aov(master$OG0011025 ~ master$status, data = master)
summary(OG0011025) 
TukeyHSD(OG0011025) # significant between healthy and disease
```
[15] "OG0011028"
```{r}
OG0011028 <- aov(master$OG0011028 ~ master$status, data = master)
summary(OG0011028) 
TukeyHSD(OG0011028) # significant between healthy and disease; healthy and exposed
```
[16] "OG0011034"
```{r}
OG0011034 <- aov(master$OG0011034 ~ master$status, data = master)
summary(OG0011034) 
TukeyHSD(OG0011034) # significant between healthy and disease
```
[17] "OG0011714"
```{r}
OG0011714 <- aov(master$OG0011714 ~ master$status, data = master)
summary(OG0011714) 
TukeyHSD(OG0011714) # significant between healthy and disease
```
[18] "OG0011721"
```{r}
OG0011721 <- aov(master$OG0011721 ~ master$status, data = master)
summary(OG0011721) 
TukeyHSD(OG0011721) # significant between healthy and disease
```

### Significant 

Make a dataframe including only the 18 significant highly variable orthologs

```{r,results='hide',tidy=TRUE}
ECM
sig_orthogroups <- c("OG0000071","OG0000128","OG0001191","OG0001630","OG0001920","OG0002093","OG0002812","OG0002951","OG0003597","OG0004233","OG0007699","OG0007911","OG0008154","OG0011025","OG0011028","OG0011034","OG0011714","OG0011721")
sig_orthogroups <- ECM[ECM$Orthogroup %in% sig_orthogroups,]
write.csv(sig_orthogroups,file="./all/sig_ECM_orthogroups.csv",row.names = FALSE)
```
Add the Tukey p value results to this file in 3 columns: EvI,CvI and CvE.
Rename this file "sig_immune_orthogroups_ANOVA.csv"

### Boxplots
```{r}
ECM_expression <- read.csv("./all/sig_ECM_orthogroups_ANOVA.csv")
names(ECM_expression)
ECM_expression <- ECM_expression[,-c(2:12)]
t_ECM_expression <- setNames(as.data.frame(t(ECM_expression[,-1])), ECM_expression[,1])
row.names(t_ECM_expression)
view(t_ECM_expression)
t_ECM_expression <- tibble::rownames_to_column(t_ECM_expression, "name")
view(t_ECM_expression)

label <- read.csv("./all/metaData.csv")
colnames(label)
label <- label[-c(53),c(1,5)] # remove pstr_d8; keep species, status
t_ECM_expression$species <- label$species
t_ECM_expression <- t_ECM_expression %>% relocate(species, .before = everything())
t_ECM_expression$status <- label$status
t_ECM_expression <- t_ECM_expression %>% relocate(status, .after = name)


t_ECM_expression$status <- gsub("healthy","control", t_ECM_expression$status)
t_ECM_expression$status <- gsub("disease","infected", t_ECM_expression$status)
t_ECM_expression$status <- factor(t_ECM_expression$status, levels = c("control","exposed","infected"))
t_ECM_expression$species <- factor(t_ECM_expression$species, levels = c("mcav","past","pstr","oann","cnat"))
master <- t_ECM_expression

# Create dataframe for Source Data file
## OG0004233 - Superoxide Dismutase
## OG0000128 - Alpha-2 Type I Collagen

source_data$OG0004233 <- master$OG0004233
source_data$OG0000128 <- master$OG0000128

write.csv(source_data,file="~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/REVISIONS/Final Revision/fig2_sourceData.csv")

```

#### ** Alpha-2 type I collagen
Type I collagen is a member of group I collagen (fibrillar forming collagen).
```{r}
OG0000128 <- aov(OG0000128~status,data=master) %>%
  tukey_hsd()
OG0000128

OG0000128_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0000128, fill = status)) +
  geom_point(aes(y = OG0000128, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Alpha-2 type I collagen") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0000128,label="p.adj",y.position=16,step.increase=0.15,inherit.aes=FALSE,size=3)

OG0000128_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0000128, fill = status)) +
  geom_point(aes(y = OG0000128, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Alpha-2 type I collagen") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0000128, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0000128_species,OG0000128_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")

OG0000128_species_stats <- ggplot_build(OG0000128_species)
OG0000128_species
OG0000128_species_stats$data

OG0000128_status_stats <- ggplot_build(OG0000128_status)
OG0000128_status
OG0000128_status_stats$data
```


#### ** Superoxide dismutase [Cu-Zn] 1
```{r, fig.width=6,fig.height=3,dpi=200}
OG0004233 <- aov(OG0004233~status,data=master) %>%
  tukey_hsd()
OG0004233

OG0004233_status <- 
ggplot(data = master, 
       aes(x = status, y = OG0004233, fill = status)) +
  geom_point(aes(y = OG0004233, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Superoxide dismutase [Cu-Zn] 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0004233,label="p.adj",y.position=11.5,step.increase=0.15,inherit.aes=FALSE,size=3)

OG0004233_species <- 
ggplot(data = master, 
       aes(x = species, y = OG0004233, fill = status)) +
  geom_point(aes(y = OG0004233, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Superoxide dismutase [Cu-Zn] 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_fill_manual(values = c("#57be8d","#68b7e6","#f37225"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0004233, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)


ggarrange(OG0004233_species,OG0004233_status,
          ncol=2, widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",align = "hv")


OG0004233_species_stats <- ggplot_build(OG0004233_species)
OG0004233_species
OG0004233_species_stats$data

OG0004233_status_stats <- ggplot_build(OG0004233_status)
OG0004233_status
OG0004233_status_stats$data
```

### Combined Figure
```{r immune orthogroups boxplots, fig.width=6, fig.height=20, dpi=200}
pdf("immunePlots_2.pdf",width = 6,height = 20)
ggarrange(OG0005460_species,OG0005460_status,
          OG0004181_species,OG0004181_status,
          OG0002812_species,OG0002812_status,
          OG0007458_species,OG0007458_status,
          OG0002316_species,OG0002316_status,
          OG0000274_species,OG0000274_status,
          OG0002093_species,OG0002093_status,
          OG0004233_species,OG0004233_status,
          ncol = 2,nrow = 8,
          widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",
          align = "hv"
          )
dev.off()
```


