# SCTLD-Transmission-Experiment-USVI

Any questions, email kelsey.beavers@mavs.uta.edu

## This repository contains 6 Rmarkdown files:
  1. Command_Line_Code - code used to filter reads, assemble and annotate transcriptomes, and quantify transcripts
  2. Coral DEG and Homolog Analysis - code used to quantify DEGs in each coral species and identify immune/ECM homologs with significant differential expression between disease states
  3. Coral Orthogroup Analysis - code used to identify immune/ECM orthogroups with significant differential expression between disease states
  4. Symbiont DEG Analysis - code used to quantify DEGs in each symbiont genera
  5. Coral EVE Analysis - code used to for analyses of highly variable and lineage-specific single-copy coral orthologs
  6. Symbiont EVE Analysis - code used to for analyses of highly variable and lineage-specific single-copy symbiont orthologs

## This repository now also contains all of the reference transcriptomes and their annotations generated from this study in the tarball *Transcriptomes.tar.gz*
  
## Not included in this repository but publicly accessible:
  1. Raw RNAseq data generated from the transmission experiment - https://www.ncbi.nlm.nih.gov/bioproject/PRJNA860922
  2. MasterCoral_db used in "Command_Line_Code" to generate coral-only de novo reference transcriptomes - Kelsey Beavers. (2023). Master Coral database used in USVI SCTLD Transmission Experiment Gene Expression Analysis [Data set]. Zenodo. https://doi.org/10.5281/zenodo.7838980
  3. Genomes used to create M. cavernosa (https://matzlab.weebly.com/data--code.html) and O. annularis (https://www.ncbi.nlm.nih.gov/genome/13173?genome_assembly_id=311351) genome-guided transcriptome assemblies. 
  
  
I hope to make these analyses as clear and reproducible as possible, so if you have any questions, please don't hesitate to ask. 
 
  
