---
title: "Differential Expression Analysis - Coral Genes"
author: "Kelsey Beavers"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Required Packages
```{r}
library(tximport)
library(tibble)
library(DESeq2)
library(dplyr)
library(PCAtools)
library(ggalt)
library(ggplot2)
library(lifecycle)
library(ggvenn)
library(VennDiagram)
library(stringr)
library(ggpubr)
library(pheatmap)
library(utils)
```


# Colpophyllia natans

## Tximport

Read in the raw counts from Salmon for all cnat samples

cnat samples: c2,c3,c4,c6,c8,d2,d3,d4,d6,d8

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/cnat/salmon/")
cnat_samples <- read.table("cnat_samples.csv", header = TRUE)
View(cnat_samples)
cnat_files <- file.path(cnat_samples$samples, "quant.sf")
names(cnat_files) <- paste0("sample", 1:10)
all(file.exists(cnat_files))
cnat_tx2gene <- read.csv("cnat_tx2gene.csv")
head(cnat_tx2gene)
cnat_txi <- tximport(cnat_files, type = 'salmon',tx2gene=cnat_tx2gene)
names(cnat_txi)
tail(cnat_txi$counts)
dim(cnat_txi$counts)
# 13,692 contigs with evalue < 0.000001
write.csv(cnat_txi, file = "cnat_txi_test.csv")
write.csv(cnat_txi$counts, file="cnat_counts.csv",quote = FALSE)
cnat_counts <- cnat_txi$counts
colnames(cnat_counts)
cnat_counts <- as.data.frame(cnat_counts)
colnames(cnat_counts) <- c("cnat_c2","cnat_c3","cnat_c4","cnat_c6","cnat_c8","cnat_d2","cnat_d3","cnat_d4","cnat_d6","cnat_d8")
cnat_counts <- tibble::rownames_to_column(cnat_counts,"Entry")
cnat_counts[2:11] <-  round(cnat_counts[2:11], digits=0)
cnat_counts <- as.data.frame(cnat_counts)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/cnat/DESeq2/")
write.csv(cnat_counts, file = "cnat_counts.csv",quote = FALSE)
```

## DESeq2

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/cnat/DESeq2/")
countData <- read.csv("cnat_counts.csv", row.names="Entry")
colnames(countData)
countData <- countData[,-c(1)]
colData <- read.csv("./../../../metaData.csv", row.names = "name")
cnatColData <- colData[colData$species == 'cnat',]
rownames(cnatColData)
colnames(countData)
# Set the design. This design will test for the differences in expression caused by treatment 
dds_treatment <- DESeqDataSetFromMatrix(countData = countData, 
                                        colData = cnatColData, 
                                        design = ~genotype + treatment )
# Filter out columns with an average number of reads less than 10. 
dds_treatment <- dds_treatment[ rowMeans(counts(dds_treatment)) > 10, ] 

test <- DESeq(dds_treatment)
res <- results(test)
res
resorderd <-res[order(res$padj),] 
write.csv(resorderd, file = "DESeq_treatment.csv")
rld <- rlog(dds_treatment, blind=FALSE) 
rrld <- assay(rld)
write.csv(rrld, file = "treatment_rlog.csv")  

# Keep significant DEGs (padj < 0.05)
DESeq_treatment_master <- read.csv("DESeq_treatment.csv")
DEGs <- DESeq_treatment_master %>% filter(padj < 0.05)
write.csv(DEGs, file = "treatment_sig_DEGs.csv")

# Annotate the significant DEGs
colnames(DEGs)[colnames(DEGs)=="X"] <- "Entry"
DEGs
dim(DEGs)
# 1350 unannotated DEGs
uniprot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/uniprot.csv")
master <- merge(DEGs, uniprot, by="Entry")
logs <- read.csv("treatment_rlog.csv")
names(logs)
names(logs)[colnames(logs)=="X"] <- "Entry"
sig_master <- merge(master,logs, by="Entry")
# 1350 annotated DEGs
write.csv(sig_master, file = "treatment_annotated_sig_DEGs.csv")
```

## PCA

```{r}
# Remove all columns except for "Entry ID" and the counts
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/cnat/DESeq2/")
countData <- read.csv("treatment_rlog.csv", row.names="X")
colnames(countData)
Gene_expression <- as.matrix(countData)
Metadata <- as.data.frame(read.csv("./../../../metaData.csv", header = TRUE, row.names = "name"))
cnatMetadata <- Metadata[Metadata$species == 'cnat',]
rownames(cnatMetadata)
colnames(Gene_expression)
p <- pca(Gene_expression, metadata = cnatMetadata,center = TRUE,removeVar = 0.1) #Remove the lower 10% of variables based on variance. 
screeplot(p)

cnatMetadata
pdf(file="rlog_PCA.pdf",height=6,width=6)
biplot(p,
       lab = cnatMetadata$sample_number,
       colby = "treatment", colkey = c('control'='blue','disease' = 'red'), colLegendTitle = 'Treatment',
       ellipse = TRUE,
       xlim=c(-450,455), ylim = c(-220,150),
       title = 'Rlog Gene Expression PCA',
       legendPosition = 'right') 
  dev.off()

pdf(file="rlog_PCA_symbiont.pdf",height=6,width=6)
biplot(p,
       lab = cnatMetadata$sample_number,
       colby = "dominant_clade",  colLegendTitle = 'Dominant \nSymbiont',
       shape = "treatment",
       ellipse = TRUE,
       xlim=c(-150,200), ylim = c(-220,220),
       title = 'Rlog Gene Expression PCA',
       legendPosition = 'right') 
dev.off()

```


## GO Enrichment Set Up

Get files ready for ShinyGO (unline GUI)

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/cnat/DESeq2/")
all_genes <- read.csv("DESeq_treatment.csv")
colnames(all_genes)[colnames(all_genes)=="X"] <- "Entry"
names(all_genes)
all_genes <- all_genes[,c(1,3,7)]

uniprot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/uniprot.csv")
names(uniprot)
uniprot <- uniprot[,c(1,3)]

master <- merge(all_genes,uniprot,by="Entry")
master
names(master)
master <- master %>% relocate(Gene.names, .after = Entry)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/cnat/DEG_GOEnrichment/")
master$Gene.names <- sub(" .*", "", master$Gene.names)
names(master)
master
write.csv(master,file="all_genes.csv")


DEGs <- read.csv("./../DESeq2/treatment_annotated_sig_DEGs.csv")
names(DEGs)
DEGs <- DEGs[,c(2,4,8,10)] #grab Entry, log2FoldChange, padj, and Gene.names
names(DEGs)
DEGs <- DEGs %>% relocate(Gene.names, .after =Entry)
DEGs$Gene.names <- sub(" .*", "", DEGs$Gene.names)
DEGs
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/cnat/DEG_GOEnrichment/")
write.csv(DEGs,file="cnat_DEGs.csv")
```

Navigate to http://bioinformatics.sdstate.edu/go/
Select "Human" from the drop down menu
Copy and paste Gene.names from "cnat_DEGs.csv" 
Click "Background" and copy/paste Gene.names from "all_genes.csv"
Select "GO Biological Process" from drop down menu
Select "0.05" for FDR cutof, "10" for # pathways to show, "2" for Pathway size: Min. and "2000" for Max. 
Select "Remove redundancy" and "Abbreviate pathways"
Wait for results and download "Top Pathways shown above" table with gene IDs as "all_enrichment.csv"

In Excel, sort "all_enrichment.csv" by decreased Fold Enrichment and keep top 5 nonredundant terms, name this file "All_enrichment_curated.csv"

Repeat with same parameters for "GO Molecular Function" drop down

# Montastraea cavernosa

## Tximport

Read in the raw counts from Salmon for all mcav samples

mcav: c1,c3,c5,c6,c7,c8,d1,d2,d3,d4,d5,d6,d8
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/mcav/salmon/")
mcav_samples <- read.table("mcav_samples.csv", header = TRUE)
mcav_samples
mcav_files <- file.path(mcav_samples$sample, "quant.sf")
names(mcav_files) <- paste0("sample", 1:13)
all(file.exists(mcav_files))
mcav_tx2gene <- read.csv("mcav_tx2gene.csv")
head(mcav_tx2gene)
mcav_txi <- tximport(mcav_files, type = 'salmon',tx2gene=mcav_tx2gene)
names(mcav_txi)
tail(mcav_txi$counts)
dim(mcav_txi$counts)
# 11,628 contigs with evalue < 0.000001
write.csv(mcav_txi, file = "mcav_txi.csv")
write.csv(mcav_txi$counts, file="mcav_counts.csv",quote = FALSE)
mcav_counts <- mcav_txi$counts
colnames(mcav_counts)
mcav_counts <- as.data.frame(mcav_counts)
colnames(mcav_counts) <- c("mcav_c1","mcav_c3","mcav_c5","mcav_c6","mcav_c7","mcav_c8","mcav_d1","mcav_d2","mcav_d3","mcav_d4","mcav_d5","mcav_d6","mcav_d8")
mcav_counts <- tibble::rownames_to_column(mcav_counts,"Entry")
mcav_counts[2:14] <-  round(mcav_counts[2:14], digits=0)
mcav_counts <- as.data.frame(mcav_counts)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/mcav/DESeq2/")
write.csv(mcav_counts, file = "mcav_counts.csv",quote = FALSE)
```

## DESeq2

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/mcav/DESeq2/")
countData <- read.csv("mcav_counts.csv", row.names="Entry")
colnames(countData)
countData <- countData[,-c(1)]
colData <- read.csv("./../../../metaData.csv", row.names = "name")
mcavColData <- colData[colData$species == 'mcav',]
rownames(mcavColData)
colnames(countData)
# Set the design. This design will test for the differences in expression caused by treatment 
# Filter out columns with an average number of reads less than 10. 
dds_treatment <- DESeqDataSetFromMatrix(countData = countData, 
                                        colData = mcavColData, 
                                        design = ~genotype + treatment )

dds_treatment <- dds_treatment[ rowMeans(counts(dds_treatment)) > 10, ] 

test <- DESeq(dds_treatment)
res <- results(test)
res
resorderd <-res[order(res$padj),] 
write.csv(resorderd, file = "DESeq_treatment.csv")
rld <- rlog(dds_treatment, blind=FALSE) 
rrld <- assay(rld)
write.csv(rrld, file = "treatment_rlog.csv")  

# Keep significant DEGs (padj < 0.05)
DESeq_treatment_master <- read.csv("DESeq_treatment.csv")
DEGs <- DESeq_treatment_master %>% filter(padj < 0.05)
# 385 DEGs
write.csv(DEGs, file = "treatment_sig_DEGs.csv")

#Annotate the significant DEGs
colnames(DEGs)[colnames(DEGs)=="X"] <- "Entry"
DEGs
dim(DEGs)
# 385 unannotated DEGs
uniprot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/uniprot.csv")
master <- merge(DEGs, uniprot, by="Entry")
logs <- read.csv("treatment_rlog.csv")
names(logs)
names(logs)[colnames(logs)=="X"] <- "Entry"
sig_master <- merge(master,logs, by="Entry")
# 385 annotated DEGs
write.csv(sig_master, file = "treatment_annotated_sig_DEGs.csv")
```

## PCA

```{r}
# Remove all columns except for "Entry ID" and the counts
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/mcav/DESeq2/")
countData <- read.csv("treatment_rlog.csv", row.names="X")
colnames(countData)
Gene_expression <- as.matrix(countData)
Metadata <- as.data.frame(read.csv("./../../../metaData.csv", header = TRUE, row.names = "name"))
mcavMetadata <- Metadata[Metadata$species == 'mcav',]
rownames(mcavMetadata)
colnames(Gene_expression)
p <- pca(Gene_expression, metadata = mcavMetadata,center = TRUE,removeVar = 0.1) #Remove the lower 10% of variables based on variance. 
screeplot(p)

mcavMetadata
pdf(file="rlog_PCA.pdf",height=6,width=6)
biplot(p,
       lab = mcavMetadata$sample_number,
       colby = "treatment", colkey = c('control'='blue','disease' = 'red'), colLegendTitle = 'Treatment',
       ellipse = TRUE,
       xlim=c(-100,75), ylim = c(-50,65),
       title = 'Rlog Gene Expression PCA',
       legendPosition = 'right') 
  dev.off()
```

## GO Enrichment Set Up

Get files ready for ShinyGO (unline GUI)

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/mcav/DESeq2/")
all_genes <- read.csv("DESeq_treatment.csv")
colnames(all_genes)[colnames(all_genes)=="X"] <- "Entry"
names(all_genes)
all_genes <- all_genes[,c(1,3,7)]

uniprot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/uniprot.csv")
names(uniprot)
uniprot <- uniprot[,c(1,3)]

master <- merge(all_genes,uniprot,by="Entry")
master
names(master)
master <- master %>% relocate(Gene.names, .after = Entry)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/mcav/DEG_GOEnrichment/")
master$Gene.names <- sub(" .*", "", master$Gene.names)
names(master)
master
write.csv(master,file="all_genes.csv")


DEGs <- read.csv("./../DESeq2/treatment_annotated_sig_DEGs.csv")
names(DEGs)
DEGs <- DEGs[,c(2,4,8,10)] #grab Entry, log2FoldChange, padj, and Gene.names
names(DEGs)
DEGs <- DEGs %>% relocate(Gene.names, .after =Entry)
DEGs$Gene.names <- sub(" .*", "", DEGs$Gene.names)
DEGs
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/mcav/DEG_GOEnrichment/")
write.csv(DEGs,file="mcav_DEGs.csv")
```

Navigate to http://bioinformatics.sdstate.edu/go/
Select "Human" from the drop down menu
Copy and paste Gene.names from "mcav_DEGs.csv" 
Click "Background" and copy/paste Gene.names from "all_genes.csv"
Select "GO Biological Process" from drop down menu
Select "0.05" for FDR cutof, "10" for # pathways to show, "2" for Pathway size: Min. and "2000" for Max. 
Select "Remove redundancy" and "Abbreviate pathways"
Wait for results and download "Top Pathways shown above" table with gene IDs as "all_enrichment.csv"

In Excel, sort "all_enrichment.csv" by decreased Fold Enrichment and keep top 5 nonredundant terms, name this file "All_enrichment_curated.csv"

Repeat with same parameters for "GO Molecular Function" drop down

# Orbicella annularis

## Tximport

Read in the raw counts from Salmon for all oann samples

oann: c1,c2,c3,c6,c7,c8,d1,d3,d6,d7,d8
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/oann/salmon/")
oann_samples <- read.table("oann_samples.csv", header = TRUE)
oann_samples
oann_files <- file.path(oann_samples$sample, "quant.sf")
names(oann_files) <- paste0("sample", 1:11)
all(file.exists(oann_files))
oann_tx2gene <- read.csv("oann_tx2gene.csv")
head(oann_tx2gene)
oann_txi <- tximport(oann_files, type = 'salmon',tx2gene=oann_tx2gene)
names(oann_txi)
tail(oann_txi$counts)
dim(oann_txi$counts)
# 13,138 contigs with evalue < 0.000001
write.csv(oann_txi, file = "oann_txi.csv")
write.csv(oann_txi$counts, file="oann_counts.csv",quote = FALSE)
oann_counts <- oann_txi$counts
colnames(oann_counts)
oann_counts <- as.data.frame(oann_counts)
colnames(oann_counts) <- c("oann_c1","oann_c2","oann_c3","oann_c6","oann_c7","oann_c8","oann_d1","oann_d3","oann_d6","oann_d7","oann_d8")
oann_counts <- tibble::rownames_to_column(oann_counts,"Entry")
oann_counts[2:12] <-  round(oann_counts[2:12], digits=0)
oann_counts <- as.data.frame(oann_counts)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/oann/DESeq2/")
write.csv(oann_counts, file = "oann_counts.csv",quote = FALSE)
```

## DESeq2

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/oann/DESeq2/")
countData <- read.csv("oann_counts.csv", row.names="Entry")
colnames(countData)
countData <- countData[,-c(1)]
colData <- read.csv("./../../../metaData.csv", row.names = "name")
oannColData <- colData[colData$species == 'oann',]
rownames(oannColData)
colnames(countData)
# Set the design. This design will test for the differences in expression caused by treatment 
# Filter out columns with an average number of reads less than 10. 
dds_treatment <- DESeqDataSetFromMatrix(countData = countData, 
                                        colData = oannColData, 
                                        design = ~genotype + treatment )
dds_treatment <- dds_treatment[ rowMeans(counts(dds_treatment)) > 10, ] 

test <- DESeq(dds_treatment)
res <- results(test)
res
resorderd <-res[order(res$padj),] 
write.csv(resorderd, file = "DESeq_treatment.csv")
rld <- rlog(dds_treatment, blind=FALSE) 
rrld <- assay(rld)
write.csv(rrld, file = "treatment_rlog.csv")  

# Keep significant DEGs (padj < 0.05)
DESeq_treatment_master <- read.csv("DESeq_treatment.csv")
DEGs <- DESeq_treatment_master %>% filter(padj < 0.05)
# 229 DEGs
write.csv(DEGs, file = "treatment_sig_DEGs.csv")

# Annotate the significant DEGs
colnames(DEGs)[colnames(DEGs)=="X"] <- "Entry"
DEGs
dim(DEGs)
# 229 unannotated DEGs
uniprot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/uniprot.csv")
master <- merge(DEGs, uniprot, by="Entry")
logs <- read.csv("treatment_rlog.csv")
names(logs)
names(logs)[colnames(logs)=="X"] <- "Entry"
sig_master <- merge(master,logs, by="Entry")
# 229 annotated DEGs
write.csv(sig_master, file = "treatment_annotated_sig_DEGs.csv")
```

## PCA

```{r}
# Remove all columns except for "Entry ID" and the counts
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/oann/DESeq2/")
countData <- read.csv("treatment_rlog.csv", row.names="X")
colnames(countData)
Gene_expression <- as.matrix(countData)
Metadata <- as.data.frame(read.csv("./../../../metaData.csv", header = TRUE, row.names = "name"))
oannMetadata <- Metadata[Metadata$species == 'oann',]
rownames(oannMetadata)
colnames(Gene_expression)
p <- pca(Gene_expression, metadata = oannMetadata,center = TRUE,removeVar = 0.1) #Remove the lower 10% of variables based on variance. 
screeplot(p)

oannMetadata
pdf(file="rlog_PCA.pdf",height=6,width=6)
biplot(p,
       lab = oannMetadata$sample_number,
       colby = "treatment", colkey = c('control'='blue','disease' = 'red'), colLegendTitle = 'Treatment',
       ellipse = TRUE,
       xlim=c(-90,100), ylim = c(-90,90),
       title = 'Rlog Gene Expression PCA',
       legendPosition = 'right') 
  dev.off()
```

## GO Enrichment Set Up

Get files ready for ShinyGO (unline GUI)

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/oann/DESeq2/")
all_genes <- read.csv("DESeq_treatment.csv")
colnames(all_genes)[colnames(all_genes)=="X"] <- "Entry"
names(all_genes)
all_genes <- all_genes[,c(1,3,7)]

uniprot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/uniprot.csv")
names(uniprot)
uniprot <- uniprot[,c(1,3)]

master <- merge(all_genes,uniprot,by="Entry")
master
names(master)
master <- master %>% relocate(Gene.names, .after = Entry)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/oann/DEG_GOEnrichment/")
master$Gene.names <- sub(" .*", "", master$Gene.names)
names(master)
master
write.csv(master,file="all_genes.csv")


DEGs <- read.csv("./../DESeq2/treatment_annotated_sig_DEGs.csv")
names(DEGs)
DEGs <- DEGs[,c(2,4,8,10)] #grab Entry, log2FoldChange, padj, and Gene.names
names(DEGs)
DEGs <- DEGs %>% relocate(Gene.names, .after =Entry)
DEGs$Gene.names <- sub(" .*", "", DEGs$Gene.names)
DEGs
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/oann/DEG_GOEnrichment/")
write.csv(DEGs,file="oann_DEGs.csv")
```

Navigate to http://bioinformatics.sdstate.edu/go/
Select "Human" from the drop down menu
Copy and paste Gene.names from "oann_DEGs.csv" 
Click "Background" and copy/paste Gene.names from "all_genes.csv"
Select "GO Biological Process" from drop down menu
Select "0.05" for FDR cutof, "10" for # pathways to show, "2" for Pathway size: Min. and "2000" for Max. 
Select "Remove redundancy" and "Abbreviate pathways"
Wait for results and download "Top Pathways shown above" table with gene IDs as "all_enrichment.csv"

In Excel, sort "all_enrichment.csv" by decreased Fold Enrichment and keep top 5 nonredundant terms, name this file "All_enrichment_curated.csv"

Repeat with same parameters for "GO Molecular Function" drop down

# Porites astreoides

## Tximport

Read in the raw counts from Salmon for all past samples

past: c1,c5,c6,c7,d1,d2,d3,d4,d5,d6,d7
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/past/salmon/")
past_samples <- read.table("past_samples.csv", header = TRUE)
past_samples
past_files <- file.path(past_samples$sample, "quant.sf")
names(past_files) <- paste0("sample", 1:11)
all(file.exists(past_files))
past_tx2gene <- read.csv("past_tx2gene.csv")
head(past_tx2gene)
past_txi <- tximport(past_files, type = 'salmon',tx2gene=past_tx2gene)
names(past_txi)
tail(past_txi$counts)
dim(past_txi$counts)
# 13271 contigs with evalue < 0.000001
write.csv(past_txi, file = "past_txi.csv")
write.csv(past_txi$counts, file="past_counts.csv",quote = FALSE)
past_counts <- past_txi$counts
colnames(past_counts)
past_counts <- as.data.frame(past_counts)
colnames(past_counts) <- c("past_c1","past_c5","past_c6","past_c7","past_d1","past_d2","past_d3","past_d4","past_d5","past_d6","past_d7")
past_counts <- tibble::rownames_to_column(past_counts,"Entry")
past_counts[2:12] <-  round(past_counts[2:12], digits=0)
past_counts <- as.data.frame(past_counts)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/past/DESeq2//")
write.csv(past_counts, file = "past_counts.csv",quote = FALSE)
```

## DESeq2

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/past/DESeq2/")
countData <- read.csv("past_counts.csv", row.names="Entry")
colnames(countData)
countData <- countData[,-c(1)]
colData <- read.csv("./../../../metaData.csv", row.names = "name")
pastColData <- colData[colData$species == 'past',]
rownames(pastColData)
colnames(countData)
# Set the design. This design will test for the differences in expression caused by treatment 
# Filter out columns with an average number of reads less than 10. 
dds_treatment <- DESeqDataSetFromMatrix(countData = countData, 
                                        colData = pastColData, 
                                        design = ~genotype + treatment )

dds_treatment <- dds_treatment[ rowMeans(counts(dds_treatment)) > 10, ] 
test <- DESeq(dds_treatment)
res <- results(test)
res
resorderd <-res[order(res$padj),] 
write.csv(resorderd, file = "DESeq_treatment.csv")
rld <- rlog(dds_treatment, blind=FALSE) 
rrld <- assay(rld)
write.csv(rrld, file = "treatment_rlog.csv")  

# Keep significant DEGs (padj < 0.05)
DESeq_treatment_master <- read.csv("DESeq_treatment.csv")
DEGs <- DESeq_treatment_master %>% filter(padj < 0.05)
# 42 DEGs
write.csv(DEGs, file = "treatment_sig_DEGs.csv")

# Annotate the significant DEGs
colnames(DEGs)[colnames(DEGs)=="X"] <- "Entry"
DEGs
dim(DEGs)
# 42 unannotated DEGs
uniprot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/uniprot.csv")
master <- merge(DEGs, uniprot, by="Entry")
logs <- read.csv("treatment_rlog.csv")
names(logs)
names(logs)[colnames(logs)=="X"] <- "Entry"
sig_master <- merge(master,logs, by="Entry")
# 42 annotated DEGs
write.csv(sig_master, file = "treatment_annotated_sig_DEGs.csv")
```

## PCA

```{r}
# Remove all columns except for "Entry ID" and the counts
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/past/DESeq2/")
countData <- read.csv("treatment_rlog.csv", row.names="X")
colnames(countData)
Gene_expression <- as.matrix(countData)
Metadata <- as.data.frame(read.csv("./../../../metaData.csv", header = TRUE, row.names = "name"))
pastMetadata <- Metadata[Metadata$species == 'past',]
rownames(pastMetadata)
colnames(Gene_expression)
p <- pca(Gene_expression, metadata = pastMetadata,center = TRUE,removeVar = 0.1) #Remove the lower 10% of variables based on variance. 
screeplot(p)

pastMetadata
pdf(file="rlog_PCA.pdf",height=6,width=6)
biplot(p,
       lab = pastMetadata$sample_number,
       colby = "treatment", colkey = c('control'='blue','disease' = 'red'), colLegendTitle = 'Treatment',
       ellipse = TRUE,
       xlim=c(-150,150), ylim = c(-150,110),
       title = 'Rlog Gene Expression PCA',
       legendPosition = 'right') 
  dev.off()
```

## GO Enrichment Set Up

Get files ready for ShinyGO (unline GUI)

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/past/DESeq2/")
all_genes <- read.csv("DESeq_treatment.csv")
colnames(all_genes)[colnames(all_genes)=="X"] <- "Entry"
names(all_genes)
all_genes <- all_genes[,c(1,3,7)]

uniprot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/uniprot.csv")
names(uniprot)
uniprot <- uniprot[,c(1,3)]

master <- merge(all_genes,uniprot,by="Entry")
master
names(master)
master <- master %>% relocate(Gene.names, .after = Entry)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/past/DEG_GOEnrichment/")
master$Gene.names <- sub(" .*", "", master$Gene.names)
names(master)
master
write.csv(master,file="all_genes.csv")


DEGs <- read.csv("./../DESeq2/treatment_annotated_sig_DEGs.csv")
names(DEGs)
DEGs <- DEGs[,c(2,4,8,10)] #grab Entry, log2FoldChange, padj, and Gene.names
names(DEGs)
DEGs <- DEGs %>% relocate(Gene.names, .after =Entry)
DEGs$Gene.names <- sub(" .*", "", DEGs$Gene.names)
DEGs
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/past/DEG_GOEnrichment/")
write.csv(DEGs,file="past_DEGs.csv")
```

Navigate to http://bioinformatics.sdstate.edu/go/
Select "Human" from the drop down menu
Copy and paste Gene.names from "past_DEGs.csv" 
Click "Background" and copy/paste Gene.names from "all_genes.csv"
Select "GO Biological Process" from drop down menu
Select "0.05" for FDR cutof, "10" for # pathways to show, "2" for Pathway size: Min. and "2000" for Max. 
Select "Remove redundancy" and "Abbreviate pathways"
Wait for results and download "Top Pathways shown above" table with gene IDs as "all_enrichment.csv"

In Excel, sort "all_enrichment.csv" by decreased Fold Enrichment and keep top 5 nonredundant terms, name this file "All_enrichment_curated.csv"

Repeat with same parameters for "GO Molecular Function" drop down

# Pseudodiploria strigosa

## Tximport

Read in the raw counts from Salmon for all pstr samples
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/past/salmon/")
past_samples <- read.table("past_samples.csv", header = TRUE)
past_samples
past_files <- file.path(past_samples$sample, "quant.sf")
names(past_files) <- paste0("sample", 1:11)
all(file.exists(past_files))
past_tx2gene <- read.csv("past_tx2gene.csv")
head(past_tx2gene)
past_txi <- tximport(past_files, type = 'salmon',tx2gene=past_tx2gene)
names(past_txi)
tail(past_txi$counts)
dim(past_txi$counts)
# 13271 contigs with evalue < 0.000001
write.csv(past_txi, file = "past_txi.csv")
write.csv(past_txi$counts, file="past_counts.csv",quote = FALSE)
past_counts <- past_txi$counts
colnames(past_counts)
past_counts <- as.data.frame(past_counts)
colnames(past_counts) <- c("past_c1","past_c5","past_c6","past_c7","past_d1","past_d2","past_d3","past_d4","past_d5","past_d6","past_d7")
past_counts <- tibble::rownames_to_column(past_counts,"Entry")
past_counts[2:12] <-  round(past_counts[2:12], digits=0)
past_counts <- as.data.frame(past_counts)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/past/DESeq2//")
write.csv(past_counts, file = "past_counts.csv",quote = FALSE)
```
pstr: c3,c5,c7,c8,d3,d5,d7
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/pstr/salmon/")
pstr_samples <- read.table("pstr_samples.csv", header = TRUE)
pstr_samples
pstr_files <- file.path(pstr_samples$sample, "quant.sf")
names(pstr_files) <- paste0("sample", 1:7)
all(file.exists(pstr_files))
pstr_tx2gene <- read.csv("pstr_tx2gene.csv")
head(pstr_tx2gene)
pstr_txi <- tximport(pstr_files, type = 'salmon',tx2gene=pstr_tx2gene)
names(pstr_txi)
tail(pstr_txi$counts)
dim(pstr_txi$counts)
# 9796 contigs with evalue < 0.000001
write.csv(pstr_txi, file = "pstr_txi.csv")
write.csv(pstr_txi$counts, file="pstr_counts.csv",quote = FALSE)
pstr_counts <- pstr_txi$counts
colnames(pstr_counts)
pstr_counts <- as.data.frame(pstr_counts)
colnames(pstr_counts) <- c("pstr_c3","pstr_c5","pstr_c7","pstr_c8","pstr_d3","pstr_d5","pstr_d7")
pstr_counts <- tibble::rownames_to_column(pstr_counts,"Entry")
pstr_counts[2:8] <-  round(pstr_counts[2:8], digits=0)
pstr_counts <- as.data.frame(pstr_counts)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/pstr/DESeq2/")
write.csv(pstr_counts, file = "pstr_counts.csv",quote = FALSE)
```

## DESeq2

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/pstr/DESeq2/")
countData <- read.csv("pstr_counts.csv", row.names="Entry")
colnames(countData)
countData <- countData[,-c(1)]
colData <- read.csv("./../../../metaData.csv", row.names = "name")
pstrColData <- colData[colData$species == 'pstr',]
rownames(pstrColData)
colnames(countData)
# Set the design. This design will test for the differences in expression caused by treatment 
# Filter out columns with an average number of reads less than 10. 
dds_treatment <- DESeqDataSetFromMatrix(countData = countData, 
                                        colData = pstrColData, 
                                        design = ~genotype + treatment )
dds_treatment <- dds_treatment[ rowMeans(counts(dds_treatment)) > 10, ] 
test <- DESeq(dds_treatment)
res <- results(test)
res
resorderd <-res[order(res$padj),] 
write.csv(resorderd, file = "DESeq_treatment.csv")
rld <- rlog(dds_treatment, blind=FALSE) 
rrld <- assay(rld)
write.csv(rrld, file = "treatment_rlog.csv")  

# Keep significant DEGs (padj < 0.05)
DESeq_treatment_master <- read.csv("DESeq_treatment.csv")
DEGs <- DESeq_treatment_master %>% filter(padj < 0.05)
# 90 DEGs
write.csv(DEGs, file = "treatment_sig_DEGs.csv")

# Annotate the significant DEGs
colnames(DEGs)[colnames(DEGs)=="X"] <- "Entry"
DEGs
dim(DEGs)
# 90 unannotated DEGs
uniprot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/uniprot.csv")
master <- merge(DEGs, uniprot, by="Entry")
logs <- read.csv("treatment_rlog.csv")
names(logs)
names(logs)[colnames(logs)=="X"] <- "Entry"
sig_master <- merge(master,logs, by="Entry")
# 90 annotated DEGs
write.csv(sig_master, file = "treatment_annotated_sig_DEGs.csv")
```

## PCA

```{r}
# Remove all columns except for "Entry ID" and the counts
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/pstr/DESeq2/")
countData <- read.csv("treatment_rlog.csv", row.names="X")
colnames(countData)
Gene_expression <- as.matrix(countData)
Metadata <- as.data.frame(read.csv("./../../../metaData.csv", header = TRUE, row.names = "name"))
pstrMetadata <- Metadata[Metadata$species == 'pstr',]
rownames(pstrMetadata)
colnames(Gene_expression)
p <- pca(Gene_expression, metadata = pstrMetadata,center = TRUE,removeVar = 0.1) #Remove the lower 10% of variables based on variance. 
screeplot(p)

pstrMetadata
pdf(file="rlog_PCA.pdf",height=6,width=6)
biplot(p,
       lab = pstrMetadata$sample_number,
       colby = "treatment", colkey = c('control'='blue','disease' = 'red'), colLegendTitle = 'Treatment',
       ellipse = TRUE,
       xlim=c(-50,80), ylim = c(-80,65),
       title = 'Rlog Gene Expression PCA',
       legendPosition = 'right') 
  dev.off()
```

## GO Enrichment Set Up

Get files ready for ShinyGO (unline GUI)

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/past/DESeq2/")
all_genes <- read.csv("DESeq_treatment.csv")
colnames(all_genes)[colnames(all_genes)=="X"] <- "Entry"
names(all_genes)
all_genes <- all_genes[,c(1,3,7)]

uniprot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/uniprot.csv")
names(uniprot)
uniprot <- uniprot[,c(1,3)]

master <- merge(all_genes,uniprot,by="Entry")
master
names(master)
master <- master %>% relocate(Gene.names, .after = Entry)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/pstr/DEG_GOEnrichment/")
master$Gene.names <- sub(" .*", "", master$Gene.names)
names(master)
master
write.csv(master,file="all_genes.csv")


DEGs <- read.csv("./../DESeq2/treatment_annotated_sig_DEGs.csv")
names(DEGs)
DEGs <- DEGs[,c(2,4,8,10)] #grab Entry, log2FoldChange, padj, and Gene.names
names(DEGs)
DEGs <- DEGs %>% relocate(Gene.names, .after =Entry)
DEGs$Gene.names <- sub(" .*", "", DEGs$Gene.names)
DEGs
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/pstr/DEG_GOEnrichment/")
write.csv(DEGs,file="pstr_DEGs.csv")
```

Navigate to http://bioinformatics.sdstate.edu/go/
Select "Human" from the drop down menu
Copy and paste Gene.names from "pstr_DEGs.csv" 
Click "Background" and copy/paste Gene.names from "all_genes.csv"
Select "GO Biological Process" from drop down menu
Select "0.05" for FDR cutof, "10" for # pathways to show, "2" for Pathway size: Min. and "2000" for Max. 
Select "Remove redundancy" and "Abbreviate pathways"
Wait for results and download "Top Pathways shown above" table with gene IDs as "all_enrichment.csv"

In Excel, sort "all_enrichment.csv" by decreased Fold Enrichment and keep top 5 nonredundant terms, name this file "All_enrichment_curated.csv"

Repeat with same parameters for "GO Molecular Function" drop down

# Analysis of Shared Genes (inferred homologs)

## Merge Raw Counts
Obtain shared genes
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/")
cnat <- read.csv("./cnat/DESeq2/cnat_counts.csv")
cnat <- cnat[,-c(1)]
names(cnat)
# [1] "Entry"   "cnat_c2" "cnat_c3" "cnat_c4" "cnat_c6" "cnat_c8" "cnat_d2" "cnat_d3" "cnat_d4" "cnat_d6" "cnat_d8"

mcav <- read.csv("./mcav/DESeq2/mcav_counts.csv")
mcav <- mcav[,-c(1)]
names(mcav)
# [1] "Entry"   "mcav_c1" "mcav_c3" "mcav_c5" "mcav_c6" "mcav_c7" "mcav_c8" "mcav_d1" "mcav_d2" "mcav_d3" "mcav_d4" "mcav_d5" "mcav_d6" "mcav_d8"

oann <- read.csv("./oann/DESeq2/oann_counts.csv")
oann <- oann[,-c(1)]
names(oann)
#[1] "Entry"   "oann_c1" "oann_c2" "oann_c3" "oann_c6" "oann_c7" "oann_c8" "oann_d1" "oann_d3" "oann_d6" "oann_d7" "oann_d8"

past <- read.csv("./past/DESeq2/past_counts.csv")
past <- past[,-c(1)]
names(past)
# [1] "Entry"   "past_c1" "past_c5" "past_c6" "past_c7" "past_d1" "past_d2" "past_d3" "past_d4" "past_d5" "past_d6" "past_d7"

pstr <- read.csv("./pstr/DESeq2/pstr_counts.csv")
pstr <- pstr[,-c(1)]
names(pstr)
# [1] "Entry"   "pstr_c3" "pstr_c5" "pstr_c7" "pstr_c8" "pstr_d3" "pstr_d5" "pstr_d7"

dim(cnat) #13,692
dim(mcav) #11,628
dim(oann) #13,138
dim(past) #13,271
dim(pstr) #9796

cnat_mcav <- merge(cnat,mcav,by="Entry") 
names(cnat_mcav)
dim(cnat_mcav) #5,561

cnat_mcav_oann <- merge(cnat_mcav,oann,by="Entry")
names(cnat_mcav_oann)
dim(cnat_mcav_oann) #4,405

cnat_mcav_oann_past <- merge(cnat_mcav_oann, past, by="Entry")
names(cnat_mcav_oann_past)
dim(cnat_mcav_oann_past) #3,121

master <- merge(cnat_mcav_oann_past,pstr,by="Entry")
names(master)
dim(master) #2,149

master <- master[ !duplicated(master$Entry), ]  
dim(master) #2,149

getwd()
setwd("./all/DESeq2/")
write.csv(master,file = "sharedGenes_counts.csv")
```

## Venn Diagram - Shared Genes
```{r}
# Manipulate 'x' with hashes and plot combinations of shared genes with ggvenn to obtain values for venn.plot (five-way Venn requires manual value input)

x <- list(
  #cnat = cnat$Entry,
  #oann = oann$Entry,
  #pstr = pstr$Entry,
  past = past$Entry,
  mcav = mcav$Entry
)

ggvenn(
  x,
  #fill_color = c("#0073C2FF", "#EFC000FF", "#868686FF", "#CD534CFF"),
  stroke_size = 0.5, set_name_size = 5
)
dev.off()

pdf(file = "sharedGenes_Venn.pdf",height = 5,width = 6)
venn.plot <- draw.quintuple.venn(
  area1 = 13692,area2 = 13138,area3 = 9796,area4 = 13271,area5 = 11628,
  n12 = 6349,n13 = 5799,n14 = 5719,n15 = 5561,
  n23 = 5140,n24 = 6408,n25 = 7401,
  n34 = 4618,n35 = 4588,
  n45 = 5908,
  n123 = 3870, n124 = 3993,n125 = 4405,
  n134 = 3396,n135 = 3415,n145 = 3632,
  n234 = 3361,n235 = 3659,n245 = 4630,
  n345 = 3045,n1234 = 2699,n1235 = 2880,n1245 = 3121,
  n1345 = 2425,n2345 = 2613,
  n12345 = 2149,
  category = c("CNAT\n n=13629", "OANN\n n=13138", "PSTR\n n=9796", "PAST\n n=13271", "MCAV\n n=11628"),
  fill = c("dodgerblue", "goldenrod1", "darkorange1", "seagreen3", "orchid3"),
  cat.col = c("dodgerblue", "goldenrod1", "darkorange1", "seagreen3", "orchid3"),
    cat.cex = 2,
  margin = 0.05,
  cex = c(
    1.5, 1.5, 1.5, 1.5, 1.5, 1, 0.8, 1, 0.8, 1, 0.8, 1, 0.8, 1, 0.8,
    1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 1, 1, 1, 1, 1.5),
  ind = TRUE)
dev.off()

```

## Venn Diagram - DEGs

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/")
cnat <- read.csv("./cnat/DESeq2/treatment_annotated_sig_DEGs.csv")
cnat <- cnat[,c(2,14:23)]
names(cnat)
# [1] "Entry"   "cnat_c2" "cnat_c3" "cnat_c4" "cnat_c6" "cnat_c8" "cnat_d2" "cnat_d3" "cnat_d4" "cnat_d6" "cnat_d8"

mcav <- read.csv("./mcav/DESeq2/treatment_annotated_sig_DEGs.csv")
mcav <- mcav[,c(2,14:26)]
names(mcav)
# [1] "Entry"   "mcav_c1" "mcav_c3" "mcav_c5" "mcav_c6" "mcav_c7" "mcav_c8" "mcav_d1" "mcav_d2" "mcav_d3" "mcav_d4" "mcav_d5" "mcav_d6" "mcav_d8"

oann <- read.csv("./oann/DESeq2/treatment_annotated_sig_DEGs.csv")
oann <- oann[,c(2,14:24)]
names(oann)
#[1] "Entry"   "oann_c1" "oann_c2" "oann_c3" "oann_c6" "oann_c7" "oann_c8" "oann_d1" "oann_d3" "oann_d6" "oann_d7" "oann_d8"

past <- read.csv("./past/DESeq2/treatment_annotated_sig_DEGs.csv")
past <- past[,c(2,14:24)]
names(past)
# [1] "Entry"   "past_c1" "past_c5" "past_c6" "past_c7" "past_d1" "past_d2" "past_d3" "past_d4" "past_d5" "past_d6" "past_d7"

pstr <- read.csv("./pstr/DESeq2/treatment_annotated_sig_DEGs.csv")
pstr <- pstr[,c(2,14:20)]
names(pstr)
# [1] "Entry"   "pstr_c3" "pstr_c5" "pstr_c7" "pstr_c8" "pstr_d3" "pstr_d5" "pstr_d7"

dim(cnat) #1350
dim(mcav) #385
dim(oann) #229
dim(past) #42
dim(pstr) #90

x <- list(
  #cnat = cnat$Entry,
  oann = oann$Entry,
  #pstr = pstr$Entry,
  #past = past$Entry,
  mcav = mcav$Entry
)

ggvenn(
  x,
  #fill_color = c("#0073C2FF", "#EFC000FF", "#868686FF", "#CD534CFF"),
  stroke_size = 0.5, set_name_size = 5
)
dev.off()

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/figures/")
pdf(file = "DEGs_Venn.pdf",height = 5,width = 6)
venn.plot <- draw.quintuple.venn(
  area1 = 1350,area2 = 229,area3 = 90,area4 = 42,area5 = 385,
  n12 = 45,n13 = 14,n14 = 9,n15 = 71,
  n23 = 4,n24 = 3,n25 = 28,
  n34 = 0,n35 = 9,
  n45 = 7,
  n123 = 3, n124 = 1,n125 = 10,
  n134 = 0,n135 = 2,n145 = 3,
  n234 = 0,n235 = 3,n245 = 2,
  n345 = 0,n1234 = 0,n1235 = 2,n1245 = 1,
  n1345 = 0,n2345 = 0,
  n12345 = 0,
  category = c("CNAT", "OANN", "PSTR", "PAST", "MCAV"),
  fill = c("dodgerblue", "goldenrod1", "darkorange1", "seagreen3", "orchid3"),
  cat.col = c("dodgerblue", "goldenrod1", "darkorange1", "seagreen3", "orchid3"),
  #  cat.cex = 2,
  margin = 0.05,
  cex = c(
    1.5, 1.5, 1.5, 1.5, 1.5, 1, 0.8, 1, 0.8, 1, 0.8, 1, 0.8, 1, 0.8,
    1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 1, 1, 1, 1, 1.5),
  ind = TRUE)
dev.off()

```

## DESeq2

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/DESeq2/")
countData <- read.csv("sharedGenes_counts.csv", row.names="Entry")
colnames(countData)
colData <- read.csv("metaData.csv", row.names = "name")
colData
rownames(colData)
colnames(countData)
countData <- countData[,-c(1)]
colnames(colData)
# Set the design. This design will test for the differences in expression caused by treatment 
# Filter out columns with an average number of reads less than 10. 
dds_treatment <- DESeqDataSetFromMatrix(countData = countData, 
                                        colData = colData, 
                                        design = ~species + treatment )
dds_treatment <- dds_treatment[ rowMeans(counts(dds_treatment)) > 10, ] 
test <- DESeq(dds_treatment)
res <- results(test)
res
resorderd <-res[order(res$padj),] 
write.csv(resorderd, file = "DESeq_treatment.csv")
rld <- rlog(dds_treatment, blind=FALSE) 
rrld <- assay(rld)
write.csv(rrld, file = "treatment_rlog.csv")  
rlog <- read.csv("treatment_rlog.csv")
```

## PCA
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/DESeq2")
countData <- read.csv("treatment_rlog.csv", row.names="X")
colnames(countData)
Gene_expression <- as.matrix(countData)
Metadata <- as.data.frame(read.csv("metaData.csv", header = TRUE, row.names = "name"))
rownames(Metadata)
colnames(Gene_expression)
p <- pca(Gene_expression, metadata = Metadata,center = TRUE,removeVar = 0.1) #Remove the lower 10% of variables based on variance. 
screeplot(p,
          components = 1:10,)

pdf(file="rlog_PCA_all.pdf",height=7,width=7)
biplot(p, 
       lab = NULL,
       #showLoadings = TRUE,
       shape = "treatment", shapeLegendTitle = "Treatment",
       ellipse = TRUE,
       colby = "species", colLegendTitle = "Species",
       xlim=c(-60,80), ylim = c(-75,80),
       title = 'Rlog Gene Expression PCA',
       legendPosition = 'right')
dev.off()
```

## Immune Gene Comparison

Prepare annotated shared genes file

Upload Entry names from "treatment_rlog.csv" from the DESeq2 analysis of shared genes into Uniprot (https://www.uniprot.org/id-mapping). Set the column names to: "Entry Name" "Gene Names" "Protein names" "Function [CC]" "Keywords" "Gene Ontology (GO)"
Save as Excel file, open in Excel, then save as "sharedGenes_uniprot.csv"
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/DESeq2/")

all_rlog <- read.csv("treatment_rlog.csv")   #2147 shared uniprot IDs
colnames(all_rlog)[colnames(all_rlog)=="X"] <- "Entry"
uniprot <- read.csv("sharedGenes_uniprot.csv")

names(all_rlog)
names(uniprot)
uniprot <- uniprot[,-c(1)] 

sharedGenesAnnot <- left_join(all_rlog,uniprot,by="Entry")
names(sharedGenesAnnot)
sharedGenesAnnot <- sharedGenesAnnot %>% relocate(Entry.name, .after = Entry)
sharedGenesAnnot <- sharedGenesAnnot %>% relocate(Gene.names, .after = Entry.name)
sharedGenesAnnot <- sharedGenesAnnot %>% relocate(Protein.names, .after = Gene.names)
sharedGenesAnnot <- sharedGenesAnnot %>% relocate(Function..CC., .after = Protein.names)
sharedGenesAnnot <- sharedGenesAnnot %>% relocate(Keywords, .after = Function..CC.)
sharedGenesAnnot <- sharedGenesAnnot %>% relocate(Gene.ontology..GO., .after = Keywords)
write.csv(sharedGenesAnnot,file="sharedGenesAnnot.csv")
```

### Immune/Viral Response
Look for immune/viral response related terms in the dataset
```{bash}
DIR=/Users/kbeavers/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/DESeq2

grep -c -e "immune" -e "immunity" -e "viral" -e "virus" sharedGenesAnnot.csv # 150
grep -e "immune" -e "immunity" -e "viral" -e "virus" sharedGenesAnnot.csv > immune_viral_sharedGenesAnnot.csv
```

Run an ANOVA on those genes and then for significant ones run Tukey tests to see if expression is significantly different between control and disease-infected, disease-infected and disease-exposed, and control and disease-exposed
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/DESeq2/")
genes <- read.csv("sharedGenesAnnot.csv")
immune <- read.csv("immune_viral_sharedGenesAnnot.csv",header = FALSE)
names(immune) <- names(genes)
colnames(immune)
immune <- immune[,-c(1,3:8)]
dim(immune)
# [1] 150  53
names(immune)
t_immune <- setNames(data.frame(t(immune[,-1])), immune[,1])
row.names(t_immune)
t_immune <- tibble::rownames_to_column(t_immune, "name")

label <- read.csv("metaData.csv")
colnames(label)
label <- label[,c(4,5,6,8)]
master <- merge(label,t_immune,by="name")
dim(master)
write.csv(master,file="immuneAndViralGenesANOVA.csv")

# Perform ANOVA on the 150 immune genes
names(master)
# Example of one test:
A0A0R4IBK5 <- aov(master$A0A0R4IBK5 ~ master$status, data = master)
summary(A0A0R4IBK5)
A7SFB5 <- aov(master$A7SFB5 ~ master$status, data = master)
summary(A7SFB5)
A7SLZ2 <- aov(master$A7SLZ2 ~ master$status, data = master)
summary(A7SLZ2)

A7YY53 <- aov(master$A7YY53 ~ master$status, data = master)
summary(A7YY53) 
TukeyHSD(A7YY53) # Significant between healthy and exposed

B0S6T2 <- aov(master$B0S6T2 ~ master$status, data = master)
summary(B0S6T2)
C6FG12 <- aov(master$C6FG12 ~ master$status, data = master)
summary(C6FG12)
E9Q5F9 <- aov(master$E9Q5F9 ~ master$status, data = master)
summary(E9Q5F9)

H9BW96 <- aov(master$H9BW96 ~ master$status, data = master)
summary(H9BW96)
TukeyHSD(H9BW96) # Significant between exposed and disease; healthy and disease

H9D1R1 <- aov(master$H9D1R1 ~ master$status, data = master)
summary(H9D1R1)
O00410 <- aov(master$O00410 ~ master$status, data = master)
summary(O00410)
O08623 <- aov(master$O08623 ~ master$status, data = master)
summary(O08623)
O14776 <- aov(master$O14776 ~ master$status, data = master)
summary(O14776)
O15118 <- aov(master$O15118 ~ master$status, data = master)
summary(O15118)
O15439 <- aov(master$O15439 ~ master$status, data = master)
summary(O15439)
O54781 <- aov(master$O54781 ~ master$status, data = master)
summary(O54781)

O60447 <- aov(master$O60447 ~ master$status, data = master)
summary(O60447)
TukeyHSD(O60447) # Significant between healthy and exposed

O60462 <- aov(master$O60462 ~ master$status, data = master)
summary(O60462)
TukeyHSD(O60462) # Significant between exposed and disease

O89110 <- aov(master$O89110 ~ master$status, data = master)
summary(O89110)
O95639 <- aov(master$O95639 ~ master$status, data = master)
summary(O95639)
P03360 <- aov(master$P03360 ~ master$status, data = master)
summary(P03360)
P04323 <- aov(master$P04323 ~ master$status, data = master)
summary(P04323)

P07314 <- aov(master$P07314 ~ master$status, data = master)
summary(P07314)
TukeyHSD(P07314) # Significant between healthy and exposed

P09057 <- aov(master$P09057 ~ master$status, data = master)
summary(P09057)

P09884 <- aov(master$P09884 ~ master$status, data = master)
summary(P09884)
TukeyHSD(P09884) # Significant between exposed and disease

P09917 <- aov(master$P09917 ~ master$status, data = master)
summary(P09917)
TukeyHSD(P09917) # Significant between healthy and disease

P10394 <- aov(master$P10394 ~ master$status, data = master)
summary(P10394)

P14316 <- aov(master$P14316 ~ master$status, data = master)
summary(P14316)
TukeyHSD(P14316) # Significant between exposed and disease

P15170 <- aov(master$P15170 ~ master$status, data = master)
summary(P15170)
TukeyHSD(P15170) # Significant between exposed and disease; healthy and disease

P19440 <- aov(master$P19440 ~ master$status, data = master)
summary(P19440)
P20693 <- aov(master$P20693 ~ master$status, data = master)
summary(P20693)
P20749 <- aov(master$P20749 ~ master$status, data = master)
summary(P20749)
P20825 <- aov(master$P20825 ~ master$status, data = master)
summary(P20825)
P22897 <- aov(master$P22897 ~ master$status, data = master)
summary(P22897)
P25228 <- aov(master$P25228 ~ master$status, data = master)
summary(P25228)
P25291 <- aov(master$P25291 ~ master$status, data = master)
summary(P25291)
P25787 <- aov(master$P25787 ~ master$status, data = master)
summary(P25787)
P26686 <- aov(master$P26686 ~ master$status, data = master)
summary(P26686)
P27590 <- aov(master$P27590 ~ master$status, data = master)
summary(P27590)
P28799 <- aov(master$P28799 ~ master$status, data = master)
summary(P28799)
P30825 <- aov(master$P30825 ~ master$status, data = master)
summary(P30825)
P31266 <- aov(master$P31266 ~ master$status, data = master)
summary(P31266)
P34528 <- aov(master$P34528 ~ master$status, data = master)
summary(P34528)
P36993 <- aov(master$P36993 ~ master$status, data = master)
summary(P36993)
P43404 <- aov(master$P43404 ~ master$status, data = master)
summary(P43404)
P49815 <- aov(master$P49815 ~ master$status, data = master)
summary(P49815)
P50591 <- aov(master$P50591 ~ master$status, data = master)
summary(P50591)
P56517 <- aov(master$P56517 ~ master$status, data = master)
summary(P56517)

P61222 <- aov(master$P61222 ~ master$status, data = master)
summary(P61222)
TukeyHSD(P61222) # Significant between exposed and disease; healthy and disease

P70196 <- aov(master$P70196 ~ master$status, data = master)
summary(P70196)
P70618 <- aov(master$P70618 ~ master$status, data = master)
summary(P70618)
P79145 <- aov(master$P79145 ~ master$status, data = master)
summary(P79145)
P92208 <- aov(master$P92208 ~ master$status, data = master)
summary(P92208)
P97864 <- aov(master$P97864 ~ master$status, data = master)
summary(P97864)
Q04721 <- aov(master$Q04721 ~ master$status, data = master)
summary(Q04721)
Q04861 <- aov(master$Q04861 ~ master$status, data = master)
summary(Q04861)
Q05086 <- aov(master$Q05086 ~ master$status, data = master)
summary(Q05086)
Q05876 <- aov(master$Q05876 ~ master$status, data = master)
summary(Q05876)
Q07113 <- aov(master$Q07113 ~ master$status, data = master)
summary(Q07113)
Q07352 <- aov(master$Q07352 ~ master$status, data = master)
summary(Q07352)
Q09328 <- aov(master$Q09328 ~ master$status, data = master)
summary(Q09328)
Q0VC71 <- aov(master$Q0VC71 ~ master$status, data = master)
summary(Q0VC71)
Q13114 <- aov(master$Q13114 ~ master$status, data = master)
summary(Q13114)
Q13217 <- aov(master$Q13217 ~ master$status, data = master)
summary(Q13217)
Q13435 <- aov(master$Q13435 ~ master$status, data = master)
summary(Q13435)
Q13951 <- aov(master$Q13951 ~ master$status, data = master)
summary(Q13951)
Q14974 <- aov(master$Q14974 ~ master$status, data = master)
summary(Q14974)
Q15075 <- aov(master$Q15075 ~ master$status, data = master)
summary(Q15075)
Q15542 <- aov(master$Q15542 ~ master$status, data = master)
summary(Q15542)
Q24180 <- aov(master$Q24180 ~ master$status, data = master)
summary(Q24180)
Q27084 <- aov(master$Q27084 ~ master$status, data = master)
summary(Q27084)
Q2EMV9 <- aov(master$Q2EMV9 ~ master$status, data = master)
summary(Q2EMV9)
Q2KJC3 <- aov(master$Q2KJC3 ~ master$status, data = master)
summary(Q2KJC3)
Q2TBJ5 <- aov(master$Q2TBJ5 ~ master$status, data = master)
summary(Q2TBJ5)
Q3TLD5 <- aov(master$Q3TLD5 ~ master$status, data = master)
summary(Q3TLD5)
Q3UDE2 <- aov(master$Q3UDE2 ~ master$status, data = master)
summary(Q3UDE2)
Q3UIR3 <- aov(master$Q3UIR3 ~ master$status, data = master)
summary(Q3UIR3)
Q3ZCC3 <- aov(master$Q3ZCC3 ~ master$status, data = master)
summary(Q3ZCC3)
Q4V7W3 <- aov(master$Q4V7W3 ~ master$status, data = master)
summary(Q4V7W3)
Q502K2 <- aov(master$Q502K2 ~ master$status, data = master)
summary(Q502K2)
Q5DU56 <- aov(master$Q5DU56 ~ master$status, data = master)
summary(Q5DU56)
Q5EA11 <- aov(master$Q5EA11 ~ master$status, data = master)
summary(Q5EA11)
Q5RG82 <- aov(master$Q5RG82 ~ master$status, data = master)
summary(Q5RG82)
Q5T4S7 <- aov(master$Q5T4S7 ~ master$status, data = master)
summary(Q5T4S7)
Q5XPI3 <- aov(master$Q5XPI3 ~ master$status, data = master)
summary(Q5XPI3)
Q5XPI3 <- aov(master$Q5XPI3 ~ master$status, data = master)
summary(Q5XPI3)

Q60803 <- aov(master$Q60803 ~ master$status, data = master)
summary(Q60803)
TukeyHSD(Q60803) # Significant between healthy and disease

Q60809 <- aov(master$Q60809 ~ master$status, data = master)
summary(Q60809)

Q60997 <- aov(master$Q60997 ~ master$status, data = master)
summary(Q60997)
TukeyHSD(Q60997) # Significant between healthy and disease

Q61382 <- aov(master$Q61382 ~ master$status, data = master)
summary(Q61382)
Q63HN8 <- aov(master$Q63HN8 ~ master$status, data = master)
summary(Q63HN8)
Q64487 <- aov(master$Q64487 ~ master$status, data = master)
summary(Q64487)
Q68F72 <- aov(master$Q68F72 ~ master$status, data = master)
summary(Q68F72)

Q6AYD3 <- aov(master$Q6AYD3 ~ master$status, data = master)
summary(Q6AYD3)
TukeyHSD(Q6AYD3) # significant between healthy and disease

Q6IQ20 <- aov(master$Q6IQ20 ~ master$status, data = master)
summary(Q6IQ20)
Q6TMK4 <- aov(master$Q6TMK4 ~ master$status, data = master)
summary(Q6TMK4)
Q6UWE0 <- aov(master$Q6UWE0 ~ master$status, data = master)
summary(Q6UWE0)
Q7LHG5 <- aov(master$Q7LHG5 ~ master$status, data = master)
summary(Q7LHG5)
Q7RTR2 <- aov(master$Q7RTR2 ~ master$status, data = master)
summary(Q7RTR2)
Q7SIC1 <- aov(master$Q7SIC1 ~ master$status, data = master)
summary(Q7SIC1)
Q86WI1 <- aov(master$Q86WI1 ~ master$status, data = master)
summary(Q86WI1)
Q8BGX0 <- aov(master$Q8BGX0 ~ master$status, data = master)
summary(Q8BGX0)

Q8BH16 <- aov(master$Q8BH16 ~ master$status, data = master)
summary(Q8BH16)
TukeyHSD(Q8BH16) # significant between healthy and disease

Q8BUH7 <- aov(master$Q8BUH7 ~ master$status, data = master)
summary(Q8BUH7)
Q8C9A2 <- aov(master$Q8C9A2 ~ master$status, data = master)
summary(Q8C9A2)

Q8CBH5 <- aov(master$Q8CBH5 ~ master$status, data = master)
summary(Q8CBH5)
TukeyHSD(Q8CBH5) # Significant between exposed and disease

Q8CFN2 <- aov(master$Q8CFN2 ~ master$status, data = master)
summary(Q8CFN2)
Q8CG09 <- aov(master$Q8CG09 ~ master$status, data = master)
summary(Q8CG09)
Q8I7P9 <- aov(master$Q8I7P9 ~ master$status, data = master)
summary(Q8I7P9)
Q8IWZ3 <- aov(master$Q8IWZ3 ~ master$status, data = master)
summary(Q8IWZ3)
Q8L7W2 <- aov(master$Q8L7W2 ~ master$status, data = master)
summary(Q8L7W2)
Q8QG61 <- aov(master$Q8QG61 ~ master$status, data = master)
summary(Q8QG61)
Q8R0J7 <- aov(master$Q8R0J7 ~ master$status, data = master)
summary(Q8R0J7)
Q8R151 <- aov(master$Q8R151 ~ master$status, data = master)
summary(Q8R151)
Q8R3H9 <- aov(master$Q8R3H9 ~ master$status, data = master)
summary(Q8R3H9)
Q8VDM6 <- aov(master$Q8VDM6 ~ master$status, data = master)
summary(Q8VDM6)
Q91X17 <- aov(master$Q91X17 ~ master$status, data = master)
summary(Q91X17)
Q920Q4 <- aov(master$Q920Q4 ~ master$status, data = master)
summary(Q920Q4)
Q92485 <- aov(master$Q92485 ~ master$status, data = master)
summary(Q92485)
Q92541 <- aov(master$Q92541 ~ master$status, data = master)
summary(Q92541)
Q92614 <- aov(master$Q92614 ~ master$status, data = master)
summary(Q92614)
Q92626 <- aov(master$Q92626 ~ master$status, data = master)
summary(Q92626)
Q92824 <- aov(master$Q92824 ~ master$status, data = master)
summary(Q92824)

Q93009 <- aov(master$Q93009 ~ master$status, data = master)
summary(Q93009)
TukeyHSD(Q93009) # significant between exposed and disease

Q95029 <- aov(master$Q95029 ~ master$status, data = master)
summary(Q95029)
Q95218 <- aov(master$Q95218 ~ master$status, data = master)
summary(Q95218)
Q95KV7 <- aov(master$Q95KV7 ~ master$status, data = master)
summary(Q95KV7)
Q96EZ8 <- aov(master$Q96EZ8 ~ master$status, data = master)
summary(Q96EZ8)
Q96J01 <- aov(master$Q96J01 ~ master$status, data = master)
summary(Q96J01)
Q96LT7 <- aov(master$Q96LT7 ~ master$status, data = master)
summary(Q96LT7)
Q99315 <- aov(master$Q99315 ~ master$status, data = master)
summary(Q99315)
Q99549 <- aov(master$Q99549 ~ master$status, data = master)
summary(Q99549)
Q99MN1 <- aov(master$Q99MN1 ~ master$status, data = master)
summary(Q99MN1)
Q9BUZ4 <- aov(master$Q9BUZ4 ~ master$status, data = master)
summary(Q9BUZ4)
Q9BXS5 <- aov(master$Q9BXS5 ~ master$status, data = master)
summary(Q9BXS5)
Q9CZT4 <- aov(master$Q9CZT4 ~ master$status, data = master)
summary(Q9CZT4)
Q9EQU5 <- aov(master$Q9EQU5 ~ master$status, data = master)
summary(Q9EQU5)

Q9H0Z9 <- aov(master$Q9H0Z9 ~ master$status, data = master)
summary(Q9H0Z9)
TukeyHSD(Q9H0Z9)  #Significant between healthy and disease

Q9H8H3 <- aov(master$Q9H8H3 ~ master$status, data = master)
summary(Q9H8H3)
Q9I7F7 <- aov(master$Q9I7F7 ~ master$status, data = master)
summary(Q9I7F7)
Q9NRW4 <- aov(master$Q9NRW4 ~ master$status, data = master)
summary(Q9NRW4)
Q9U8W7 <- aov(master$Q9U8W7 ~ master$status, data = master)
summary(Q9U8W7)
Q9UBU9 <- aov(master$Q9UBU9 ~ master$status, data = master)
summary(Q9UBU9)
Q9UBV2 <- aov(master$Q9UBV2 ~ master$status, data = master)
summary(Q9UBV2)
Q9UDY8 <- aov(master$Q9UDY8 ~ master$status, data = master)
summary(Q9UDY8)
Q9UGM3 <- aov(master$Q9UGM3 ~ master$status, data = master)
summary(Q9UGM3)
Q9UHR5 <- aov(master$Q9UHR5 ~ master$status, data = master)
summary(Q9UHR5)
Q9VMU8 <- aov(master$Q9VMU8 ~ master$status, data = master)
summary(Q9VMU8)
Q9VQ93 <- aov(master$Q9VQ93 ~ master$status, data = master)
summary(Q9VQ93)

Q9VW71 <- aov(master$Q9VW71 ~ master$status, data = master)
summary(Q9VW71)
TukeyHSD(Q9VW71) # Significant between exposed and disease

Q9Y385 <- aov(master$Q9Y385 ~ master$status, data = master)
summary(Q9Y385)
Q9Y5R2 <- aov(master$Q9Y5R2 ~ master$status, data = master)
summary(Q9Y5R2)

# 18 out of the 150 genes with GO terms that include the word "immune" or "immunity" or "virus" or "viral" show significant differences in expression between immune states:

#A7YY53,H9BW96,O60447,O60462,P07314,P09884,P09917,P14316,P15170,P61222,Q60803,Q60997,Q6AYD3,Q8BH16,Q8CBH5,Q93009,Q9H0Z9,Q9VW71

master$status <- factor(master$status, levels = c("healthy","exposed","disease"))


# Make a dataframe including only the significant immune/viral response genes
names(master)
sig_genes <- select(master,c(name,A7YY53,H9BW96,O60447,O60462,P07314,P09884,P09917,P14316,P15170,P61222,Q60803,Q60997,Q6AYD3,Q8BH16,Q8CBH5,Q93009,Q9H0Z9,Q9VW71))

names(master)
dim(sig_genes)  # 52  19
sig_genes <- tibble::column_to_rownames(sig_genes, "name")
t_sig_genes <- as.data.frame(t(sig_genes))
t_sig_genes <- tibble::rownames_to_column(t_sig_genes,"Entry")
getwd()
write.csv(t_sig_genes,file="SigImmuneAndViralGenesRlogs.csv")

t_sig_genes
uniprot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/uniprot.csv")
master <- left_join(t_sig_genes, uniprot,by="Entry")
names(master)
master <- master %>% relocate(Entry.name, .after = Entry)
master <- master %>% relocate(Gene.names, .after = Entry.name)
master <- master %>% relocate(Protein.names, .after = Gene.names)
master <- master %>% relocate(Organism, .after = Protein.names)
master <- master %>% relocate(Gene.ontology..GO., .after = Organism)
master
write.csv(master,file="SigImmuneAndViralGenesRlogsAnnot.csv")

```

Two genes removed because it did not contain the GO terms "immune" "immunity" "viral" or "virus" (O60447,Q93009)

#### Boxplots
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/DESeq2/")
master <- read.csv("immuneAndViralGenesANOVA.csv")
master <- master[,-c(1)]
label <- read.csv("metaData.csv")
colnames(label)
label <- label[,c(1)]
master$species <- label
master <- master %>% relocate(species, .before = everything())

master$status <- gsub("healthy","control", master$status)
master$status <- gsub("disease","infected", master$status)
master$status <- factor(master$status, levels = c("control","exposed","infected"))
master$species <- factor(master$species, levels = c("mcav","past","pstr","oann","cnat"))
```

##### Manganese-dependent ADP-ribose/CDP-alcohol diphosphatase
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
A7YY53 <- compare_means(A7YY53~status,data=master,method = "t.test")
A7YY53

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

A7YY53_status <- 
ggplot(data = master, 
       aes(x = status, y = A7YY53, fill = status)) +
  geom_point(aes(y = A7YY53, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "ADPRibase-Mn") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(A7YY53, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

A7YY53_species <- 
ggplot(data = master, 
       aes(x = species, y = A7YY53, fill = status)) +
  geom_point(aes(y = A7YY53, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "ADPRibase-Mn") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(A7YY53, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
#A7YY53_all <- 
  ggarrange(A7YY53_species,A7YY53_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("ADPRibase-Mn.pdf")
```

##### ** Ras-related protein rab7
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
H9BW96 <- compare_means(H9BW96~status,data=master,method = "t.test")
H9BW96

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

H9BW96_status <- 
ggplot(data = master, 
       aes(x = status, y = H9BW96, fill = status)) +
  geom_point(aes(y = H9BW96, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Ras-related protein rab7") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(H9BW96, y.position = 11.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

H9BW96_species <- 
ggplot(data = master, 
       aes(x = species, y = H9BW96, fill = status)) +
  geom_point(aes(y = H9BW96, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Ras-related protein rab7") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(H9BW96, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/virus/")
#H9BW96_all <- 
  ggarrange(H9BW96_species,H9BW96_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("rab7.pdf")

H9BW96 <- aov(master$H9BW96 ~ master$status, data = master)
H9BW96_Tukey <- TukeyHSD(H9BW96)
TukeyHSD(H9BW96)
plot(H9BW96_Tukey)
```

##### Neuropilin-2
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
O60462 <- compare_means(O60462~status,data=master,method = "t.test")
O60462

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

O60462_status <- 
ggplot(data = master, 
       aes(x = status, y = O60462, fill = status)) +
  geom_point(aes(y = O60462, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Neuropilin-2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(O60462, y.position = 11, step.increase = 0.15, inherit.aes = FALSE,size = 3)

O60462_species <- 
ggplot(data = master, 
       aes(x = species, y = O60462, fill = status)) +
  geom_point(aes(y = O60462, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Neuropilin-2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(O60462, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/virus/")
#O60462_all <- 
  ggarrange(O60462_species,O60462_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("NRP2.pdf")
```

##### Glutathione hydrolase 1 proenzyme
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
P07314 <- compare_means(P07314~status,data=master,method = "t.test")
P07314

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

P07314_status <- 
ggplot(data = master, 
       aes(x = status, y = P07314, fill = status)) +
  geom_point(aes(y = P07314, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Glutathione hydrolase 1 proenzyme") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(P07314, y.position = 10.2, step.increase = 0.15, inherit.aes = FALSE,size = 3)

P07314_species <- 
ggplot(data = master, 
       aes(x = species, y = P07314, fill = status)) +
  geom_point(aes(y = P07314, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Glutathione hydrolase 1 proenzyme") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(A7YY53, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
P07314_all <- 
  ggarrange(P07314_species,P07314_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("Ggt1.pdf")
```

##### DNA polymerase alpha catalytic subunit
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
P09884 <- compare_means(P09884~status,data=master,method = "t.test")
P09884

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

P09884_status <- 
ggplot(data = master, 
       aes(x = status, y = P09884, fill = status)) +
  geom_point(aes(y = P09884, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "DNA polymerase alpha catalytic subunit") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(P09884, y.position = 10, step.increase = 0.15, inherit.aes = FALSE,size = 3)

P09884_species <- 
ggplot(data = master, 
       aes(x = species, y = P09884, fill = status)) +
  geom_point(aes(y = P09884, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "DNA polymerase alpha catalytic subunit") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(P09884, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/virus/")
#P09884_all <- 
  ggarrange(P09884_species,P09884_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("POLA1.pdf")
```

##### ** Polyunsaturated fatty acid 5-lipoxygenase:
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
P09917 <- compare_means(P09917~status,data=master,method = "t.test")
P09917

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

P09917_status <- 
ggplot(data = master, 
       aes(x = status, y = P09917, fill = status)) +
  geom_point(aes(y = P09917, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Polyunsaturated fatty acid 5-lipoxygenase") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(P09917, y.position = 11.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

P09917_species <- 
ggplot(data = master, 
       aes(x = species, y = P09917, fill = status)) +
  geom_point(aes(y = P09917, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Polyunsaturated fatty acid 5-lipoxygenase") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(A7YY53, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
#P09917_all <- 
  ggarrange(P09917_species,P09917_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("ALOX5.pdf")

P09917 <- aov(master$P09917 ~ master$status, data = master)
summary(P09917)
P09917_Tukey <- TukeyHSD(P09917)
TukeyHSD(P09917)
plot(P09917_Tukey)
```

##### ** Interferon regulatory factor 2, IRF-2:
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
P14316 <- compare_means(P14316~status,data=master,method = "t.test")
P14316

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

P14316_status <- 
ggplot(data = master, 
       aes(x = status, y = P14316, fill = status)) +
  geom_point(aes(y = P14316, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Interferon regulatory factor 2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(P14316, y.position = 11.9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

P14316_species <- 
ggplot(data = master, 
       aes(x = species, y = P14316, fill = status)) +
  geom_point(aes(y = P14316, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Interferon regulatory factor 2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(A7YY53, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
P14316_all <- 
  ggarrange(P14316_species,P14316_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("IRF2.pdf")

P14316 <- aov(master$P14316 ~ master$status, data = master)
summary(P14316)
P14316_Tukey <- TukeyHSD(P14316)
TukeyHSD(P14316)
plot(P14316_Tukey)
```

##### Eukaryotic peptide chain release factor subunit 3a, eRF3a
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
P15170 <- compare_means(P15170~status,data=master,method = "t.test")
P15170

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

P15170_status <- 
ggplot(data = master, 
       aes(x = status, y = P15170, fill = status)) +
  geom_point(aes(y = P15170, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Eukaryotic peptide chain release factor subunit 3a, eRF3a") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(P15170, y.position = 10, step.increase = 0.15, inherit.aes = FALSE,size = 3)

P15170_species <- 
ggplot(data = master, 
       aes(x = species, y = P15170, fill = status)) +
  geom_point(aes(y = P15170, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Eukaryotic peptide chain release factor subunit 3a, eRF3a") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(P15170, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/virus/")
P15170_all <- 
  ggarrange(P15170_species,P15170_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("GSP1.pdf")
```

##### ** ATP-binding cassette sub-family E member 1 
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
P61222 <- compare_means(P61222~status,data=master,method = "t.test")
P61222

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

P61222_status <- 
ggplot(data = master, 
       aes(x = status, y = P61222, fill = status)) +
  geom_point(aes(y = P61222, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "ATP-binding cassette sub-family E member 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(P61222, y.position = 9.7, step.increase = 0.15, inherit.aes = FALSE,size = 3)

P61222_species <- 
ggplot(data = master, 
       aes(x = species, y = P61222, fill = status)) +
  geom_point(aes(y = P61222, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "ATP-binding cassette sub-family E member 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(P61222, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/virus/")
#P61222_all <- 
  ggarrange(P61222_species,P61222_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("Abce1.pdf")

P61222 <- aov(master$P61222 ~ master$status, data = master)
TukeyHSD(P61222)
P61222_Tukey <- TukeyHSD(P61222)
plot(P61222_Tukey)
```

##### ** TNF receptor-associated factor 3:
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
Q60803 <- compare_means(Q60803~status,data=master,method = "t.test")
Q60803

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

Q60803_status <- 
ggplot(data = master, 
       aes(x = status, y = Q60803, fill = status)) +
  geom_point(aes(y = Q60803, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "TNF receptor-associated factor 3") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(Q60803, y.position = 12.4, step.increase = 0.15, inherit.aes = FALSE,size = 3)

Q60803_species <- 
ggplot(data = master, 
       aes(x = species, y = Q60803, fill = status)) +
  geom_point(aes(y = Q60803, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "TNF receptor-associated factor 3") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(A7YY53, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
#Q60803_all <- 
  ggarrange(Q60803_species,Q60803_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("TRAF3.pdf")

Q60803 <- aov(master$Q60803 ~ master$status, data = master)
summary(Q60803)
TukeyHSD(Q60803)
Q60803_Tukey <- TukeyHSD(Q60803)
plot(Q60803_Tukey)
```

##### ** Deleted in malignant brain tumors 1 protein:
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
Q60997 <- compare_means(Q60997~status,data=master,method = "t.test")
Q60997

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

Q60997_status <- 
ggplot(data = master, 
       aes(x = status, y = Q60997, fill = status)) +
  geom_point(aes(y = Q60997, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Deleted in malignant brain tumors 1 ") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(Q60997, y.position = 10, step.increase = 0.15, inherit.aes = FALSE,size = 3)

Q60997_species <- 
ggplot(data = master, 
       aes(x = species, y = Q60997, fill = status)) +
  geom_point(aes(y = Q60997, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Deleted in malignant brain tumors 1 ") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(A7YY53, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
Q60997_all <- 
  ggarrange(Q60997_species,Q60997_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("DMBT1.pdf")

Q60997 <- aov(master$Q60997 ~ master$status, data = master)
summary(Q60997)
TukeyHSD(Q60997)
Q60997_Tukey <- TukeyHSD(Q60997)
plot(Q60997_Tukey)
```

#####  Proliferation-associated protein 2G4 
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
Q6AYD3 <- compare_means(Q6AYD3~status,data=master,method = "t.test")
Q6AYD3

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

Q6AYD3_status <- 
ggplot(data = master, 
       aes(x = status, y = Q6AYD3, fill = status)) +
  geom_point(aes(y = Q6AYD3, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Proliferation-associated protein 2G4") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(Q6AYD3, y.position = 9.9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

Q6AYD3_species <- 
ggplot(data = master, 
       aes(x = species, y = Q6AYD3, fill = status)) +
  geom_point(aes(y = Q6AYD3, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Proliferation-associated protein 2G4") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(Q6AYD3, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/virus/")
#Q6AYD3_all <- 
  ggarrange(Q6AYD3_species,Q6AYD3_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("Pa2g4.pdf")

Q6AYD3 <- aov(master$Q6AYD3 ~ master$status, data = master)
summary(Q6AYD3)
TukeyHSD(Q6AYD3)
Q6AYD3_Tukey <- TukeyHSD(Q6AYD3)
plot(Q6AYD3_Tukey)
```

##### F-box/LRR-repeat protein 2
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
Q8BH16 <- compare_means(Q8BH16~status,data=master,method = "t.test")
Q8BH16

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

Q8BH16_status <- 
ggplot(data = master, 
       aes(x = status, y = Q8BH16, fill = status)) +
  geom_point(aes(y = Q8BH16, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "F-box/LRR-repeat protein 2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(Q8BH16, y.position = 7.9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

Q8BH16_species <- 
ggplot(data = master, 
       aes(x = species, y = Q8BH16, fill = status)) +
  geom_point(aes(y = Q8BH16, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "F-box/LRR-repeat protein 2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(Q8BH16, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/virus/")
Q8BH16_all <- 
  ggarrange(Q8BH16_species,Q8BH16_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("Fbxl2.pdf")
```

##### Major facilitator superfamily domain-containing protein 6:
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
Q8CBH5 <- compare_means(Q8CBH5~status,data=master,method = "t.test")
Q8CBH5

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

Q8CBH5_status <- 
ggplot(data = master, 
       aes(x = status, y = Q8CBH5, fill = status)) +
  geom_point(aes(y = Q8CBH5, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Major facilitator superfamily domain-containing protein 6") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(Q8CBH5, y.position = 8.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

Q8CBH5_species <- 
ggplot(data = master, 
       aes(x = species, y = Q8CBH5, fill = status)) +
  geom_point(aes(y = Q8CBH5, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Major facilitator superfamily domain-containing protein 6") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(A7YY53, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
Q8CBH5_all <- 
  ggarrange(Q8CBH5_species,Q8CBH5_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("Mfsd6.pdf")
```

##### ** RNA-binding protein 38:
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
Q9H0Z9 <- compare_means(Q9H0Z9~status,data=master,method = "t.test")
Q9H0Z9

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

Q9H0Z9_status <- 
ggplot(data = master, 
       aes(x = status, y = Q9H0Z9, fill = status)) +
  geom_point(aes(y = Q9H0Z9, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "RNA-binding protein 38") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(Q9H0Z9, y.position = 12.3, step.increase = 0.15, inherit.aes = FALSE,size = 3)

Q9H0Z9_species <- 
ggplot(data = master, 
       aes(x = species, y = Q9H0Z9, fill = status)) +
  geom_point(aes(y = Q9H0Z9, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "RNA-binding protein 38") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(Q9H0Z9, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/virus/")
Q9H0Z9_all <- 
  ggarrange(Q9H0Z9_species,Q9H0Z9_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("RBM38.pdf")

Q9H0Z9 <- aov(master$Q9H0Z9 ~ master$status, data = master)
TukeyHSD(Q9H0Z9)
Q9H0Z9_Tukey <- TukeyHSD(Q9H0Z9)
plot(Q9H0Z9_Tukey)
```

##### Fat-like cadherin-related tumor suppressor homolog:
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
Q9VW71 <- compare_means(Q9VW71~status,data=master,method = "t.test")
Q9VW71

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

Q9VW71_status <- 
ggplot(data = master, 
       aes(x = status, y = Q9VW71, fill = status)) +
  geom_point(aes(y = Q9VW71, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Fat-like cadherin-related tumor suppressor homolog") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(Q9VW71, y.position = 11.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

Q9VW71_species <- 
ggplot(data = master, 
       aes(x = species, y = Q9VW71, fill = status)) +
  geom_point(aes(y = Q9VW71, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Fat-like cadherin-related tumor suppressor homolog") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(A7YY53, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
Q9VW71_all <- 
  ggarrange(Q9VW71_species,Q9VW71_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("kug.pdf")
```

### Extracellular matrix 
Look for ECM related terms in the dataset
```{bash}
DIR=/Users/kbeavers/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/DESeq2

grep -c "extracellular matrix structural constituent" sharedGenesAnnot.csv # 17
grep "extracellular matrix structural constituent" sharedGenesAnnot.csv > ECM_sharedGenesAnnot.csv

```

Run an ANOVA on those genes and then for significant ones run Tukey tests to see if expression is significantly different between control and disease-infected, disease-infected and disease-exposed, and control and disease-exposed

```{r}
library(tibble)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/DESeq2/")
genes <- read.csv("sharedGenesAnnot.csv")
ECM <- read.csv("ECM_sharedGenesAnnot.csv",header = FALSE)
names(ECM) <- names(genes)
colnames(ECM)
ECM <- ECM[,-c(1,3:8)]
dim(ECM)
# [1] 17  53
names(ECM)
t_ECM <- setNames(data.frame(t(ECM[,-1])), ECM[,1])
row.names(t_ECM)
t_ECM <- tibble::rownames_to_column(t_ECM, "name")

label <- read.csv("metaData.csv")
colnames(label)
label <- label[,-c(1,2,6,12:22)]
master <- merge(label,t_ECM,by="name")
dim(master)   #52   79
write.csv(master,file="ECM_ANOVA.csv")

# Perform ANOVA on the 71 ECM genes
names(master)

A1L224 <- aov(master$A1L224 ~ master$status, data = master)
summary(A1L224)
TukeyHSD(A1L224) # Significant between healthy and disease

A2AAY5 <- aov(master$A2AAY5 ~ master$status, data = master)
summary(A2AAY5)
A2AJ76 <- aov(master$A2AJ76 ~ master$status, data = master)
summary(A2AJ76)
A2AVA0 <- aov(master$A2AVA0 ~ master$status, data = master)
summary(A2AVA0)
A6H584 <- aov(master$A6H584 ~ master$status, data = master)
summary(A6H584)
A6NMZ7 <- aov(master$A6NMZ7 ~ master$status, data = master)
summary(A6NMZ7)

B8V7R6 <- aov(master$B8V7R6 ~ master$status, data = master)
summary(B8V7R6)
Collagen
Collagen <- TukeyHSD(B8V7R6) # Significant between exposed and disease; healthy and disease
Collagen$`master$status`[,"p adj"]

D3YXG0 <- aov(master$D3YXG0 ~ master$status, data = master)
summary(D3YXG0)
O00339 <- aov(master$O00339 ~ master$status, data = master)
summary(O00339)
O08807 <- aov(master$O08807 ~ master$status, data = master)
summary(O08807)
O70589 <- aov(master$O70589 ~ master$status, data = master)
summary(O70589)
P07911 <- aov(master$P07911 ~ master$status, data = master)
summary(P07911)
P08254 <- aov(master$P08254 ~ master$status, data = master)
summary(P08254)
P09487 <- aov(master$P09487 ~ master$status, data = master)
summary(P09487)
P10079 <- aov(master$P10079 ~ master$status, data = master)
summary(P10079)
P11532 <- aov(master$P11532 ~ master$status, data = master)
summary(P11532)
P13608 <- aov(master$P13608 ~ master$status, data = master)
summary(P13608)
P13944 <- aov(master$P13944 ~ master$status, data = master)
summary(P13944)
P14211 <- aov(master$P14211 ~ master$status, data = master)
summary(P14211)
P15535 <- aov(master$P15535 ~ master$status, data = master)
summary(P15535)
P20749 <- aov(master$P20749 ~ master$status, data = master)
summary(P20749)
P21956 <- aov(master$P21956 ~ master$status, data = master)
summary(P21956)
P22105 <- aov(master$P22105 ~ master$status, data = master)
summary(P22105)
P22757 <- aov(master$P22757 ~ master$status, data = master)
summary(P22757)
P25291 <- aov(master$P25291 ~ master$status, data = master)
summary(P25291)
P27590 <- aov(master$P27590 ~ master$status, data = master)
summary(P27590)
P35555 <- aov(master$P35555 ~ master$status, data = master)
summary(P35555)
P51655 <- aov(master$P51655 ~ master$status, data = master)
summary(P51655)
P97259 <- aov(master$P97259 ~ master$status, data = master)
summary(P97259)

# 3 out of the 17 genes with GO term "extracellular matrix structural constituent" show significant differences in expression between immune states

#B8V7R6, Q75N90, Q99715

master$status <- factor(master$status, levels = c("healthy","exposed","disease"))


```

Make a dataframe including only the significant ECM genes

```{r}
library(dplyr)
names(master)
sig_genes <- select(master,c(name,B8V7R6,Q75N90,Q99715))

names(master)
dim(sig_genes)  # 52  4
sig_genes <- tibble::column_to_rownames(sig_genes, "name")
t_sig_genes <- as.data.frame(t(sig_genes))
t_sig_genes <- tibble::rownames_to_column(t_sig_genes,"Entry")
getwd()
write.csv(t_sig_genes,file="ECMGenesRlogs.csv")

t_sig_genes
uniprot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/uniprot.csv")
master <- left_join(t_sig_genes, uniprot,by="Entry")
names(master)
master <- master %>% relocate(Entry.name, .after = Entry)
master <- master %>% relocate(Gene.names, .after = Entry.name)
master <- master %>% relocate(Protein.names, .after = Gene.names)
master <- master %>% relocate(Organism, .after = Protein.names)
master <- master %>% relocate(Gene.ontology..GO., .after = Organism)
master
write.csv(master,file="ECMRlogsAnnot.csv")

```

#### Boxplots
```{r}
library(dplyr)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/DESeq2/")
master <- read.csv("ECM_ANOVA.csv")
master <- master[,-c(1)]
label <- read.csv("metaData.csv")
colnames(label)
label <- label[,c(1)]
master$species <- label
master <- master %>% relocate(species, .before = everything())

master$status <- gsub("healthy","control", master$status)
master$status <- gsub("disease","infected", master$status)
master$status <- factor(master$status, levels = c("control","exposed","infected"))
master$species <- factor(master$species, levels = c("mcav","past","pstr","oann","cnat"))
```

##### ** Collagen alpha chain (fragment)
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
B8V7R6 <- compare_means(B8V7R6~status,data=master,method = "t.test")
B8V7R6

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

#B8V7R6_status <- 
ggplot(data = master, 
       aes(x = status, y = B8V7R6, fill = status)) +
  geom_point(aes(y = B8V7R6, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Collagen alpha chain (fragment)") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(B8V7R6, y.position = 16.1, step.increase = 0.15, inherit.aes = FALSE,size = 3)

#B8V7R6_species <- 
ggplot(data = master, 
       aes(x = species, y = B8V7R6, fill = status)) +
  geom_point(aes(y = B8V7R6, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "ADPRibase-Mn") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(B8V7R6, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/DESeq2/figures")
pdf("collagen.pdf",width = 4,height = 6)
ggarrange(B8V7R6_species,B8V7R6_status,
          ncol = 2,nrow = 1,
          widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",
          align = "hv"
          )
dev.off()

```

##### Fibrillin-3
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
Q75N90 <- compare_means(Q75N90~status,data=master,method = "t.test")
Q75N90

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

#Q75N90_status <- 
ggplot(data = master, 
       aes(x = status, y = Q75N90, fill = status)) +
  geom_point(aes(y = Q75N90, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Fibrillin-3") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(Q75N90, y.position = 12.7, step.increase = 0.15, inherit.aes = FALSE,size = 3)

#Q75N90_species <- 
ggplot(data = master, 
       aes(x = species, y = Q75N90, fill = status)) +
  geom_point(aes(y = Q75N90, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Fibrillin-3") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(Q75N90, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

```

##### Collagen alpha-1(XII) chain
```{r}
library(ggpubr)
library(tibble)
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
Q99715 <- compare_means(Q99715~status,data=master,method = "t.test")
Q99715

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

#Q99715_status <- 
ggplot(data = master, 
       aes(x = status, y = Q99715, fill = status)) +
  geom_point(aes(y = Q99715, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Collagen alpha-1(XII) chain") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(Q99715, y.position = 12.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

#Q99715_species <- 
ggplot(data = master, 
       aes(x = species, y = Q99715, fill = status)) +
  geom_point(aes(y = Q99715, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Collagen alpha-1(XII) chain") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(Q99715, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)




ggarrange(P09917_species,P09917_status,
          P14316_species,P14316_status,
          Q60803_species,Q60803_status,
          Q60997_species,Q60997_status,
          H9BW96_species,H9BW96_status,
          P61222_species,P61222_status,
          Q6AYD3_species,Q6AYD3_status,
          Q9H0Z9_species,Q9H0Z9_status,
          ncol = 2,nrow = 8,
          widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",
          align = "hv"
          )
```

### Combined Figure
```{r}
getwd()
setwd("./figures")
pdf("immunePlots_2.pdf",width = 6,height = 20)
ggarrange(P09917_species,P09917_status,
          P14316_species,P14316_status,
          Q60803_species,Q60803_status,
          Q60997_species,Q60997_status,
          H9BW96_species,H9BW96_status,
          P61222_species,P61222_status,
          Q6AYD3_species,Q6AYD3_status,
          Q9H0Z9_species,Q9H0Z9_status,
          ncol = 2,nrow = 8,
          widths = c(1.75,1.25),
          common.legend = TRUE, legend = "bottom",
          align = "hv"
          )
dev.off()

# Remember to change the pvalues and add the ones from the TukeyHSD post hoc test
```

## GO Enrichment on DEGs

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/cnat/DEG_GOEnrichment/BP/")
Cnat_BP_All <- read.csv("All_enrichment_curated.csv")
names(Cnat_BP_All)
Cnat_BP_All <- Cnat_BP_All[,-c(6,7)]
Cnat_BP_All$GO <- "BP"
Cnat_BP_All$Species <- "Cnat"
names(Cnat_BP_All)
Cnat_BP_All$DEGs <- "1350"
Cnat_BP_All$DEGs <- as.numeric(Cnat_BP_All$DEGs) 
Cnat_BP_All$Percent.DEGs <- Cnat_BP_All$nGenes / Cnat_BP_All$DEGs #divide number of genes in pathway by total number of DEGs

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/oann/DEG_GOEnrichment/BP/")
Oann_BP_All <- read.csv("All_enrichment_curated.csv")
names(Oann_BP_All)
Oann_BP_All <- Oann_BP_All[,-c(6,7)]
Oann_BP_All$GO <- "BP"
Oann_BP_All$Species <- "Oann"
names(Oann_BP_All)
Oann_BP_All$DEGs <- "229"
Oann_BP_All$DEGs <- as.numeric(Oann_BP_All$DEGs) 
Oann_BP_All$Percent.DEGs <- Oann_BP_All$nGenes / Oann_BP_All$DEGs #divide number of genes in pathway by total number of DEGs

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/pstr/DEG_GOEnrichment/BP/")
Pstr_BP_All <- read.csv("All_enrichment_curated.csv")
names(Pstr_BP_All)
Pstr_BP_All <- Pstr_BP_All[,-c(6,7)]
Pstr_BP_All$GO <- "BP"
Pstr_BP_All$Species <- "Pstr"
names(Pstr_BP_All)
Pstr_BP_All$DEGs <- "90"
Pstr_BP_All$DEGs <- as.numeric(Pstr_BP_All$DEGs) 
Pstr_BP_All$Percent.DEGs <- Pstr_BP_All$nGenes / Pstr_BP_All$DEGs #divide number of genes in pathway by total number of DEGs

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/past/DEG_GOEnrichment/BP/")
Past_BP_All <- read.csv("All_enrichment_curated.csv")
names(Past_BP_All)
Past_BP_All <- Past_BP_All[,-c(6,7)]
Past_BP_All$GO <- "BP"
Past_BP_All$Species <- "Past"
names(Past_BP_All)
Past_BP_All$DEGs <- "42"
Past_BP_All$DEGs <- as.numeric(Past_BP_All$DEGs) 
Past_BP_All$Percent.DEGs <- Past_BP_All$nGenes / Past_BP_All$DEGs #divide number of genes in pathway by total number of DEGs

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/mcav/DEG_GOEnrichment/BP/")
Mcav_BP_All <- read.csv("All_enrichment_curated.csv")
names(Mcav_BP_All)
Mcav_BP_All <- Mcav_BP_All[,-c(6,7)]
Mcav_BP_All$GO <- "BP"
Mcav_BP_All$Species <- "Mcav"
names(Mcav_BP_All)
Mcav_BP_All$DEGs <- "385"
Mcav_BP_All$DEGs <- as.numeric(Mcav_BP_All$DEGs) 
Mcav_BP_All$Percent.DEGs <- Mcav_BP_All$nGenes / Mcav_BP_All$DEGs #divide number of genes in pathway by total number of DEGs

all_BP <- rbind(Cnat_BP_All,Oann_BP_All,Pstr_BP_All,Past_BP_All,Mcav_BP_All)
all_BP$Species <- factor(all_BP$Species, levels = c("Cnat","Oann","Pstr","Mcav","Past"))
names(all_BP)



setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/cnat/DEG_GOEnrichment/MF/")
Cnat_MF_All <- read.csv("All_enrichment_curated.csv")
names(Cnat_MF_All)
Cnat_MF_All <- Cnat_MF_All[,-c(6,7)]
Cnat_MF_All$GO <- "MF"
Cnat_MF_All$Species <- "Cnat"
names(Cnat_MF_All)
Cnat_MF_All$DEGs <- "1350"
Cnat_MF_All$DEGs <- as.numeric(Cnat_MF_All$DEGs) 
Cnat_MF_All$Percent.DEGs <- Cnat_MF_All$nGenes / Cnat_MF_All$DEGs #divide number of genes in pathway by total number of DEGs

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/oann/DEG_GOEnrichment/MF/")
Oann_MF_All <- read.csv("All_enrichment_curated.csv")
names(Oann_MF_All)
Oann_MF_All <- Oann_MF_All[,-c(6,7)]
Oann_MF_All$GO <- "MF"
Oann_MF_All$Species <- "Oann"
names(Oann_MF_All)
Oann_MF_All$DEGs <- "229"
Oann_MF_All$DEGs <- as.numeric(Oann_MF_All$DEGs) 
Oann_MF_All$Percent.DEGs <- Oann_MF_All$nGenes / Oann_MF_All$DEGs #divide number of genes in pathway by total number of DEGs

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/pstr/DEG_GOEnrichment/MF/")
Pstr_MF_All <- read.csv("All_enrichment_curated.csv")
names(Pstr_MF_All)
Pstr_MF_All <- Pstr_MF_All[,-c(6,7)]
Pstr_MF_All$GO <- "MF"
Pstr_MF_All$Species <- "Pstr"
names(Pstr_MF_All)
Pstr_MF_All$DEGs <- "90"
Pstr_MF_All$DEGs <- as.numeric(Pstr_MF_All$DEGs) 
Pstr_MF_All$Percent.DEGs <- Pstr_MF_All$nGenes / Pstr_MF_All$DEGs #divide number of genes in pathway by total number of DEGs

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/past/DEG_GOEnrichment/MF/")
Past_MF_All <- read.csv("All_enrichment_curated.csv")
names(Past_MF_All)
Past_MF_All <- Past_MF_All[,-c(6,7)]
Past_MF_All$GO <- "MF"
Past_MF_All$Species <- "Past"
names(Past_MF_All)
Past_MF_All$DEGs <- "42"
Past_MF_All$DEGs <- as.numeric(Past_MF_All$DEGs) 
Past_MF_All$Percent.DEGs <- Past_MF_All$nGenes / Past_MF_All$DEGs #divide number of genes in pathway by total number of DEGs

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/mcav/DEG_GOEnrichment/MF/")
Mcav_MF_All <- read.csv("All_enrichment_curated.csv")
names(Mcav_MF_All)
Mcav_MF_All <- Mcav_MF_All[,-c(6,7)]
Mcav_MF_All$GO <- "MF"
Mcav_MF_All$Species <- "Mcav"
names(Mcav_MF_All)
Mcav_MF_All$DEGs <- "385"
Mcav_MF_All$DEGs <- as.numeric(Mcav_MF_All$DEGs) 
Mcav_MF_All$Percent.DEGs <- Mcav_MF_All$nGenes / Mcav_MF_All$DEGs #divide number of genes in pathway by total number of DEGs

all_MF <- rbind(Cnat_MF_All,Oann_MF_All,Pstr_MF_All,Past_MF_All,Mcav_MF_All)
all_MF$Species <- factor(all_MF$Species, levels = c("Cnat","Oann","Pstr","Mcav","Past"))
names(all_MF)

cnat_all <- rbind(Cnat_BP_All,Cnat_MF_All)
oann_all <- rbind(Oann_BP_All,Oann_MF_All)
past_all <- rbind(Past_BP_All,Past_MF_All)
pstr_all <- rbind(Pstr_BP_All,Pstr_MF_All)
mcav_all <- rbind(Mcav_BP_All,Mcav_MF_All)


cnat <- 
  ggplot(data=cnat_all,aes(x=reorder(Pathway,Percent.DEGs),y=Percent.DEGs,fill=Species))+
   geom_point(aes(color=Species,size=-log(Enrichment.FDR)))+
   geom_segment(aes(xend=Pathway,yend=0,color=Species),size=0.75)+
   facet_grid(GO~ ., scales = "free_y")+
   labs(title = "C.natans")+
   ylab("Proportion of DEGs")+
   xlab(NULL)+
   theme_gray()+
   scale_size_continuous(limits=c(1,20), breaks = c(5,10,15,20),name="-log padj")+
   theme(axis.text.y = element_text(angle = 0, size = 4))+
   #scale_x_discrete(labels = function(x) str_wrap(x,width=70))+
   coord_flip()+
   scale_color_manual(values=c("#ff6666"))
cnat

oann <-ggplot(data=oann_all,aes(x=reorder(Pathway,Percent.DEGs),y=Percent.DEGs,fill=Species))+
   geom_point(aes(color=Species,size=-log(Enrichment.FDR)))+
   geom_segment(aes(xend=Pathway,yend=0,color=Species),size=0.75)+
   facet_grid(GO~ ., scales = "free_y")+
   labs(title = "O.annularis")+
   ylab("Proportion of DEGs")+
   xlab(NULL)+
   theme_gray()+
   scale_size_continuous(limits=c(1,20), breaks = c(5,10,15,20),name="-log padj")+
   theme(axis.text.y = element_text(angle = 0, size = 4))+
   #scale_x_discrete(labels = function(x) str_wrap(x,width=70))+
   coord_flip()+
  scale_color_manual(values=c("#ffc125"))
oann

past <- ggplot(data=past_all,aes(x=reorder(Pathway,Percent.DEGs),y=Percent.DEGs,fill=Species))+
   geom_point(aes(color=Species,size=-log(Enrichment.FDR)))+
   geom_segment(aes(xend=Pathway,yend=0,color=Species),size=0.75)+
   facet_grid(GO~ ., scales = "free_y")+
   labs(title = "P.astreoides")+
   ylab("Proportion of DEGs")+
   xlab(NULL)+
   theme_gray()+
   scale_size_continuous(limits=c(1,20), breaks = c(5,10,15,20),name="-log padj")+
   theme(axis.text.y = element_text(angle = 0, size = 4))+
   #scale_x_discrete(labels = function(x) str_wrap(x,width=70))+
   coord_flip()+
    scale_color_manual(values=c("#ff66ff"))
past

pstr <- ggplot(data=pstr_all,aes(x=reorder(Pathway,Percent.DEGs),y=Percent.DEGs,fill=Species))+
   geom_point(aes(color=Species,size=-log(Enrichment.FDR)))+
   geom_segment(aes(xend=Pathway,yend=0,color=Species),size=0.75)+
   facet_grid(GO~ ., scales = "free_y")+
   labs(title = "P.strigosa")+
   ylab("Proportion of DEGs")+
   xlab(NULL)+
   theme_gray()+
   scale_size_continuous(limits=c(1,20), breaks = c(5,10,15,20),name="-log padj")+
   theme(axis.text.y = element_text(angle = 0, size = 4))+
   #scale_x_discrete(labels = function(x) str_wrap(x,width=70))+
   coord_flip()+
   scale_color_manual(values=c("#00cc66"))
pstr

mcav <- ggplot(data=mcav_all,aes(x=reorder(Pathway,Percent.DEGs),y=Percent.DEGs,fill=Species))+
   geom_point(aes(color=Species,size=-log(Enrichment.FDR)))+
   geom_segment(aes(xend=Pathway,yend=0,color=Species),size=0.75)+
   facet_grid(GO~ ., scales = "free_y")+
   labs(title = "M.cavernosa")+
   ylab("Proportion of DEGs")+
   xlab(NULL)+
   theme_gray()+
   scale_size_continuous(limits=c(1,20), breaks = c(5,10,15,20),name="-log padj")+
   theme(axis.text.y = element_text(angle = 0, size = 4))+
   #scale_x_discrete(labels = function(x) str_wrap(x,width=70))+
   coord_flip()+
   scale_color_manual(values=c("#0099ff"))
mcav

ggarrange(cnat,
          oann,
          pstr,
          mcav,
          past,
          ncol = 1,
          common.legend = TRUE, legend = "right",
          align = "hv")
```

# C. natans Study

### PCA
```{r}
# Remove all columns except for "Entry ID" and the counts
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/cnat/DESeq2/")
countData <- read.csv("treatment_rlog.csv", row.names="X")
colnames(countData)
Gene_expression <- as.matrix(countData)
Metadata <- as.data.frame(read.csv("./../../../metaData.csv", header = TRUE, row.names = "name"))
cnatMetadata <- Metadata[Metadata$species == 'cnat',]
rownames(cnatMetadata)
colnames(Gene_expression)
p <- pca(Gene_expression, metadata = cnatMetadata,center = TRUE,removeVar = 0.1) #Remove the lower 10% of variables based on variance. 
screeplot(p)

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/cnat/Symbiont_study/")
pdf(file="cnat_symbiont_PCA.pdf",height = 8,width = 8)
  biplot(p,
       lab = cnatMetadata$sample_number,
       ntopLoadings = 2,
       showLoadings = FALSE,
       sizeLoadingsNames = 2.5,
       fillBoxedLoadings = alpha("white",1/4),
       drawConnectorsLoadings = TRUE,
       lengthLoadingsArrowsFactor = 1.7,
       colby = "dominant_clade", colkey = c('C'='#41B6C4','D' = '#225EA8'), colLegendTitle = 'Dominant \nClade',
       shape = "treatment",
       ellipse = TRUE,
       #hline = 0, vline = 0,
       xlim=c(-145,200), ylim = c(-200,215),
       title = 'Rlog Gene Expression PCA',
       #showLoadings = TRUE,
       legendPosition = 'right') 
dev.off()


# Access the variable loadings
p$loadings[1:nrow(p$loadings),1:3]$PC1
cnat_loadings <- p$loadings[1:nrow(p$loadings),1:3]
getwd()
write.csv(cnat_loadings,file="cnat_symbiont_pca_loadings.csv")
```

Open "cnat_symbiont_pca_loadings.csv" in Excel
Copy top 100 Uniprot IDs and search using "ID mapping" at https://www.uniprot.org/id-mapping
Filter by taxonomy and select "Metazoa"
Set columns to: Entry   Entry.name    Protein.name    Gene.name 
Download uncompressed .xlsx file, open, and save as "metazoaLoadings.csv"


### Heatmap
Plot the expression of the top 20 loadings
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/cnat/Symbiont_study/")
loadings <- read.csv("cnat_symbiont_pca_loadings.csv")
colnames(loadings)[colnames(loadings)=="X"] <- "Entry"
metazoaLoadings <- read.csv("metazoaLoadings.csv")
loadingsMaster <- right_join(loadings, metazoaLoadings,by="Entry")
loadingsMaster <- loadingsMaster[sort(abs(loadingsMaster$PC1),decreasing = T, index.return=T)[[2]],]

loadings <- loadingsMaster[c(1:20),]
rlogs <- read.csv("./../DESeq2/treatment_rlog.csv")
colnames(rlogs)[colnames(rlogs)=="X"] <- "Entry"
master <- left_join(loadings,rlogs,by="Entry")
master <- tibble::column_to_rownames(master,"Protein.name")
names(master)
master <- master[,-c(1:7)]

label <- read.csv("./../../../metaData.csv")
cnatLabel <- label[label$species == 'cnat',]
head(cnatLabel)
names(cnatLabel)
cnatLabel <- cnatLabel[,c(3,7,8)] #name, LGR, dominant_clade
cnatLabel <- tibble::column_to_rownames(cnatLabel,"name")

getwd()
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/cnat/Symbiont_study/")
pdf(file="loadingsHeatmap20metazoa.pdf",height=10,width=12)
pheatmap(master,
         cluster_cols = TRUE,
         cluster_rows = TRUE,
         scale = "none",
         color = colorRampPalette(c("white","#ffeda0","#fed976","#feb24c","#fd8d3c","#fc4e2a","#e31a1c","#bd0026","#800026"))(200),
         border_color = "black",
         annotation_col = cnatLabel,
         legend = TRUE,
         legend_labels = cnatLabel,
         fontsize_row = 6,
         fontsize_col = 6,
         cellwidth = 12,
         cellheight = 8,
         treeheight_col = 10,
         treeheight_row = 50
         )
dev.off()
```
