---
title: "EVE Analysis - Coral Single-Copy Orthologs"
author: "Kelsey Beavers"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Required Packages
```{r}
library(dplyr)
library(tximport)
library(DESeq2)
library(readr)
library(dplyr)
library(ggrepel)
library(PCAtools)
library(tibble)
library(ggalt)
library(evemodel) 
library(ape)
library(ggsignif)
library(ggplot2)
library(tidyverse)
library(pheatmap)
library(ggpubr)
library(cowplot)
library(ggvenn)
```

# TXimport
## Step 1: Set up 
This time, instead of using Uniprot IDs as our gene ID in our tx2gene files, we will use the single-copy ortholog ID from OrthoFinder. We will use "Orthogroup_Sequences.csv" from the "Command_Line_Code.Rmd" file to create these new tx2 genes files for each species. 
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Orthogroups/")
orthogroupSequenceNames <- read.csv("Orthogroup_Sequences.csv",header = TRUE)
dim(orthogroupSequenceNames)
#1817 single copy orthologs
colnames(orthogroupSequenceNames)

cnat <- orthogroupSequenceNames[,c(1,2)]
names(cnat)
write.csv(cnat,file="cnat_contig_to_orthogroup.csv")

mcav <- orthogroupSequenceNames[,c(1,3)]
names(mcav)
write.csv(mcav,file="mcav_contig_to_orthogroup.csv")

oann <- orthogroupSequenceNames[,c(1,4)]
names(oann)
write.csv(oann,file="oann_contig_to_orthogroup.csv")

past <- orthogroupSequenceNames[,c(1,5)]
names(past)
write.csv(past,file="past_contig_to_orthogroup.csv")

pstr <- orthogroupSequenceNames[,c(1,6)]
names(pstr)
write.csv(pstr,file="pstr_contig_to_orthogroup.csv")
```

## Step 2: Import read quants from Salmon
Read in the raw counts from Salmon for ALL coral samples
```{r}
# C. natans
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/cnat/salmon/")
Cnat_samples <- read.table("Cnat_samples.csv", header = TRUE)
Cnat_samples
Cnat_files <- file.path(Cnat_samples$samples, "quant.sf")
names(Cnat_files) <- paste0("sample", 1:10)
all(file.exists(Cnat_files))
Cnat_tx2gene <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Orthogroups/cnat_contig_to_orthogroup.csv")
names(Cnat_tx2gene)
Cnat_tx2gene <- Cnat_tx2gene[,c(3,2)]
Cnat_tx2gene$cnat_ref_proteome_cdhit <- gsub("\\..*","",Cnat_tx2gene$cnat_ref_proteome_cdhit)
Cnat_txi <- tximport(Cnat_files, type = 'salmon',tx2gene=Cnat_tx2gene)
names(Cnat_txi)
tail(Cnat_txi$counts)
dim(Cnat_txi$counts)   #1812
dim(Cnat_tx2gene)  #1817
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Counts/")
write.csv(Cnat_txi$counts, file = "Cnat_counts.csv")

# M. cavernosa
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/mcav/salmon/")
Mcav_samples <- read.table("mcav_samples.csv", header = TRUE)
Mcav_samples
Mcav_files <- file.path(Mcav_samples$sample, "quant.sf")
names(Mcav_files) <- paste0("sample", 1:13)
all(file.exists(Mcav_files))
Mcav_tx2gene <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Orthogroups/mcav_contig_to_orthogroup.csv")
names(Mcav_tx2gene)
Mcav_tx2gene <- Mcav_tx2gene[,c(3,2)]
Mcav_tx2gene$mcav_ref_proteome_cdhit <- gsub("\\..*","",Mcav_tx2gene$mcav_ref_proteome_cdhit)
Mcav_txi <- tximport(Mcav_files, type = 'salmon',tx2gene=Mcav_tx2gene)
names(Mcav_txi)
tail(Mcav_txi$counts)
dim(Mcav_txi$counts)   #1803
dim(Mcav_tx2gene)  #1817
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Counts/")
write.csv(Mcav_txi$counts, file = "Mcav_counts.csv")

#O.annularis
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/Oann/salmon/")
Oann_samples <- read.table("Oann_samples.csv", header = TRUE)
Oann_samples
Oann_files <- file.path(Oann_samples$sample, "quant.sf")
names(Oann_files) <- paste0("sample", 1:11)
all(file.exists(Oann_files))
Oann_tx2gene <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Orthogroups/Oann_contig_to_orthogroup.csv")
names(Oann_tx2gene)
Oann_tx2gene <- Oann_tx2gene[,c(3,2)]
Oann_tx2gene$oann_ref_proteome_cdhit <- gsub("\\..*","",Oann_tx2gene$oann_ref_proteome_cdhit)
Oann_txi <- tximport(Oann_files, type = 'salmon',tx2gene=Oann_tx2gene)
names(Oann_txi)
tail(Oann_txi$counts)
dim(Oann_txi$counts)   #1814
dim(Oann_tx2gene)  #1817
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Counts/")
write.csv(Oann_txi$counts, file = "Oann_counts.csv")

# P.astreoides
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/Past/salmon/")
Past_samples <- read.table("Past_samples.csv", header = TRUE)
Past_samples
Past_files <- file.path(Past_samples$sample, "quant.sf")
names(Past_files) <- paste0("sample", 1:11)
all(file.exists(Past_files))
Past_tx2gene <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Orthogroups/Past_contig_to_orthogroup.csv")
names(Past_tx2gene)
Past_tx2gene <- Past_tx2gene[,c(3,2)]
Past_tx2gene$past_ref_proteome_cdhit <- gsub("\\..*","",Past_tx2gene$past_ref_proteome_cdhit)
Past_txi <- tximport(Past_files, type = 'salmon',tx2gene=Past_tx2gene)
names(Past_txi)
tail(Past_txi$counts)
dim(Past_txi$counts)   #1808
dim(Past_tx2gene)  #1817
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Counts/")
write.csv(Past_txi$counts, file = "Past_counts.csv")

#P.strigosa
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/Pstr/salmon/")
Pstr_samples <- read.table("Pstr_samples.csv", header = TRUE)
Pstr_samples
Pstr_files <- file.path(Pstr_samples$sample, "quant.sf")
names(Pstr_files) <- paste0("sample", 1:7)
all(file.exists(Pstr_files))
Pstr_tx2gene <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Orthogroups/Pstr_contig_to_orthogroup.csv")
names(Pstr_tx2gene)
Pstr_tx2gene <- Pstr_tx2gene[,c(3,2)]
Pstr_tx2gene$pstr_ref_proteome_cdhit <- gsub("\\..*","",Pstr_tx2gene$pstr_ref_proteome_cdhit)
Pstr_txi <- tximport(Pstr_files, type = 'salmon',tx2gene=Pstr_tx2gene)
names(Pstr_txi)
tail(Pstr_txi$counts)
dim(Pstr_txi$counts)   #1804
dim(Pstr_tx2gene)  #1817
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Counts/")
write.csv(Pstr_txi$counts, file = "Pstr_counts.csv")
```
Now merge the count data
```{r}
Cnat <- as.data.frame(Cnat_txi$counts)
Cnat <- tibble::rownames_to_column(Cnat,"Orthogroup")
colnames(Cnat) <- c("Orthogroup","cnat_c2","cnat_c3","cnat_c4","cnat_c6","cnat_c8","cnat_d2","cnat_d3","cnat_d4","cnat_d6","cnat_d8")

Mcav <- as.data.frame(Mcav_txi$counts)
Mcav <- tibble::rownames_to_column(Mcav,"Orthogroup")
colnames(Mcav) <- c("Orthogroup","mcav_c1","mcav_c3","mcav_c5","mcav_c6","mcav_c7","mcav_c8","mcav_d1","mcav_d2","mcav_d3","mcav_d4","mcav_d5","mcav_d6","mcav_d8")

Oann <- as.data.frame(Oann_txi$counts)
Oann <- tibble::rownames_to_column(Oann,"Orthogroup")
colnames(Oann) <- c("Orthogroup","oann_c1","oann_c2","oann_c3","oann_c6","oann_c7","oann_c8","oann_d1","oann_d3","oann_d6","oann_d7","oann_d8")

Past <- as.data.frame(Past_txi$counts)
Past <- tibble::rownames_to_column(Past,"Orthogroup")
colnames(Past) <- c("Orthogroup","past_c1","past_c5","past_c6","past_c7","past_d1","past_d2","past_d3","past_d4","past_d5","past_d6","past_d7")

Pstr <- as.data.frame(Pstr_txi$counts)
Pstr <- tibble::rownames_to_column(Pstr,"Orthogroup")
colnames(Pstr) <- c("Orthogroup","pstr_c3","pstr_c5","pstr_c7","pstr_c8","pstr_d3","pstr_d5","pstr_d7")

Mcav_Cnat <- inner_join(Mcav,Cnat, by ="Orthogroup")
Mcav_Cnat_Oann <- inner_join(Mcav_Cnat,Oann,by="Orthogroup")
Mcav_Cnat_Oann_Past <- inner_join(Mcav_Cnat_Oann,Past,by="Orthogroup")
all_species <- inner_join(Mcav_Cnat_Oann_Past,Pstr,by="Orthogroup")
dim(all_species)
names(all_species)
# 1781 single-copy orthologs expressed and shared between all 5 species
```

## Step 3: Annotate the single-copy orthologs

```{r}
# Merge in the names of the mcav proteome sequences
mcav_ortho_to_contig <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Orthogroups/mcav_SC_ortholog_sequences.csv")
View(mcav_ortho_to_contig)
names(mcav_ortho_to_contig)[2]<-paste("mcav_contig")

all_species_contig <- left_join(all_species,mcav_ortho_to_contig,by="Orthogroup")
all_species_contig <- all_species_contig %>% relocate(mcav_contig, .after = Orthogroup)

# Annotate with Uniprot Entry ID's with "mcav_sc_orthologs_annotated.txt"
annot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Annotation/mcav_sc_orthologs_annotated.csv")
master <- left_join(all_species_contig,annot,by="mcav_contig")
names(master)
master <- master %>% relocate(Entry, .after = mcav_contig)
master <- master %>% relocate(Entry.name, .after = Entry)
master <- master %>% relocate(Evalue, .after = Entry.name)
View(master)
dim(master)  #1781 

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Counts/new")
write.csv(master,file="masterCounts_1.csv")
```

# DESeq2

Reorganize "masterCounts_1.csv" so that columns name match row names in "metaData.csv"
Make counts integers.
Run DESeq2 on Orthogroups to get Rlog normalized counts.

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/")
countData <- read.csv("./Counts/masterCounts_1.csv", row.names="Orthogroup")
colnames(countData)
countData <- countData[,-c(1:5)]
colData <- read.csv("metaData.csv",row.names = "name")
colData
rownames(colData)
colnames(countData)
# Reorganize countData so that columns match row names in colData:
countData <- countData %>% select(order(colnames(countData)))

# Make counts in countData integers:
names(countData)
countData[1:52] <-  round(countData[1:52], digits=0)
countData <- as.data.frame(countData)

# Set the design
dds_treatment <- DESeqDataSetFromMatrix(countData = countData, 
                                        colData = colData, 
                                        design = ~species + treatment )

# Filter out columns with an average number of reads less than 10
dds_treatment <- dds_treatment[ rowMeans(counts(dds_treatment)) > 10, ] 

# Run DESeq2
test <- DESeq(dds_treatment)
res <- results(test)
res
resorderd <-res[order(res$padj),] 
setwd("./DESeq2/")
write.csv(resorderd, file = "DESeq_treatment_1.csv")
rld <- rlog(dds_treatment, blind=FALSE) 
rrld <- assay(rld)
dim(rrld)  # 1766 after filtering out low counts
write.csv(rrld, file = "treatment_rlog_1.csv")  
```

# EVE

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/DESeq2/")
exprTbl <- read.csv("treatment_rlog_1.csv")
dim(exprTbl)
colnames(exprTbl)
colnames(exprTbl)[colnames(exprTbl)=="X"] <- "Ortholog"
head(exprTbl)

# The table needs to be converted to a matrix:
exprMat <- as.matrix(exprTbl[,-1])
rownames(exprMat) <- exprTbl$Ortholog
exprMat
dim(exprMat)  #1766  52

# Species phylogeny can be obtained from OrthoFinder output in DIR=/OrthoFinder/Results/Species_Tree/SpeciesTree_rooted.txt

# I find it is best to run OrthoFinder with an outgroup species to get a better species tree. I included a Cassiopea proteome (generated the same way as the other proteomes) as my outgroup to obtain the following rooted species tree:
# (Cassiopea_proteome:0.405291,(past_reference_proteome:0.207024,((mcav_reference_proteome:0.0938366,oann_reference_proteome:0.0723831)0.262775:0.0844971,(pstr_reference_proteome:0.0954775,cnat_reference_proteome:0.106944)0.514097:0.105377)0.45022:0.114876)1:0.405291); 
# I then remove the out group from the rooted tree and abbreviate the species names
speciesTree <-read.tree(text="((past:0.207024,((mcav:0.0938366,oann:0.0723831)0.262775:0.0844971,(pstr:0.0954775,cnat:0.106944)0.514097:0.105377)0.45022:0.114876)1:0.405291);")

# Check validity of species tree
plot(speciesTree,
     type = "phylogram",
     use.edge.length = TRUE,
     node.pos = NULL,
     show.tip.label = TRUE,
     root.edge = TRUE)

# EVE needs to know which species each column in the expression matrix belongs to. This is done by creating a vector of species names corresponding to the tip labels in the species tree

speciesTree$tip.label
colnames(exprMat)
colSpecies <- sub("_.*$","",colnames(exprMat))
colSpecies

# Run EVE
res <- betaSharedTest(tree = speciesTree, gene.data = exprMat, colSpecies = colSpecies)

# plot likelihood ratio test statistic histogram
hist(res$LRT,freq = F)

# Plot the chi-squared distribution with one degree of freedom
x = seq(0.5,10,length.out = 100)
y = dchisq(x,df = 1)
lines(x,y,col="red")

# P-value can then be calculated using:
pval = pchisq(res$LRT,df = 1,lower.tail = F)
pval

# The shared beta:
res$sharedBeta
# [1] 0.3377453
log(0.3377453)
# [1] -1.085463

#Combine LRT, beta, theta, sigma2, alpha:
head(cbind(res$LRT,res$indivBetaRes$par,pval))
colnames(exprMat)
dim(exprMat)
# [1]  1766   52
a <- cbind(res$LRT,res$indivBetaRes$par,pval,exprMat[,1:52])
rownames(a) <- exprTbl$Ortholog

# Give Column Name to LRT column
colnames(a)
colnames(a)[colnames(a)==""] <- "LRT"
head(a)
a <- as.data.frame(a)
a <- tibble::rownames_to_column(a, "Orthogroup")



#Export results
getwd()
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/EVE/new")

write.csv(a, file="R_EVE_results_SCTLD_5species.csv")
# Add three columns in excel:
  # 1: "type" column after "pval" with formula:
    # =IF(F2<0.3375006,"Lineage Specific","Highly Variable")    # F2 = "beta" column
  # 2: "significant" column after "type" with formula:
    # =IF(G2<0.05,"Significant","Not Significant")              # G2 = "pval" column
  # 3: "category" after "significant" column with formula:
    # =IF(I2="Significant",H2,"NS")                             # I2 = "significance", H2 = "type"

#Visualize LRT v Beta by volcano plot for gene data
library(DESeq2)
library(ggplot2)
library(dplyr)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/EVE/new")
Shared_EVE <- read.csv("R_EVE_results_SCTLD_5species.csv")
colnames(Shared_EVE)
head(Shared_EVE)
Shared_EVE_sig <- Shared_EVE %>% filter(pval <= 0.05)
write.csv(Shared_EVE_sig, file="Signficant_SCTLD_eve.csv")
# Make a basic volcano plot
names(Shared_EVE)
names(Shared_EVE_sig)
Shared_EVE_LS <- Shared_EVE_sig %>% filter(log(beta)< (-1.085463))
dim(Shared_EVE_LS)  #85
Shared_EVE_HV <- Shared_EVE_sig %>% filter(log(beta)> (-1.085463))
dim(Shared_EVE_HV)  #216
216+85  # 301
getwd()
pdf(file="EVE_volcanoPlot_orthologs.pdf",height=6,width=6)
p <- ggplot(data = Shared_EVE,
       aes(x=log(beta),y=-log(pval),color=category))+
  geom_point()+
  scale_color_manual(values=c("#F8766D","#00b9e3","black"))+
  geom_vline(xintercept = -1.086188, col="black", linetype="dashed")+
  annotate("text", x = 0.9, y = 9.5, label = "Shared beta", size = 3)+
  geom_hline(yintercept = -log(0.05), col="black",linetype = "dashed")+
  annotate("text", x = -4, y = 2.5, label = "p = 0.05", size = 3)+
  xlim(-5,5)+
  theme_light()
p + labs(color = "EVE Gene Category",title = "EVE Genes Volcano Plot",subtitle = "85 Lineage-Specific and 216 Highly Variable Genes") +
  theme(plot.title = element_text(face = "bold"), plot.subtitle = element_text(size = 9))
dev.off()
```

## Highly Variable Genes

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/EVE/new")
sig_EVE <- read.csv("Signficant_SCTLD_eve.csv")
colnames(sig_EVE)
sig_EVE <- sig_EVE[,c(2,7,8,11)]   #Orthogroup,beta, pval, category
dim(sig_EVE)   #301 significant EVE genes
names(sig_EVE)
rlogs <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/DESeq2/treatment_rlog_1.csv")
names(rlogs)
colnames(rlogs)[1] <- "Orthogroup"
sig_EVE_rlogs <- left_join(sig_EVE,rlogs,by="Orthogroup")    #301
names(sig_EVE_rlogs)
write.csv(sig_EVE_rlogs,file="sig_EVE_rlogs.csv")


plasticGenes <- subset(sig_EVE_rlogs, sig_EVE$category == "Highly Variable")
names(plasticGenes)
plasticGenes <- plasticGenes[,-c(2:4)]
dim(plasticGenes)
# [1] 216  53
names(plasticGenes)
t_plasticGenes <- setNames(data.frame(t(plasticGenes[,-1])), plasticGenes[,1])
row.names(t_plasticGenes)
t_plasticGenes <- tibble::rownames_to_column(t_plasticGenes, "name")

label <- read.csv("./../../metaData.csv")
colnames(label)
label <- label[,-c(1,2,6,12:22)]
master <- merge(label,t_plasticGenes,by="name")
master
dim(master)
# [1]  52 224
write.csv(master,file="highlyVariable_rlogs_ortholog.csv")
```

### ANOVA

```{r}
results <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/EVE/new/highlyVariable_rlogs_ortholog.csv")
names(results)
results <- results[,c(4,10:ncol(results))]
names(results)
dim(results)
# 52 217

# ANOVA loop significant results
getwd()
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/EVE/new")
for(i in 2:ncol(results))
{
  column <- names(results[i])
  #tidy will summarize and return neat format
  avz <- broom::tidy(aov(results[,i] ~ status, data = results))
  
  # Add this condition if you only want aov with P < 0.05
  if(avz$p.value[1] < 0.05) {
    
    print(column)
  }
}


[1] "OG0009610"
OG0009610 <- aov(results$OG0009610 ~ results$status, data = results)
summary(OG0009610) 
TukeyHSD(OG0009610) # Significant between healthy and exposed

[1] "OG0009628"
OG0009628 <- aov(results$OG0009628 ~ results$status, data = results)
summary(OG0009628) 
TukeyHSD(OG0009628) # Significant between exposed and disease

[1] "OG0009900"
OG0009900 <- aov(results$OG0009900 ~ results$status, data = results)
summary(OG0009900) 
TukeyHSD(OG0009900) # Significant between healthy and disease

[1] "OG0009952"
OG0009952 <- aov(results$OG0009952 ~ results$status, data = results)
summary(OG0009952) 
TukeyHSD(OG0009952) # Significant between healthy and disease

[1] "OG0009999"
OG0009999 <- aov(results$OG0009999 ~ results$status, data = results)
summary(OG0009999) 
TukeyHSD(OG0009999) # Significant between healthy and disease

[1] "OG0010018"
OG0010018 <- aov(results$OG0010018 ~ results$status, data = results)
summary(OG0010018) 
TukeyHSD(OG0010018) # Significant between healthy and disease

[1] "OG0010040"
OG0010040 <- aov(results$OG0010040 ~ results$status, data = results)
summary(OG0010040) 
TukeyHSD(OG0010040) # Significant between healthy and disease

[1] "OG0010080"
OG0010080 <- aov(results$OG0010080 ~ results$status, data = results)
summary(OG0010080) 
TukeyHSD(OG0010080) # NOT significant

[1] "OG0010117"
OG0010117 <- aov(results$OG0010117 ~ results$status, data = results)
summary(OG0010117) 
TukeyHSD(OG0010117) # Significant between healthy and disease

[1] "OG0010145"
OG0010145 <- aov(results$OG0010145 ~ results$status, data = results)
summary(OG0010145) 
TukeyHSD(OG0010145) # Significant between healthy and disease;healthy and exposed

[1] "OG0010176"
OG0010176 <- aov(results$OG0010176 ~ results$status, data = results)
summary(OG0010176) 
TukeyHSD(OG0010176) # Significant between healthy and disease

[1] "OG0010177"
OG0010177 <- aov(results$OG0010177 ~ results$status, data = results)
summary(OG0010177) 
TukeyHSD(OG0010177) # Significant between healthy and disease; exposed and disease

[1] "OG0010196"
OG0010196 <- aov(results$OG0010196 ~ results$status, data = results)
summary(OG0010196) 
TukeyHSD(OG0010196) # Significant between healthy and disease

[1] "OG0010223"
OG0010223 <- aov(results$OG0010223 ~ results$status, data = results)
summary(OG0010223) 
TukeyHSD(OG0010223) # Significant between healthy and disease

[1] "OG0010246"
OG0010246 <- aov(results$OG0010246 ~ results$status, data = results)
summary(OG0010246) 
TukeyHSD(OG0010246) # NOT significant

[1] "OG0010267"
OG0010267 <- aov(results$OG0010267 ~ results$status, data = results)
summary(OG0010267) 
TukeyHSD(OG0010267) # NOT significant

[1] "OG0010270"
OG0010270 <- aov(results$OG0010270 ~ results$status, data = results)
summary(OG0010270) 
TukeyHSD(OG0010270) # Significant between healthy and disease

[1] "OG0010317"
OG0010317 <- aov(results$OG0010317 ~ results$status, data = results)
summary(OG0010317) 
TukeyHSD(OG0010317) # Significant between healthy and disease

[1] "OG0010335"
OG0010335 <- aov(results$OG0010335 ~ results$status, data = results)
summary(OG0010335) 
TukeyHSD(OG0010335) # Significant between healthy and disease

[1] "OG0010347"
OG0010347 <- aov(results$OG0010347 ~ results$status, data = results)
summary(OG0010347) 
TukeyHSD(OG0010347) # Significant between healthy and disease

[1] "OG0010377"
OG0010377 <- aov(results$OG0010377 ~ results$status, data = results)
summary(OG0010377) 
TukeyHSD(OG0010377) # Significant between healthy and disease

[1] "OG0010385"
OG0010385 <- aov(results$OG0010385 ~ results$status, data = results)
summary(OG0010385) 
TukeyHSD(OG0010385) # Significant between healthy and disease

[1] "OG0010386"
OG0010386 <- aov(results$OG0010386 ~ results$status, data = results)
summary(OG0010386) 
TukeyHSD(OG0010386) # Significant between healthy and disease

[1] "OG0010389"
OG0010389 <- aov(results$OG0010389 ~ results$status, data = results)
summary(OG0010389) 
TukeyHSD(OG0010389) # Significant between healthy and disease

[1] "OG0010405"
OG0010405 <- aov(results$OG0010405 ~ results$status, data = results)
summary(OG0010405) 
TukeyHSD(OG0010405) # Significant between healthy and disease

[1] "OG0010416"
OG0010416 <- aov(results$OG0010416 ~ results$status, data = results)
summary(OG0010416) 
TukeyHSD(OG0010416) # Significant between healthy and disease

[1] "OG0010441"
OG0010441 <- aov(results$OG0010441 ~ results$status, data = results)
summary(OG0010441) 
TukeyHSD(OG0010441) # Significant between healthy and disease

[1] "OG0010453"
OG0010453 <- aov(results$OG0010453 ~ results$status, data = results)
summary(OG0010453) 
TukeyHSD(OG0010453) # Significant between healthy and disease

[1] "OG0010458"
OG0010458 <- aov(results$OG0010458 ~ results$status, data = results)
summary(OG0010458) 
TukeyHSD(OG0010458) # Significant between healthy and disease

[1] "OG0010466"
OG0010466 <- aov(results$OG0010466 ~ results$status, data = results)
summary(OG0010466) 
TukeyHSD(OG0010466) # Significant between healthy and disease; exposed and disease

[1] "OG0010477"
OG0010477 <- aov(results$OG0010477 ~ results$status, data = results)
summary(OG0010477) 
TukeyHSD(OG0010477) # Significant between healthy and disease

[1] "OG0010479"
OG0010479 <- aov(results$OG0010479 ~ results$status, data = results)
summary(OG0010479) 
TukeyHSD(OG0010479) # Significant between healthy and exposed

[1] "OG0010517"
OG0010517 <- aov(results$OG0010517 ~ results$status, data = results)
summary(OG0010517) 
TukeyHSD(OG0010517) # Significant between healthy and disease

[1] "OG0010529"
OG0010529 <- aov(results$OG0010529 ~ results$status, data = results)
summary(OG0010529) 
TukeyHSD(OG0010529) # Significant between healthy and disease

[1] "OG0010687"
OG0010687 <- aov(results$OG0010687 ~ results$status, data = results)
summary(OG0010687) 
TukeyHSD(OG0010687) # Significant between healthy and disease; exposed and disease

[1] "OG0010700"
OG0010700 <- aov(results$OG0010700 ~ results$status, data = results)
summary(OG0010700) 
TukeyHSD(OG0010700) # Significant between healthy and exposed

[1] "OG0010712"
OG0010712 <- aov(results$OG0010712 ~ results$status, data = results)
summary(OG0010712) 
TukeyHSD(OG0010712) # Significant between healthy and disease

[1] "OG0010714"
OG0010714 <- aov(results$OG0010714 ~ results$status, data = results)
summary(OG0010714) 
TukeyHSD(OG0010714) # NOT significant

[1] "OG0010721"
OG0010721 <- aov(results$OG0010721 ~ results$status, data = results)
summary(OG0010721) 
TukeyHSD(OG0010721) # Significant between healthy and exposed

[1] "OG0010742"
OG0010742 <- aov(results$OG0010742 ~ results$status, data = results)
summary(OG0010742) 
TukeyHSD(OG0010742) # Significant between healthy and disease

[1] "OG0010781"
OG0010781 <- aov(results$OG0010781 ~ results$status, data = results)
summary(OG0010781) 
TukeyHSD(OG0010781) # Significant between healthy and disease

[1] "OG0010806"
OG0010806 <- aov(results$OG0010806 ~ results$status, data = results)
summary(OG0010806) 
TukeyHSD(OG0010806) # Significant between healthy and disease; exposed and disease

[1] "OG0010856"
OG0010856 <- aov(results$OG0010856 ~ results$status, data = results)
summary(OG0010856) 
TukeyHSD(OG0010856) # Significant between healthy and disease

[1] "OG0010888"
OG0010888 <- aov(results$OG0010888 ~ results$status, data = results)
summary(OG0010888) 
TukeyHSD(OG0010888) # Significant between exposed and disease

[1] "OG0010904"
OG0010904 <- aov(results$OG0010904 ~ results$status, data = results)
summary(OG0010904) 
TukeyHSD(OG0010904) # Significant between healthy and disease; exposed and disease

[1] "OG0010954"
OG0010954 <- aov(results$OG0010954 ~ results$status, data = results)
summary(OG0010954) 
TukeyHSD(OG0010954) # NOT significant

[1] "OG0010962"
OG0010962 <- aov(results$OG0010962 ~ results$status, data = results)
summary(OG0010962) 
TukeyHSD(OG0010962) # Significant between healthy and disease

[1] "OG0011004"
OG0011004 <- aov(results$OG0011004 ~ results$status, data = results)
summary(OG0011004) 
TukeyHSD(OG0011004) # Significant between healthy and disease

[1] "OG0011009"
OG0011009 <- aov(results$OG0011009 ~ results$status, data = results)
summary(OG0011009) 
TukeyHSD(OG0011009) # Significant between healthy and disease

[1] "OG0011043"
OG0011043 <- aov(results$OG0011043 ~ results$status, data = results)
summary(OG0011043) 
TukeyHSD(OG0011043) # Significant between healthy and disease

[1] "OG0011062"
OG0011062 <- aov(results$OG0011062 ~ results$status, data = results)
summary(OG0011062) 
TukeyHSD(OG0011062) # Significant between healthy and disease; exposed and disease

[1] "OG0011066"
OG0011066 <- aov(results$OG0011066 ~ results$status, data = results)
summary(OG0011066) 
TukeyHSD(OG0011066) # Significant between healthy and disease; exposed and disease

[1] "OG0011070"
OG0011070 <- aov(results$OG0011070 ~ results$status, data = results)
summary(OG0011070) 
TukeyHSD(OG0011070) # Significant between healthy and exposed

[1] "OG0011132"
OG0011132 <- aov(results$OG0011132 ~ results$status, data = results)
summary(OG0011132) 
TukeyHSD(OG0011132) # Significant between healthy and disease

[1] "OG0011136"
OG0011136 <- aov(results$OG0011136 ~ results$status, data = results)
summary(OG0011136) 
TukeyHSD(OG0011136) # Significant between healthy and disease

[1] "OG0011140"
OG0011140 <- aov(results$OG0011140 ~ results$status, data = results)
summary(OG0011140) 
TukeyHSD(OG0011140) # Significant between healthy and disease

[1] "OG0011223"
OG0011223 <- aov(results$OG0011223 ~ results$status, data = results)
summary(OG0011223) 
TukeyHSD(OG0011223) # Significant between healthy and disease

[1] "OG0011246"
OG0011246 <- aov(results$OG0011246 ~ results$status, data = results)
summary(OG0011246) 
TukeyHSD(OG0011246) # Significant between exposed and disease

[1] "OG0011249"
OG0011249 <- aov(results$OG0011249 ~ results$status, data = results)
summary(OG0011249) 
TukeyHSD(OG0011249) # Significant between healthy and disease

[1] "OG0011283"
OG0011283 <- aov(results$OG0011283 ~ results$status, data = results)
summary(OG0011283) 
TukeyHSD(OG0011283) # Significant between healthy and disease; exposed and disease

[1] "OG0011345"
OG0011345 <- aov(results$OG0011345 ~ results$status, data = results)
summary(OG0011345) 
TukeyHSD(OG0011345) # Significant between healthy and disease; exposed and disease

[1] "OG0011346"
OG0011346 <- aov(results$OG0011346 ~ results$status, data = results)
summary(OG0011346) 
TukeyHSD(OG0011346) # Significant between healthy and disease

[1] "OG0011380"
OG0011380 <- aov(results$OG0011380 ~ results$status, data = results)
summary(OG0011380) 
TukeyHSD(OG0011380) # Significant between exposed and disease

[1] "OG0011444"
OG0011444 <- aov(results$OG0011444 ~ results$status, data = results)
summary(OG0011444) 
TukeyHSD(OG0011444) # Significant between healthy and disease

[1] "OG0011563"
OG0011563 <- aov(results$OG0011563 ~ results$status, data = results)
summary(OG0011563) 
TukeyHSD(OG0011563) # Significant between healthy and disease

[1] "OG0011572"
OG0011572 <- aov(results$OG0011572 ~ results$status, data = results)
summary(OG0011572) 
TukeyHSD(OG0011572) # Significant between healthy and disease; exposed and disease

[1] "OG0011610"
OG0011610 <- aov(results$OG0011610 ~ results$status, data = results)
summary(OG0011610) 
TukeyHSD(OG0011610) # Significant between healthy and disease; healthy and exposed

[1] "OG0011693"
OG0011693 <- aov(results$OG0011693 ~ results$status, data = results)
summary(OG0011693) 
TukeyHSD(OG0011693) # Significant between healthy and disease

[1] "OG0011783"
OG0011783 <- aov(results$OG0011783 ~ results$status, data = results)
summary(OG0011783) 
TukeyHSD(OG0011783) # Significant between healthy and disease; exposed and disease

[1] "OG0011934"
OG0011934 <- aov(results$OG0011934 ~ results$status, data = results)
summary(OG0011934) 
TukeyHSD(OG0011934) # Significant between healthy and disease; exposed and disease

[1] "OG0011988"
OG0011988 <- aov(results$OG0011988 ~ results$status, data = results)
summary(OG0011988) 
TukeyHSD(OG0011988) # Significant between healthy and exposed

[1] "OG0012035"
OG0012035 <- aov(results$OG0012035 ~ results$status, data = results)
summary(OG0012035) 
TukeyHSD(OG0012035) # Significant between healthy and disease; healthy and exposed

[1] "OG0012049"
OG0012049 <- aov(results$OG0012049 ~ results$status, data = results)
summary(OG0012049) 
TukeyHSD(OG0012049) # Significant between healthy and exposed

[1] "OG0012059"
OG0012059 <- aov(results$OG0012059 ~ results$status, data = results)
summary(OG0012059) 
TukeyHSD(OG0012059) # Significant between healthy and disease

[1] "OG0012131"
OG0012131 <- aov(results$OG0012131 ~ results$status, data = results)
summary(OG0012131) 
TukeyHSD(OG0012131) # Significant between healthy and disease

[1] "OG0012135"
OG0012135 <- aov(results$OG0012135 ~ results$status, data = results)
summary(OG0012135) 
TukeyHSD(OG0012135) # Significant between healthy and disease

[1] "OG0012234"
OG0012234 <- aov(results$OG0012234 ~ results$status, data = results)
summary(OG0012234) 
TukeyHSD(OG0012234) # Significant between healthy and disease

[1] "OG0012259"
OG0012259 <- aov(results$OG0012259 ~ results$status, data = results)
summary(OG0012259) 
TukeyHSD(OG0012259) # Significant between healthy and disease

[1] "OG0012265"
OG0012265 <- aov(results$OG0012265 ~ results$status, data = results)
summary(OG0012265) 
TukeyHSD(OG0012265) # Significant between healthy and disease; exposed and disease

[1] "OG0012277"
OG0012277 <- aov(results$OG0012277 ~ results$status, data = results)
summary(OG0012277) 
TukeyHSD(OG0012277) # Significant between healthy and exposed

[1] "OG0012396"
OG0012396 <- aov(results$OG0012396 ~ results$status, data = results)
summary(OG0012396) 
TukeyHSD(OG0012396) # Significant between healthy and disease; exposed and disease

[1] "OG0012472"
OG0012472 <- aov(results$OG0012472 ~ results$status, data = results)
summary(OG0012472) 
TukeyHSD(OG0012472) # Significant between healthy and disease

[1] "OG0012839"
OG0012839 <- aov(results$OG0012839 ~ results$status, data = results)
summary(OG0012839) 
TukeyHSD(OG0012839) # Significant between healthy and disease
```

#### Significant 
Make a dataframe including only the significant highly variable orthologs

```{r}
names(results)
sig_orthologs <- select(results,c(OG0009610,OG0009628,OG0009900,OG0009952,OG0009999,OG0010018,OG0010040,OG0010117,OG0010145,OG0010176,OG0010177,OG0010196,OG0010223,OG0010270,OG0010317,OG0010335,OG0010347,OG0010377,OG0010385,OG0010386,OG0010389,OG0010405,OG0010416,OG0010441,OG0010453,OG0010458,OG0010466,OG0010477,OG0010479,OG0010517,OG0010529,OG0010687,OG0010700,OG0010712,OG0010721,OG0010742,OG0010781,OG0010806,OG0010856,OG0010888,OG0010904,OG0010962,OG0011004,OG0011009,OG0011043,OG0011062,OG0011066,OG0011070,OG0011132,OG0011136,OG0011140,OG0011223,OG0011246,OG0011249,OG0011283,OG0011345,OG0011346,OG0011380,OG0011444,OG0011563,OG0011572,OG0011610,OG0011693,OG0011783,OG0011934,OG0011988,OG0012035,OG0012049,OG0012059,OG0012131,OG0012135,OG0012234,OG0012259,OG0012265,OG0012277,OG0012396,OG0012472,OG0012839))

names(results)

write.csv(sig_orthologs,file="sig_HV_orthologs.csv")

dim(sig_orthologs)
t_sig_orthologs <- as.data.frame(t(sig_orthologs))
label <- read.csv("./../../metaData.csv")
names(label)
names(t_sig_orthologs) <- label$name
dim(t_sig_orthologs)
getwd()

past_annot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Annotation/past_sc_orthologs_annotated.csv")
past_contig <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Orthogroups/past_contig_to_orthogroup.csv")
past_contig <- past_contig[,-c(1)]
colnames(past_contig)[colnames(past_contig)=="past_ref_proteome_cdhit"] <- "past_contig"
past_info <- merge(past_annot,past_contig,by="past_contig")
names(past_info)
names(t_sig_orthologs)
t_sig_orthologs <- tibble::rownames_to_column(t_sig_orthologs, "Orthogroup")

results <- left_join(t_sig_orthologs,past_info,by="Orthogroup")
dim(results)
# [1] 78 57
names(results)
results <- results %>% relocate(past_contig, .after = Orthogroup)
results <- results %>% relocate(Entry, .after = past_contig)
results <- results %>% relocate(Entry.name, .after = Entry)
results <- results %>% relocate(Evalue, .after = Entry.name)

getwd()
write.csv(results,file="sig_HV_ANOVA_annot.csv")

results
goodEval <- results %>% filter(results$Evalue < 0.000001)
write.csv(goodEval,file="sig_HV_ANOVA_annot_goodEval.csv")

uniprot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/uniprot.csv")

names(goodEval)
names(uniprot)

annot <- left_join(goodEval,uniprot,by="Entry")
names(annot)
colnames(annot)[colnames(annot)=="Entry.name.x"] <- "Entry.name"
annot <- annot[,-c(58)]
annot <- annot %>% relocate(Gene.names, .after = Evalue)
annot <- annot %>% relocate(Protein.names, .after = Gene.names)
annot <- annot %>% relocate(Organism, .after = Protein.names)
annot <- annot %>% relocate(Gene.ontology..GO., .after = Organism)
getwd()
write.csv(annot,file="sig_HV_ANOVA_annot_goodEval_annot.csv")
```

Add the Tukey p value results to this file in 3 columns: EvI,CvI and CvE

##### Heatmap

```{r}
library(dplyr)
HV_genes <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/EVE/new/highlyVariable_rlogs_ortholog.csv")
dim(HV_genes)
sig_orthologs <- select(HV_genes,c(name,status,OG0012259,OG0011004,OG0011070,OG0011066,OG0010040,OG0010712,OG0009952,OG0012131,OG0012472,OG0011444,OG0011693,OG0012839,OG0011988,OG0010529,OG0012234,OG0012135,OG0010888,OG0010177,OG0010517,OG0011246,OG0012396,OG0012059,OG0010721,OG0010904,OG0010386,OG0009628,OG0011572,OG0011380,OG0011345,OG0012035,OG0012049,OG0010453,OG0010176,OG0010962,OG0010458,OG0010806,OG0010117,OG0010385,OG0009999,OG0010405,OG0010317,OG0010196,OG0012265,OG0011563,OG0011934,OG0010687,OG0011062,OG0010466,OG0011283,OG0012277,OG0010477,OG0010223,OG0010335,OG0010479,OG0010856))

View(sig_orthologs)
dim(sig_orthologs)
sig_orthologs_ordered <- sig_orthologs[order(sig_orthologs$status),]
View(sig_orthologs_ordered)
sig_orthologs_ordered$status
HV_sig_infected <- sig_orthologs_ordered[c(1:19),-c(1:2)]
HV_sig_exposed <- sig_orthologs_ordered[c(20:27),-c(1:2)]
HV_sig_control <- sig_orthologs_ordered[c(28:52),-c(1:2)]

HV_sig_control_avg <- colMeans(HV_sig_control)
HV_sig_exposed_avg <- colMeans(HV_sig_exposed)
HV_sig_infected_avg <- colMeans(HV_sig_infected)
HV_sig_genes_avg <- rbind(HV_sig_control_avg,HV_sig_exposed_avg,HV_sig_infected_avg)
row.names(HV_sig_genes_avg) <- c("control","exposed","infected")
HV_sig_genes_avg
t_HV_sig_genes_avg <- data.frame(t(HV_sig_genes_avg))
t_HV_sig_genes_avg <- tibble::rownames_to_column(t_HV_sig_genes_avg, "Orthogroup")
#write.csv(t_HV_sig_genes_avg,file="sig_HV_genes_avg_expression_3treatments.csv")

t_HV_sig_genes_avg <- tibble::column_to_rownames(t_HV_sig_genes_avg, "Orthogroup")

library(pheatmap)
getwd()
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/EVE/new")
pdf(file = "sig_HV_genes_heatmap.pdf",width = 10,height = 10)
pheatmap(t_HV_sig_genes_avg,
         scale="row",
         cluster_cols = FALSE,
         cluster_rows = TRUE,
         cellwidth = 10,
         cellheight = 8,
         color = colorRampPalette(c("#0571b0","#92c5de","white","#f4a582","#ca0020"))(200),
         treeheight_row = 20,
         treeheight_col = NULL,
         fontsize_row = 5,
         fontsize_col = 5
)
dev.off()




HV_genes <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/EVE/highlyVariable_rlogs_ortholog.csv")
dim(HV_genes)
sig_orthologs <- select(HV_genes,c(name,status,OG0010466,OG0010177,OG0011934,OG0012265,OG0011066,OG0010904,OG0010806,OG0012396,OG0012839,OG0010117,OG0010405,OG0010040,OG0011062,OG0011345,OG0010517,OG0010176,OG0012059,OG0010477,OG0010453,OG0012131,OG0010458,OG0011283,OG0011693,OG0010962,OG0012035,OG0012234,OG0011444,OG0009999,OG0010712,OG0012259,OG0011246))

View(sig_orthologs)
dim(sig_orthologs)  #52  33
sig_orthologs_ordered <- sig_orthologs[order(sig_orthologs$status),]
View(sig_orthologs_ordered)
sig_orthologs_ordered$status
HV_sig_infected <- sig_orthologs_ordered[c(1:19),-c(1:2)]
HV_sig_exposed <- sig_orthologs_ordered[c(20:27),-c(1:2)]
HV_sig_control <- sig_orthologs_ordered[c(28:52),-c(1:2)]

HV_sig_control_avg <- colMeans(HV_sig_control)
HV_sig_exposed_avg <- colMeans(HV_sig_exposed)
HV_sig_infected_avg <- colMeans(HV_sig_infected)
HV_sig_genes_avg <- rbind(HV_sig_control_avg,HV_sig_exposed_avg,HV_sig_infected_avg)
row.names(HV_sig_genes_avg) <- c("control","exposed","infected")
HV_sig_genes_avg
t_HV_sig_genes_avg <- data.frame(t(HV_sig_genes_avg))
t_HV_sig_genes_avg <- tibble::rownames_to_column(t_HV_sig_genes_avg, "Orthogroup")
#write.csv(t_HV_sig_genes_avg,file="sig_HV_genes_avg_expression_3treatments.csv")

t_HV_sig_genes_avg <- tibble::column_to_rownames(t_HV_sig_genes_avg, "Orthogroup")

getwd()
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/EVE/new")
pdf(file = "super_sig_HV_genes_heatmap.pdf",width = 10,height = 10)
pheatmap(t_HV_sig_genes_avg,
         scale="row",
         cluster_cols = FALSE,
         cluster_rows = TRUE,
         cellwidth = 10,
         cellheight = 8,
         color = colorRampPalette(c("#0571b0","#92c5de","white","#f4a582","#ca0020"))(200),
         treeheight_row = 20,
         treeheight_col = NULL,
         fontsize_row = 5,
         fontsize_col = 5
)
dev.off()


```

##### Boxplots

```{r}
results <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/EVE/new/sig_HV_ANOVA_annot_goodEval_annot.csv")
names(results)
results <- results[,-c(1)]
dim(results)    # 54 significant Highly Variable EVE genes with good evalue
names(results)

resorderd <-results[order(results$CvI),] 
resorderd <- resorderd %>% filter(CvI < (0.05)) 
dim(resorderd)  #45 significantly different between control and infected

resorderd <-results[order(results$CvE),] 
resorderd <- resorderd %>% filter(CvE < (0.05)) 
dim(resorderd)  #6 significantly different between control and exposed

resorderd <-results[order(results$EvI),] 
resorderd <- resorderd %>% filter(EvI < (0.05)) 
dim(resorderd)  #18 significantly different between control and exposed

names(resorderd)
resorderd <- resorderd[,-c(2:12)]

names(results)
results <- results[,-c(2:12)]
t_resordered_rlogs <- setNames(as.data.frame(t(results[,-1])), results[,1])
row.names(t_resordered_rlogs)
t_resordered_rlogs <- tibble::rownames_to_column(t_resordered_rlogs, "name")
view(t_resordered_rlogs)


label <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/DESeq2/metaData.csv")
colnames(label)
label <- label[,c(1,4,6)] # keep species,name, and status
results <- merge(label,t_resordered_rlogs,by="name")
view(results)


results$status <- gsub("healthy","control", results$status)
results$status <- gsub("disease","infected", results$status)
results$status <- factor(results$status, levels = c("control","exposed","infected"))
results$species <- factor(results$species, levels = c("mcav","past","pstr","oann","cnat"))

```

###### Tpm4
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0010466 <- compare_means(OG0010466~status,data=results,method = "t.test")
OG0010466

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0010466_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0010466, fill = status)) +
  geom_point(aes(y = OG0010466, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Tropomyosin alpha-4 chain") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0010466, y.position = 12, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0010466_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0010466, fill = status)) +
  geom_point(aes(y = OG0010466, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Tropomyosin alpha-4 chain") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0010466, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
#OG0010466_all <- 
  ggarrange(OG0010466_species,OG0010466_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("Tpm4.pdf")
```

###### FOXL2

```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0010177 <- compare_means(OG0010177~status,data=results,method = "t.test")
OG0010177

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0010177_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0010177, fill = status)) +
  geom_point(aes(y = OG0010177, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Forkhead box protein L2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0010177, y.position = 10, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0010177_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0010177, fill = status)) +
  geom_point(aes(y = OG0010177, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Forkhead box protein L2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0010177, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
#OG0010177_all <- 
  ggarrange(OG0010177_species,OG0010177_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("FOXL2.pdf")
```

###### Klhl18 
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0011934 <- compare_means(OG0011934~status,data=results,method = "t.test")
OG0011934

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0011934_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0011934, fill = status)) +
  geom_point(aes(y = OG0011934, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Kelch-like protein 18") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0011934, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0011934_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0011934, fill = status)) +
  geom_point(aes(y = OG0011934, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Kelch-like protein 18") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0011934, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
#OG0011934_all <- 
  ggarrange(OG0011934_species,OG0011934_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("Klhl18.pdf")
```

###### ADAMTS7 
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0012265 <- compare_means(OG0012265~status,data=results,method = "t.test")
OG0012265

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0012265_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0012265, fill = status)) +
  geom_point(aes(y = OG0012265, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "ADAM-TS 7") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0012265, y.position = 12.7, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0012265_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0012265, fill = status)) +
  geom_point(aes(y = OG0012265, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "ADAM-TS 7") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0012265, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
#OG0012265_all <- 
  ggarrange(OG0012265_species,OG0012265_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("ADAMTS7.pdf")
```

###### Pacrg
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0011066 <- compare_means(OG0011066~status,data=results,method = "t.test")
OG0011066

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0011066_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0011066, fill = status)) +
  geom_point(aes(y = OG0011066, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Parkin coregulated gene protein homolog") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0011066, y.position = 11, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0011066_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0011066, fill = status)) +
  geom_point(aes(y = OG0011066, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Parkin coregulated gene protein homolog") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0011066, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
#OG0011066_all <- 
  ggarrange(OG0011066_species,OG0011066_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("Pacrg.pdf")
```

###### P4HTM
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0010904 <- compare_means(OG0010904~status,data=results,method = "t.test")
OG0010904

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0010904_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0010904, fill = status)) +
  geom_point(aes(y = OG0010904, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Transmembrane prolyl 4-hydroxylase, P4H-TM") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0010904, y.position = 10, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0010904_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0010904, fill = status)) +
  geom_point(aes(y = OG0010904, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Transmembrane prolyl 4-hydroxylase, P4H-TM") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0010904, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
#OG0010904_all <- 
  ggarrange(OG0010904_species,OG0010904_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("P4HTM.pdf")
```

###### NSMCE4A
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0012396 <- compare_means(OG0012396~status,data=results,method = "t.test")
OG0012396

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

#OG0012396_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0012396, fill = status)) +
  geom_point(aes(y = OG0012396, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Non-SMC element 4 homolog A") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0012396, y.position = 8.3, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0012396_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0012396, fill = status)) +
  geom_point(aes(y = OG0012396, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Non-SMC element 4 homolog A") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0012396, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
#OG0012396_all <- 
  ggarrange(OG0012396_species,OG0012396_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("NSMCE4A.pdf")
```

###### Pdcl3
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0012839 <- compare_means(OG0012839~status,data=results,method = "t.test")
OG0012839

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0012839_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0012839, fill = status)) +
  geom_point(aes(y = OG0012839, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Phosducin-like protein 3") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0012839, y.position = 8.9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0012839_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0012839, fill = status)) +
  geom_point(aes(y = OG0012839, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Phosducin-like protein 3") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0012839, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
#OG0012839_all <- 
  ggarrange(OG0012839_species,OG0012839_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("Pdcl3.pdf")
```

###### ELK1
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0010405 <- compare_means(OG0010405~status,data=results,method = "t.test")
OG0010405

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0010405_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0010405, fill = status)) +
  geom_point(aes(y = OG0010405, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "ETS domain-containing protein Elk-1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0010405, y.position = 11, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0010405_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0010405, fill = status)) +
  geom_point(aes(y = OG0010405, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "ETS domain-containing protein Elk-1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0010405, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
#OG0010405_all <- 
  ggarrange(OG0010405_species,OG0010405_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("ELK1.pdf")
```

######ADORA2A
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0011062 <- compare_means(OG0011062~status,data=results,method = "t.test")
OG0011062

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0011062_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0011062, fill = status)) +
  geom_point(aes(y = OG0011062, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Adenosine receptor A2a") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0011062, y.position = 9.5, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0011062_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0011062, fill = status)) +
  geom_point(aes(y = OG0011062, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Adenosine receptor A2a") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0011062, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
#OG0011062_all <- 
  ggarrange(OG0011062_species,OG0011062_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("ADORA2A.pdf")
```

###### Folh1
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0012035 <- compare_means(OG0012035~status,data=results,method = "t.test")
OG0012035

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0012035_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0012035, fill = status)) +
  geom_point(aes(y = OG0012035, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Glutamate carboxypeptidase 2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0012035, y.position = 11.3, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0012035_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0012035, fill = status)) +
  geom_point(aes(y = OG0012035, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Glutamate carboxypeptidase 2") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0012035, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
#OG0012035_all <- 
  ggarrange(OG0012035_species,OG0012035_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("Folh1.pdf")
```



TPM4:       OG0010466_species,OG0010466_status        Cytoskeleton
PACRG:      OG0011066_species,OG0011066_status        Immunity
P4HM:       OG0010904_species,OG0010904_status        Hypoxia
Pdcl3:      OG0012839_species,OG0012839_status        Apoptosis
ADORA2A:    OG0011062_species,OG0011062_status        GPCR signaling
Folh1:      OG0012035_species,OG0012035_status        Folate metabolism


###### Irf2bp1

```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0012131 <- compare_means(OG0012131~status,data=results,method = "t.test")
OG0012131

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0012131_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0012131, fill = status)) +
  geom_point(aes(y = OG0012131, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Interferon regulatory factor 2-binding protein 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0012131, y.position = 11.3, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0012131_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0012131, fill = status)) +
  geom_point(aes(y = OG0012131, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Interferon regulatory factor 2-binding protein 1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0012131, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
#OG0012131_all <- 
  ggarrange(OG0012131_species,OG0012131_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("Irf2bp1.pdf")
```

###### BTG1
```{r}
# just do this function to get the brackets and pvalues, we will replace these pvalues with Tukey results manually in illustrator
OG0010176 <- compare_means(OG0010176~status,data=results,method = "t.test")
OG0010176

my_comparisons <- list(c('control','exposed'),c('exposed','infected'),c('control','infected'))

OG0010176_status <- 
ggplot(data = results, 
       aes(x = status, y = OG0010176, fill = status)) +
  geom_point(aes(y = OG0010176, color = status),
             position =position_dodge(width = 0.47),size = 1.5, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Protein BTG1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))+
  stat_pvalue_manual(OG0010176, y.position = 12, step.increase = 0.15, inherit.aes = FALSE,size = 3)

OG0010176_species <- 
ggplot(data = results, 
       aes(x = species, y = OG0010176, fill = status)) +
  geom_point(aes(y = OG0010176, color = status),
             position =position_dodge(width = 0.47),size = 1, alpha = 2) +
  geom_boxplot(width = 0.5, outlier.shape = NA, alpha = 0.5) +
  labs(y = "Log Expression", x = NULL, fill = "status", col= "status", title = "Protein BTG1") +
  scale_x_discrete(labels=c("control"="C","exposed"="E","infected"="I"))+
  scale_color_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_fill_manual(values = c("#00BA38","#619CFF","#F8766D"))+
  scale_y_continuous(expand = expansion(mult = c(0.05,0.1)))
 # stat_pvalue_manual(OG0010176, y.position = 9, step.increase = 0.15, inherit.aes = FALSE,size = 3)

setwd("./figures/")
#OG0010176_all <- 
  ggarrange(OG0010176_species,OG0010176_status,
            ncol=2, widths = c(1.75,1.25),
            common.legend = TRUE, legend = "bottom",align = "hv")
ggsave("Irf2bp1.pdf")
```

###### Together
```{r}
getwd()
pdf(file="boxplots_new.pdf",height=14,width=6)
ggarrange(OG0010466_species,OG0010466_status,
          OG0011066_species,OG0011066_status,
          OG0010904_species,OG0010904_status,
          OG0012265_species,OG0012265_status,
          OG0011062_species,OG0011062_status,
          OG0010176_species,OG0010176_status,
  ncol=2, nrow=6,widths = c(1.75,1.25),
  common.legend = TRUE, legend = "right",align = "hv")
dev.off()
```

#### Venn Diagram

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/EVE/new")
sharedHV <- read.csv("sig_HV_ANOVA_annot_goodEval_annot.csv")
dim(sharedHV)
view(sharedHV)
names(sharedHV)
sharedHV <- sharedHV[,-c(1)]

EvI <-sharedHV[order(sharedHV$EvI),] 
EvI <- EvI %>% filter(EvI <= 0.05)
dim(EvI)  #18

CvI <-sharedHV[order(sharedHV$CvI),] 
CvI <- CvI %>% filter(CvI <= 0.05)
dim(CvI)  #45

CvE <-sharedHV[order(sharedHV$CvE),] 
CvE <- CvE %>% filter(CvE <= 0.05)
dim(CvE)  #6


x <- list(
  EvI = EvI$Orthogroup,
  CvI = CvI$Orthogroup,
  CvE = CvE$Orthogroup
  )

getwd()
pdf(file = "sharedHVGenes_Venn_SYMBIONT.pdf",height=6,width = 6)
ggvenn(
  x,
  fill_color = c("#2ECFCA", "#F8DF77", "#FF4E62"),
  show_percentage = FALSE,
  stroke_size = 0.3, set_name_size = 3
)
dev.off()
```

## Lineage Specific Genes

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/EVE/new")
sig_EVE <- read.csv("sig_EVE_rlogs.csv")
LineageGenes <- subset(sig_EVE, sig_EVE$category == "Lineage Specific")
names(LineageGenes)
dim(LineageGenes)  # 85 57
LineageGenes <- LineageGenes[,-c(1,3:5)]
dim(LineageGenes)
# [1] 85 53
#write.csv(LineageGenes,file="lineage_rlogs.csv")

#Average the rlog counts across species
colnames(LineageGenes)
LineageGenes$cnat <- rowMeans(LineageGenes[,c(2:11)])
LineageGenes$mcav <- rowMeans(LineageGenes[,c(12:24)])
LineageGenes$oann <- rowMeans(LineageGenes[,c(25:35)])
LineageGenes$past <- rowMeans(LineageGenes[,c(36:46)])
LineageGenes$pstr <- rowMeans(LineageGenes[,c(47:53)])
names(LineageGenes)
LSavg <- LineageGenes[,c(1,54:58)]
names(LSavg)
dim(LSavg)  # 85 6 
write.csv(LSavg,file="lineage_species_avg.csv")


past_annot <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Annotation/past_sc_orthologs_annotated.csv")
past_contig <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Orthogroups/past_contig_to_orthogroup.csv")
past_contig <- past_contig[,-c(1)]
colnames(past_contig)[colnames(past_contig)=="past_ref_proteome_cdhit"] <- "past_contig"
past_info <- merge(past_annot,past_contig,by="past_contig")
names(past_info)
names(LSavg)
#LSavg <- tibble::rownames_to_column(LSavg, "Orthogroup")
LSavg <- left_join(LSavg,past_info,by="Orthogroup")

LSavg <- LSavg %>% relocate(past_contig, .after = Orthogroup)
LSavg <- LSavg %>% relocate(Entry, .after = past_contig)
LSavg <- LSavg %>% relocate(Entry.name, .after = Entry)
LSavg <- LSavg %>% relocate(Evalue, .after = Entry.name)

getwd()
write.csv(LSavg,file="sig_LS_avg_annot.csv")

goodEval <- LSavg %>% filter(LSavg$Evalue < 0.000001)
write.csv(goodEval,file="sig_LS_avg_annot_goodEval.csv")

```


### Correlate to Relative Risk
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/EVE/new")
LSavg <- read.csv("sig_LS_avg_annot_goodEval.csv",row.names = "Entry")
names(LSavg)
LSavg <- LSavg[,-c(1:5)]
LSavg <- LSavg %>% relocate(oann, .after = cnat)
LSavg <- LSavg %>% relocate(pstr, .after = oann)
LSavg <- LSavg %>% relocate(past, .after = pstr)
head(LSavg)

LS_rlogs <- as.data.frame(t(LSavg))
colnames(LS_rlogs)
LS_rlogs$Species <- row.names(LS_rlogs)
head(LS_rlogs)

label <- read.csv("./../../../EVE/meta_sampleInfo.csv", row.names = "Species")
label
label <- label[c("cnat","oann","pstr","past","mcav"),]
rownames(label)
rownames(LS_rlogs)

LS_rlogs$RelativeRisk <- label$Median_RR
names(LS_rlogs)
LS_rlogs <- LS_rlogs %>% relocate(Species, everything())
LS_rlogs <- LS_rlogs %>% relocate(RelativeRisk, .after = Species)

names(LS_rlogs)

##### Correlate to Relative Risk

# Loop through all tests, print estimates and p-values if p <=0.05
for (i in 3:length(LS_rlogs)){
   a <- cor.test(LS_rlogs$RelativeRisk, LS_rlogs[,i],method = c("pearson"))
   if (a$p.value <= 0.05){
      print(paste(colnames(LS_rlogs)[i], " corr:", a$estimate, " p-value:", a$p.value))
   }
}
```
[1] "Q6NWF4  corr: 0.969015234625614  p-value: 0.00651669430351937"
[1] "Q8SQ72  corr: 0.884915512995971  p-value: 0.0460485845095573"
[1] "Q6PDL0  corr: 0.911168059252092  p-value: 0.0313555524128266"

##### Scatterplots

```{r}   
# Plot the genes
LS_rlogs$Species <- factor(LS_rlogs$Species, levels = c("cnat","oann","pstr","past","mcav"))


#Q8SQ72 <- 
   ggplot(LS_rlogs,aes(x=RelativeRisk, y=Q8SQ72))+
   geom_point(aes(x=RelativeRisk, y=Q8SQ72, color=Species), position = "identity",size=3.5)+
   geom_smooth(method=glm, color="black")+
   #geom_text(label=LS_rlogs$species, nudge_y = -0.5)+
   labs(y="Mean Expression", x = "Relative Risk", title = "Gamma-sarcoglycan")+
   stat_cor(label.x.npc = "left",label.y = 10.1)

#Q6PDL0 <- 
   ggplot(LS_rlogs,aes(x=RelativeRisk, y=Q6PDL0))+
   geom_point(aes(x=RelativeRisk, y=Q6PDL0, color=Species), position = "identity",size=3)+
   geom_smooth(method=glm, color="black")+
   #geom_text(label=row.names(LS_rlogs), nudge_y = -0.5, size =3 )+
   labs(y="Mean Expression", x = "Relative Risk", title = "Cytoplasmic dynein 1 light intermediate chain 2")+
   stat_cor(label.x.npc = "left",label.y = 10.5)
   
#Q6NWF4 <- 
   ggplot(LS_rlogs,aes(x=RelativeRisk, y=Q6NWF4))+
   geom_point(aes(x=RelativeRisk, y=Q6NWF4, color=Species), position = "identity",size=3)+
   geom_smooth(method=glm, color="black")+
   #geom_text(label=row.names(LS_rlogs), nudge_y = -0.5,size=3)+
   labs(y="Mean Expression", x = "Relative Risk", title = "Vacuolar protein-sorting-associated protein 25")+
   stat_cor(label.x.npc = "left",label.y = 8.3)

```

### Correlate to LGR
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/EVE/new")
LSavg <- read.csv("sig_LS_avg_annot_goodEval.csv",row.names = "Entry")
names(LSavg)
LSavg <- LSavg[,-c(1:5)]
LSavg <- LSavg %>% relocate(oann, .after = cnat)
LSavg <- LSavg %>% relocate(pstr, .after = oann)
LSavg <- LSavg %>% relocate(past, .after = pstr)
head(LSavg)

LS_rlogs <- as.data.frame(t(LSavg))
colnames(LS_rlogs)
LS_rlogs$Species <- row.names(LS_rlogs)
head(LS_rlogs)

label <- read.csv("./../../../EVE/meta_sampleInfo.csv", row.names = "Species")
label
label <- label[c("cnat","oann","pstr","past","mcav"),]
rownames(label)
rownames(LS_rlogs)

LS_rlogs$LGR <- label$Avg_LGR
names(LS_rlogs)
LS_rlogs <- LS_rlogs %>% relocate(Species, everything())
LS_rlogs <- LS_rlogs %>% relocate(LGR, .after = Species)


names(LS_rlogs)

##### Correlate to Relative Risk

# Here is an example test:
cor.test(LS_rlogs$Q99JY9,LS_rlogs$LGR, method=c("pearson"))

# Loop through all tests, print estimates and p-values if p <= 0.05
for (i in 3:length(LS_rlogs)){
   a <- cor.test(LS_rlogs$LGR, LS_rlogs[,i])
   if (a$p.value <= 0.05){
      print(paste(colnames(LS_rlogs)[i], " corr:", a$estimate, " p-value:", a$p.value))
   }
}

```
[1] "Q63ZP1  corr: -0.889928346915833  p-value: 0.0431065703379798"
[1] "Q8VEE0  corr: -0.890625079928523  p-value: 0.0427025637944924"
[1] "Q8VBV7  corr: -0.910697691560168  p-value: 0.0316026274652564" 
[1] "Q9D428  corr: -0.895959996497787  p-value: 0.0396495315775774"
[1] "Q8TBH0  corr: 0.990478243445025  p-value: 0.00111375125995564"

##### Scatterplots
```{r}   
# Plot the genes
library(ggplot2)
library(ggpubr)
library(cowplot)
LS_rlogs$Species <- factor(LS_rlogs$Species, levels = c("cnat","oann","pstr","past","mcav"))

#Q63ZP1 <- 
   ggplot(LS_rlogs,aes(x=LGR, y=Q63ZP1))+
   geom_point(aes(x=LGR, y=Q63ZP1, color=Species), position = "identity",size=3)+
   geom_smooth(method=glm, color="black")+
   #geom_text(label=row.names(LS_rlogs), nudge_y = -0.5)+
   labs(y="Mean Expression", x = "Avg. Lesion Growth Rate", title = "PHD finger protein 10")+
   stat_cor(label.x.npc = "left",label.y = 3.7)

#Q8VEE0 <- 
   ggplot(LS_rlogs,aes(x=LGR, y=Q8VEE0))+
   geom_point(aes(x=LGR, y=Q8VEE0, color=Species), position = "identity",size=3)+
   geom_smooth(method=glm, color="black")+
   #geom_text(label=row.names(LS_rlogs), nudge_y = -0.5)+
   labs(y="Mean Expression", x = "Avg. Lesion Growth Rate", title = "Ribulose-phosphate 3-epimerase")+
   stat_cor(label.x.npc = "left",label.y = .8)

#Q8VBV7 <- 
   ggplot(LS_rlogs,aes(x=LGR, y=Q8VBV7))+
   geom_point(aes(x=LGR, y=Q8VBV7, color=Species), position = "identity",size=3)+
   geom_smooth(method=glm, color="black")+
   #geom_text(label=row.names(LS_rlogs), nudge_y = -0.5)+
   labs(y="Mean Expression", x = "Avg. Lesion Growth Rate", title = "COP9 signalosome complex subunit 8")+
   stat_cor(label.x.npc = "left",label.y = 6.5)

Q9D428 <- 
   ggplot(LS_rlogs,aes(x=LGR, y=Q9D428))+
   geom_point(aes(x=LGR, y=Q9D428, color=Species), position = "identity",size=3)+
   geom_smooth(method=glm, color="black")+
#   geom_text(label=row.names(LS_rlogs), nudge_y = -0.5)+
   labs(y="Mean Expression", x = "Avg. Lesion Growth Rate", title = "Golgin subfamily A member 7B")+
   stat_cor(label.x.npc = "left",label.y = .2)

#Q8TBH0 <- 
   ggplot(LS_rlogs,aes(x=LGR, y=Q8TBH0))+
   geom_point(aes(x=LGR, y=Q8TBH0, color=Species), position = "identity",size=4)+
   geom_smooth(method=glm, color="black")+
#   geom_text(label=row.names(LS_rlogs), nudge_y = -0.5)+
   labs(y="Mean Expression", x = "Avg. Lesion Growth Rate", title = "Arrestin domain-containing protein 2")+
   stat_cor(label.x.npc = "left",label.y = 9.3)

   
   
getwd()
pdf(file="LS_correlations_1.pdf",height = 8,width=7)
ggarrange(Q96LD1,Q4R5P6,Q6NWF4,
          Q63ZP1,Q8VEE0,Q8VBV7,Q9D428,Q8TBH0,
          ncol = 3, nrow = 3, legend = "none",
          common.legend = FALSE,
          align = "hv"
          )
dev.off()

```

#### Heatmap

```{r}
library(tibble)
library(dplyr)
library(ggpubr)
library(pheatmap)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/EVE/")
LSavg <- read.csv("sig_LS_avg_annot_goodEval.csv",row.names = "Entry")
names(LSavg)
LSavg <- LSavg[,-c(1:5)]
LSavg <- LSavg %>% relocate(oann, .after = cnat)
LSavg <- LSavg %>% relocate(pstr, .after = oann)
LSavg <- LSavg %>% relocate(past, .after = pstr)
head(LSavg)
names(LSavg)
#write.csv(LSavg,file="LS_species_avg_entry.csv")



LSavg <- read.csv("LS_species_avg_entry.csv")
colnames(LSavg)[colnames(LSavg)=="X"] <- "Entry"
LSavg <- tibble::column_to_rownames(LSavg, "Entry")

label <- read.csv("./../../EVE/meta_sampleInfo.csv", row.names = "Species")
label
label <- label[c("cnat","oann","pstr","past","mcav"),c(2,3)]
colnames(label)
rownames(label)
colnames(LSavg)


LSavg <- LSavg[,c("mcav","past","pstr","oann","cnat")]
RRGenes <- LSavg[c("Q96LD1","Q4R5P6","Q6NWF4"),]

breaksList=seq(0,12,by=1)

getwd()
pdf(file = "LS_Genes_RR_heatmap.pdf",height = 6,width = 6)
RR <- 
pheatmap(RRGenes,
         cluster_cols = FALSE,
         cluster_rows = TRUE,
         scale = "none",
         color = colorRampPalette(c("white","#fed976","#fc4e2a","#67000d"))(12),
         border_color = "grey50",
         annotation_col = label,
         legend = TRUE,
         breaks = breaksList,
         legend_labels = label,
         fontsize_row = 7,
         fontsize_col = 10,
         cellwidth = 20,
         cellheight = 20
         )
dev.off()

LSavg <- LSavg[,c("past","mcav","pstr","oann","cnat")]
LGRgenes <- LSavg[c("Q63ZP1","Q8VEE0","Q8VBV7","Q9D428","Q8TBH0"),]


pdf(file = "LS_Genes_LGR_heatmap.pdf",height = 6,width = 6)
LGR <- 
pheatmap(LGRgenes,
         cluster_cols = FALSE,
         cluster_rows = TRUE,
         scale = "none",
         color = colorRampPalette(c("white","#fed976","#fc4e2a","#67000d"))(12),
         border_color = "grey50",
         annotation_col = label,
         legend = TRUE,
         breaks = breaksList,
         legend_labels = label,
         fontsize_row = 7,
         fontsize_col = 10,
         cellwidth = 20,
         cellheight = 20
         )
dev.off()
```

