---
title: "EVE Analysis - Coral Single-Copy Orthologs"
author: "Kelsey Beavers"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Required Packages
```{r}
library(dplyr)
library(tximport)
library(DESeq2)
library(readr)
library(dplyr)
library(ggrepel)
library(PCAtools)
library(tibble)
library(ggalt)
library(evemodel) 
library(ape)
library(ggsignif)
library(ggplot2)
library(tidyverse)
library(pheatmap)
library(ggpubr)
library(cowplot)
```

# TXimport
## Step 1: Set up 
This time, instead of using Uniprot IDs as our gene ID in our tx2gene files, we will use the single-copy ortholog ID from OrthoFinder. We will use "Orthogroup_Sequences.csv" from the "Command_Line_Code.Rmd" file to create these new tx2 genes files for each species. 
```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Orthogroups/")
orthogroupSequenceNames <- read.csv("Orthogroup_Sequences.csv",header = TRUE)
dim(orthogroupSequenceNames)
#1817 single copy orthologs
colnames(orthogroupSequenceNames)

cnat <- orthogroupSequenceNames[,c(1,2)]
names(cnat)
write.csv(cnat,file="cnat_contig_to_orthogroup.csv")

mcav <- orthogroupSequenceNames[,c(1,3)]
names(mcav)
write.csv(mcav,file="mcav_contig_to_orthogroup.csv")

oann <- orthogroupSequenceNames[,c(1,4)]
names(oann)
write.csv(oann,file="oann_contig_to_orthogroup.csv")

past <- orthogroupSequenceNames[,c(1,5)]
names(past)
write.csv(past,file="past_contig_to_orthogroup.csv")

pstr <- orthogroupSequenceNames[,c(1,6)]
names(pstr)
write.csv(pstr,file="pstr_contig_to_orthogroup.csv")
```

## Step 2: Import read quants from Salmon
Read in the raw counts from Salmon for ALL coral samples
```{r}
# C. natans
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/cnat/salmon/")
Cnat_samples <- read.table("Cnat_samples.csv", header = TRUE)
Cnat_samples
Cnat_files <- file.path(Cnat_samples$samples, "quant.sf")
names(Cnat_files) <- paste0("sample", 1:10)
all(file.exists(Cnat_files))
Cnat_tx2gene <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Orthogroups/cnat_contig_to_orthogroup.csv")
names(Cnat_tx2gene)
Cnat_tx2gene <- Cnat_tx2gene[,c(3,2)]
Cnat_tx2gene$cnat_ref_proteome_cdhit <- gsub("\\..*","",Cnat_tx2gene$cnat_ref_proteome_cdhit)
Cnat_txi <- tximport(Cnat_files, type = 'salmon',tx2gene=Cnat_tx2gene)
names(Cnat_txi)
tail(Cnat_txi$counts)
dim(Cnat_txi$counts)   #1812
dim(Cnat_tx2gene)  #1817
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Counts/")
write.csv(Cnat_txi$counts, file = "Cnat_counts.csv")

# M. cavernosa
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/mcav/salmon/")
Mcav_samples <- read.table("mcav_samples.csv", header = TRUE)
Mcav_samples
Mcav_files <- file.path(Mcav_samples$sample, "quant.sf")
names(Mcav_files) <- paste0("sample", 1:13)
all(file.exists(Mcav_files))
Mcav_tx2gene <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Orthogroups/mcav_contig_to_orthogroup.csv")
names(Mcav_tx2gene)
Mcav_tx2gene <- Mcav_tx2gene[,c(3,2)]
Mcav_tx2gene$mcav_ref_proteome_cdhit <- gsub("\\..*","",Mcav_tx2gene$mcav_ref_proteome_cdhit)
Mcav_txi <- tximport(Mcav_files, type = 'salmon',tx2gene=Mcav_tx2gene)
names(Mcav_txi)
tail(Mcav_txi$counts)
dim(Mcav_txi$counts)   #1803
dim(Mcav_tx2gene)  #1817
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Counts/")
write.csv(Mcav_txi$counts, file = "Mcav_counts.csv")

#O.annularis
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/Oann/salmon/")
Oann_samples <- read.table("Oann_samples.csv", header = TRUE)
Oann_samples
Oann_files <- file.path(Oann_samples$sample, "quant.sf")
names(Oann_files) <- paste0("sample", 1:11)
all(file.exists(Oann_files))
Oann_tx2gene <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Orthogroups/Oann_contig_to_orthogroup.csv")
names(Oann_tx2gene)
Oann_tx2gene <- Oann_tx2gene[,c(3,2)]
Oann_tx2gene$oann_ref_proteome_cdhit <- gsub("\\..*","",Oann_tx2gene$oann_ref_proteome_cdhit)
Oann_txi <- tximport(Oann_files, type = 'salmon',tx2gene=Oann_tx2gene)
names(Oann_txi)
tail(Oann_txi$counts)
dim(Oann_txi$counts)   #1814
dim(Oann_tx2gene)  #1817
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Counts/")
write.csv(Oann_txi$counts, file = "Oann_counts.csv")

# P.astreoides
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/Past/salmon/")
Past_samples <- read.table("Past_samples.csv", header = TRUE)
Past_samples
Past_files <- file.path(Past_samples$sample, "quant.sf")
names(Past_files) <- paste0("sample", 1:11)
all(file.exists(Past_files))
Past_tx2gene <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Orthogroups/Past_contig_to_orthogroup.csv")
names(Past_tx2gene)
Past_tx2gene <- Past_tx2gene[,c(3,2)]
Past_tx2gene$past_ref_proteome_cdhit <- gsub("\\..*","",Past_tx2gene$past_ref_proteome_cdhit)
Past_txi <- tximport(Past_files, type = 'salmon',tx2gene=Past_tx2gene)
names(Past_txi)
tail(Past_txi$counts)
dim(Past_txi$counts)   #1808
dim(Past_tx2gene)  #1817
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Counts/")
write.csv(Past_txi$counts, file = "Past_counts.csv")

#P.strigosa
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/Pstr/salmon/")
Pstr_samples <- read.table("Pstr_samples.csv", header = TRUE)
Pstr_samples
Pstr_files <- file.path(Pstr_samples$sample, "quant.sf")
names(Pstr_files) <- paste0("sample", 1:7)
all(file.exists(Pstr_files))
Pstr_tx2gene <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Orthogroups/Pstr_contig_to_orthogroup.csv")
names(Pstr_tx2gene)
Pstr_tx2gene <- Pstr_tx2gene[,c(3,2)]
Pstr_tx2gene$pstr_ref_proteome_cdhit <- gsub("\\..*","",Pstr_tx2gene$pstr_ref_proteome_cdhit)
Pstr_txi <- tximport(Pstr_files, type = 'salmon',tx2gene=Pstr_tx2gene)
names(Pstr_txi)
tail(Pstr_txi$counts)
dim(Pstr_txi$counts)   #1804
dim(Pstr_tx2gene)  #1817
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Counts/")
write.csv(Pstr_txi$counts, file = "Pstr_counts.csv")
```
Now merge the count data
```{r}
Cnat <- as.data.frame(Cnat_txi$counts)
Cnat <- tibble::rownames_to_column(Cnat,"Orthogroup")
colnames(Cnat) <- c("Orthogroup","cnat_c2","cnat_c3","cnat_c4","cnat_c6","cnat_c8","cnat_d2","cnat_d3","cnat_d4","cnat_d6","cnat_d8")

Mcav <- as.data.frame(Mcav_txi$counts)
Mcav <- tibble::rownames_to_column(Mcav,"Orthogroup")
colnames(Mcav) <- c("Orthogroup","mcav_c1","mcav_c3","mcav_c5","mcav_c6","mcav_c7","mcav_c8","mcav_d1","mcav_d2","mcav_d3","mcav_d4","mcav_d5","mcav_d6","mcav_d8")

Oann <- as.data.frame(Oann_txi$counts)
Oann <- tibble::rownames_to_column(Oann,"Orthogroup")
colnames(Oann) <- c("Orthogroup","oann_c1","oann_c2","oann_c3","oann_c6","oann_c7","oann_c8","oann_d1","oann_d3","oann_d6","oann_d7","oann_d8")

Past <- as.data.frame(Past_txi$counts)
Past <- tibble::rownames_to_column(Past,"Orthogroup")
colnames(Past) <- c("Orthogroup","past_c1","past_c5","past_c6","past_c7","past_d1","past_d2","past_d3","past_d4","past_d5","past_d6","past_d7")

Pstr <- as.data.frame(Pstr_txi$counts)
Pstr <- tibble::rownames_to_column(Pstr,"Orthogroup")
colnames(Pstr) <- c("Orthogroup","pstr_c3","pstr_c5","pstr_c7","pstr_c8","pstr_d3","pstr_d5","pstr_d7")

Mcav_Cnat <- inner_join(Mcav,Cnat, by ="Orthogroup")
Mcav_Cnat_Oann <- inner_join(Mcav_Cnat,Oann,by="Orthogroup")
Mcav_Cnat_Oann_Past <- inner_join(Mcav_Cnat_Oann,Past,by="Orthogroup")
all_species <- inner_join(Mcav_Cnat_Oann_Past,Pstr,by="Orthogroup")
dim(all_species)
names(all_species)
# 1781 single-copy orthologs expressed and shared between all 5 species
```

## Step 3: Annotate the single-copy orthologs

```{r}
# Merge in the names of the past proteome sequences
past_ortho_to_contig <- read.csv("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Orthogroups/past_SC_ortholog_sequences.csv")
View(past_ortho_to_contig)
past_ortho_to_contig <- past_ortho_to_contig[,-c(1)]
names(past_ortho_to_contig)[2]<-paste("past_contig")

all_species_contig <- left_join(all_species,past_ortho_to_contig,by="Orthogroup")
all_species_contig <- all_species_contig %>% relocate(past_contig, .after = Orthogroup)
dim(all_species_contig)  

# Annotate with Uniprot Entry ID's with "past_sc_orthologs_annotated.txt"
annot <- read.table("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Annotation/past_sc_orthologs_annotated.txt",sep="\t", quote="", comment.char="",header=TRUE)
master <- left_join(all_species_contig,annot,by="past_contig")
names(master)
master <- master %>% relocate(Entry, .after = past_contig)
master <- master %>% relocate(Entry.name, .after = Entry)
master <- master %>% relocate(Evalue, .after = Entry.name)
View(master)
dim(master)  #1781 

setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/Counts/")
write.csv(master,file="masterCounts_1.csv")
```

# DESeq2

Reorganize "masterCounts_1.csv" so that columns name match row names in "metaData.csv"
Make counts integers.
Run DESeq2 on Orthogroups to get Rlog normalized counts.

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/")
library(DESeq2)
library(dplyr)
countData <- read.csv("./Counts/masterCounts_1.csv", row.names="Orthogroup")
colnames(countData)
countData <- countData[,-c(1:5)]
colData <- read.csv("metaData.csv",row.names = "name")
colData
rownames(colData)
colnames(countData)
# Reorganize countData so that columns match row names in colData:
countData <- countData %>% select(order(colnames(countData)))

# Make counts in countData integers:
names(countData)
countData[1:52] <-  round(countData[1:52], digits=0)
countData <- as.data.frame(countData)

# Set the design
dds_treatment <- DESeqDataSetFromMatrix(countData = countData, 
                                        colData = colData, 
                                        design = ~species + treatment )

# Filter out columns with an average number of reads less than 10
dds_treatment <- dds_treatment[ rowMeans(counts(dds_treatment)) > 10, ] 

# Run DESeq2
test <- DESeq(dds_treatment)
res <- results(test)
res
resorderd <-res[order(res$padj),] 
setwd("./DESeq2/")
write.csv(resorderd, file = "DESeq_treatment_1.csv")
rld <- rlog(dds_treatment, blind=FALSE) 
rrld <- assay(rld)
dim(rrld)  # 1765 after filtering out low counts
write.csv(rrld, file = "treatment_rlog_1.csv")  
```

# EVE

```{r}
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/DESeq2/")
exprTbl <- read.csv("treatment_rlog.csv")
dim(exprTbl)
colnames(exprTbl)
colnames(exprTbl)[colnames(exprTbl)=="X"] <- "Ortholog"
head(exprTbl)

# The table needs to be converted to a matrix:
exprMat <- as.matrix(exprTbl[,-1])
rownames(exprMat) <- exprTbl$Ortholog
exprMat
dim(exprMat)  #1766  52

# Species phylogeny can be obtained from OrthoFinder output in DIR=/OrthoFinder/Results/Species_Tree/SpeciesTree_rooted.txt

# I find it is best to run OrthoFinder with an outgroup species to get a better species tree. I included a Cassiopea proteome (generated the same way as the other proteomes) as my outgroup to obtain the following rooted species tree:
# (Cassiopea_proteome:0.405291,(past_reference_proteome:0.207024,((mcav_reference_proteome:0.0938366,oann_reference_proteome:0.0723831)0.262775:0.0844971,(pstr_reference_proteome:0.0954775,cnat_reference_proteome:0.106944)0.514097:0.105377)0.45022:0.114876)1:0.405291); 
# I then remove the out group from the rooted tree and abbreviate the species names
speciesTree <-read.tree(text="((past:0.207024,((mcav:0.0938366,oann:0.0723831)0.262775:0.0844971,(pstr:0.0954775,cnat:0.106944)0.514097:0.105377)0.45022:0.114876)1:0.405291);")

# Check validity of species tree
plot(speciesTree,
     type = "phylogram",
     use.edge.length = TRUE,
     node.pos = NULL,
     show.tip.label = TRUE,
     root.edge = TRUE)

# EVE needs to know which species each column in the expression matrix belongs to. This is done by creating a vector of species names corresponding to the tip labels in the species tree

speciesTree$tip.label
colnames(exprMat)
colSpecies <- sub("_.*$","",colnames(exprMat))
colSpecies

# Run EVE
res <- betaSharedTest(tree = speciesTree, gene.data = exprMat, colSpecies = colSpecies)

# plot likelihood ratio test statistic histogram
hist(res$LRT,freq = F)

# Plot the chi-squared distribution with one degree of freedom
x = seq(0.5,10,length.out = 100)
y = dchisq(x,df = 1)
lines(x,y,col="red")

# P-value can then be calculated using:
pval = pchisq(res$LRT,df = 1,lower.tail = F)
pval

# The shared beta:
res$sharedBeta
# [1] 0.3375006
# [1] 0.3377453
log(0.3375006)
# [1] -1.086188

#Combine LRT, beta, theta, sigma2, alpha:
head(cbind(res$LRT,res$indivBetaRes$par,pval))
colnames(exprMat)
dim(exprMat)
# [1]  1766   52
a <- cbind(res$LRT,res$indivBetaRes$par,pval,exprMat[,1:52])
rownames(a) <- exprTbl$Ortholog

# Give Column Name to LRT column
colnames(a)
colnames(a)[colnames(a)==""] <- "LRT"
head(a)

#Export results
getwd()
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/EVE/")

write.csv(a, file="R_EVE_results_SCTLD_5species.csv")
# Add three columns in excel:
  # 1: "type" column with formula:
    # =IF(F2<0.3375006,"Lineage Specific","Highly Variable")    # F2 = "beta" column
  # 2: "significant column with formula:
    # =IF(G2<0.05,"Significant","Not Significant")              # G2 = "pval" column
  # 3: "category" column with formula:
    # =IF(I2="Significant",H2,"NS")                             # I2 = "significance", H2 = "type"

#Visualize LRT v Beta by volcano plot for gene data
library(DESeq2)
library(ggplot2)
library(dplyr)
setwd("~/OneDrive - University of Texas at Arlington/Dissertation/SCTLD/Files_for_R/January_2022/SCTLD/host/all/Orthofinder_cdhit/EVE/")
Shared_EVE <- read.csv("R_EVE_results_SCTLD_5species.csv")
colnames(Shared_EVE)
head(Shared_EVE)
Shared_EVE_sig <- Shared_EVE %>% filter(pval <= 0.05)
write.csv(Shared_EVE_sig, file="Signficant_SCTLD_eve.csv")
# Make a basic volcano plot
names(Shared_EVE)
names(Shared_EVE_sig)
Shared_EVE_LS <- Shared_EVE_sig %>% filter(log(beta)< (-1.127714))
dim(Shared_EVE_LS)  #85
Shared_EVE_HV <- Shared_EVE_sig %>% filter(log(beta)> (-1.127714))
dim(Shared_EVE_HV)  #216
216+85  # 301
getwd()
pdf(file="EVE_volcanoPlot_orthologs.pdf",height=6,width=6)
p <- ggplot(data = Shared_EVE,
       aes(x=log(beta),y=-log(pval),color=category))+
  geom_point()+
  scale_color_manual(values=c("#F8766D","#00b9e3","black"))+
  geom_vline(xintercept = -1.086188, col="black", linetype="dashed")+
  annotate("text", x = 0.9, y = 9.5, label = "Shared beta", size = 3)+
  geom_hline(yintercept = -log(0.05), col="black",linetype = "dashed")+
  annotate("text", x = -4, y = 2.5, label = "p = 0.05", size = 3)+
  xlim(-5,5)+
  theme_light()
p + labs(color = "EVE Gene Category",title = "EVE Genes Volcano Plot",subtitle = "85 Lineage-Specific and 216 Highly Variable Genes") +
  theme(plot.title = element_text(face = "bold"), plot.subtitle = element_text(size = 9))
dev.off()
```